{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import vcf\n",
    "import io\n",
    "import pandas as pd\n",
    "import pysam\n",
    "from cyvcf2 import VCF\n",
    "from gtfparse import read_gtf\n",
    "from tqdm import tqdm\n",
    "pd.set_option('display.max_columns', None)\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "from collections import Counter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/software/miniconda3/envs/Jupyter/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3058: DtypeWarning: Columns (5) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n",
      "INFO:numexpr.utils:Note: NumExpr detected 12 cores but \"NUMEXPR_MAX_THREADS\" not set, so enforcing safe limit of 8.\n",
      "INFO:numexpr.utils:NumExpr defaulting to 8 threads.\n",
      "INFO:root:Extracted GTF attributes: ['gene_id', 'db_xref', 'description', 'gbkey', 'gene', 'gene_biotype', 'pseudo', 'transcript_id', 'product', 'exon_number', 'gene_synonym', 'tag', 'protein_id', 'exception', 'inference', 'note', 'partial', 'The', 'transl_except', 'isoform', 'anticodon', 'non-AUG', 'standard_name', 'source', 'similar', 'transferase', 'codons', '12S', '16S', 'transl_table', 'ATPase']\n"
     ]
    }
   ],
   "source": [
    "trans=[]#create list of PID with rna-seq data in heidelberg\n",
    "with open('trans.txt')as s:\n",
    "    v=s.readlines()[1:]\n",
    "    for rec in v:\n",
    "        rec=rec.rstrip()\n",
    "        trans.append(rec.split(' ')[-1].split('-')[1])\n",
    "\n",
    "\n",
    "df=pd.read_csv('2020-01-28-AR-IntronExonJunctions_387_KGE_Genes.variant_info.vcf',skiprows=108,sep='\\t')\n",
    "df_genes=pd.read_excel('2021-01-28_cancer_gene_list_germline_splice.xlsx')\n",
    "\n",
    "\n",
    "\n",
    "df['PID_trans']=df['PID'].str.split('-',expand=True)[1]\n",
    "df=df[df['PID_trans'].isin(trans)]\n",
    "\n",
    "\n",
    "#naga=set(df['HUGO_Symbol']) ##gene symbols which cann ot be matched between naga and arne, no hits or renaming ?\n",
    "#for g,temp in zip(df_genes['Approved symbol (HGNC)'],df_genes['Splice Project Gene Priority']):\n",
    " #   if g not in naga:\n",
    "  #      print(g,temp)\n",
    "#naga=set(df['HUGO_Symbol']) #gene symbols which cann ot be matched between naga and arne, no hits or renaming ?\n",
    "#for g in df_genes['Approved symbol (HGNC)']:\n",
    " #   if g not in naga:\n",
    "  #      print(g)\n",
    "\n",
    "df_merge=pd.merge(df,df_genes,left_on='HUGO_Symbol',right_on='Approved symbol (HGNC)', how='left')\n",
    "df_merge.dropna(subset=['Splice Project Gene Priority'],inplace=True)\n",
    "\n",
    "\n",
    "\n",
    "hgnc=pd.read_csv('HGNC_trans_symbolLookUp.txt', sep='\\t')\n",
    "\n",
    "\n",
    "#for x,y in hgnc['Approved symbol'].value_counts().items(): # genes with no clear hgnc in my database, dropdupli, keep first\n",
    "#    if y >1:\n",
    " #       print(x)\n",
    "\n",
    "hgnc=hgnc.drop_duplicates(subset=['Approved symbol'])\n",
    "\n",
    "replaceDict={'NM_003220':'NM_001042425','NM_004241':'NM_032776','NM_022976':'NM_001144914'}\n",
    "for _ in replaceDict:\n",
    "    i=hgnc[hgnc['RefSeq accession']==_].index.values[0]\n",
    "    hgnc.at[i,'RefSeq accession']=replaceDict[_]\n",
    "#discrepancy between https://biomart.genenames.org/ and https://genome.ucsc.edu/cgi-bin/hgTracks?db=hg19\n",
    "# need to update the refseq trans\n",
    "\n",
    "\n",
    "df_merge=pd.merge(df_merge,hgnc,left_on='HUGO_Symbol', right_on='Approved symbol', how='inner')\n",
    "\n",
    "\n",
    "\n",
    "#set(df_merge[~df_merge['RefSeq accession'].isin(set(gtf_df['transcript_id']))]['RefSeq accession']) genes were hgnc trans can not be matched between gtf and df_merge\n",
    "\n",
    "gtf_df=read_gtf('/mnt/g27prist/CMTD/Stephan/bcbio_installation/genomes/Hsapiens/GRCh37/StephanDatabase/GRCH37/annotation/GRCh37_latest_genomic_replace.gtf')\n",
    "gtf_df['transcript_id']=gtf_df['transcript_id'].str.split('.',expand=True)[0]\n",
    "gtf_df=gtf_df[gtf_df['transcript_id'].isin(set(df_merge['RefSeq accession']))]\n",
    "\n",
    "df_merge=df_merge.drop_duplicates(subset=['#CHROM','POS','ALT','PID'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 19062/19062 [02:07<00:00, 149.53it/s]\n"
     ]
    }
   ],
   "source": [
    "def exonPos(trans,pos):\n",
    "    temp=gtf_df[(gtf_df['transcript_id']==trans)&(gtf_df['seqname'].str.len()<=3)&(gtf_df['feature']=='CDS')]\n",
    "    if temp.empty:\n",
    "        print(trans)\n",
    "        return('err','err')\n",
    "    strand=temp['strand'].values[0]\n",
    "\n",
    "    if strand=='+':\n",
    "        StartEx=temp['start'].min()\n",
    "        StopEx=temp['end'].max()\n",
    "    if strand=='-':\n",
    "        StartEx=temp['end'].max()\n",
    "        StopEx=temp['start'].min()\n",
    "    \n",
    "    \n",
    "    if strand == '+':\n",
    "        if pos < StartEx and pos > (StartEx -30):\n",
    "            dist=StartEx-pos\n",
    "            return (dist, '5utr_startcodon')\n",
    "        if pos > StopEx and pos < (StopEx + 30):\n",
    "            dist=pos-StopEx\n",
    "            return (dist, '3utr_stopcodon')\n",
    "        \n",
    "        OutsideAcceptor=temp[((temp['start']-30)<=pos)&((temp['start'])>pos)]\n",
    "        InsideAcceptor=temp[((temp['start']+30)>=pos)&((temp['start'])<=pos)]\n",
    "        \n",
    "        OutsideDonor=temp[((temp['end']+30)>=pos)&((temp['end'])<pos)]\n",
    "        InsideDonor=temp[((temp['end']-30)<=pos)&((temp['end'])>=pos)]\n",
    "        \n",
    "        if not OutsideAcceptor.empty:\n",
    "            dist=OutsideAcceptor['start'].values[0]-pos\n",
    "            return (dist,'outsideAcceptorSite')\n",
    "        if not InsideAcceptor.empty:\n",
    "            dist=pos-InsideAcceptor['start'].values[0]\n",
    "            return (dist,'insideAcceptorSite')\n",
    "            \n",
    "        \n",
    "        if not OutsideDonor.empty:\n",
    "            dist=pos-OutsideDonor['end'].values[0]\n",
    "            return (dist,'outsideDonorSite')\n",
    "            \n",
    "        if not InsideDonor.empty:\n",
    "            dist=InsideDonor['end'].values[0]-pos\n",
    "            return (dist,'insideDonorSite')\n",
    "    \n",
    "    \n",
    "    if strand =='-':\n",
    "        if pos > StartEx and pos < (StartEx + 30):\n",
    "            dist=pos-StartEx\n",
    "            return (dist, '5utr_startcodon')\n",
    "        if pos < StopEx and pos > (StopEx+30):\n",
    "            dist=StopEx-pos\n",
    "            return (dist, '3utr_stopcodon')\n",
    "        \n",
    "        OutsideDonor=temp[((temp['start']-30)<=pos)&((temp['start'])>pos)]\n",
    "        InsideDonor=temp[((temp['start']+30)>=pos)&((temp['start'])<=pos)]\n",
    "        \n",
    "        OutsideAcceptor=temp[((temp['end']+30)>=pos)&((temp['end'])<pos)]\n",
    "        InsideAcceptor=temp[((temp['end']-30)<=pos)&((temp['end'])>=pos)]\n",
    "        \n",
    "        if not OutsideAcceptor.empty:\n",
    "            dist=pos-OutsideAcceptor['end'].values[0]\n",
    "            return (dist,'outsideAcceptorSite')\n",
    "        if not InsideAcceptor.empty:\n",
    "            dist=InsideAcceptor['end'].values[0]-pos\n",
    "            return (dist,'insideAcceptorSite')\n",
    "            \n",
    "        \n",
    "        if not OutsideDonor.empty:\n",
    "            dist=OutsideDonor['start'].values[0]-pos\n",
    "            return (dist,'outsideDonor')\n",
    "            \n",
    "        if not InsideDonor.empty:\n",
    "            dist=pos-InsideDonor['start'].values[0]\n",
    "            return (dist,'insideDonor')\n",
    "    \n",
    "    return ('01','01')\n",
    "            \n",
    "    \n",
    "    \n",
    "    \n",
    "        \n",
    "    \n",
    "posRelativeRefSeq = []\n",
    "Utr_cod=[]\n",
    "\n",
    "for pos,trans in tqdm(zip(df_merge[\"POS\"],df_merge[\"RefSeq accession\"]),total=len(df_merge)):\n",
    "    _=exonPos(trans,int(pos))\n",
    "    posRelativeRefSeq.append(_[0])\n",
    "    Utr_cod.append(_[1])\n",
    "#df_merge['PosExonRefSeqAccession']=posRelativeRefSeq\n",
    "\n",
    "df_merge['PosExonRefSeqAccession']=posRelativeRefSeq\n",
    "df_merge['PosExon_type']=Utr_cod\n",
    "\n",
    "df_merge=df_merge[df_merge['PosExon_type']!='01']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_merge.to_pickle('dfmerge')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>Control_GT</th>\n",
       "      <th>Tumor_GT</th>\n",
       "      <th>HUGO_Symbol</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>HGVSg</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>Allele_Frequency</th>\n",
       "      <th>VEP_Most_Severe_Consequence</th>\n",
       "      <th>Positive_Evidence</th>\n",
       "      <th>Negative_Evidence</th>\n",
       "      <th>Positive_CharGer_Score</th>\n",
       "      <th>Negative_CharGer_Score</th>\n",
       "      <th>CharGer_Score</th>\n",
       "      <th>ClinVar_Pathogenicity</th>\n",
       "      <th>ACMG_Classification</th>\n",
       "      <th>CharGer_Classification</th>\n",
       "      <th>PubMed_Link</th>\n",
       "      <th>ClinVar_Traits</th>\n",
       "      <th>CharGer_Summary</th>\n",
       "      <th>Control_VAF</th>\n",
       "      <th>Tumor_VAF</th>\n",
       "      <th>Tumor_dpALT</th>\n",
       "      <th>Tumor_dp</th>\n",
       "      <th>Control_dpALT</th>\n",
       "      <th>Control_dp</th>\n",
       "      <th>GT_Classification</th>\n",
       "      <th>max_gnomAD_AF</th>\n",
       "      <th>exome_control_AF</th>\n",
       "      <th>local_control_AF</th>\n",
       "      <th>PID</th>\n",
       "      <th>overlap</th>\n",
       "      <th>canopyCluster</th>\n",
       "      <th>TiN_Class</th>\n",
       "      <th>CHR_ARM_INFO</th>\n",
       "      <th>AI_cluster</th>\n",
       "      <th>thresholdCluster</th>\n",
       "      <th>GIT_VERSION</th>\n",
       "      <th>ALL_FUNCTIONAL_CONSENSUS</th>\n",
       "      <th>RNA_EXPRESSION</th>\n",
       "      <th>RNA_INFO</th>\n",
       "      <th>RNA_VAF</th>\n",
       "      <th>PID_trans</th>\n",
       "      <th>Approved symbol (HGNC)</th>\n",
       "      <th>Splice Project Gene Priority</th>\n",
       "      <th>Origin (Bertelsen 2019 (PMID: 31263571), Schrader 2016 (PMC5477989), Huang 2018 (PMID: 29625052), Groebner 2018 (PMID: 29489754), Priestley 2019 (PMID: 31645765))</th>\n",
       "      <th>NCBI gene ID (HGNC)</th>\n",
       "      <th>RefSeq (HGNC)</th>\n",
       "      <th>Total Exon Count (hg19)</th>\n",
       "      <th>Coding Exon Count (hg19)</th>\n",
       "      <th>Protein size [aa] (hg19)</th>\n",
       "      <th>Gene Basket (NCT)</th>\n",
       "      <th>Gene classification (most likely mechanism)</th>\n",
       "      <th>Gene group (KGE)</th>\n",
       "      <th>inheritance cancer predisposition</th>\n",
       "      <th>OMIM-G Phenotypes</th>\n",
       "      <th>non-cancer related phenotype</th>\n",
       "      <th>final proposal</th>\n",
       "      <th>technical details gene</th>\n",
       "      <th>Huang_KGE_Cancer syndrome(s)</th>\n",
       "      <th>Huang_KGE_Major associated tumor types</th>\n",
       "      <th>Literature</th>\n",
       "      <th>HGNC ID</th>\n",
       "      <th>Status</th>\n",
       "      <th>Approved symbol</th>\n",
       "      <th>Approved name</th>\n",
       "      <th>RefSeq accession</th>\n",
       "      <th>PosExonRefSeqAccession</th>\n",
       "      <th>PosExon_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>169853169</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>2584</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.26;FR=0.5;HP=1;HapScore=2;MGOF=1;MMLQ=38...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-133.02,0,-122.02:1:99:102:55</td>\n",
       "      <td>1/0:-130.1,0,-172.6:0:99:115:52</td>\n",
       "      <td>ABCB11</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>2:g.169853169T&gt;C</td>\n",
       "      <td>ENST00000263817:c.453T&gt;C</td>\n",
       "      <td>ENSP00000263817:p.151</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>PM2</td>\n",
       "      <td>.</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>.</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PM2(Low/absent allele frequency 9.479e-06&lt;=0.0...</td>\n",
       "      <td>0.452174</td>\n",
       "      <td>0.539216</td>\n",
       "      <td>55</td>\n",
       "      <td>102</td>\n",
       "      <td>52</td>\n",
       "      <td>115</td>\n",
       "      <td>GermlineInBoth</td>\n",
       "      <td>1.0025e-05</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>D021-592D9X</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>Germline_AI_neutral</td>\n",
       "      <td>2q</td>\n",
       "      <td>1</td>\n",
       "      <td>Germline</td>\n",
       "      <td>v0.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>592D9X</td>\n",
       "      <td>ABCB11</td>\n",
       "      <td>Cold (387-ACMG-MASTER)</td>\n",
       "      <td>P, G, H</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cholestasis, progressive familial intrahepatic...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HGNC:42</td>\n",
       "      <td>Approved</td>\n",
       "      <td>ABCB11</td>\n",
       "      <td>ATP binding cassette subfamily B member 11</td>\n",
       "      <td>NM_003742</td>\n",
       "      <td>24</td>\n",
       "      <td>insideDonor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>169826057</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>2625</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.1;FR=0.5;HP=1;HapScore=2;MGOF=1;MMLQ=37;...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-118.32,0,-133.02:1:99:78:37</td>\n",
       "      <td>1/0:-148.9,0,-170.7:0:99:103:49</td>\n",
       "      <td>ABCB11</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>2:g.169826057T&gt;G</td>\n",
       "      <td>ENST00000263817:c.1814T&gt;G</td>\n",
       "      <td>ENSP00000263817:p.Q605P</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>PM2,PP2,PP3</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>.</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PM2(Low/absent allele frequency 0.0&lt;=0.0005) -...</td>\n",
       "      <td>0.474359</td>\n",
       "      <td>0.475728</td>\n",
       "      <td>49</td>\n",
       "      <td>103</td>\n",
       "      <td>37</td>\n",
       "      <td>78</td>\n",
       "      <td>GermlineInBoth</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>H021-06H1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>Germline_AI_neutral</td>\n",
       "      <td>2q</td>\n",
       "      <td>1</td>\n",
       "      <td>Germline</td>\n",
       "      <td>v0.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>06H1</td>\n",
       "      <td>ABCB11</td>\n",
       "      <td>Cold (387-ACMG-MASTER)</td>\n",
       "      <td>P, G, H</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cholestasis, progressive familial intrahepatic...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HGNC:42</td>\n",
       "      <td>Approved</td>\n",
       "      <td>ABCB11</td>\n",
       "      <td>ATP binding cassette subfamily B member 11</td>\n",
       "      <td>NM_003742</td>\n",
       "      <td>4</td>\n",
       "      <td>insideAcceptorSite</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>169870004</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2157</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.18;FR=0.5;HP=2;HapScore=2;MGOF=5;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-110.9,0,-139.5:5:99:79:34</td>\n",
       "      <td>1/0:-109.48,0,-138.28:5:99:80:36</td>\n",
       "      <td>ABCB11</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>2:g.169870004G&gt;A</td>\n",
       "      <td>ENST00000263817:c.167G&gt;A</td>\n",
       "      <td>ENSP00000263817:p.S56L</td>\n",
       "      <td>0.000555</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>PP2,PP3</td>\n",
       "      <td>BP4</td>\n",
       "      <td>2</td>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>.</td>\n",
       "      <td>not specified;not provided</td>\n",
       "      <td>PP2(Missense variant in gene from PP2 gene lis...</td>\n",
       "      <td>0.430380</td>\n",
       "      <td>0.450000</td>\n",
       "      <td>36</td>\n",
       "      <td>80</td>\n",
       "      <td>34</td>\n",
       "      <td>79</td>\n",
       "      <td>GermlineInBoth</td>\n",
       "      <td>0.00054664</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>H021-13JVXT</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>Germline_AI_neutral</td>\n",
       "      <td>2q</td>\n",
       "      <td>1</td>\n",
       "      <td>Germline</td>\n",
       "      <td>v0.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>13JVXT</td>\n",
       "      <td>ABCB11</td>\n",
       "      <td>Cold (387-ACMG-MASTER)</td>\n",
       "      <td>P, G, H</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cholestasis, progressive familial intrahepatic...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HGNC:42</td>\n",
       "      <td>Approved</td>\n",
       "      <td>ABCB11</td>\n",
       "      <td>ATP binding cassette subfamily B member 11</td>\n",
       "      <td>NM_003742</td>\n",
       "      <td>16</td>\n",
       "      <td>insideAcceptorSite</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2</td>\n",
       "      <td>169791916</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2965</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.11;FR=0.5;HP=1;HapScore=2;MGOF=7;MMLQ=36...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>0/1:-252.04,0,-216.74:7:99:180:95</td>\n",
       "      <td>0/1:-247.73,0,-292.93:4:99:209:97</td>\n",
       "      <td>ABCB11</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>2:g.169791916C&gt;T</td>\n",
       "      <td>ENST00000263817:c.2834C&gt;T</td>\n",
       "      <td>ENSP00000263817:p.S945N</td>\n",
       "      <td>0.000125</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>PM2,PP2,PP3</td>\n",
       "      <td>BP4</td>\n",
       "      <td>4</td>\n",
       "      <td>-1</td>\n",
       "      <td>3</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>.</td>\n",
       "      <td>Progressive familial intrahepatic cholestasis;...</td>\n",
       "      <td>PM2(Low/absent allele frequency 0.0001248&lt;=0.0...</td>\n",
       "      <td>0.527778</td>\n",
       "      <td>0.464115</td>\n",
       "      <td>97</td>\n",
       "      <td>209</td>\n",
       "      <td>95</td>\n",
       "      <td>180</td>\n",
       "      <td>GermlineInBoth</td>\n",
       "      <td>0.00025483</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>H021-24ZM</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>Germline_AI_neutral</td>\n",
       "      <td>2q</td>\n",
       "      <td>1</td>\n",
       "      <td>Germline</td>\n",
       "      <td>v0.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24ZM</td>\n",
       "      <td>ABCB11</td>\n",
       "      <td>Cold (387-ACMG-MASTER)</td>\n",
       "      <td>P, G, H</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cholestasis, progressive familial intrahepatic...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HGNC:42</td>\n",
       "      <td>Approved</td>\n",
       "      <td>ABCB11</td>\n",
       "      <td>ATP binding cassette subfamily B member 11</td>\n",
       "      <td>NM_003742</td>\n",
       "      <td>19</td>\n",
       "      <td>insideAcceptorSite</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>169826566</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1304</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0;FR=0.5;HP=2;HapScore=1;MGOF=1;MMLQ=38;MQ...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-47.38,0,-35.08:1:99:28:15</td>\n",
       "      <td>1/0:-87.67,0,-22.27:1:99:39:30</td>\n",
       "      <td>ABCB11</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>2:g.169826566C&gt;A</td>\n",
       "      <td>ENST00000263817:c.1798C&gt;A</td>\n",
       "      <td>ENSP00000263817:p.V600L</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>PM2,PP2,PP3</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>.</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PM2(Low/absent allele frequency 0.0&lt;=0.0005) -...</td>\n",
       "      <td>0.535714</td>\n",
       "      <td>0.769231</td>\n",
       "      <td>30</td>\n",
       "      <td>39</td>\n",
       "      <td>15</td>\n",
       "      <td>28</td>\n",
       "      <td>GermlineInBoth</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>H021-35W4SQ</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>Germline_AI_neutral</td>\n",
       "      <td>2q</td>\n",
       "      <td>3</td>\n",
       "      <td>Germline</td>\n",
       "      <td>v0.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>35W4SQ</td>\n",
       "      <td>ABCB11</td>\n",
       "      <td>Cold (387-ACMG-MASTER)</td>\n",
       "      <td>P, G, H</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cholestasis, progressive familial intrahepatic...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HGNC:42</td>\n",
       "      <td>Approved</td>\n",
       "      <td>ABCB11</td>\n",
       "      <td>ATP binding cassette subfamily B member 11</td>\n",
       "      <td>NM_003742</td>\n",
       "      <td>11</td>\n",
       "      <td>insideDonor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60105</th>\n",
       "      <td>15</td>\n",
       "      <td>34634331</td>\n",
       "      <td>.</td>\n",
       "      <td>TATG</td>\n",
       "      <td>T</td>\n",
       "      <td>2970</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.22;FR=0.5;HP=1;HapScore=1;MGOF=4;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>0/1:-173.1,0,-286.7:4:99:98:38</td>\n",
       "      <td>0/1:-293.24,0,-300:4:99:179:63</td>\n",
       "      <td>NOP10</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>15:g.34634332_34634334delATG</td>\n",
       "      <td>ENST00000328848:c.delATG</td>\n",
       "      <td>ENSP00000332198:p.</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>PM2</td>\n",
       "      <td>.</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>.</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PM2(Low/absent allele frequency 1.193e-05&lt;=0.0...</td>\n",
       "      <td>0.387755</td>\n",
       "      <td>0.351955</td>\n",
       "      <td>63</td>\n",
       "      <td>179</td>\n",
       "      <td>38</td>\n",
       "      <td>98</td>\n",
       "      <td>GermlineInBoth</td>\n",
       "      <td>4.221e-06</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>H021-TZTQ81</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>Germline_AI_neutral</td>\n",
       "      <td>15q</td>\n",
       "      <td>2</td>\n",
       "      <td>Germline</td>\n",
       "      <td>v0.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TZTQ81</td>\n",
       "      <td>NOP10</td>\n",
       "      <td>Cold (387-ACMG-MASTER)</td>\n",
       "      <td>P, G, H</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Dyskeratosis congenita, autosomal recessive 1,...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HGNC:14378</td>\n",
       "      <td>Approved</td>\n",
       "      <td>NOP10</td>\n",
       "      <td>NOP10 ribonucleoprotein</td>\n",
       "      <td>NM_018648</td>\n",
       "      <td>22</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60106</th>\n",
       "      <td>8</td>\n",
       "      <td>55371619</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>2818</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.26;FR=0.5;HP=2;HapScore=1;MGOF=15;MMLQ=3...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>0/1:-141.07,0,-137.57:15:99:87:43</td>\n",
       "      <td>0/1:-145.37,0,-261.17:15:99:126:46</td>\n",
       "      <td>SOX17</td>\n",
       "      <td>splice_region_variant</td>\n",
       "      <td>8:g.55371619C&gt;G</td>\n",
       "      <td>ENST00000297316:c.309C&gt;G</td>\n",
       "      <td>ENSP00000297316:p.103</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>splice_region_variant</td>\n",
       "      <td>PM2</td>\n",
       "      <td>.</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>.</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PM2(Low/absent allele frequency 0.0&lt;=0.0005)</td>\n",
       "      <td>0.494253</td>\n",
       "      <td>0.365079</td>\n",
       "      <td>46</td>\n",
       "      <td>126</td>\n",
       "      <td>43</td>\n",
       "      <td>87</td>\n",
       "      <td>GermlineInBoth</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>H021-R7S1NJ</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>Germline_AI_down</td>\n",
       "      <td>8q</td>\n",
       "      <td>4</td>\n",
       "      <td>Germline</td>\n",
       "      <td>v0.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R7S1NJ</td>\n",
       "      <td>SOX17</td>\n",
       "      <td>Cold (387-ACMG-MASTER)</td>\n",
       "      <td>S</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Vesicoureteral reflux 3, 613674 (3), Autosomal...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HGNC:18122</td>\n",
       "      <td>Approved</td>\n",
       "      <td>SOX17</td>\n",
       "      <td>SRY-box transcription factor 17</td>\n",
       "      <td>NM_022454</td>\n",
       "      <td>1</td>\n",
       "      <td>insideAcceptorSite</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60107</th>\n",
       "      <td>8</td>\n",
       "      <td>55372558</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2195</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.22;FR=0.5;HP=1;HapScore=2;MGOF=15;MMLQ=3...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-86.75,0,-102.85:11:99:65:30</td>\n",
       "      <td>1/0:-137.38,0,-125.58:15:99:90:46</td>\n",
       "      <td>SOX17</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>8:g.55372558G&gt;A</td>\n",
       "      <td>ENST00000297316:c.G&gt;A</td>\n",
       "      <td>ENSP00000297316:p.</td>\n",
       "      <td>0.000164</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>PM2</td>\n",
       "      <td>.</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>.</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PM2(Low/absent allele frequency 0.0001637&lt;=0.0...</td>\n",
       "      <td>0.461538</td>\n",
       "      <td>0.511111</td>\n",
       "      <td>46</td>\n",
       "      <td>90</td>\n",
       "      <td>30</td>\n",
       "      <td>65</td>\n",
       "      <td>GermlineInBoth</td>\n",
       "      <td>0.00016931</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>H021-TQ37KJ</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>Germline_AI_neutral</td>\n",
       "      <td>8q</td>\n",
       "      <td>1</td>\n",
       "      <td>Germline</td>\n",
       "      <td>v0.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TQ37KJ</td>\n",
       "      <td>SOX17</td>\n",
       "      <td>Cold (387-ACMG-MASTER)</td>\n",
       "      <td>S</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Vesicoureteral reflux 3, 613674 (3), Autosomal...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HGNC:18122</td>\n",
       "      <td>Approved</td>\n",
       "      <td>SOX17</td>\n",
       "      <td>SRY-box transcription factor 17</td>\n",
       "      <td>NM_022454</td>\n",
       "      <td>6</td>\n",
       "      <td>3utr_stopcodon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60108</th>\n",
       "      <td>3</td>\n",
       "      <td>181431123</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2696</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.2;FR=0.5;HP=5;HapScore=2;MGOF=10;MMLQ=32...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-129.61,0,-128.21:9:99:76:35</td>\n",
       "      <td>1/0:-144.71,0,-118.91:10:99:85:45</td>\n",
       "      <td>SOX2</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>3:g.181431123G&gt;A</td>\n",
       "      <td>ENST00000325404:c.G&gt;A</td>\n",
       "      <td>ENSP00000323588:p.</td>\n",
       "      <td>0.000046</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>PM2</td>\n",
       "      <td>.</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>.</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PM2(Low/absent allele frequency 4.588e-05&lt;=0.0...</td>\n",
       "      <td>0.460526</td>\n",
       "      <td>0.529412</td>\n",
       "      <td>45</td>\n",
       "      <td>85</td>\n",
       "      <td>35</td>\n",
       "      <td>76</td>\n",
       "      <td>GermlineInBoth</td>\n",
       "      <td>4.6947e-05</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>H021-S4CLSR</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>Germline_AI_neutral</td>\n",
       "      <td>3q</td>\n",
       "      <td>1</td>\n",
       "      <td>Germline</td>\n",
       "      <td>v0.2.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S4CLSR</td>\n",
       "      <td>SOX2</td>\n",
       "      <td>Cold (387-ACMG-MASTER)</td>\n",
       "      <td>S</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Microphthalmia, syndromic 3, 206900 (3), Autos...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HGNC:11195</td>\n",
       "      <td>Approved</td>\n",
       "      <td>SOX2</td>\n",
       "      <td>SRY-box transcription factor 2</td>\n",
       "      <td>NM_003106</td>\n",
       "      <td>24</td>\n",
       "      <td>3utr_stopcodon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60109</th>\n",
       "      <td>2</td>\n",
       "      <td>220437370</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2965</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.23;FR=0.5;HP=1;HapScore=2;MGOF=3;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-150.81,0,-139.91:2:99:92:46</td>\n",
       "      <td>1/0:-151.93,0,-133.43:3:99:94:49</td>\n",
       "      <td>INHA</td>\n",
       "      <td>splice_region_variant</td>\n",
       "      <td>2:g.220437370G&gt;A</td>\n",
       "      <td>ENST00000243786:c.G&gt;A</td>\n",
       "      <td>ENSP00000243786:p.</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>splice_region_variant</td>\n",
       "      <td>PM2</td>\n",
       "      <td>.</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>.</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PM2(Low/absent allele frequency 1.312e-05&lt;=0.0...</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.521277</td>\n",
       "      <td>49</td>\n",
       "      <td>94</td>\n",
       "      <td>46</td>\n",
       "      <td>92</td>\n",
       "      <td>GermlineInBoth</td>\n",
       "      <td>1.3465e-05</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>H021-WS9DU9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>Germline_AI_neutral</td>\n",
       "      <td>2q</td>\n",
       "      <td>1</td>\n",
       "      <td>Germline</td>\n",
       "      <td>v0.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WS9DU9</td>\n",
       "      <td>INHA</td>\n",
       "      <td>Cold (387-ACMG-MASTER)</td>\n",
       "      <td>B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HGNC:6065</td>\n",
       "      <td>Approved</td>\n",
       "      <td>INHA</td>\n",
       "      <td>inhibin subunit alpha</td>\n",
       "      <td>NM_002191</td>\n",
       "      <td>6</td>\n",
       "      <td>outsideDonorSite</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18168 rows × 78 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #CHROM        POS ID   REF ALT  QUAL FILTER  \\\n",
       "0          2  169853169  .     T   C  2584   PASS   \n",
       "1          2  169826057  .     T   G  2625   PASS   \n",
       "2          2  169870004  .     G   A  2157   PASS   \n",
       "3          2  169791916  .     C   T  2965   PASS   \n",
       "4          2  169826566  .     C   A  1304   PASS   \n",
       "...      ...        ... ..   ...  ..   ...    ...   \n",
       "60105     15   34634331  .  TATG   T  2970   PASS   \n",
       "60106      8   55371619  .     C   G  2818   PASS   \n",
       "60107      8   55372558  .     G   A  2195   PASS   \n",
       "60108      3  181431123  .     G   A  2696   PASS   \n",
       "60109      2  220437370  .     G   A  2965   PASS   \n",
       "\n",
       "                                                    INFO              FORMAT  \\\n",
       "0      BRF=0.26;FR=0.5;HP=1;HapScore=2;MGOF=1;MMLQ=38...  GT:GL:GOF:GQ:NR:NV   \n",
       "1      BRF=0.1;FR=0.5;HP=1;HapScore=2;MGOF=1;MMLQ=37;...  GT:GL:GOF:GQ:NR:NV   \n",
       "2      BRF=0.18;FR=0.5;HP=2;HapScore=2;MGOF=5;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "3      BRF=0.11;FR=0.5;HP=1;HapScore=2;MGOF=7;MMLQ=36...  GT:GL:GOF:GQ:NR:NV   \n",
       "4      BRF=0;FR=0.5;HP=2;HapScore=1;MGOF=1;MMLQ=38;MQ...  GT:GL:GOF:GQ:NR:NV   \n",
       "...                                                  ...                 ...   \n",
       "60105  BRF=0.22;FR=0.5;HP=1;HapScore=1;MGOF=4;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "60106  BRF=0.26;FR=0.5;HP=2;HapScore=1;MGOF=15;MMLQ=3...  GT:GL:GOF:GQ:NR:NV   \n",
       "60107  BRF=0.22;FR=0.5;HP=1;HapScore=2;MGOF=15;MMLQ=3...  GT:GL:GOF:GQ:NR:NV   \n",
       "60108  BRF=0.2;FR=0.5;HP=5;HapScore=2;MGOF=10;MMLQ=32...  GT:GL:GOF:GQ:NR:NV   \n",
       "60109  BRF=0.23;FR=0.5;HP=1;HapScore=2;MGOF=3;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "\n",
       "                              Control_GT                            Tumor_GT  \\\n",
       "0      1/0:-133.02,0,-122.02:1:99:102:55     1/0:-130.1,0,-172.6:0:99:115:52   \n",
       "1       1/0:-118.32,0,-133.02:1:99:78:37     1/0:-148.9,0,-170.7:0:99:103:49   \n",
       "2         1/0:-110.9,0,-139.5:5:99:79:34    1/0:-109.48,0,-138.28:5:99:80:36   \n",
       "3      0/1:-252.04,0,-216.74:7:99:180:95   0/1:-247.73,0,-292.93:4:99:209:97   \n",
       "4         1/0:-47.38,0,-35.08:1:99:28:15      1/0:-87.67,0,-22.27:1:99:39:30   \n",
       "...                                  ...                                 ...   \n",
       "60105     0/1:-173.1,0,-286.7:4:99:98:38      0/1:-293.24,0,-300:4:99:179:63   \n",
       "60106  0/1:-141.07,0,-137.57:15:99:87:43  0/1:-145.37,0,-261.17:15:99:126:46   \n",
       "60107   1/0:-86.75,0,-102.85:11:99:65:30   1/0:-137.38,0,-125.58:15:99:90:46   \n",
       "60108   1/0:-129.61,0,-128.21:9:99:76:35   1/0:-144.71,0,-118.91:10:99:85:45   \n",
       "60109   1/0:-150.81,0,-139.91:2:99:92:46    1/0:-151.93,0,-133.43:3:99:94:49   \n",
       "\n",
       "      HUGO_Symbol Variant_Classification                         HGVSg  \\\n",
       "0          ABCB11     synonymous_variant              2:g.169853169T>C   \n",
       "1          ABCB11       missense_variant              2:g.169826057T>G   \n",
       "2          ABCB11       missense_variant              2:g.169870004G>A   \n",
       "3          ABCB11       missense_variant              2:g.169791916C>T   \n",
       "4          ABCB11       missense_variant              2:g.169826566C>A   \n",
       "...           ...                    ...                           ...   \n",
       "60105       NOP10         intron_variant  15:g.34634332_34634334delATG   \n",
       "60106       SOX17  splice_region_variant               8:g.55371619C>G   \n",
       "60107       SOX17    3_prime_UTR_variant               8:g.55372558G>A   \n",
       "60108        SOX2    3_prime_UTR_variant              3:g.181431123G>A   \n",
       "60109        INHA  splice_region_variant              2:g.220437370G>A   \n",
       "\n",
       "                           HGVSc                    HGVSp  Allele_Frequency  \\\n",
       "0       ENST00000263817:c.453T>C    ENSP00000263817:p.151          0.000009   \n",
       "1      ENST00000263817:c.1814T>G  ENSP00000263817:p.Q605P          0.000000   \n",
       "2       ENST00000263817:c.167G>A   ENSP00000263817:p.S56L          0.000555   \n",
       "3      ENST00000263817:c.2834C>T  ENSP00000263817:p.S945N          0.000125   \n",
       "4      ENST00000263817:c.1798C>A  ENSP00000263817:p.V600L          0.000000   \n",
       "...                          ...                      ...               ...   \n",
       "60105   ENST00000328848:c.delATG       ENSP00000332198:p.          0.000012   \n",
       "60106   ENST00000297316:c.309C>G    ENSP00000297316:p.103          0.000000   \n",
       "60107      ENST00000297316:c.G>A       ENSP00000297316:p.          0.000164   \n",
       "60108      ENST00000325404:c.G>A       ENSP00000323588:p.          0.000046   \n",
       "60109      ENST00000243786:c.G>A       ENSP00000243786:p.          0.000013   \n",
       "\n",
       "      VEP_Most_Severe_Consequence Positive_Evidence Negative_Evidence  \\\n",
       "0              synonymous_variant               PM2                 .   \n",
       "1                missense_variant       PM2,PP2,PP3                 .   \n",
       "2                missense_variant           PP2,PP3               BP4   \n",
       "3                missense_variant       PM2,PP2,PP3               BP4   \n",
       "4                missense_variant       PM2,PP2,PP3                 .   \n",
       "...                           ...               ...               ...   \n",
       "60105              intron_variant               PM2                 .   \n",
       "60106       splice_region_variant               PM2                 .   \n",
       "60107         3_prime_UTR_variant               PM2                 .   \n",
       "60108         3_prime_UTR_variant               PM2                 .   \n",
       "60109       splice_region_variant               PM2                 .   \n",
       "\n",
       "       Positive_CharGer_Score  Negative_CharGer_Score  CharGer_Score  \\\n",
       "0                           2                       0              2   \n",
       "1                           4                       0              4   \n",
       "2                           2                      -1              1   \n",
       "3                           4                      -1              3   \n",
       "4                           4                       0              4   \n",
       "...                       ...                     ...            ...   \n",
       "60105                       2                       0              2   \n",
       "60106                       2                       0              2   \n",
       "60107                       2                       0              2   \n",
       "60108                       2                       0              2   \n",
       "60109                       2                       0              2   \n",
       "\n",
       "        ClinVar_Pathogenicity     ACMG_Classification  CharGer_Classification  \\\n",
       "0                           .  Uncertain Significance  Uncertain Significance   \n",
       "1                           .  Uncertain Significance  Uncertain Significance   \n",
       "2      Uncertain Significance  Uncertain Significance  Uncertain Significance   \n",
       "3      Uncertain Significance  Uncertain Significance  Uncertain Significance   \n",
       "4                           .  Uncertain Significance  Uncertain Significance   \n",
       "...                       ...                     ...                     ...   \n",
       "60105                       .  Uncertain Significance  Uncertain Significance   \n",
       "60106                       .  Uncertain Significance  Uncertain Significance   \n",
       "60107                       .  Uncertain Significance  Uncertain Significance   \n",
       "60108                       .  Uncertain Significance  Uncertain Significance   \n",
       "60109                       .  Uncertain Significance  Uncertain Significance   \n",
       "\n",
       "      PubMed_Link                                     ClinVar_Traits  \\\n",
       "0               .                                                  .   \n",
       "1               .                                                  .   \n",
       "2               .                         not specified;not provided   \n",
       "3               .  Progressive familial intrahepatic cholestasis;...   \n",
       "4               .                                                  .   \n",
       "...           ...                                                ...   \n",
       "60105           .                                                  .   \n",
       "60106           .                                                  .   \n",
       "60107           .                                                  .   \n",
       "60108           .                                                  .   \n",
       "60109           .                                                  .   \n",
       "\n",
       "                                         CharGer_Summary  Control_VAF  \\\n",
       "0      PM2(Low/absent allele frequency 9.479e-06<=0.0...     0.452174   \n",
       "1      PM2(Low/absent allele frequency 0.0<=0.0005) -...     0.474359   \n",
       "2      PP2(Missense variant in gene from PP2 gene lis...     0.430380   \n",
       "3      PM2(Low/absent allele frequency 0.0001248<=0.0...     0.527778   \n",
       "4      PM2(Low/absent allele frequency 0.0<=0.0005) -...     0.535714   \n",
       "...                                                  ...          ...   \n",
       "60105  PM2(Low/absent allele frequency 1.193e-05<=0.0...     0.387755   \n",
       "60106       PM2(Low/absent allele frequency 0.0<=0.0005)     0.494253   \n",
       "60107  PM2(Low/absent allele frequency 0.0001637<=0.0...     0.461538   \n",
       "60108  PM2(Low/absent allele frequency 4.588e-05<=0.0...     0.460526   \n",
       "60109  PM2(Low/absent allele frequency 1.312e-05<=0.0...     0.500000   \n",
       "\n",
       "       Tumor_VAF  Tumor_dpALT  Tumor_dp  Control_dpALT  Control_dp  \\\n",
       "0       0.539216           55       102             52         115   \n",
       "1       0.475728           49       103             37          78   \n",
       "2       0.450000           36        80             34          79   \n",
       "3       0.464115           97       209             95         180   \n",
       "4       0.769231           30        39             15          28   \n",
       "...          ...          ...       ...            ...         ...   \n",
       "60105   0.351955           63       179             38          98   \n",
       "60106   0.365079           46       126             43          87   \n",
       "60107   0.511111           46        90             30          65   \n",
       "60108   0.529412           45        85             35          76   \n",
       "60109   0.521277           49        94             46          92   \n",
       "\n",
       "      GT_Classification max_gnomAD_AF exome_control_AF local_control_AF  \\\n",
       "0        GermlineInBoth    1.0025e-05                .                .   \n",
       "1        GermlineInBoth             .                .                .   \n",
       "2        GermlineInBoth    0.00054664                .                .   \n",
       "3        GermlineInBoth    0.00025483                .                .   \n",
       "4        GermlineInBoth             .                .                .   \n",
       "...                 ...           ...              ...              ...   \n",
       "60105    GermlineInBoth     4.221e-06                .                .   \n",
       "60106    GermlineInBoth             .                .                .   \n",
       "60107    GermlineInBoth    0.00016931                .                .   \n",
       "60108    GermlineInBoth    4.6947e-05                .                .   \n",
       "60109    GermlineInBoth    1.3465e-05                .                .   \n",
       "\n",
       "               PID  overlap  canopyCluster            TiN_Class CHR_ARM_INFO  \\\n",
       "0      D021-592D9X        1              1  Germline_AI_neutral           2q   \n",
       "1        H021-06H1        1              1  Germline_AI_neutral           2q   \n",
       "2      H021-13JVXT        1              1  Germline_AI_neutral           2q   \n",
       "3        H021-24ZM        1              1  Germline_AI_neutral           2q   \n",
       "4      H021-35W4SQ        1              1  Germline_AI_neutral           2q   \n",
       "...            ...      ...            ...                  ...          ...   \n",
       "60105  H021-TZTQ81        1              3  Germline_AI_neutral          15q   \n",
       "60106  H021-R7S1NJ        1              3     Germline_AI_down           8q   \n",
       "60107  H021-TQ37KJ        1              1  Germline_AI_neutral           8q   \n",
       "60108  H021-S4CLSR        1              3  Germline_AI_neutral           3q   \n",
       "60109  H021-WS9DU9        1              1  Germline_AI_neutral           2q   \n",
       "\n",
       "      AI_cluster thresholdCluster GIT_VERSION ALL_FUNCTIONAL_CONSENSUS  \\\n",
       "0              1         Germline      v0.2.2                      NaN   \n",
       "1              1         Germline      v0.2.2                      NaN   \n",
       "2              1         Germline      v0.2.2                      NaN   \n",
       "3              1         Germline      v0.2.2                      NaN   \n",
       "4              3         Germline      v0.2.2                      NaN   \n",
       "...          ...              ...         ...                      ...   \n",
       "60105          2         Germline      v0.2.2                      NaN   \n",
       "60106          4         Germline      v0.2.2                      NaN   \n",
       "60107          1         Germline      v0.2.2                      NaN   \n",
       "60108          1         Germline      v0.2.4                      NaN   \n",
       "60109          1         Germline      v0.2.2                      NaN   \n",
       "\n",
       "      RNA_EXPRESSION RNA_INFO RNA_VAF PID_trans Approved symbol (HGNC)  \\\n",
       "0                NaN      NaN     NaN    592D9X                 ABCB11   \n",
       "1                NaN      NaN     NaN      06H1                 ABCB11   \n",
       "2                NaN      NaN     NaN    13JVXT                 ABCB11   \n",
       "3                NaN      NaN     NaN      24ZM                 ABCB11   \n",
       "4                NaN      NaN     NaN    35W4SQ                 ABCB11   \n",
       "...              ...      ...     ...       ...                    ...   \n",
       "60105            NaN      NaN     NaN    TZTQ81                  NOP10   \n",
       "60106            NaN      NaN     NaN    R7S1NJ                  SOX17   \n",
       "60107            NaN      NaN     NaN    TQ37KJ                  SOX17   \n",
       "60108            NaN      NaN     NaN    S4CLSR                   SOX2   \n",
       "60109            NaN      NaN     NaN    WS9DU9                   INHA   \n",
       "\n",
       "      Splice Project Gene Priority  \\\n",
       "0           Cold (387-ACMG-MASTER)   \n",
       "1           Cold (387-ACMG-MASTER)   \n",
       "2           Cold (387-ACMG-MASTER)   \n",
       "3           Cold (387-ACMG-MASTER)   \n",
       "4           Cold (387-ACMG-MASTER)   \n",
       "...                            ...   \n",
       "60105       Cold (387-ACMG-MASTER)   \n",
       "60106       Cold (387-ACMG-MASTER)   \n",
       "60107       Cold (387-ACMG-MASTER)   \n",
       "60108       Cold (387-ACMG-MASTER)   \n",
       "60109       Cold (387-ACMG-MASTER)   \n",
       "\n",
       "      Origin (Bertelsen 2019 (PMID: 31263571), Schrader 2016 (PMC5477989), Huang 2018 (PMID: 29625052), Groebner 2018 (PMID: 29489754), Priestley 2019 (PMID: 31645765))  \\\n",
       "0                                                P, G, H                                                                                                                   \n",
       "1                                                P, G, H                                                                                                                   \n",
       "2                                                P, G, H                                                                                                                   \n",
       "3                                                P, G, H                                                                                                                   \n",
       "4                                                P, G, H                                                                                                                   \n",
       "...                                                  ...                                                                                                                   \n",
       "60105                                            P, G, H                                                                                                                   \n",
       "60106                                                  S                                                                                                                   \n",
       "60107                                                  S                                                                                                                   \n",
       "60108                                                  S                                                                                                                   \n",
       "60109                                                  B                                                                                                                   \n",
       "\n",
       "       NCBI gene ID (HGNC) RefSeq (HGNC)  Total Exon Count (hg19)  \\\n",
       "0                      NaN           NaN                      NaN   \n",
       "1                      NaN           NaN                      NaN   \n",
       "2                      NaN           NaN                      NaN   \n",
       "3                      NaN           NaN                      NaN   \n",
       "4                      NaN           NaN                      NaN   \n",
       "...                    ...           ...                      ...   \n",
       "60105                  NaN           NaN                      NaN   \n",
       "60106                  NaN           NaN                      NaN   \n",
       "60107                  NaN           NaN                      NaN   \n",
       "60108                  NaN           NaN                      NaN   \n",
       "60109                  NaN           NaN                      NaN   \n",
       "\n",
       "       Coding Exon Count (hg19)  Protein size [aa] (hg19) Gene Basket (NCT)  \\\n",
       "0                           NaN                       NaN               NaN   \n",
       "1                           NaN                       NaN               NaN   \n",
       "2                           NaN                       NaN               NaN   \n",
       "3                           NaN                       NaN               NaN   \n",
       "4                           NaN                       NaN               NaN   \n",
       "...                         ...                       ...               ...   \n",
       "60105                       NaN                       NaN               NaN   \n",
       "60106                       NaN                       NaN               NaN   \n",
       "60107                       NaN                       NaN               NaN   \n",
       "60108                       NaN                       NaN               NaN   \n",
       "60109                       NaN                       NaN               NaN   \n",
       "\n",
       "      Gene classification (most likely mechanism) Gene group (KGE)  \\\n",
       "0                                             NaN              NaN   \n",
       "1                                             NaN              NaN   \n",
       "2                                             NaN              NaN   \n",
       "3                                             NaN              NaN   \n",
       "4                                             NaN              NaN   \n",
       "...                                           ...              ...   \n",
       "60105                                         NaN              NaN   \n",
       "60106                                         NaN              NaN   \n",
       "60107                                         NaN              NaN   \n",
       "60108                                         NaN              NaN   \n",
       "60109                                         NaN              NaN   \n",
       "\n",
       "      inheritance cancer predisposition  \\\n",
       "0                                   NaN   \n",
       "1                                   NaN   \n",
       "2                                   NaN   \n",
       "3                                   NaN   \n",
       "4                                   NaN   \n",
       "...                                 ...   \n",
       "60105                               NaN   \n",
       "60106                               NaN   \n",
       "60107                               NaN   \n",
       "60108                               NaN   \n",
       "60109                               NaN   \n",
       "\n",
       "                                       OMIM-G Phenotypes  \\\n",
       "0      Cholestasis, progressive familial intrahepatic...   \n",
       "1      Cholestasis, progressive familial intrahepatic...   \n",
       "2      Cholestasis, progressive familial intrahepatic...   \n",
       "3      Cholestasis, progressive familial intrahepatic...   \n",
       "4      Cholestasis, progressive familial intrahepatic...   \n",
       "...                                                  ...   \n",
       "60105  Dyskeratosis congenita, autosomal recessive 1,...   \n",
       "60106  Vesicoureteral reflux 3, 613674 (3), Autosomal...   \n",
       "60107  Vesicoureteral reflux 3, 613674 (3), Autosomal...   \n",
       "60108  Microphthalmia, syndromic 3, 206900 (3), Autos...   \n",
       "60109                                                  0   \n",
       "\n",
       "       non-cancer related phenotype final proposal technical details gene  \\\n",
       "0                               NaN            NaN                    NaN   \n",
       "1                               NaN            NaN                    NaN   \n",
       "2                               NaN            NaN                    NaN   \n",
       "3                               NaN            NaN                    NaN   \n",
       "4                               NaN            NaN                    NaN   \n",
       "...                             ...            ...                    ...   \n",
       "60105                           NaN            NaN                    NaN   \n",
       "60106                           NaN            NaN                    NaN   \n",
       "60107                           NaN            NaN                    NaN   \n",
       "60108                           NaN            NaN                    NaN   \n",
       "60109                           NaN            NaN                    NaN   \n",
       "\n",
       "      Huang_KGE_Cancer syndrome(s) Huang_KGE_Major associated tumor types  \\\n",
       "0                              NaN                                    NaN   \n",
       "1                              NaN                                    NaN   \n",
       "2                              NaN                                    NaN   \n",
       "3                              NaN                                    NaN   \n",
       "4                              NaN                                    NaN   \n",
       "...                            ...                                    ...   \n",
       "60105                          NaN                                    NaN   \n",
       "60106                          NaN                                    NaN   \n",
       "60107                          NaN                                    NaN   \n",
       "60108                          NaN                                    NaN   \n",
       "60109                          NaN                                    NaN   \n",
       "\n",
       "      Literature     HGNC ID    Status Approved symbol  \\\n",
       "0            NaN     HGNC:42  Approved          ABCB11   \n",
       "1            NaN     HGNC:42  Approved          ABCB11   \n",
       "2            NaN     HGNC:42  Approved          ABCB11   \n",
       "3            NaN     HGNC:42  Approved          ABCB11   \n",
       "4            NaN     HGNC:42  Approved          ABCB11   \n",
       "...          ...         ...       ...             ...   \n",
       "60105        NaN  HGNC:14378  Approved           NOP10   \n",
       "60106        NaN  HGNC:18122  Approved           SOX17   \n",
       "60107        NaN  HGNC:18122  Approved           SOX17   \n",
       "60108        NaN  HGNC:11195  Approved            SOX2   \n",
       "60109        NaN   HGNC:6065  Approved            INHA   \n",
       "\n",
       "                                    Approved name RefSeq accession  \\\n",
       "0      ATP binding cassette subfamily B member 11        NM_003742   \n",
       "1      ATP binding cassette subfamily B member 11        NM_003742   \n",
       "2      ATP binding cassette subfamily B member 11        NM_003742   \n",
       "3      ATP binding cassette subfamily B member 11        NM_003742   \n",
       "4      ATP binding cassette subfamily B member 11        NM_003742   \n",
       "...                                           ...              ...   \n",
       "60105                     NOP10 ribonucleoprotein        NM_018648   \n",
       "60106             SRY-box transcription factor 17        NM_022454   \n",
       "60107             SRY-box transcription factor 17        NM_022454   \n",
       "60108              SRY-box transcription factor 2        NM_003106   \n",
       "60109                       inhibin subunit alpha        NM_002191   \n",
       "\n",
       "      PosExonRefSeqAccession         PosExon_type  \n",
       "0                         24          insideDonor  \n",
       "1                          4   insideAcceptorSite  \n",
       "2                         16   insideAcceptorSite  \n",
       "3                         19   insideAcceptorSite  \n",
       "4                         11          insideDonor  \n",
       "...                      ...                  ...  \n",
       "60105                     22  outsideAcceptorSite  \n",
       "60106                      1   insideAcceptorSite  \n",
       "60107                      6       3utr_stopcodon  \n",
       "60108                     24       3utr_stopcodon  \n",
       "60109                      6     outsideDonorSite  \n",
       "\n",
       "[18168 rows x 78 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
