{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "843b3e60",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "56c89982",
   "metadata": {},
   "source": [
    "## Functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "69581f91",
   "metadata": {},
   "outputs": [],
   "source": [
    "# gus = genes of unknown significance\n",
    "gus = ['AKT2', 'AKT3', 'ATR', 'ATRX', 'CDKN1A', 'CDKN2B', 'CDKN2C', 'CDKN2D', 'CEBPA', 'CEP57', 'DAXX', 'EGFR', \n",
    "       'EGLN1', 'EGLN2', 'EPAS1', 'EPHB2', 'ABRAXAS1', 'FANCM', 'GPRC5A', 'HNF1A', 'HORMAD1', 'HORMAD2', 'JAK2', \n",
    "       'KIF1B', 'MAP3K1', 'MAP3K6', 'MDH2', 'MLH3', 'MMS19', 'MN1', 'MRE11', 'PALLD', 'PIK3C2G', 'PIK3R2', 'PMS1', \n",
    "       'PRF1', 'PTCH2', 'RAD50', 'RAD51', 'RECQL', 'RINT1']\n",
    "\n",
    "# make df .xlsx conform for xlsx_df function with more columns\n",
    "def conf_xlsx(df):\n",
    "    dfn = df.copy()\n",
    "    for i in ['consensus_sequence_IGV_region_left', 'consensus_sequence_IGV_sequence_left', \n",
    "              'consensus_sequence_IGV_region_right', 'consensus_sequence_IGV_sequence_right', \n",
    "              'INFO_IGV', 'too_low_coverage', 'splice_change']:\n",
    "        dfn[i] = np.nan\n",
    "    dfn['shortcut'] = dfn['#CHROM'].astype(str)+'_'+dfn['POS'].astype(str)+'_'+dfn['REF']+'_'+dfn['ALT']\n",
    "    return dfn\n",
    "\n",
    "# function to make excel tables with one variant per row\n",
    "cols_cpra = ['#CHROM', 'POS', 'REF', 'ALT']\n",
    "columns_df_prior = ['#CHROM', 'POS', 'REF', 'ALT', 'MaxEntScan_alt', 'MaxEntScan_diff', 'MaxEntScan_ref', \n",
    "                    'CADD_PHRED', 'SpliceAI_pred_DP_AG', 'SpliceAI_pred_DP_AL', 'SpliceAI_pred_DP_DG', \n",
    "                    'SpliceAI_pred_DP_DL', 'SpliceAI_pred_DS_AG', 'SpliceAI_pred_DS_AL', 'SpliceAI_pred_DS_DG', \n",
    "                    'SpliceAI_pred_DS_DL', 'SpliceAI_pred_SYMBOL', 'ada_score', 'rf_score',\n",
    "                    'PosExonRefSeqAccession', 'PosExon_type', 'ClinVar_Pathogenicity', \n",
    "                    'consensus_sequence_IGV_region_left', 'consensus_sequence_IGV_sequence_left', \n",
    "                    'consensus_sequence_IGV_region_right', 'consensus_sequence_IGV_sequence_right', 'hgsv_new',\n",
    "                    'INFO_IGV', 'too_low_coverage', 'splice_change', 'shortcut', 'ClinVar_Traits', \n",
    "                    'Splice Project Gene Priority', 'PID']\n",
    "columns_df_xlsx = columns_df_prior[:-1]\n",
    "columns_df_xlsx.extend(['PIDs', 'ct_PIDs'])\n",
    "\n",
    "def xlsx_df(prio_df, name):\n",
    "    lists = []\n",
    "    for p,df in prio_df.groupby(by=['#CHROM', 'POS', 'REF', 'ALT']):\n",
    "        df = df.reset_index(drop=True)\n",
    "        list_df = [df.iloc[0][i] for i in columns_df_prior[:-1]]\n",
    "        list_df.extend([list(df['PID']), len(df['PID'])])\n",
    "        lists.append(list_df)\n",
    "    new_df = pd.DataFrame(lists, columns=columns_df_xlsx)\n",
    "    new_df['#CHROM'] = pd.Categorical(new_df['#CHROM'],\n",
    "                                      categories=['1','2','3','4','5','6','7','8','9','10','11','12', '13', '14', \n",
    "                                                  '15', '16', '17', '18', '19', '20', '21', '22', 'X', 'Y'],\n",
    "                                      ordered=True)\n",
    "    new_df.sort_values(by=['ct_PIDs', '#CHROM', 'POS', 'REF', 'ALT'], inplace=True, ignore_index=True, \n",
    "                       ascending=[False, True, True, True, True])\n",
    "    new_df.drop(axis='columns', columns='ct_PIDs', inplace=True)\n",
    "    new_df.to_excel('01_xlsx_files/%s.xlsx'%name, index=False)\n",
    "\n",
    "def xlsx_vars(prio_vars_df, name):\n",
    "    merged = pd.merge(all_vars[columns_df_prior], prio_vars_df, how='inner')\n",
    "    lists = []\n",
    "    for p,df in merged.groupby(by=['#CHROM', 'POS', 'REF', 'ALT']):\n",
    "        df = df.reset_index(drop=True)\n",
    "        list_df = [df.iloc[0][i] for i in columns_df_prior[:-1]]\n",
    "        list_df.extend([list(df['PID']), len(df['PID'])])\n",
    "        lists.append(list_df)\n",
    "    new_df = pd.DataFrame(lists, columns=columns_df_xlsx)\n",
    "    new_df['#CHROM'] = pd.Categorical(new_df['#CHROM'],\n",
    "                                      categories=['1','2','3','4','5','6','7','8','9','10','11','12', '13', '14', \n",
    "                                                  '15', '16', '17', '18', '19', '20', '21', '22', 'X', 'Y'],\n",
    "                                      ordered=True)\n",
    "    new_df.sort_values(by=['ct_PIDs', '#CHROM', 'POS', 'REF', 'ALT'], inplace=True, ignore_index=True, \n",
    "                       ascending=[False, True, True, True, True])\n",
    "    new_df.drop(axis='columns', columns='ct_PIDs', inplace=True)\n",
    "    new_df.to_excel('01_xlsx_files/%s.xlsx'%name, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "10286278",
   "metadata": {},
   "outputs": [],
   "source": [
    "# function to search for specific name in column\n",
    "def search_col(df, inp):\n",
    "    return [i for i in df.columns if inp in i.lower()]\n",
    "\n",
    "# function to make spliceAI prediction scores into floats\n",
    "def make_float(df):\n",
    "    for i in ['AG', 'AL', 'DG', 'DL']:\n",
    "        df[('SpliceAI_pred_DS_'+i)]=df[('SpliceAI_pred_DS_'+i)].astype(float)\n",
    "    for i in ['ada_score', 'rf_score', 'MaxEntScan_diff', 'MaxEntScan_alt', 'MaxEntScan_ref']:\n",
    "        df[i] = df[i].astype(float)\n",
    "    return df\n",
    "\n",
    "# function to merge in IGV seen variant .xlsx files with all_vars with all infos\n",
    "def merge_xlsx(file_name):\n",
    "    df = pd.read_excel('01_xlsx_files/'+file_name)\n",
    "    merged = pd.merge(df[['#CHROM', 'POS', 'REF', 'ALT', 'consensus_sequence_IGV_region_left',\n",
    "                          'consensus_sequence_IGV_sequence_left', 'consensus_sequence_IGV_region_right',\n",
    "                          'consensus_sequence_IGV_sequence_right', 'exon_skipping', 'intron_retention', \n",
    "                          \"3'acceptor_loss\", \"3'acceptor_gain\", \"5'donor_loss\", \"5'donor_gain\", 'frameshift',\n",
    "                          'premat_stop_cod', 'hgsv_new', 'INFO_IGV', 'too_low_coverage',\n",
    "                          'splice_change', 'var_in_blood_perc', 'var_in_tumor_perc', 'counts_coverage', \n",
    "                          'perc_coverage', 'pos_last', 'pos_next', 'transc_pos_last', 'transc_pos_next', 'kind', \n",
    "                          'info_cDNA', 'more_consequences']], all_vars, on=cols_cpra, \n",
    "                      how='left').reset_index(drop=True)\n",
    "    merged['shortcut'] = (merged['#CHROM'].astype(str)+'_'+merged['POS'].astype(str)+\n",
    "                          '_'+merged['REF']+'_'+merged['ALT'])\n",
    "    merged['ClinVar_Pathogenicity'] = merged['ClinVar_Pathogenicity'].replace(\n",
    "        {'uncertain_significance':'Uncertain Significance', 'Uncertain significance':'Uncertain Significance'})\n",
    "    merged['PosExon_type'] = merged['PosExon_type'].replace({'insideDonor':'insideDonorSite', \n",
    "                                                             'outsideDonor':'outsideDonorSite'})\n",
    "    merged[['exon_skipping', 'intron_retention', \"3'acceptor_loss\", \"3'acceptor_gain\", \"5'donor_loss\", \n",
    "            \"5'donor_gain\"]] = merged[['exon_skipping', 'intron_retention', \"3'acceptor_loss\", \"3'acceptor_gain\", \n",
    "                                       \"5'donor_loss\", \"5'donor_gain\"]].replace({'x':True, np.nan:False})\n",
    "    merged['strand'] = merged['info_cDNA'].str.split(',', expand=True)[0]\n",
    "    merged = merged.drop(columns='info_cDNA')\n",
    "    merged['#CHROM'] = pd.Categorical(merged['#CHROM'],\n",
    "                                      categories=['1','2','3','4','5','6','7','8','9','10','11','12', '13', '14', \n",
    "                                                  '15', '16', '17', '18', '19', '20', '21', '22', 'X', 'Y'],\n",
    "                                      ordered=True)\n",
    "    merged.sort_values(by=['#CHROM', 'POS', 'REF', 'ALT'], inplace=True, ignore_index=True, \n",
    "                       ascending=True)\n",
    "    merged = make_float(merged)\n",
    "    return merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "49789acd",
   "metadata": {},
   "outputs": [],
   "source": [
    "# categorize variants into groups\n",
    "# spliceAI score > 0.9\n",
    "def splai_o09(df, coverage, splice_change):\n",
    "    return len(df[((df['SpliceAI_pred_DS_AG']>0.9)|\n",
    "                   (df['SpliceAI_pred_DS_AL']>0.9)|\n",
    "                   (df['SpliceAI_pred_DS_DG']>0.9)|\n",
    "                   (df['SpliceAI_pred_DS_DL']>0.9))&\n",
    "                  (df['too_low_coverage']==coverage)&\n",
    "                  (df['splice_change']==splice_change)].groupby(cols_cpra))\n",
    "\n",
    "# ada/rf score > 0.6\n",
    "def ada_rf_o06(df, coverage, splice_change):\n",
    "    return len(df[(df['ada_score']>0.9)&\n",
    "                  (df['rf_score']>0.9)&\n",
    "                  (df['too_low_coverage']==coverage)&\n",
    "                  (df['splice_change']==splice_change)].groupby(cols_cpra))\n",
    "\n",
    "# MaxEntScan high disruption of native splice site\n",
    "def MES_disr(df, coverage, splice_change):\n",
    "    return len(df[(df['MaxEntScan_diff']>=0)&\n",
    "                  (df['MaxEntScan_alt']<6.2)&\n",
    "                  (df['too_low_coverage']==coverage)&\n",
    "                  (df['splice_change']==splice_change)].groupby(cols_cpra))\n",
    "\n",
    "# MaxEntScan high possibility of creating new splice site\n",
    "def MES_new(df, coverage, splice_change):\n",
    "    return len(df[(df['MaxEntScan_diff']<0)&\n",
    "                  (df['MaxEntScan_alt']>8.5)&\n",
    "                  (df['too_low_coverage']==coverage)&\n",
    "                  (df['splice_change']==splice_change)].groupby(cols_cpra))\n",
    "\n",
    "# position relative to exon/intron boundary less than 3 and in intron\n",
    "def intron_pos(df, coverage, splice_change):\n",
    "    return len(df[(df['PosExonRefSeqAccession']<3)&\n",
    "                  (df['PosExon_type'].isin(['outsideAcceptorSite', 'outsideDonorSite', 'outsideDonor']))&\n",
    "                  (df['too_low_coverage']==coverage)&\n",
    "                  (df['splice_change']==splice_change)].groupby(cols_cpra))\n",
    "\n",
    "# variant categorization per hot/warm/cold genes\n",
    "def gene_prio(df):\n",
    "    hot_lst, warm_gus_lst, warm_lst, cold_lst = ({} for i in range(4))\n",
    "    for i,lst,genes in zip(['Hot (ACMG / MASTER)', 'Warm (MASTER-ACMG)', 'Warm (MASTER-ACMG)', \n",
    "                            'Cold (387-ACMG-MASTER)'], \n",
    "                           [[''], gus, list(set(df['HUGO_Symbol'][df['Splice Project Gene Priority']==\n",
    "                                                                  'Warm (MASTER-ACMG)'])^set(gus)), ['']], \n",
    "                           [hot_lst, warm_gus_lst, warm_lst, cold_lst]):\n",
    "        for coverage, splice_change, cat in zip(['no', 'no', 'no', 'yes', 'yes', 'yes'], \n",
    "                                            ['yes', 'maybe', 'no', 'yes', 'maybe', 'no'], \n",
    "                                            ['sufficient, yes', 'sufficient, maybe', 'sufficient, no', \n",
    "                                             'not sufficient, yes', 'not sufficient, maybe', 'not sufficient, no']):\n",
    "            genes[cat] = len(df[(df['Splice Project Gene Priority']==i)&\n",
    "                                (~df['HUGO_Symbol'].isin(lst))&\n",
    "                                (df['too_low_coverage']==coverage)&\n",
    "                                (df['splice_change']==splice_change)].groupby(cols_cpra))\n",
    "    newdf = pd.DataFrame([hot_lst, warm_gus_lst, warm_lst, cold_lst], \n",
    "                        index=['hot genes', 'warm genes - GUS', 'GUS', 'cold genes'])\n",
    "    newdf = newdf.reset_index(drop=False)\n",
    "    newdf = newdf.rename(columns={'index':'category'})\n",
    "    return newdf\n",
    "\n",
    "# variant categorization for specific scores\n",
    "def categ_vars(df):\n",
    "    splAI_lst, ada_rf_lst, MES_disr_lst, MES_new_lst, intron_pos_lst = ({} for i in range(5))\n",
    "    for coverage, splice_change, cat in zip(['no', 'no', 'no', 'yes', 'yes', 'yes'], \n",
    "                                            ['yes', 'maybe', 'no', 'yes', 'maybe', 'no'], \n",
    "                                            ['sufficient, yes', 'sufficient, maybe', 'sufficient, no', \n",
    "                                             'not sufficient, yes', 'not sufficient, maybe', 'not sufficient, no']):\n",
    "        splAI_lst[cat] = splai_o09(df, coverage, splice_change)\n",
    "        ada_rf_lst[cat] = ada_rf_o06(df, coverage, splice_change)\n",
    "        MES_disr_lst[cat] = MES_disr(df, coverage, splice_change)\n",
    "        MES_new_lst[cat] = MES_new(df, coverage, splice_change)\n",
    "        intron_pos_lst[cat] = intron_pos(df, coverage, splice_change)\n",
    "        \n",
    "    newdf = pd.DataFrame([splAI_lst, ada_rf_lst, MES_disr_lst, MES_new_lst, intron_pos_lst], \n",
    "                        index=['spliceAI > 0.9', 'ada, rf score > 0.9', 'MES high disruption', 'MES high new ss', \n",
    "                               'pos. rel. to ss < 3, intron'])\n",
    "    newdf = newdf.reset_index(drop=False)\n",
    "    newdf = newdf.rename(columns={'index':'category'})\n",
    "    return newdf\n",
    "\n",
    "# merge variant categorization for gene priority and specific scores, create percentages/sums\n",
    "def perc_cat(df):\n",
    "    dfs = gene_prio(df).append(categ_vars(df), ignore_index=True)\n",
    "    dfs['sum'] = (dfs['sufficient, yes']+dfs['sufficient, maybe']+dfs['sufficient, no']+\n",
    "                  dfs['not sufficient, yes']+dfs['not sufficient, maybe']+dfs['not sufficient, no'])\n",
    "    sum_row = {i:(dfs.iloc[0][i]+dfs.iloc[2][i]+dfs.iloc[3][i]) for i in dfs.columns[1:8]}\n",
    "    dfs = dfs.append({'category':'sums'}|sum_row, ignore_index=True)\n",
    "    for i in ['sufficient, yes', 'sufficient, maybe', 'sufficient, no']:\n",
    "        dfs['% '+i] = dfs[i]/(dfs['sufficient, yes']+dfs['sufficient, maybe']+dfs['sufficient, no'])\n",
    "    dfs['sufficient to not sufficient'] = ((dfs['sufficient, yes']+dfs['sufficient, maybe']+dfs['sufficient, no'])/\n",
    "                                           (dfs['sum']))\n",
    "    return dfs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "1075c58b",
   "metadata": {},
   "outputs": [],
   "source": [
    "def right_type(df):\n",
    "    dfn = df.copy()\n",
    "    dfn[['#CHROM', 'REF', 'ALT']] = dfn[['#CHROM', 'REF', 'ALT']].astype(str)\n",
    "    dfn['POS'] = dfn['POS'].astype(int)\n",
    "    return dfn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "507178ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "# dataframe with all variants\n",
    "all_vars = pd.read_pickle('00_dataframes/all_annotated_vars')\n",
    "\n",
    "all_vars['#CHROM'] = pd.Categorical(all_vars['#CHROM'],\n",
    "                                    categories=['1','2','3','4','5','6','7','8','9','10','11','12', '13', '14', \n",
    "                                                '15', '16', '17', '18', '19', '20', '21', '22', 'X', 'Y'],\n",
    "                                    ordered=True)\n",
    "all_vars = all_vars.sort_values(by=['#CHROM', 'POS', 'REF', 'ALT'], ignore_index=True)\n",
    "all_vars = right_type(all_vars)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "37b8638a",
   "metadata": {},
   "outputs": [],
   "source": [
    "igv_all = merge_xlsx('2021-08-19_all_vars_analyzed.xlsx')\n",
    "igv_wo_dup = igv_all.copy().drop_duplicates(cols_cpra)\n",
    "#igv_all.to_pickle('00_dataframes/2021-11-19_igv_vars')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4b3bf483",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "557cad8e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import xml.etree.ElementTree as ET\n",
    "root = ET.parse('01_xlsx_files/response_1637323300036.xml').getroot()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "382bfd02",
   "metadata": {},
   "outputs": [],
   "source": [
    "chrom = t.find('chromosome').text\n",
    "    pos = t.find('position').text\n",
    "    ref = t.find('reference').text\n",
    "    alt = t.find('alternate').text\n",
    "    gene = t.find('gene').text\n",
    "    gt = t.find('genotype').text\n",
    "    assessment = t.find('assessment').text\n",
    "    #cadd = t.find('cadd').text\n",
    "    variation = t.find('variation').text\n",
    "    pt = t.find('phenotype').text\n",
    "    function = t.find('function').text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "91779c53",
   "metadata": {},
   "outputs": [
    {
     "ename": "IndexError",
     "evalue": "list index out of range",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mIndexError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-19-33a7a95b0fda>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     40\u001b[0m             \u001b[0mcrit\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mline\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m' id=\"'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'\"'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m': '\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     41\u001b[0m             \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mline\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m' '\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 42\u001b[0;31m                 \u001b[0mcrit\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0mi\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'=\"'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstrip\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'\"'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m', '\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     43\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     44\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0mline\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstartswith\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'<text'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mIndexError\u001b[0m: list index out of range"
     ]
    }
   ],
   "source": [
    "with open('01_xlsx_files/response_1637323300036.xml') as f:\n",
    "    ct = 0\n",
    "    ct_cl = 0\n",
    "    qci_dict = {}\n",
    "    for line in f.readlines():\n",
    "        line = line.lstrip()\n",
    "        if line.startswith('<chromosome'):\n",
    "            chrom = line.split('>')[1].split('<')[0]\n",
    "        elif line.startswith('<position'):\n",
    "            pos = line.split('>')[1].split('<')[0]\n",
    "        elif line.startswith('<reference'):\n",
    "            ref = line.split('>')[1].split('<')[0]\n",
    "        elif line.startswith('<alternate'):\n",
    "            alt = line.split('>')[1].split('<')[0]\n",
    "        elif line.startswith('<genotype'):\n",
    "            gt = line.split('>')[1].split('<')[0]\n",
    "        #elif line.startswith('<computedAssessment'):\n",
    "            #cp_ass = line.split('=\"')[1].split('\"')[0]\n",
    "        #elif line.startswith('<assessment'):\n",
    "            #ass = line.split('>')[1].split('<')[0]\n",
    "        elif line.startswith('<phenotype'):\n",
    "            pt = line.split('>')[1].split('<')[0]\n",
    "        elif line.startswith('<classification '):\n",
    "            ct += 1\n",
    "            ct_cl = 0\n",
    "            in_dict = {}\n",
    "            qci_dict[ct] = in_dict\n",
    "            \n",
    "            for k,v in zip(['chromosome', 'position', 'reference', \n",
    "                            'alternate', 'genotype', 'computedAssessment', \n",
    "                            'assessment', 'phenotype'], \n",
    "                           [chrom, pos, ref, alt, gt, pt]):\n",
    "                in_dict[k]=v\n",
    "            \n",
    "            in_dict['classified_by'] = line.split(' type=\"')[1].split('\"')[0]\n",
    "            in_dict['inheritance'] = line.split(' value=\"')[1].split('\"')[0]\n",
    "            in_dict['criteria'] = ''\n",
    "            \n",
    "        elif line.startswith('<criterion'):\n",
    "            crit = line.split(' id=\"')[1].split('\"')[0]+': '\n",
    "            for i in line.split(' ')[2:]:\n",
    "                crit += i.split('=\"')[1].strip('\"')[0]+', '\n",
    "        \n",
    "        elif line.startswith('<text'):\n",
    "            crit += line.split('>')[1].split('<')[0]+', '\n",
    "            in_dict['criteria'] += crit\n",
    "        #elif line.startswith('<rationale '):\n",
    "            #crit += line.split('text=\"')[1].split('\"')[0]\n",
    "            \n",
    "        '''\n",
    "        elif line.startswith('<cadd'):\n",
    "            cadd = line.split('>')[1].split('<')[0]\n",
    "        elif line.startswith('<variation'):\n",
    "            var = line.split('>')[1].split('<')[0]\n",
    "        elif line.startswith('<gene'):\n",
    "            gene = line.split('>')[1].split('<')[0]\n",
    "        elif \n",
    "        \n",
    "        elif line.startswith('<function'):\n",
    "            fct = line.split('>')[1].split('<')[0]\n",
    "            '''\n",
    "# counts, \"category\"+str(ct_cl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a6353fa6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "f2bd1e16",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{1: {'chromosome': '3',\n",
       "  'position': '52439929',\n",
       "  'reference': '0',\n",
       "  'alternate': 'A',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Likely Pathogenic',\n",
       "  'assessment': 'Pathogenic',\n",
       "  'phenotype': 'Tumor predisposition syndrome',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '35.00',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'BAP1',\n",
       "  'function': 'loss'},\n",
       " 2: {'chromosome': '3',\n",
       "  'position': '52440267',\n",
       "  'reference': '0',\n",
       "  'alternate': 'G',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Pathogenic',\n",
       "  'assessment': 'Likely Pathogenic',\n",
       "  'phenotype': 'Tumor predisposition syndrome',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '33.00',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'BAP1',\n",
       "  'function': 'loss'},\n",
       " 3: {'chromosome': '11',\n",
       "  'position': '108186639',\n",
       "  'reference': '0',\n",
       "  'alternate': 'A',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Pathogenic',\n",
       "  'assessment': 'Pathogenic',\n",
       "  'phenotype': 'Ataxia-telangiectasia',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '33.00',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'ATM',\n",
       "  'function': 'loss'},\n",
       " 4: {'chromosome': '11',\n",
       "  'position': '108224490',\n",
       "  'reference': '0',\n",
       "  'alternate': 'G',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Uncertain Significance',\n",
       "  'assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Ataxia-telangiectasia',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '23.70',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'ATM',\n",
       "  'function': 'loss'},\n",
       " 5: {'chromosome': '3',\n",
       "  'position': '52443569',\n",
       "  'reference': '0',\n",
       "  'alternate': 'T',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Pathogenic',\n",
       "  'assessment': 'Pathogenic',\n",
       "  'phenotype': 'Tumor predisposition syndrome',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '35.00',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'BAP1',\n",
       "  'function': 'loss'},\n",
       " 6: {'chromosome': '17',\n",
       "  'position': '41234419',\n",
       "  'reference': '0',\n",
       "  'alternate': 'C',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Pathogenic',\n",
       "  'assessment': 'Pathogenic',\n",
       "  'phenotype': 'Hereditary breast and/or ovarian cancer',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '33.00',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'BRCA1',\n",
       "  'function': 'loss'},\n",
       " 7: {'chromosome': '17',\n",
       "  'position': '41222939',\n",
       "  'reference': '0',\n",
       "  'alternate': 'G',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Pathogenic',\n",
       "  'assessment': 'Pathogenic',\n",
       "  'phenotype': 'Hereditary breast and/or ovarian cancer',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '22.90',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'BRCA1',\n",
       "  'function': 'loss'},\n",
       " 8: {'chromosome': '17',\n",
       "  'position': '59934594',\n",
       "  'reference': '0',\n",
       "  'alternate': 'C',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Pathogenic',\n",
       "  'assessment': 'Likely Pathogenic',\n",
       "  'phenotype': 'Hereditary breast and/or ovarian cancer',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '33.00',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'BRIP1',\n",
       "  'function': 'loss'},\n",
       " 9: {'chromosome': '16',\n",
       "  'position': '68849662',\n",
       "  'reference': '0',\n",
       "  'alternate': 'C',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Pathogenic',\n",
       "  'assessment': 'Pathogenic',\n",
       "  'phenotype': 'Hereditary diffuse malignant gastric tumor',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '34.00',\n",
       "  'variation': 'Deletion',\n",
       "  'gene': 'CDH1',\n",
       "  'function': 'loss'},\n",
       " 10: {'chromosome': '22',\n",
       "  'position': '29121230',\n",
       "  'reference': '0',\n",
       "  'alternate': 'T',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Pathogenic',\n",
       "  'assessment': 'Pathogenic',\n",
       "  'phenotype': 'Hereditary breast cancer',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '33.00',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'CHEK2',\n",
       "  'function': 'loss'},\n",
       " 11: {'chromosome': '1',\n",
       "  'position': '45797760',\n",
       "  'reference': '0',\n",
       "  'alternate': 'C',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Pathogenic',\n",
       "  'assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'MYH-associated polyposis',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '33.00',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'MUTYH',\n",
       "  'function': 'loss'},\n",
       " 12: {'chromosome': '1',\n",
       "  'position': '45797835',\n",
       "  'reference': '0',\n",
       "  'alternate': 'G',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Likely Pathogenic',\n",
       "  'assessment': 'Pathogenic',\n",
       "  'phenotype': 'MYH-associated polyposis',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '23.20',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'MUTYH',\n",
       "  'function': 'loss'},\n",
       " 13: {'chromosome': '7',\n",
       "  'position': '66459197',\n",
       "  'reference': '0',\n",
       "  'alternate': 'G',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Pathogenic',\n",
       "  'assessment': 'Pathogenic',\n",
       "  'phenotype': 'Shwachman syndrome',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '33.00',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'SBDS',\n",
       "  'function': 'loss'},\n",
       " 14: {'chromosome': '16',\n",
       "  'position': '3650977',\n",
       "  'reference': '0',\n",
       "  'alternate': 'TA',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Pathogenic',\n",
       "  'assessment': 'Pathogenic',\n",
       "  'phenotype': 'Fanconi anemia',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '19.68',\n",
       "  'variation': 'Insertion',\n",
       "  'gene': 'SLX4',\n",
       "  'function': 'normal'},\n",
       " 15: {'chromosome': '17',\n",
       "  'position': '7579699',\n",
       "  'reference': '0',\n",
       "  'alternate': 'T',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Pathogenic',\n",
       "  'assessment': 'Likely Pathogenic',\n",
       "  'phenotype': 'Li-Fraumeni syndrome',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '33.00',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'TP53',\n",
       "  'function': 'loss'},\n",
       " 16: {'chromosome': '17',\n",
       "  'position': '7578370',\n",
       "  'reference': '0',\n",
       "  'alternate': 'T',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Pathogenic',\n",
       "  'assessment': 'Pathogenic',\n",
       "  'phenotype': 'Li-Fraumeni syndrome',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '33.00',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'TP53',\n",
       "  'function': 'loss'},\n",
       " 17: {'chromosome': '19',\n",
       "  'position': '45871887',\n",
       "  'reference': '0',\n",
       "  'alternate': 'T',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Likely Pathogenic',\n",
       "  'assessment': 'Likely Pathogenic',\n",
       "  'phenotype': 'Xeroderma pigmentosum complementation group D',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '34.00',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'ERCC2',\n",
       "  'function': 'loss'},\n",
       " 18: {'chromosome': '17',\n",
       "  'position': '29482996',\n",
       "  'reference': '0',\n",
       "  'alternate': 'T',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Likely Pathogenic',\n",
       "  'assessment': 'Likely Pathogenic',\n",
       "  'phenotype': 'Neurofibromatosis 1',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'variation': 'Deletion',\n",
       "  'gene': 'NF1',\n",
       "  'function': 'normal'},\n",
       " 19: {'chromosome': '13',\n",
       "  'position': '48881547',\n",
       "  'reference': '0',\n",
       "  'alternate': 'A',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Likely Pathogenic',\n",
       "  'assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Familial retinoblastoma',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '22.10',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'RB1',\n",
       "  'function': 'normal'},\n",
       " 20: {'chromosome': '13',\n",
       "  'position': '48953712',\n",
       "  'reference': '0',\n",
       "  'alternate': 'C',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Uncertain Significance',\n",
       "  'assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Familial retinoblastoma',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'variation': 'Deletion',\n",
       "  'gene': 'RB1',\n",
       "  'function': 'normal'},\n",
       " 21: {'chromosome': '8',\n",
       "  'position': '145738768',\n",
       "  'reference': '0',\n",
       "  'alternate': 'T',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Likely Pathogenic',\n",
       "  'assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Baller-gerold syndrome',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '24.20',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'RECQL4',\n",
       "  'function': 'normal'},\n",
       " 22: {'chromosome': '1',\n",
       "  'position': '17345454',\n",
       "  'reference': '0',\n",
       "  'alternate': 'G',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Likely Pathogenic',\n",
       "  'assessment': 'Likely Pathogenic',\n",
       "  'phenotype': 'Hereditary paraganglioma-pheochromocytoma syndrome',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '34.00',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'SDHB',\n",
       "  'function': 'loss'},\n",
       " 23: {'chromosome': '9',\n",
       "  'position': '135779206',\n",
       "  'reference': '0',\n",
       "  'alternate': 'C',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Likely Pathogenic',\n",
       "  'assessment': 'Likely Pathogenic',\n",
       "  'phenotype': 'Tuberous sclerosis complex',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '33.00',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'TSC1',\n",
       "  'function': 'loss'},\n",
       " 24: {'chromosome': '16',\n",
       "  'position': '2121936',\n",
       "  'reference': '0',\n",
       "  'alternate': 'GT',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Likely Pathogenic',\n",
       "  'assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Tuberous sclerosis complex',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'variation': 'Insertion',\n",
       "  'gene': 'TSC2',\n",
       "  'function': 'loss'},\n",
       " 25: {'chromosome': '16',\n",
       "  'position': '2138446',\n",
       "  'reference': '0',\n",
       "  'alternate': 'C',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Uncertain Significance',\n",
       "  'assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Tuberous sclerosis complex',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '32.00',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'TSC2',\n",
       "  'function': 'loss'},\n",
       " 26: {'chromosome': '16',\n",
       "  'position': '2137924',\n",
       "  'reference': '0',\n",
       "  'alternate': 'T',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Uncertain Significance',\n",
       "  'assessment': 'Likely Benign',\n",
       "  'phenotype': 'Tuberous sclerosis complex',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '33.00',\n",
       "  'variation': 'Deletion',\n",
       "  'gene': 'TSC2',\n",
       "  'function': 'loss'},\n",
       " 27: {'chromosome': '9',\n",
       "  'position': '100437871',\n",
       "  'reference': '0',\n",
       "  'alternate': 'C',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Likely Pathogenic',\n",
       "  'assessment': 'Likely Pathogenic',\n",
       "  'phenotype': 'Xeroderma pigmentosum',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '29.60',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'XPA',\n",
       "  'function': 'loss'},\n",
       " 28: {'chromosome': '3',\n",
       "  'position': '14212051',\n",
       "  'reference': '0',\n",
       "  'alternate': 'G',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Likely Pathogenic',\n",
       "  'assessment': 'Likely Pathogenic',\n",
       "  'phenotype': 'Xeroderma pigmentosum',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '32.00',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'XPC',\n",
       "  'function': 'loss'},\n",
       " 29: {'chromosome': '3',\n",
       "  'position': '14206927',\n",
       "  'reference': '0',\n",
       "  'alternate': 'A',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Likely Pathogenic',\n",
       "  'assessment': 'Likely Pathogenic',\n",
       "  'phenotype': 'Xeroderma pigmentosum',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '35.00',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'XPC',\n",
       "  'function': 'loss'},\n",
       " 30: {'chromosome': '3',\n",
       "  'position': '14190057',\n",
       "  'reference': '0',\n",
       "  'alternate': 'T',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Uncertain Significance',\n",
       "  'assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Xeroderma pigmentosum',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '22.30',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'XPC',\n",
       "  'function': 'loss'},\n",
       " 31: {'chromosome': '14',\n",
       "  'position': '105243112',\n",
       "  'reference': '0',\n",
       "  'alternate': 'T',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Uncertain Significance',\n",
       "  'assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Cowden syndrome',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '17.34',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'AKT1',\n",
       "  'function': 'normal'},\n",
       " 32: {'chromosome': '11',\n",
       "  'position': '44255793',\n",
       "  'reference': '0',\n",
       "  'alternate': 'A',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Uncertain Significance',\n",
       "  'assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Multiple hereditary exostoses',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '23.80',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'EXT2',\n",
       "  'function': 'loss'},\n",
       " 33: {'chromosome': '2',\n",
       "  'position': '48033789',\n",
       "  'reference': '0',\n",
       "  'alternate': 'T',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Uncertain Significance',\n",
       "  'assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Lynch Syndrome',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '32.00',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'MSH6',\n",
       "  'function': 'loss'},\n",
       " 34: {'chromosome': '3',\n",
       "  'position': '52443725',\n",
       "  'reference': '0',\n",
       "  'alternate': 'G',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Uncertain Significance',\n",
       "  'assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Cancers and Tumors',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '23.20',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'PHF7',\n",
       "  'function': 'loss'},\n",
       " 35: {'chromosome': '19',\n",
       "  'position': '50918678',\n",
       "  'reference': '0',\n",
       "  'alternate': 'A',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Uncertain Significance',\n",
       "  'assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Familial colorectal cancer',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'POLD1',\n",
       "  'function': 'normal'},\n",
       " 36: {'chromosome': '12',\n",
       "  'position': '133254296',\n",
       "  'reference': '0',\n",
       "  'alternate': 'A',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Uncertain Significance',\n",
       "  'assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Susceptibility to colorectal cancer type 12',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '17.71',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'POLE',\n",
       "  'function': 'normal'},\n",
       " 37: {'chromosome': '17',\n",
       "  'position': '56809887',\n",
       "  'reference': '0',\n",
       "  'alternate': 'G',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Uncertain Significance',\n",
       "  'assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Hereditary breast and/or ovarian cancer',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '16.73',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'RAD51C',\n",
       "  'function': 'normal'},\n",
       " 38: {'chromosome': '17',\n",
       "  'position': '74470455',\n",
       "  'reference': '0',\n",
       "  'alternate': 'T',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Uncertain Significance',\n",
       "  'assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Tylosis with esophageal cancer syndrome',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '23.40',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'RHBDF2',\n",
       "  'function': 'normal'},\n",
       " 39: {'chromosome': '17',\n",
       "  'position': '74469327',\n",
       "  'reference': '0',\n",
       "  'alternate': 'A',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Likely Benign',\n",
       "  'assessment': 'Likely Benign',\n",
       "  'phenotype': 'Tylosis with esophageal cancer syndrome',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'RHBDF2',\n",
       "  'function': 'normal'},\n",
       " 40: {'chromosome': '10',\n",
       "  'position': '104389820',\n",
       "  'reference': '0',\n",
       "  'alternate': 'A',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Uncertain Significance',\n",
       "  'assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Basal cell nevus syndrome',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '23.80',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'SUFU',\n",
       "  'function': 'normal'},\n",
       " 41: {'chromosome': '22',\n",
       "  'position': '24175755',\n",
       "  'reference': '0',\n",
       "  'alternate': 'A',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Likely Benign',\n",
       "  'assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Coffin-Siris syndrome',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'SMARCB1',\n",
       "  'function': 'normal'},\n",
       " 42: {'chromosome': '11',\n",
       "  'position': '108202604',\n",
       "  'reference': '0',\n",
       "  'alternate': 'C',\n",
       "  'genotype': 'Het',\n",
       "  'computedAssessment': 'Pathogenic',\n",
       "  'assessment': 'Pathogenic',\n",
       "  'phenotype': 'Ataxia-telangiectasia',\n",
       "  'classified_by': 'QIAGEN',\n",
       "  'cadd': '34.00',\n",
       "  'variation': 'SNV',\n",
       "  'gene': 'ATM',\n",
       "  'function': 'loss'}}"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "qci_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "f56f03da",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PP4 - Patient's phenotype or family history is highly specific for a disease with a single genetic etiology (supporting)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 797.61; 95% confidence interval = (169.04, 3763.54); FET 2-tail p-value < 0.0001; affected individual count = 8] (Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PP4 - Patient's phenotype or family history is highly specific for a disease with a single genetic etiology (supporting)\n",
      "- BS4 - Lack of segregation in affected members of a family (Strong)\n",
      "+ PP5 - Reputable source recently reports variant as pathogenic, but the evidence is not available to the laboratory to perform an independent evaluation (Supporting)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 797.61; 95% confidence interval = (169.04, 3763.54); FET 2-tail p-value < 0.0001; affected individual count = 8] (Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 18.67; 95% confidence interval = (0.90, 388.84); FET 2-tail p-value = 4.464E-2; affected individual count = 2; Note: Small number of affected individuals with variant.] (Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency falls below the max recessive frequency (0.50%) expected for this phenotype] (Moderate)\n",
      "+ PP5 - Reputable source recently reports variant as pathogenic, but the evidence is not available to the laboratory to perform an independent evaluation (Supporting)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 18.67; 95% confidence interval = (0.90, 388.84); FET 2-tail p-value = 4.464E-2; affected individual count = 2; Note: Small number of affected individuals with variant.] (Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency falls below the max recessive frequency (0.50%) expected for this phenotype] (Moderate)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency falls below the max recessive frequency (0.50%) expected for this phenotype] (Moderate)\n",
      "+ PP3 - Multiple lines of computational evidence support a deleterious effect on the gene or gene product [CADD = 23.7, MaxEntScan] (Supporting)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency falls below the max recessive frequency (0.50%) expected for this phenotype] (Moderate)\n",
      "+ PP3 - Multiple lines of computational evidence support a deleterious effect on the gene or gene product [CADD = 23.7, MaxEntScan] (Supporting)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 978.58; 95% confidence interval = (46.93, 20404.50); FET 2-tail p-value < 0.0001; affected individual count = 2; Note: Small number of affected individuals with variant.] (Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PP5 - Reputable source recently reports variant as pathogenic, but the evidence is not available to the laboratory to perform an independent evaluation (Supporting)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 978.58; 95% confidence interval = (46.93, 20404.50); FET 2-tail p-value < 0.0001; affected individual count = 2; Note: Small number of affected individuals with variant.] (Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PP5 - Reputable source recently reports variant as pathogenic, but the evidence is not available to the laboratory to perform an independent evaluation (Supporting)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 17.95; 95% confidence interval = (1.01, 318.65); FET 2-tail p-value = 5.493E-3; affected individual count = 6] (Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PP5 - Reputable source recently reports variant as pathogenic, but the evidence is not available to the laboratory to perform an independent evaluation (Supporting)\n",
      "- BP5 - Variant found in a case with an alternate molecular basis for disease (Supporting)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 17.95; 95% confidence interval = (1.01, 318.65); FET 2-tail p-value = 5.493E-3; affected individual count = 6] (Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PP5 - Reputable source recently reports variant as pathogenic, but the evidence is not available to the laboratory to perform an independent evaluation (Supporting)\n",
      "- BP5 - Variant found in a case with an alternate molecular basis for disease (Supporting)\n",
      "+ PS3 - Well-established in vitro or in vivo functional studies supportive of a damaging effect on the gene or gene product (Strong)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 114.02; 95% confidence interval = (7.01, 1853.56); FET 2-tail p-value < 0.0001; affected individual count = 41] (Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PP3 - Multiple lines of computational evidence support a deleterious effect on the gene or gene product [CADD = 22.9, MaxEntScan] (Supporting)\n",
      "+ PP5 - Reputable source recently reports variant as pathogenic, but the evidence is not available to the laboratory to perform an independent evaluation (Supporting)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PS3 - Well-established in vitro or in vivo functional studies supportive of a damaging effect on the gene or gene product (Strong)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 114.02; 95% confidence interval = (7.01, 1853.56); FET 2-tail p-value < 0.0001; affected individual count = 41] (Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PP3 - Multiple lines of computational evidence support a deleterious effect on the gene or gene product [CADD = 22.9, MaxEntScan] (Supporting)\n",
      "+ PP5 - Reputable source recently reports variant as pathogenic, but the evidence is not available to the laboratory to perform an independent evaluation (Supporting)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PS3 - Well-established in vitro or in vivo functional studies supportive of a damaging effect on the gene or gene product (Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PS3 - Well-established in vitro or in vivo functional studies supportive of a damaging effect on the gene or gene product (Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 26.80; 95% confidence interval = (1.29, 558.28); FET 2-tail p-value = 2.472E-2; affected individual count = 2; Note: Small number of affected individuals with variant.] (Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PP5 - Reputable source recently reports variant as pathogenic, but the evidence is not available to the laboratory to perform an independent evaluation (Supporting)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 26.80; 95% confidence interval = (1.29, 558.28); FET 2-tail p-value = 2.472E-2; affected individual count = 2; Note: Small number of affected individuals with variant.] (Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PP5 - Reputable source recently reports variant as pathogenic, but the evidence is not available to the laboratory to perform an independent evaluation (Supporting)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PS3 - Well-established in vitro or in vivo functional studies supportive of a damaging effect on the gene or gene product (Strong)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 58.63; 95% confidence interval = (42.45, 80.97); FET 2-tail p-value < 0.0001; affected individual count = 479] (Strong)\n",
      "+ PP5 - Reputable source recently reports variant as pathogenic, but the evidence is not available to the laboratory to perform an independent evaluation (Supporting)\n",
      "- BP2 - Observed in trans with a pathogenic variant for a dominant disorder (Supporting)\n",
      "- BP5 - Variant found in a case with an alternate molecular basis for disease (Supporting)\n",
      "+ PM7 - At least 3 independent somatic observations of the alteration in literature (Moderate)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PS3 - Well-established in vitro or in vivo functional studies supportive of a damaging effect on the gene or gene product (Strong)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 58.63; 95% confidence interval = (42.45, 80.97); FET 2-tail p-value < 0.0001; affected individual count = 479] (Strong)\n",
      "+ PP5 - Reputable source recently reports variant as pathogenic, but the evidence is not available to the laboratory to perform an independent evaluation (Supporting)\n",
      "- BP2 - Observed in trans with a pathogenic variant for a dominant disorder (Supporting)\n",
      "- BP5 - Variant found in a case with an alternate molecular basis for disease (Supporting)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PS3 - Well-established in vitro or in vivo functional studies supportive of a damaging effect on the gene or gene product (Strong)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 6.41; 95% confidence interval = (5.33, 7.70); FET 2-tail p-value < 0.0001; affected individual count = 183] (Strong)\n",
      "- BS1 - Allele frequency is greater than expected for disorder [Max Recessive Expected Frequency: 1.000%; gnomAD Frequency: 1.539%] (Strong)\n",
      "- BS2 - Observed in a healthy adult individual for a recessive (homozygous), dominant (heterozygous), or X-linked (hemizygous) disorder with full penetrance expected at an early age (Strong)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PS3 - Well-established in vitro or in vivo functional studies supportive of a damaging effect on the gene or gene product (Strong)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 6.41; 95% confidence interval = (5.33, 7.70); FET 2-tail p-value < 0.0001; affected individual count = 183] (Strong)\n",
      "- BS1 - Allele frequency is greater than expected for disorder [Max Recessive Expected Frequency: 1.000%; gnomAD Frequency: 1.539%] (Strong)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 76.75; 95% confidence interval = (48.59, 121.21); FET 2-tail p-value < 0.0001; affected individual count = 150] (Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency falls below 1% for this recessive phenotype] (Moderate)\n",
      "+ PM3 - For recessive disorders, detected in trans with a pathogenic variant (Moderate)\n",
      "+ PP3 - Multiple lines of computational evidence support a deleterious effect on the gene or gene product [CADD = 23.2, MaxEntScan] (Supporting)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 76.75; 95% confidence interval = (48.59, 121.21); FET 2-tail p-value < 0.0001; affected individual count = 150] (Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency falls below 1% for this recessive phenotype] (Moderate)\n",
      "+ PM3 - For recessive disorders, detected in trans with a pathogenic variant (Moderate)\n",
      "+ PP3 - Multiple lines of computational evidence support a deleterious effect on the gene or gene product [CADD = 23.2, MaxEntScan] (Supporting)\n",
      "+ PA2 - Established common pathogenic founder mutation (Standalone)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PS3 - Well-established in vitro or in vivo functional studies supportive of a damaging effect on the gene or gene product (Strong)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 99.74; 95% confidence interval = (88.99, 111.80); FET 2-tail p-value < 0.0001; affected individual count = 720] (Strong)\n",
      "+ PM3 - For recessive disorders, detected in trans with a pathogenic variant (Moderate)\n",
      "+ PP5 - Reputable source recently reports variant as pathogenic, but the evidence is not available to the laboratory to perform an independent evaluation (Supporting)\n",
      "- BS1 - Allele frequency is greater than expected for disorder [Max Recessive Expected Frequency: 0.224%; gnomAD Frequency: 0.518%] (Strong)\n",
      "- BS2 - Observed in a healthy adult individual for a recessive (homozygous), dominant (heterozygous), or X-linked (hemizygous) disorder with full penetrance expected at an early age [# homozygous control individuals: 2 (gnomAD)] (Strong)\n",
      "- BS4 - Lack of segregation in affected members of a family (Strong)\n",
      "+ PA2 - Established common pathogenic founder mutation (Standalone)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PS3 - Well-established in vitro or in vivo functional studies supportive of a damaging effect on the gene or gene product (Strong)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 99.74; 95% confidence interval = (88.99, 111.80); FET 2-tail p-value < 0.0001; affected individual count = 720] (Strong)\n",
      "+ PM3 - For recessive disorders, detected in trans with a pathogenic variant (Moderate)\n",
      "+ PP5 - Reputable source recently reports variant as pathogenic, but the evidence is not available to the laboratory to perform an independent evaluation (Supporting)\n",
      "- BS1 - Allele frequency is greater than expected for disorder [Max Recessive Expected Frequency: 0.224%; gnomAD Frequency: 0.518%] (Strong)\n",
      "- BS2 - Observed in a healthy adult individual for a recessive (homozygous), dominant (heterozygous), or X-linked (hemizygous) disorder with full penetrance expected at an early age [# homozygous control individuals: 2 (gnomAD)] (Strong)\n",
      "- BS4 - Lack of segregation in affected members of a family (Strong)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 1022.11; 95% confidence interval = (52.75, 19803.80); FET 2-tail p-value < 0.0001; affected individual count = 3; Note: Small number of affected individuals with variant.] (Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency falls below the max recessive frequency (0.25%) expected for this phenotype] (Moderate)\n",
      "+ PM3 - For recessive disorders, detected in trans with a pathogenic variant (Moderate)\n",
      "+ PS3 - Well-established in vitro or in vivo functional studies supportive of a damaging effect on the gene or gene product (Strong)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 1022.11; 95% confidence interval = (52.75, 19803.80); FET 2-tail p-value < 0.0001; affected individual count = 3; Note: Small number of affected individuals with variant.] (Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency falls below the max recessive frequency (0.25%) expected for this phenotype] (Moderate)\n",
      "+ PM3 - For recessive disorders, detected in trans with a pathogenic variant (Moderate)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PS3 - Well-established in vitro or in vivo functional studies supportive of a damaging effect on the gene or gene product (Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PM7 - At least 3 independent somatic observations of the alteration in literature (Moderate)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PS3 - Well-established in vitro or in vivo functional studies supportive of a damaging effect on the gene or gene product (Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PM7 - At least 3 independent somatic observations of the alteration in literature (Moderate)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PVS7 - At least 20 independent somatic observations of the alteration in literature (Very Strong)\n",
      "+ PS3 - Well-established in vitro or in vivo functional studies supportive of a damaging effect on the gene or gene product (Strong)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 13.79; 95% confidence interval = (0.76, 249.34); FET 2-tail p-value = 1.722E-2; affected individual count = 5] (Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PP5 - Reputable source recently reports variant as pathogenic, but the evidence is not available to the laboratory to perform an independent evaluation (Supporting)\n",
      "- BP2 - Observed in cis or trans with a pathogenic variant (phase unknown) for a dominant disorder (Supporting)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PVS7 - At least 20 independent somatic observations of the alteration in literature (Very Strong)\n",
      "+ PS3 - Well-established in vitro or in vivo functional studies supportive of a damaging effect on the gene or gene product (Strong)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 13.79; 95% confidence interval = (0.76, 249.34); FET 2-tail p-value = 1.722E-2; affected individual count = 5] (Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PP5 - Reputable source recently reports variant as pathogenic, but the evidence is not available to the laboratory to perform an independent evaluation (Supporting)\n",
      "- BP2 - Observed in cis or trans with a pathogenic variant (phase unknown) for a dominant disorder (Supporting)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency falls below the max recessive frequency (0.03%) expected for this phenotype] (Moderate)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency falls below the max recessive frequency (0.03%) expected for this phenotype] (Moderate)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PS3 - Well-established in vitro or in vivo functional studies supportive of a damaging effect on the gene or gene product (Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PP3 - Multiple lines of computational evidence support a deleterious effect on the gene or gene product [CADD = 22.1, MaxEntScan] (Supporting)\n",
      "+ PS3 - Well-established in vitro or in vivo functional studies supportive of a damaging effect on the gene or gene product (Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PP3 - Multiple lines of computational evidence support a deleterious effect on the gene or gene product [CADD = 22.1, MaxEntScan] (Supporting)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency falls below the max recessive frequency (0.03%) expected for this phenotype] (Moderate)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency falls below the max recessive frequency (0.03%) expected for this phenotype] (Moderate)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PP4 - Patient's phenotype or family history is highly specific for a disease with a single genetic etiology (supporting)\n",
      "+ PP3 - Multiple lines of computational evidence support a deleterious effect on the gene or gene product (Supporting)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "- BS1 - Allele frequency is greater than expected for disorder [Max Dominant Expected Frequency: 0.006%; gnomAD Frequency: 0.011%] (Strong)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "- BS1 - Allele frequency is greater than expected for disorder [Max Dominant Expected Frequency: 0.006%; gnomAD Frequency: 0.011%] (Strong)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PM6 - Assumed de novo, but without confirmation of paternity and maternity (Moderate)\n",
      "- BS1 - Allele frequency is greater than expected for disorder [Max Dominant Expected Frequency: 0.006%; gnomAD Frequency: 0.310%] (Strong)\n",
      "- BP2 - Observed in cis or trans with a pathogenic variant (phase unknown) for a dominant disorder (Supporting)\n",
      "- BP6 - Reputable source recently reports variant as benign, but the evidence is not available to the laboratory to perform an independent evaluation (Supporting)\n",
      "+ PP3 - Multiple lines of computational evidence support a deleterious effect on the gene or gene product (Supporting)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PM6 - Assumed de novo, but without confirmation of paternity and maternity (Moderate)\n",
      "- BS1 - Allele frequency is greater than expected for disorder [Max Dominant Expected Frequency: 0.006%; gnomAD Frequency: 0.310%] (Strong)\n",
      "- BP2 - Observed in cis or trans with a pathogenic variant (phase unknown) for a dominant disorder (Supporting)\n",
      "- BP6 - Reputable source recently reports variant as benign, but the evidence is not available to the laboratory to perform an independent evaluation (Supporting)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency falls below the max recessive frequency (0.15%) expected for this phenotype] (Moderate)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency falls below the max recessive frequency (0.15%) expected for this phenotype] (Moderate)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency falls below the max recessive frequency (0.15%) expected for this phenotype] (Moderate)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency falls below the max recessive frequency (0.15%) expected for this phenotype] (Moderate)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency falls below the max recessive frequency (0.15%) expected for this phenotype] (Moderate)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency falls below the max recessive frequency (0.15%) expected for this phenotype] (Moderate)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency falls below the max recessive frequency (0.15%) expected for this phenotype] (Moderate)\n",
      "+ PP3 - Multiple lines of computational evidence support a deleterious effect on the gene or gene product [CADD = 22.3, MaxEntScan] (Supporting)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency falls below the max recessive frequency (0.15%) expected for this phenotype] (Moderate)\n",
      "+ PP3 - Multiple lines of computational evidence support a deleterious effect on the gene or gene product [CADD = 22.3, MaxEntScan] (Supporting)\n",
      "+ PP3 - Multiple lines of computational evidence support a deleterious effect on the gene or gene product [MaxEntScan] (Supporting)\n",
      "- BS1 - Allele frequency is greater than expected for disorder [Max Dominant Expected Frequency: 0.000%; gnomAD Frequency: 0.003%] (Strong)\n",
      "+ PP3 - Multiple lines of computational evidence support a deleterious effect on the gene or gene product [MaxEntScan] (Supporting)\n",
      "- BS1 - Allele frequency is greater than expected for disorder [Max Dominant Expected Frequency: 0.000%; gnomAD Frequency: 0.003%] (Strong)\n",
      "+ PP3 - Multiple lines of computational evidence support a deleterious effect on the gene or gene product [CADD = 23.8, MaxEntScan] (Supporting)\n",
      "- BS1 - Allele frequency is greater than expected for disorder [Max Dominant Expected Frequency: 0.001%; gnomAD Frequency: 0.004%] (Strong)\n",
      "+ PP3 - Multiple lines of computational evidence support a deleterious effect on the gene or gene product [CADD = 23.8, MaxEntScan] (Supporting)\n",
      "- BS1 - Allele frequency is greater than expected for disorder [Max Dominant Expected Frequency: 0.001%; gnomAD Frequency: 0.004%] (Strong)\n",
      "+ PM1 - Located in a mutational hot spot (Moderate)\n",
      "+ PM5 - Novel missense change at an amino acid residue where a different missense change (p.R1334Q) determined to be pathogenic has been seen before (Moderate)\n",
      "+ PP3 - Multiple lines of computational evidence support a deleterious effect on the gene or gene product [CADD = 32.0, MaxEntScan] (Supporting)\n",
      "+ PM1 - Located in a mutational hot spot (Moderate)\n",
      "+ PM5 - Novel missense change at an amino acid residue where a different missense change (p.R1334Q) determined to be pathogenic has been seen before (Moderate)\n",
      "+ PP3 - Multiple lines of computational evidence support a deleterious effect on the gene or gene product [CADD = 32.0, MaxEntScan] (Supporting)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 39.05; 95% confidence interval = (4.56, 334.76); FET 2-tail p-value = 3.023E-2; affected individual count = 1; Note: Small number of affected individuals with variant.] (Strong)\n",
      "+ PP3 - Multiple lines of computational evidence support a deleterious effect on the gene or gene product [CADD = 23.2, MaxEntScan] (Supporting)\n",
      "- BP5 - Variant found in a case with an alternate molecular basis for disease (Supporting)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 39.05; 95% confidence interval = (4.56, 334.76); FET 2-tail p-value = 3.023E-2; affected individual count = 1; Note: Small number of affected individuals with variant.] (Strong)\n",
      "+ PP3 - Multiple lines of computational evidence support a deleterious effect on the gene or gene product [CADD = 23.2, MaxEntScan] (Supporting)\n",
      "- BP5 - Variant found in a case with an alternate molecular basis for disease (Supporting)\n",
      "+ PP3 - Multiple lines of computational evidence support a deleterious effect on the gene or gene product [MaxEntScan] (Supporting)\n",
      "- BP4 - Multiple lines of computational evidence suggest no impact on the gene or gene product, CADD < 10 (Supporting)\n",
      "+ PP3 - Multiple lines of computational evidence support a deleterious effect on the gene or gene product [MaxEntScan] (Supporting)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PP2 - Missense variant in a gene that has a low rate of benign missense variation and in which missense variants are a common mechanism of disease (Supporting)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PP2 - Missense variant in a gene that has a low rate of benign missense variation and in which missense variants are a common mechanism of disease (Supporting)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "- BP7 - A synonymous variant which does not create a donor (GT) or acceptor (AG) sequence and the nucleotide is not highly conserved (Supporting)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "- BP7 - A synonymous variant which does not create a donor (GT) or acceptor (AG) sequence and the nucleotide is not highly conserved (Supporting)\n",
      "+ PP3 - Multiple lines of computational evidence support a deleterious effect on the gene or gene product [CADD = 23.4, MaxEntScan] (Supporting)\n",
      "- BS1 - Allele frequency is greater than expected for disorder [Max Dominant Expected Frequency: 0.000%; gnomAD Frequency: 0.011%] (Strong)\n",
      "+ PP3 - Multiple lines of computational evidence support a deleterious effect on the gene or gene product [CADD = 23.4, MaxEntScan] (Supporting)\n",
      "- BS1 - Allele frequency is greater than expected for disorder [Max Dominant Expected Frequency: 0.000%; gnomAD Frequency: 0.011%] (Strong)\n",
      "- BS1 - Allele frequency is greater than expected for disorder [Max Dominant Expected Frequency: 0.000%; gnomAD Frequency: 0.461%] (Strong)\n",
      "- BP4 - Multiple lines of computational evidence suggest no impact on the gene or gene product, CADD < 10 (Supporting)\n",
      "- BP6 - Reputable source recently reports variant as benign, but the evidence is not available to the laboratory to perform an independent evaluation (Supporting)\n",
      "- BS1 - Allele frequency is greater than expected for disorder [Max Dominant Expected Frequency: 0.000%; gnomAD Frequency: 0.461%] (Strong)\n",
      "- BP4 - Multiple lines of computational evidence suggest no impact on the gene or gene product, CADD < 10 (Supporting)\n",
      "- BP6 - Reputable source recently reports variant as benign, but the evidence is not available to the laboratory to perform an independent evaluation (Supporting)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PP3 - Multiple lines of computational evidence support a deleterious effect on the gene or gene product [CADD = 23.8, MaxEntScan] (Supporting)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency is 0% or <= 0.001%] (Moderate)\n",
      "+ PP3 - Multiple lines of computational evidence support a deleterious effect on the gene or gene product [CADD = 23.8, MaxEntScan] (Supporting)\n",
      "- BS1 - Allele frequency is greater than expected for disorder [Max Dominant Expected Frequency: 0.000%; gnomAD Frequency: 0.030%] (Strong)\n",
      "- BP4 - Multiple lines of computational evidence suggest no impact on the gene or gene product, CADD < 10 (Supporting)\n",
      "- BS1 - Allele frequency is greater than expected for disorder [Max Dominant Expected Frequency: 0.000%; gnomAD Frequency: 0.030%] (Strong)\n",
      "- BP4 - Multiple lines of computational evidence suggest no impact on the gene or gene product, CADD < 10 (Supporting)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PS3 - Well-established in vitro or in vivo functional studies supportive of a damaging effect on the gene or gene product (Strong)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 100.90; 95% confidence interval = (31.79, 320.25); FET 2-tail p-value < 0.0001; affected individual count = 72] (Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency falls below the max recessive frequency (0.50%) expected for this phenotype] (Moderate)\n",
      "+ PM3 - For recessive disorders, detected in trans with a pathogenic variant (Moderate)\n",
      "+ PP5 - Reputable source recently reports variant as pathogenic, but the evidence is not available to the laboratory to perform an independent evaluation (Supporting)\n",
      "+ PVS1 - Null variant (nonsense, frameshift, canonical +/-1 or 2 splice sites, initiation codon, copy number loss, single or multi exon deletion) in a gene where loss of function (LOF) is a known mechanism of disease (Very Strong)\n",
      "+ PS3 - Well-established in vitro or in vivo functional studies supportive of a damaging effect on the gene or gene product (Strong)\n",
      "+ PS4 - The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls [odds ratio = 100.90; 95% confidence interval = (31.79, 320.25); FET 2-tail p-value < 0.0001; affected individual count = 72] (Strong)\n",
      "+ PM2 - Absent from controls (or at extremely low frequency if recessive) in gnomAD [In these sources of population frequency data, this variant's frequency falls below the max recessive frequency (0.50%) expected for this phenotype] (Moderate)\n",
      "+ PM3 - For recessive disorders, detected in trans with a pathogenic variant (Moderate)\n",
      "+ PP5 - Reputable source recently reports variant as pathogenic, but the evidence is not available to the laboratory to perform an independent evaluation (Supporting)\n"
     ]
    }
   ],
   "source": [
    "newr = []\n",
    "for t in root.findall('variant'):\n",
    "    for i in t.findall('classifications'):\n",
    "        for s in i.findall('classification'):\n",
    "            for o in s.findall('criteria'):\n",
    "                for n in o.findall('criterion'):\n",
    "                    print(n.find('text').text)\n",
    "        #d_new = {}\n",
    "        #for s in i.findall('classification'):\n",
    "        #    d_new[s.find]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "79ad95cf",
   "metadata": {},
   "outputs": [],
   "source": [
    "rows = []\n",
    "\n",
    "for t in root.findall('variant'):\n",
    "    chrom = t.find('chromosome').text\n",
    "    pos = t.find('position').text\n",
    "    ref = t.find('reference').text\n",
    "    alt = t.find('alternate').text\n",
    "    gene = t.find('gene').text\n",
    "    gt = t.find('genotype').text\n",
    "    assessment = t.find('assessment').text\n",
    "    #cadd = t.find('cadd').text\n",
    "    variation = t.find('variation').text\n",
    "    pt = t.find('phenotype').text\n",
    "    function = t.find('function').text\n",
    "    for i in t.findall('transcriptchange'):\n",
    "        transc = i.find('transcript').text+':'+i.find('change').text\n",
    "        ex_num = i.find('exonNumber').text\n",
    "        region = i.find('region').text\n",
    "    for i in t.findall('classifications'):\n",
    "        class_dict = {}\n",
    "        for s in i.findall('classification'):\n",
    "            \n",
    "            \n",
    "            for l in \n",
    "        \n",
    "    \n",
    "    \n",
    "    rows.append({'#CHROM': chrom, 'POS':pos, 'REF':ref, 'ALT':alt, 'Gene':gene, \n",
    "                 'Genotype':gt, \n",
    "                 'Assessment':assessment, 'Phenotype':pt, 'Transcript':transc, \n",
    "                 'Exon_number':ex_num, 'Region':region, 'Function':function, \n",
    "                 'Variation':variation, })\n",
    "\n",
    "qci_df = pd.DataFrame(rows)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "id": "c5876ba5",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Assessment</th>\n",
       "      <th>Phenotype</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>Exon_number</th>\n",
       "      <th>Region</th>\n",
       "      <th>Function</th>\n",
       "      <th>Variation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>52439929</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>BAP1</td>\n",
       "      <td>Het</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Tumor predisposition syndrome</td>\n",
       "      <td>NM_004656.4:c.784-1G&gt;T</td>\n",
       "      <td>10</td>\n",
       "      <td>Splice Site</td>\n",
       "      <td>loss</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>52440267</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>BAP1</td>\n",
       "      <td>Het</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>Tumor predisposition syndrome</td>\n",
       "      <td>NM_004656.4:c.783+2T&gt;C</td>\n",
       "      <td>9</td>\n",
       "      <td>Splice Site</td>\n",
       "      <td>loss</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>11</td>\n",
       "      <td>108186639</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ATM</td>\n",
       "      <td>Het</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Ataxia-telangiectasia</td>\n",
       "      <td>NM_000051.4:c.6095+1G&gt;A</td>\n",
       "      <td>41</td>\n",
       "      <td>Splice Site</td>\n",
       "      <td>loss</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>11</td>\n",
       "      <td>108224490</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>ATM</td>\n",
       "      <td>Het</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Ataxia-telangiectasia</td>\n",
       "      <td>NM_000051.4:c.8672-3T&gt;G</td>\n",
       "      <td>59</td>\n",
       "      <td>Intron</td>\n",
       "      <td>loss</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3</td>\n",
       "      <td>52443569</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>BAP1</td>\n",
       "      <td>Het</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Tumor predisposition syndrome</td>\n",
       "      <td>NM_004656.4:c.122+1G&gt;A</td>\n",
       "      <td>3</td>\n",
       "      <td>Splice Site</td>\n",
       "      <td>loss</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>17</td>\n",
       "      <td>41234419</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>Het</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Hereditary breast and/or ovarian cancer</td>\n",
       "      <td>NM_007294.4:c.4357+2T&gt;G</td>\n",
       "      <td>12</td>\n",
       "      <td>Splice Site</td>\n",
       "      <td>loss</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>17</td>\n",
       "      <td>41222939</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>Het</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Hereditary breast and/or ovarian cancer</td>\n",
       "      <td>NM_007294.4:c.4986+6T&gt;C</td>\n",
       "      <td>15</td>\n",
       "      <td>Intron</td>\n",
       "      <td>loss</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>17</td>\n",
       "      <td>59934594</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>BRIP1</td>\n",
       "      <td>Het</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>Hereditary breast and/or ovarian cancer</td>\n",
       "      <td>NM_032043.3:c.206-2A&gt;G</td>\n",
       "      <td>4</td>\n",
       "      <td>Splice Site</td>\n",
       "      <td>loss</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>16</td>\n",
       "      <td>68849662</td>\n",
       "      <td>CG</td>\n",
       "      <td>C</td>\n",
       "      <td>CDH1</td>\n",
       "      <td>Het</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Hereditary diffuse malignant gastric tumor</td>\n",
       "      <td>NM_001317184.2:c.1382+1delG</td>\n",
       "      <td>9</td>\n",
       "      <td>Splice Site</td>\n",
       "      <td>loss</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>22</td>\n",
       "      <td>29121230</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>Het</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Hereditary breast cancer</td>\n",
       "      <td>NM_001005735.2:c.573+1G&gt;A</td>\n",
       "      <td>4</td>\n",
       "      <td>Splice Site</td>\n",
       "      <td>loss</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>1</td>\n",
       "      <td>45797760</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>MUTYH</td>\n",
       "      <td>Het</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>MYH-associated polyposis</td>\n",
       "      <td>NM_001048171.2:c.850-2A&gt;G</td>\n",
       "      <td>11</td>\n",
       "      <td>Splice Site</td>\n",
       "      <td>loss</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>1</td>\n",
       "      <td>45797835</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>MUTYH</td>\n",
       "      <td>Het</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>MYH-associated polyposis</td>\n",
       "      <td>NM_001048171.2:c.849+3A&gt;C</td>\n",
       "      <td>10</td>\n",
       "      <td>Intron</td>\n",
       "      <td>loss</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>7</td>\n",
       "      <td>66459197</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SBDS</td>\n",
       "      <td>Het</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Shwachman syndrome</td>\n",
       "      <td>NM_016038.4:c.258+2T&gt;C</td>\n",
       "      <td>2</td>\n",
       "      <td>Splice Site</td>\n",
       "      <td>loss</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>16</td>\n",
       "      <td>3650977</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>SLX4</td>\n",
       "      <td>Het</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Fanconi anemia</td>\n",
       "      <td>NM_032444.4:c.1163+2dupT</td>\n",
       "      <td>5</td>\n",
       "      <td>Splice Site</td>\n",
       "      <td>normal</td>\n",
       "      <td>Insertion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>17</td>\n",
       "      <td>7579699</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TP53</td>\n",
       "      <td>Het</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>Li-Fraumeni syndrome</td>\n",
       "      <td>NM_000546.6:c.96+1G&gt;A</td>\n",
       "      <td>3</td>\n",
       "      <td>Splice Site</td>\n",
       "      <td>loss</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>17</td>\n",
       "      <td>7578370</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TP53</td>\n",
       "      <td>Het</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Li-Fraumeni syndrome</td>\n",
       "      <td>NM_000546.6:c.559+1G&gt;A</td>\n",
       "      <td>5</td>\n",
       "      <td>Splice Site</td>\n",
       "      <td>loss</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>19</td>\n",
       "      <td>45871887</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>ERCC2</td>\n",
       "      <td>Het</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>Xeroderma pigmentosum complementation group D</td>\n",
       "      <td>NM_000400.4:c.360+1G&gt;A</td>\n",
       "      <td>5</td>\n",
       "      <td>Splice Site</td>\n",
       "      <td>loss</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>17</td>\n",
       "      <td>29482996</td>\n",
       "      <td>TTCAGCTTCCAATA</td>\n",
       "      <td>T</td>\n",
       "      <td>NF1</td>\n",
       "      <td>Het</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>Neurofibromatosis 1</td>\n",
       "      <td>NM_000267.3:c.61-4_69delTCAGCTTCCAATA</td>\n",
       "      <td>1</td>\n",
       "      <td>Intron</td>\n",
       "      <td>normal</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>13</td>\n",
       "      <td>48881547</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>RB1</td>\n",
       "      <td>Het</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Familial retinoblastoma</td>\n",
       "      <td>NM_000321.3:c.264+5G&gt;A</td>\n",
       "      <td>2</td>\n",
       "      <td>Intron</td>\n",
       "      <td>normal</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>13</td>\n",
       "      <td>48953712</td>\n",
       "      <td>CTTTTT</td>\n",
       "      <td>C</td>\n",
       "      <td>RB1</td>\n",
       "      <td>Het</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Familial retinoblastoma</td>\n",
       "      <td>NM_000321.3:c.1333-17_1333-13delTTTTT</td>\n",
       "      <td>13</td>\n",
       "      <td>Intron</td>\n",
       "      <td>normal</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>8</td>\n",
       "      <td>145738768</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>RECQL4</td>\n",
       "      <td>Het</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Baller-gerold syndrome</td>\n",
       "      <td>ENST00000617875.6:c.2296+1C&gt;A</td>\n",
       "      <td>14</td>\n",
       "      <td>Splice Site</td>\n",
       "      <td>normal</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>1</td>\n",
       "      <td>17345454</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SDHB</td>\n",
       "      <td>Het</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>Hereditary paraganglioma-pheochromocytoma synd...</td>\n",
       "      <td>NM_003000.3:c.766-1G&gt;C</td>\n",
       "      <td>8</td>\n",
       "      <td>Splice Site</td>\n",
       "      <td>loss</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>9</td>\n",
       "      <td>135779206</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>TSC1</td>\n",
       "      <td>Het</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>Tuberous sclerosis complex</td>\n",
       "      <td>NM_000368.5:c.2042-2A&gt;G</td>\n",
       "      <td>17</td>\n",
       "      <td>Splice Site</td>\n",
       "      <td>loss</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>16</td>\n",
       "      <td>2121936</td>\n",
       "      <td>G</td>\n",
       "      <td>GT</td>\n",
       "      <td>TSC2</td>\n",
       "      <td>Het</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Tuberous sclerosis complex</td>\n",
       "      <td>NM_000548.5:c.2097+2dupT</td>\n",
       "      <td>19</td>\n",
       "      <td>Splice Site</td>\n",
       "      <td>loss</td>\n",
       "      <td>Insertion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>16</td>\n",
       "      <td>2138446</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>TSC2</td>\n",
       "      <td>Het</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Tuberous sclerosis complex</td>\n",
       "      <td>NM_000548.5:c.5260-1G&gt;C</td>\n",
       "      <td>42</td>\n",
       "      <td>Splice Site</td>\n",
       "      <td>loss</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>16</td>\n",
       "      <td>2137924</td>\n",
       "      <td>TCCCTGCAGTGCAGGAAAGGTAGGGCCGGGTGGGG</td>\n",
       "      <td>T</td>\n",
       "      <td>TSC2</td>\n",
       "      <td>Het</td>\n",
       "      <td>Likely Benign</td>\n",
       "      <td>Tuberous sclerosis complex</td>\n",
       "      <td>NM_000548.5:c.5068+27_5069-47delCAGGAAAGGTAGGG...</td>\n",
       "      <td>39</td>\n",
       "      <td>Intron</td>\n",
       "      <td>loss</td>\n",
       "      <td>Deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>9</td>\n",
       "      <td>100437871</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>XPA</td>\n",
       "      <td>Het</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>Xeroderma pigmentosum</td>\n",
       "      <td>NM_000380.4:c.674-2A&gt;G</td>\n",
       "      <td>6</td>\n",
       "      <td>Splice Site</td>\n",
       "      <td>loss</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>3</td>\n",
       "      <td>14212051</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>XPC</td>\n",
       "      <td>Het</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>Xeroderma pigmentosum</td>\n",
       "      <td>NM_004628.5:c.300-1G&gt;C</td>\n",
       "      <td>3</td>\n",
       "      <td>Splice Site</td>\n",
       "      <td>loss</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>3</td>\n",
       "      <td>14206927</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>XPC</td>\n",
       "      <td>Het</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>Xeroderma pigmentosum</td>\n",
       "      <td>NM_004628.5:c.779+1G&gt;T</td>\n",
       "      <td>6</td>\n",
       "      <td>Splice Site</td>\n",
       "      <td>loss</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>3</td>\n",
       "      <td>14190057</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>XPC</td>\n",
       "      <td>Het</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Xeroderma pigmentosum</td>\n",
       "      <td>NM_004628.5:c.2420+5G&gt;A</td>\n",
       "      <td>13</td>\n",
       "      <td>Intron</td>\n",
       "      <td>loss</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>14</td>\n",
       "      <td>105243112</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>Het</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Cowden syndrome</td>\n",
       "      <td>NM_005163.2:c.176-5C&gt;A</td>\n",
       "      <td>3</td>\n",
       "      <td>Intron</td>\n",
       "      <td>normal</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>11</td>\n",
       "      <td>44255793</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>EXT2</td>\n",
       "      <td>Het</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Multiple hereditary exostoses</td>\n",
       "      <td>NM_207122.2:c.1935G&gt;A</td>\n",
       "      <td>12</td>\n",
       "      <td>Exon</td>\n",
       "      <td>loss</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>2</td>\n",
       "      <td>48033789</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>MSH6</td>\n",
       "      <td>Het</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Lynch Syndrome</td>\n",
       "      <td>NM_000179.3:c.4000C&gt;T</td>\n",
       "      <td>9</td>\n",
       "      <td>Exon</td>\n",
       "      <td>loss</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>3</td>\n",
       "      <td>52443725</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>PHF7</td>\n",
       "      <td>Het</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Cancers and Tumors</td>\n",
       "      <td>NM_016483.7:c.-1608C&gt;G</td>\n",
       "      <td>0</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>loss</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>19</td>\n",
       "      <td>50918678</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>POLD1</td>\n",
       "      <td>Het</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Familial colorectal cancer</td>\n",
       "      <td>NM_001256849.1:c.2565-17G&gt;A</td>\n",
       "      <td>20</td>\n",
       "      <td>Intron</td>\n",
       "      <td>normal</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>12</td>\n",
       "      <td>133254296</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>POLE</td>\n",
       "      <td>Het</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Susceptibility to colorectal cancer type 12</td>\n",
       "      <td>NM_006231.4:c.588G&gt;T</td>\n",
       "      <td>7</td>\n",
       "      <td>Exon</td>\n",
       "      <td>normal</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>17</td>\n",
       "      <td>56809887</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>RAD51C</td>\n",
       "      <td>Het</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Hereditary breast and/or ovarian cancer</td>\n",
       "      <td>NM_058216.3:c.1008A&gt;G</td>\n",
       "      <td>8</td>\n",
       "      <td>Exon</td>\n",
       "      <td>normal</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>17</td>\n",
       "      <td>74470455</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>RHBDF2</td>\n",
       "      <td>Het</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Tylosis with esophageal cancer syndrome</td>\n",
       "      <td>NM_001005498.4:c.1464G&gt;A</td>\n",
       "      <td>12</td>\n",
       "      <td>Exon</td>\n",
       "      <td>normal</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>17</td>\n",
       "      <td>74469327</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>RHBDF2</td>\n",
       "      <td>Het</td>\n",
       "      <td>Likely Benign</td>\n",
       "      <td>Tylosis with esophageal cancer syndrome</td>\n",
       "      <td>NM_001005498.4:c.1809+7C&gt;T</td>\n",
       "      <td>16</td>\n",
       "      <td>Intron</td>\n",
       "      <td>normal</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>10</td>\n",
       "      <td>104389820</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>SUFU</td>\n",
       "      <td>Het</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Basal cell nevus syndrome</td>\n",
       "      <td>NM_016169.4:c.1366-3C&gt;A</td>\n",
       "      <td>11</td>\n",
       "      <td>Intron</td>\n",
       "      <td>normal</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>22</td>\n",
       "      <td>24175755</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SMARCB1</td>\n",
       "      <td>Het</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Coffin-Siris syndrome</td>\n",
       "      <td>NM_003073.5:c.987-4G&gt;A</td>\n",
       "      <td>7</td>\n",
       "      <td>Intron</td>\n",
       "      <td>normal</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>11</td>\n",
       "      <td>108202604</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>ATM</td>\n",
       "      <td>Het</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Ataxia-telangiectasia</td>\n",
       "      <td>NM_000051.4:c.7630-2A&gt;C</td>\n",
       "      <td>52</td>\n",
       "      <td>Splice Site</td>\n",
       "      <td>loss</td>\n",
       "      <td>SNV</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   #CHROM        POS                                  REF ALT     Gene  \\\n",
       "0       3   52439929                                    C   A     BAP1   \n",
       "1       3   52440267                                    A   G     BAP1   \n",
       "2      11  108186639                                    G   A      ATM   \n",
       "3      11  108224490                                    T   G      ATM   \n",
       "4       3   52443569                                    C   T     BAP1   \n",
       "5      17   41234419                                    A   C    BRCA1   \n",
       "6      17   41222939                                    A   G    BRCA1   \n",
       "7      17   59934594                                    T   C    BRIP1   \n",
       "8      16   68849662                                   CG   C     CDH1   \n",
       "9      22   29121230                                    C   T    CHEK2   \n",
       "10      1   45797760                                    T   C    MUTYH   \n",
       "11      1   45797835                                    T   G    MUTYH   \n",
       "12      7   66459197                                    A   G     SBDS   \n",
       "13     16    3650977                                    T  TA     SLX4   \n",
       "14     17    7579699                                    C   T     TP53   \n",
       "15     17    7578370                                    C   T     TP53   \n",
       "16     19   45871887                                    C   T    ERCC2   \n",
       "17     17   29482996                       TTCAGCTTCCAATA   T      NF1   \n",
       "18     13   48881547                                    G   A      RB1   \n",
       "19     13   48953712                               CTTTTT   C      RB1   \n",
       "20      8  145738768                                    G   T   RECQL4   \n",
       "21      1   17345454                                    C   G     SDHB   \n",
       "22      9  135779206                                    T   C     TSC1   \n",
       "23     16    2121936                                    G  GT     TSC2   \n",
       "24     16    2138446                                    G   C     TSC2   \n",
       "25     16    2137924  TCCCTGCAGTGCAGGAAAGGTAGGGCCGGGTGGGG   T     TSC2   \n",
       "26      9  100437871                                    T   C      XPA   \n",
       "27      3   14212051                                    C   G      XPC   \n",
       "28      3   14206927                                    C   A      XPC   \n",
       "29      3   14190057                                    C   T      XPC   \n",
       "30     14  105243112                                    G   T     AKT1   \n",
       "31     11   44255793                                    G   A     EXT2   \n",
       "32      2   48033789                                    C   T     MSH6   \n",
       "33      3   52443725                                    C   G     PHF7   \n",
       "34     19   50918678                                    G   A    POLD1   \n",
       "35     12  133254296                                    C   A     POLE   \n",
       "36     17   56809887                                    A   G   RAD51C   \n",
       "37     17   74470455                                    C   T   RHBDF2   \n",
       "38     17   74469327                                    G   A   RHBDF2   \n",
       "39     10  104389820                                    C   A     SUFU   \n",
       "40     22   24175755                                    G   A  SMARCB1   \n",
       "41     11  108202604                                    A   C      ATM   \n",
       "\n",
       "   Genotype              Assessment  \\\n",
       "0       Het              Pathogenic   \n",
       "1       Het       Likely Pathogenic   \n",
       "2       Het              Pathogenic   \n",
       "3       Het  Uncertain Significance   \n",
       "4       Het              Pathogenic   \n",
       "5       Het              Pathogenic   \n",
       "6       Het              Pathogenic   \n",
       "7       Het       Likely Pathogenic   \n",
       "8       Het              Pathogenic   \n",
       "9       Het              Pathogenic   \n",
       "10      Het  Uncertain Significance   \n",
       "11      Het              Pathogenic   \n",
       "12      Het              Pathogenic   \n",
       "13      Het              Pathogenic   \n",
       "14      Het       Likely Pathogenic   \n",
       "15      Het              Pathogenic   \n",
       "16      Het       Likely Pathogenic   \n",
       "17      Het       Likely Pathogenic   \n",
       "18      Het  Uncertain Significance   \n",
       "19      Het  Uncertain Significance   \n",
       "20      Het  Uncertain Significance   \n",
       "21      Het       Likely Pathogenic   \n",
       "22      Het       Likely Pathogenic   \n",
       "23      Het  Uncertain Significance   \n",
       "24      Het  Uncertain Significance   \n",
       "25      Het           Likely Benign   \n",
       "26      Het       Likely Pathogenic   \n",
       "27      Het       Likely Pathogenic   \n",
       "28      Het       Likely Pathogenic   \n",
       "29      Het  Uncertain Significance   \n",
       "30      Het  Uncertain Significance   \n",
       "31      Het  Uncertain Significance   \n",
       "32      Het  Uncertain Significance   \n",
       "33      Het  Uncertain Significance   \n",
       "34      Het  Uncertain Significance   \n",
       "35      Het  Uncertain Significance   \n",
       "36      Het  Uncertain Significance   \n",
       "37      Het  Uncertain Significance   \n",
       "38      Het           Likely Benign   \n",
       "39      Het  Uncertain Significance   \n",
       "40      Het  Uncertain Significance   \n",
       "41      Het              Pathogenic   \n",
       "\n",
       "                                            Phenotype  \\\n",
       "0                       Tumor predisposition syndrome   \n",
       "1                       Tumor predisposition syndrome   \n",
       "2                               Ataxia-telangiectasia   \n",
       "3                               Ataxia-telangiectasia   \n",
       "4                       Tumor predisposition syndrome   \n",
       "5             Hereditary breast and/or ovarian cancer   \n",
       "6             Hereditary breast and/or ovarian cancer   \n",
       "7             Hereditary breast and/or ovarian cancer   \n",
       "8          Hereditary diffuse malignant gastric tumor   \n",
       "9                            Hereditary breast cancer   \n",
       "10                           MYH-associated polyposis   \n",
       "11                           MYH-associated polyposis   \n",
       "12                                 Shwachman syndrome   \n",
       "13                                     Fanconi anemia   \n",
       "14                               Li-Fraumeni syndrome   \n",
       "15                               Li-Fraumeni syndrome   \n",
       "16      Xeroderma pigmentosum complementation group D   \n",
       "17                                Neurofibromatosis 1   \n",
       "18                            Familial retinoblastoma   \n",
       "19                            Familial retinoblastoma   \n",
       "20                             Baller-gerold syndrome   \n",
       "21  Hereditary paraganglioma-pheochromocytoma synd...   \n",
       "22                         Tuberous sclerosis complex   \n",
       "23                         Tuberous sclerosis complex   \n",
       "24                         Tuberous sclerosis complex   \n",
       "25                         Tuberous sclerosis complex   \n",
       "26                              Xeroderma pigmentosum   \n",
       "27                              Xeroderma pigmentosum   \n",
       "28                              Xeroderma pigmentosum   \n",
       "29                              Xeroderma pigmentosum   \n",
       "30                                    Cowden syndrome   \n",
       "31                      Multiple hereditary exostoses   \n",
       "32                                     Lynch Syndrome   \n",
       "33                                 Cancers and Tumors   \n",
       "34                         Familial colorectal cancer   \n",
       "35        Susceptibility to colorectal cancer type 12   \n",
       "36            Hereditary breast and/or ovarian cancer   \n",
       "37            Tylosis with esophageal cancer syndrome   \n",
       "38            Tylosis with esophageal cancer syndrome   \n",
       "39                          Basal cell nevus syndrome   \n",
       "40                              Coffin-Siris syndrome   \n",
       "41                              Ataxia-telangiectasia   \n",
       "\n",
       "                                           Transcript Exon_number  \\\n",
       "0                              NM_004656.4:c.784-1G>T          10   \n",
       "1                              NM_004656.4:c.783+2T>C           9   \n",
       "2                             NM_000051.4:c.6095+1G>A          41   \n",
       "3                             NM_000051.4:c.8672-3T>G          59   \n",
       "4                              NM_004656.4:c.122+1G>A           3   \n",
       "5                             NM_007294.4:c.4357+2T>G          12   \n",
       "6                             NM_007294.4:c.4986+6T>C          15   \n",
       "7                              NM_032043.3:c.206-2A>G           4   \n",
       "8                         NM_001317184.2:c.1382+1delG           9   \n",
       "9                           NM_001005735.2:c.573+1G>A           4   \n",
       "10                          NM_001048171.2:c.850-2A>G          11   \n",
       "11                          NM_001048171.2:c.849+3A>C          10   \n",
       "12                             NM_016038.4:c.258+2T>C           2   \n",
       "13                           NM_032444.4:c.1163+2dupT           5   \n",
       "14                              NM_000546.6:c.96+1G>A           3   \n",
       "15                             NM_000546.6:c.559+1G>A           5   \n",
       "16                             NM_000400.4:c.360+1G>A           5   \n",
       "17              NM_000267.3:c.61-4_69delTCAGCTTCCAATA           1   \n",
       "18                             NM_000321.3:c.264+5G>A           2   \n",
       "19              NM_000321.3:c.1333-17_1333-13delTTTTT          13   \n",
       "20                      ENST00000617875.6:c.2296+1C>A          14   \n",
       "21                             NM_003000.3:c.766-1G>C           8   \n",
       "22                            NM_000368.5:c.2042-2A>G          17   \n",
       "23                           NM_000548.5:c.2097+2dupT          19   \n",
       "24                            NM_000548.5:c.5260-1G>C          42   \n",
       "25  NM_000548.5:c.5068+27_5069-47delCAGGAAAGGTAGGG...          39   \n",
       "26                             NM_000380.4:c.674-2A>G           6   \n",
       "27                             NM_004628.5:c.300-1G>C           3   \n",
       "28                             NM_004628.5:c.779+1G>T           6   \n",
       "29                            NM_004628.5:c.2420+5G>A          13   \n",
       "30                             NM_005163.2:c.176-5C>A           3   \n",
       "31                              NM_207122.2:c.1935G>A          12   \n",
       "32                              NM_000179.3:c.4000C>T           9   \n",
       "33                             NM_016483.7:c.-1608C>G           0   \n",
       "34                        NM_001256849.1:c.2565-17G>A          20   \n",
       "35                               NM_006231.4:c.588G>T           7   \n",
       "36                              NM_058216.3:c.1008A>G           8   \n",
       "37                           NM_001005498.4:c.1464G>A          12   \n",
       "38                         NM_001005498.4:c.1809+7C>T          16   \n",
       "39                            NM_016169.4:c.1366-3C>A          11   \n",
       "40                             NM_003073.5:c.987-4G>A           7   \n",
       "41                            NM_000051.4:c.7630-2A>C          52   \n",
       "\n",
       "         Region Function  Variation  \n",
       "0   Splice Site     loss        SNV  \n",
       "1   Splice Site     loss        SNV  \n",
       "2   Splice Site     loss        SNV  \n",
       "3        Intron     loss        SNV  \n",
       "4   Splice Site     loss        SNV  \n",
       "5   Splice Site     loss        SNV  \n",
       "6        Intron     loss        SNV  \n",
       "7   Splice Site     loss        SNV  \n",
       "8   Splice Site     loss   Deletion  \n",
       "9   Splice Site     loss        SNV  \n",
       "10  Splice Site     loss        SNV  \n",
       "11       Intron     loss        SNV  \n",
       "12  Splice Site     loss        SNV  \n",
       "13  Splice Site   normal  Insertion  \n",
       "14  Splice Site     loss        SNV  \n",
       "15  Splice Site     loss        SNV  \n",
       "16  Splice Site     loss        SNV  \n",
       "17       Intron   normal   Deletion  \n",
       "18       Intron   normal        SNV  \n",
       "19       Intron   normal   Deletion  \n",
       "20  Splice Site   normal        SNV  \n",
       "21  Splice Site     loss        SNV  \n",
       "22  Splice Site     loss        SNV  \n",
       "23  Splice Site     loss  Insertion  \n",
       "24  Splice Site     loss        SNV  \n",
       "25       Intron     loss   Deletion  \n",
       "26  Splice Site     loss        SNV  \n",
       "27  Splice Site     loss        SNV  \n",
       "28  Splice Site     loss        SNV  \n",
       "29       Intron     loss        SNV  \n",
       "30       Intron   normal        SNV  \n",
       "31         Exon     loss        SNV  \n",
       "32         Exon     loss        SNV  \n",
       "33     Promoter     loss        SNV  \n",
       "34       Intron   normal        SNV  \n",
       "35         Exon   normal        SNV  \n",
       "36         Exon   normal        SNV  \n",
       "37         Exon   normal        SNV  \n",
       "38       Intron   normal        SNV  \n",
       "39       Intron   normal        SNV  \n",
       "40       Intron   normal        SNV  \n",
       "41  Splice Site     loss        SNV  "
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "qci_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "d46ad668",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'#CHROM': '3',\n",
       "  'POS': '52439929',\n",
       "  'REF': 'C',\n",
       "  'ALT': 'A',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Pathogenic',\n",
       "  'phenotype': 'Tumor predisposition syndrome'},\n",
       " {'#CHROM': '3',\n",
       "  'POS': '52440267',\n",
       "  'REF': 'A',\n",
       "  'ALT': 'G',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Likely Pathogenic',\n",
       "  'phenotype': 'Tumor predisposition syndrome'},\n",
       " {'#CHROM': '11',\n",
       "  'POS': '108186639',\n",
       "  'REF': 'G',\n",
       "  'ALT': 'A',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Pathogenic',\n",
       "  'phenotype': 'Ataxia-telangiectasia'},\n",
       " {'#CHROM': '11',\n",
       "  'POS': '108224490',\n",
       "  'REF': 'T',\n",
       "  'ALT': 'G',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Ataxia-telangiectasia'},\n",
       " {'#CHROM': '3',\n",
       "  'POS': '52443569',\n",
       "  'REF': 'C',\n",
       "  'ALT': 'T',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Pathogenic',\n",
       "  'phenotype': 'Tumor predisposition syndrome'},\n",
       " {'#CHROM': '17',\n",
       "  'POS': '41234419',\n",
       "  'REF': 'A',\n",
       "  'ALT': 'C',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Pathogenic',\n",
       "  'phenotype': 'Hereditary breast and/or ovarian cancer'},\n",
       " {'#CHROM': '17',\n",
       "  'POS': '41222939',\n",
       "  'REF': 'A',\n",
       "  'ALT': 'G',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Pathogenic',\n",
       "  'phenotype': 'Hereditary breast and/or ovarian cancer'},\n",
       " {'#CHROM': '17',\n",
       "  'POS': '59934594',\n",
       "  'REF': 'T',\n",
       "  'ALT': 'C',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Likely Pathogenic',\n",
       "  'phenotype': 'Hereditary breast and/or ovarian cancer'},\n",
       " {'#CHROM': '16',\n",
       "  'POS': '68849662',\n",
       "  'REF': 'CG',\n",
       "  'ALT': 'C',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Pathogenic',\n",
       "  'phenotype': 'Hereditary diffuse malignant gastric tumor'},\n",
       " {'#CHROM': '22',\n",
       "  'POS': '29121230',\n",
       "  'REF': 'C',\n",
       "  'ALT': 'T',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Pathogenic',\n",
       "  'phenotype': 'Hereditary breast cancer'},\n",
       " {'#CHROM': '1',\n",
       "  'POS': '45797760',\n",
       "  'REF': 'T',\n",
       "  'ALT': 'C',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'MYH-associated polyposis'},\n",
       " {'#CHROM': '1',\n",
       "  'POS': '45797835',\n",
       "  'REF': 'T',\n",
       "  'ALT': 'G',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Pathogenic',\n",
       "  'phenotype': 'MYH-associated polyposis'},\n",
       " {'#CHROM': '7',\n",
       "  'POS': '66459197',\n",
       "  'REF': 'A',\n",
       "  'ALT': 'G',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Pathogenic',\n",
       "  'phenotype': 'Shwachman syndrome'},\n",
       " {'#CHROM': '16',\n",
       "  'POS': '3650977',\n",
       "  'REF': 'T',\n",
       "  'ALT': 'TA',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Pathogenic',\n",
       "  'phenotype': 'Fanconi anemia'},\n",
       " {'#CHROM': '17',\n",
       "  'POS': '7579699',\n",
       "  'REF': 'C',\n",
       "  'ALT': 'T',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Likely Pathogenic',\n",
       "  'phenotype': 'Li-Fraumeni syndrome'},\n",
       " {'#CHROM': '17',\n",
       "  'POS': '7578370',\n",
       "  'REF': 'C',\n",
       "  'ALT': 'T',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Pathogenic',\n",
       "  'phenotype': 'Li-Fraumeni syndrome'},\n",
       " {'#CHROM': '19',\n",
       "  'POS': '45871887',\n",
       "  'REF': 'C',\n",
       "  'ALT': 'T',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Likely Pathogenic',\n",
       "  'phenotype': 'Xeroderma pigmentosum complementation group D'},\n",
       " {'#CHROM': '17',\n",
       "  'POS': '29482996',\n",
       "  'REF': 'TTCAGCTTCCAATA',\n",
       "  'ALT': 'T',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Likely Pathogenic',\n",
       "  'phenotype': 'Neurofibromatosis 1'},\n",
       " {'#CHROM': '13',\n",
       "  'POS': '48881547',\n",
       "  'REF': 'G',\n",
       "  'ALT': 'A',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Familial retinoblastoma'},\n",
       " {'#CHROM': '13',\n",
       "  'POS': '48953712',\n",
       "  'REF': 'CTTTTT',\n",
       "  'ALT': 'C',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Familial retinoblastoma'},\n",
       " {'#CHROM': '8',\n",
       "  'POS': '145738768',\n",
       "  'REF': 'G',\n",
       "  'ALT': 'T',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Baller-gerold syndrome'},\n",
       " {'#CHROM': '1',\n",
       "  'POS': '17345454',\n",
       "  'REF': 'C',\n",
       "  'ALT': 'G',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Likely Pathogenic',\n",
       "  'phenotype': 'Hereditary paraganglioma-pheochromocytoma syndrome'},\n",
       " {'#CHROM': '9',\n",
       "  'POS': '135779206',\n",
       "  'REF': 'T',\n",
       "  'ALT': 'C',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Likely Pathogenic',\n",
       "  'phenotype': 'Tuberous sclerosis complex'},\n",
       " {'#CHROM': '16',\n",
       "  'POS': '2121936',\n",
       "  'REF': 'G',\n",
       "  'ALT': 'GT',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Tuberous sclerosis complex'},\n",
       " {'#CHROM': '16',\n",
       "  'POS': '2138446',\n",
       "  'REF': 'G',\n",
       "  'ALT': 'C',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Tuberous sclerosis complex'},\n",
       " {'#CHROM': '16',\n",
       "  'POS': '2137924',\n",
       "  'REF': 'TCCCTGCAGTGCAGGAAAGGTAGGGCCGGGTGGGG',\n",
       "  'ALT': 'T',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Likely Benign',\n",
       "  'phenotype': 'Tuberous sclerosis complex'},\n",
       " {'#CHROM': '9',\n",
       "  'POS': '100437871',\n",
       "  'REF': 'T',\n",
       "  'ALT': 'C',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Likely Pathogenic',\n",
       "  'phenotype': 'Xeroderma pigmentosum'},\n",
       " {'#CHROM': '3',\n",
       "  'POS': '14212051',\n",
       "  'REF': 'C',\n",
       "  'ALT': 'G',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Likely Pathogenic',\n",
       "  'phenotype': 'Xeroderma pigmentosum'},\n",
       " {'#CHROM': '3',\n",
       "  'POS': '14206927',\n",
       "  'REF': 'C',\n",
       "  'ALT': 'A',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Likely Pathogenic',\n",
       "  'phenotype': 'Xeroderma pigmentosum'},\n",
       " {'#CHROM': '3',\n",
       "  'POS': '14190057',\n",
       "  'REF': 'C',\n",
       "  'ALT': 'T',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Xeroderma pigmentosum'},\n",
       " {'#CHROM': '14',\n",
       "  'POS': '105243112',\n",
       "  'REF': 'G',\n",
       "  'ALT': 'T',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Cowden syndrome'},\n",
       " {'#CHROM': '11',\n",
       "  'POS': '44255793',\n",
       "  'REF': 'G',\n",
       "  'ALT': 'A',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Multiple hereditary exostoses'},\n",
       " {'#CHROM': '2',\n",
       "  'POS': '48033789',\n",
       "  'REF': 'C',\n",
       "  'ALT': 'T',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Lynch Syndrome'},\n",
       " {'#CHROM': '3',\n",
       "  'POS': '52443725',\n",
       "  'REF': 'C',\n",
       "  'ALT': 'G',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Cancers and Tumors'},\n",
       " {'#CHROM': '19',\n",
       "  'POS': '50918678',\n",
       "  'REF': 'G',\n",
       "  'ALT': 'A',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Familial colorectal cancer'},\n",
       " {'#CHROM': '12',\n",
       "  'POS': '133254296',\n",
       "  'REF': 'C',\n",
       "  'ALT': 'A',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Susceptibility to colorectal cancer type 12'},\n",
       " {'#CHROM': '17',\n",
       "  'POS': '56809887',\n",
       "  'REF': 'A',\n",
       "  'ALT': 'G',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Hereditary breast and/or ovarian cancer'},\n",
       " {'#CHROM': '17',\n",
       "  'POS': '74470455',\n",
       "  'REF': 'C',\n",
       "  'ALT': 'T',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Tylosis with esophageal cancer syndrome'},\n",
       " {'#CHROM': '17',\n",
       "  'POS': '74469327',\n",
       "  'REF': 'G',\n",
       "  'ALT': 'A',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Likely Benign',\n",
       "  'phenotype': 'Tylosis with esophageal cancer syndrome'},\n",
       " {'#CHROM': '10',\n",
       "  'POS': '104389820',\n",
       "  'REF': 'C',\n",
       "  'ALT': 'A',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Basal cell nevus syndrome'},\n",
       " {'#CHROM': '22',\n",
       "  'POS': '24175755',\n",
       "  'REF': 'G',\n",
       "  'ALT': 'A',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Uncertain Significance',\n",
       "  'phenotype': 'Coffin-Siris syndrome'},\n",
       " {'#CHROM': '11',\n",
       "  'POS': '108202604',\n",
       "  'REF': 'A',\n",
       "  'ALT': 'C',\n",
       "  'genotype': 'Het',\n",
       "  'computed_assessment': 'Pathogenic',\n",
       "  'phenotype': 'Ataxia-telangiectasia'}]"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "14a768ef",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "3d0b5006",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from xml.dom import minidom\n",
    "xmldoc = minidom.parse('01_xlsx_files/response_1637323300036.xml')\n",
    "itemlist = xmldoc.getElementsByTagName('variant')\n",
    "itemlist.count()\n",
    "#for s in itemlist:\n",
    "    #print(s.attr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e27e6800",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fe42727d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c2bf0453",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "74553500",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "867aadd8",
   "metadata": {},
   "outputs": [
    {
     "ename": "FeatureNotFound",
     "evalue": "Couldn't find a tree builder with the features you requested: xml. Do you need to install a parser library?",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFeatureNotFound\u001b[0m                           Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-47-5624624253de>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      2\u001b[0m     \u001b[0mqci\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 4\u001b[0;31m \u001b[0mBs_data\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mbs\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mqci\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'xml'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      5\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0;31m# Finding all instances of tag\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.9/site-packages/bs4/__init__.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, markup, features, builder, parse_only, from_encoding, exclude_encodings, element_classes, **kwargs)\u001b[0m\n\u001b[1;32m    243\u001b[0m             \u001b[0mbuilder_class\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mbuilder_registry\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlookup\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0mfeatures\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    244\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mbuilder_class\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 245\u001b[0;31m                 raise FeatureNotFound(\n\u001b[0m\u001b[1;32m    246\u001b[0m                     \u001b[0;34m\"Couldn't find a tree builder with the features you \"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    247\u001b[0m                     \u001b[0;34m\"requested: %s. Do you need to install a parser library?\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mFeatureNotFound\u001b[0m: Couldn't find a tree builder with the features you requested: xml. Do you need to install a parser library?"
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "Bs_data = bs(qci, 'xml')\n",
    " \n",
    "# Finding all instances of tag\n",
    "# `unique`\n",
    "Bs_data.find_all('unique')\n",
    " \n",
    "# Using find() to extract attributes\n",
    "# of the first instance of the tag\n",
    "#b_name = Bs_data.find('child', {'name':'Frank'})\n",
    " \n",
    "# Extracting the data stored in a\n",
    "# specific attribute of the\n",
    "# `child` tag\n",
    "#value = b_name.get('test')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "444ccdaa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>consensus_sequence_IGV_region_left</th>\n",
       "      <th>consensus_sequence_IGV_sequence_left</th>\n",
       "      <th>consensus_sequence_IGV_region_right</th>\n",
       "      <th>consensus_sequence_IGV_sequence_right</th>\n",
       "      <th>exon_skipping</th>\n",
       "      <th>intron_retention</th>\n",
       "      <th>...</th>\n",
       "      <th>note_hboc</th>\n",
       "      <th>add_note_hboc</th>\n",
       "      <th>splice_pred_alamut_hboc</th>\n",
       "      <th>prediction_hboc</th>\n",
       "      <th>literature_hboc</th>\n",
       "      <th>evidence_level_lit_hboc</th>\n",
       "      <th>comment_hboc</th>\n",
       "      <th>recomm_action_hboc</th>\n",
       "      <th>shortcut</th>\n",
       "      <th>strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>10384956</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr1:10384914-10384953</td>\n",
       "      <td>CCTTCTATGATCGGTTCCACTGGTTCAAACTTGTGGGGAG</td>\n",
       "      <td>chr1:10394578-10394617</td>\n",
       "      <td>GGCATTTGTTTACCTGAGCAATCTGCTGTATCCCGTGCCC</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1_10384956_A_G</td>\n",
       "      <td>plus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>10394695</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1_10394695_A_C</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>10399821</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1_10399821_A_T</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>11740670</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1_11740670_C_A</td>\n",
       "      <td>minus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>15767087</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1_15767087_G_C</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1458</th>\n",
       "      <td>X</td>\n",
       "      <td>47045028</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X_47045028_A_G</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1459</th>\n",
       "      <td>X</td>\n",
       "      <td>53246977</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X_53246977_C_T</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1460</th>\n",
       "      <td>X</td>\n",
       "      <td>70349523</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X_70349523_A_G</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1461</th>\n",
       "      <td>X</td>\n",
       "      <td>147019617</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X_147019617_G_A</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1462</th>\n",
       "      <td>X</td>\n",
       "      <td>147019617</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X_147019617_G_A</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1463 rows × 162 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #CHROM        POS REF ALT consensus_sequence_IGV_region_left  \\\n",
       "0         1   10384956   A   G             chr1:10384914-10384953   \n",
       "1         1   10394695   A   C                                NaN   \n",
       "2         1   10399821   A   T                                NaN   \n",
       "3         1   11740670   C   A                                NaN   \n",
       "4         1   15767087   G   C                                NaN   \n",
       "...     ...        ...  ..  ..                                ...   \n",
       "1458      X   47045028   A   G                                NaN   \n",
       "1459      X   53246977   C   T                                NaN   \n",
       "1460      X   70349523   A   G                                NaN   \n",
       "1461      X  147019617   G   A                                NaN   \n",
       "1462      X  147019617   G   A                                NaN   \n",
       "\n",
       "          consensus_sequence_IGV_sequence_left  \\\n",
       "0     CCTTCTATGATCGGTTCCACTGGTTCAAACTTGTGGGGAG   \n",
       "1                                          NaN   \n",
       "2                                          NaN   \n",
       "3                                          NaN   \n",
       "4                                          NaN   \n",
       "...                                        ...   \n",
       "1458                                       NaN   \n",
       "1459                                       NaN   \n",
       "1460                                       NaN   \n",
       "1461                                       NaN   \n",
       "1462                                       NaN   \n",
       "\n",
       "     consensus_sequence_IGV_region_right  \\\n",
       "0                 chr1:10394578-10394617   \n",
       "1                                    NaN   \n",
       "2                                    NaN   \n",
       "3                                    NaN   \n",
       "4                                    NaN   \n",
       "...                                  ...   \n",
       "1458                                 NaN   \n",
       "1459                                 NaN   \n",
       "1460                                 NaN   \n",
       "1461                                 NaN   \n",
       "1462                                 NaN   \n",
       "\n",
       "         consensus_sequence_IGV_sequence_right  exon_skipping  \\\n",
       "0     GGCATTTGTTTACCTGAGCAATCTGCTGTATCCCGTGCCC           True   \n",
       "1                                          NaN          False   \n",
       "2                                          NaN          False   \n",
       "3                                          NaN           True   \n",
       "4                                          NaN          False   \n",
       "...                                        ...            ...   \n",
       "1458                                       NaN          False   \n",
       "1459                                       NaN          False   \n",
       "1460                                       NaN          False   \n",
       "1461                                       NaN          False   \n",
       "1462                                       NaN          False   \n",
       "\n",
       "      intron_retention  ...  note_hboc  add_note_hboc  \\\n",
       "0                False  ...        NaN            NaN   \n",
       "1                False  ...        NaN            NaN   \n",
       "2                False  ...        NaN            NaN   \n",
       "3                 True  ...        NaN            NaN   \n",
       "4                False  ...        NaN            NaN   \n",
       "...                ...  ...        ...            ...   \n",
       "1458             False  ...        NaN            NaN   \n",
       "1459              True  ...        NaN            NaN   \n",
       "1460             False  ...        NaN            NaN   \n",
       "1461             False  ...        NaN            NaN   \n",
       "1462             False  ...        NaN            NaN   \n",
       "\n",
       "      splice_pred_alamut_hboc  prediction_hboc literature_hboc  \\\n",
       "0                         NaN              NaN             NaN   \n",
       "1                         NaN              NaN             NaN   \n",
       "2                         NaN              NaN             NaN   \n",
       "3                         NaN              NaN             NaN   \n",
       "4                         NaN              NaN             NaN   \n",
       "...                       ...              ...             ...   \n",
       "1458                      NaN              NaN             NaN   \n",
       "1459                      NaN              NaN             NaN   \n",
       "1460                      NaN              NaN             NaN   \n",
       "1461                      NaN              NaN             NaN   \n",
       "1462                      NaN              NaN             NaN   \n",
       "\n",
       "     evidence_level_lit_hboc comment_hboc recomm_action_hboc         shortcut  \\\n",
       "0                        NaN          NaN                NaN   1_10384956_A_G   \n",
       "1                        NaN          NaN                NaN   1_10394695_A_C   \n",
       "2                        NaN          NaN                NaN   1_10399821_A_T   \n",
       "3                        NaN          NaN                NaN   1_11740670_C_A   \n",
       "4                        NaN          NaN                NaN   1_15767087_G_C   \n",
       "...                      ...          ...                ...              ...   \n",
       "1458                     NaN          NaN                NaN   X_47045028_A_G   \n",
       "1459                     NaN          NaN                NaN   X_53246977_C_T   \n",
       "1460                     NaN          NaN                NaN   X_70349523_A_G   \n",
       "1461                     NaN          NaN                NaN  X_147019617_G_A   \n",
       "1462                     NaN          NaN                NaN  X_147019617_G_A   \n",
       "\n",
       "     strand  \n",
       "0      plus  \n",
       "1       NaN  \n",
       "2       NaN  \n",
       "3     minus  \n",
       "4       NaN  \n",
       "...     ...  \n",
       "1458    NaN  \n",
       "1459    NaN  \n",
       "1460    NaN  \n",
       "1461    NaN  \n",
       "1462    NaN  \n",
       "\n",
       "[1463 rows x 162 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "igv_all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9734116b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "ca6a1f2a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>category</th>\n",
       "      <th>sufficient, yes</th>\n",
       "      <th>sufficient, maybe</th>\n",
       "      <th>sufficient, no</th>\n",
       "      <th>not sufficient, yes</th>\n",
       "      <th>not sufficient, maybe</th>\n",
       "      <th>not sufficient, no</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>spliceAI &gt; 0.9</td>\n",
       "      <td>50</td>\n",
       "      <td>12</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ada, rf score &gt; 0.9</td>\n",
       "      <td>51</td>\n",
       "      <td>17</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MES high disruption</td>\n",
       "      <td>63</td>\n",
       "      <td>33</td>\n",
       "      <td>77</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MES high new ss</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>55</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>pos. rel. to ss &lt; 3, intron</td>\n",
       "      <td>37</td>\n",
       "      <td>14</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      category  sufficient, yes  sufficient, maybe  \\\n",
       "0               spliceAI > 0.9               50                 12   \n",
       "1          ada, rf score > 0.9               51                 17   \n",
       "2          MES high disruption               63                 33   \n",
       "3              MES high new ss                1                  1   \n",
       "4  pos. rel. to ss < 3, intron               37                 14   \n",
       "\n",
       "   sufficient, no  not sufficient, yes  not sufficient, maybe  \\\n",
       "0              19                    0                     14   \n",
       "1              23                    0                     19   \n",
       "2              77                    0                     22   \n",
       "3              55                    0                      0   \n",
       "4              19                    0                     11   \n",
       "\n",
       "   not sufficient, no  \n",
       "0                  21  \n",
       "1                  35  \n",
       "2                  55  \n",
       "3                  15  \n",
       "4                  25  "
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "categ_vars(igv_wo_dup)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "679d40f6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>shortcut</th>\n",
       "      <th>HUGO_Symbol</th>\n",
       "      <th>SpliceAI_pred_DP_AG</th>\n",
       "      <th>SpliceAI_pred_DP_AL</th>\n",
       "      <th>SpliceAI_pred_DP_DG</th>\n",
       "      <th>SpliceAI_pred_DP_DL</th>\n",
       "      <th>SpliceAI_pred_DS_AG</th>\n",
       "      <th>SpliceAI_pred_DS_AL</th>\n",
       "      <th>SpliceAI_pred_DS_DG</th>\n",
       "      <th>SpliceAI_pred_DS_DL</th>\n",
       "      <th>SpliceAI_pred_SYMBOL</th>\n",
       "      <th>ClinVar_Pathogenicity</th>\n",
       "      <th>ClinVar_Traits</th>\n",
       "      <th>PosExonRefSeqAccession</th>\n",
       "      <th>PosExon_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>349</th>\n",
       "      <td>1_243736225_TAC_T</td>\n",
       "      <td>AKT3</td>\n",
       "      <td>-14</td>\n",
       "      <td>3</td>\n",
       "      <td>-14</td>\n",
       "      <td>3</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.64</td>\n",
       "      <td>1.00</td>\n",
       "      <td>AKT3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>outsideDonorSite</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>384</th>\n",
       "      <td>2_47641558_GTA_G</td>\n",
       "      <td>MSH2</td>\n",
       "      <td>0</td>\n",
       "      <td>-26</td>\n",
       "      <td>-25</td>\n",
       "      <td>-1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>MSH2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>outsideDonorSite</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>415</th>\n",
       "      <td>2_225378349_TGCGCCTCTGTC_T</td>\n",
       "      <td>CUL3</td>\n",
       "      <td>-35</td>\n",
       "      <td>6</td>\n",
       "      <td>-6</td>\n",
       "      <td>6</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.94</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>CUL3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>insideAcceptorSite</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432</th>\n",
       "      <td>3_37053354_G_A</td>\n",
       "      <td>MLH1</td>\n",
       "      <td>50</td>\n",
       "      <td>-43</td>\n",
       "      <td>-44</td>\n",
       "      <td>-1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.99</td>\n",
       "      <td>MLH1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>outsideDonorSite</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>590</th>\n",
       "      <td>5_79968062_G_T</td>\n",
       "      <td>MSH3</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>15</td>\n",
       "      <td>0.97</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>MSH3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>600</th>\n",
       "      <td>5_80064822_GGTAAT_G</td>\n",
       "      <td>MSH3</td>\n",
       "      <td>0</td>\n",
       "      <td>-13</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.92</td>\n",
       "      <td>MSH3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>insideDonorSite</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>654</th>\n",
       "      <td>7_95820419_A_C</td>\n",
       "      <td>SLC25A13</td>\n",
       "      <td>-28</td>\n",
       "      <td>50</td>\n",
       "      <td>-28</td>\n",
       "      <td>2</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.99</td>\n",
       "      <td>SLC25A13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>outsideDonorSite</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>657</th>\n",
       "      <td>7_105190699_T_G</td>\n",
       "      <td>RINT1</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>-49</td>\n",
       "      <td>49</td>\n",
       "      <td>0.45</td>\n",
       "      <td>0.94</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>RINT1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>723</th>\n",
       "      <td>9_432158_T_A</td>\n",
       "      <td>DOCK8</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>27</td>\n",
       "      <td>-39</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.94</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>DOCK8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>765</th>\n",
       "      <td>9_137650127_C_T</td>\n",
       "      <td>COL5A1</td>\n",
       "      <td>15</td>\n",
       "      <td>-38</td>\n",
       "      <td>-2</td>\n",
       "      <td>15</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.07</td>\n",
       "      <td>COL5A1</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Ehlers-Danlos syndrome, classic type</td>\n",
       "      <td>15</td>\n",
       "      <td>insideDonorSite</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>859</th>\n",
       "      <td>11_95546754_G_C</td>\n",
       "      <td>CEP57</td>\n",
       "      <td>3</td>\n",
       "      <td>-1</td>\n",
       "      <td>3</td>\n",
       "      <td>-1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.05</td>\n",
       "      <td>1.00</td>\n",
       "      <td>CEP57</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>outsideDonorSite</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960</th>\n",
       "      <td>13_20763744_T_G</td>\n",
       "      <td>GJB2</td>\n",
       "      <td>36</td>\n",
       "      <td>-2</td>\n",
       "      <td>36</td>\n",
       "      <td>-16</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.91</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>GJB2</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>Deafness, autosomal recessive 1A;not specified...</td>\n",
       "      <td>24</td>\n",
       "      <td>5utr_startcodon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>973</th>\n",
       "      <td>13_48936949_A_G</td>\n",
       "      <td>RB1</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>49</td>\n",
       "      <td>0.60</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>RB1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>993</th>\n",
       "      <td>14_68331714_TTTTA_T</td>\n",
       "      <td>RAD51B</td>\n",
       "      <td>13</td>\n",
       "      <td>6</td>\n",
       "      <td>22</td>\n",
       "      <td>13</td>\n",
       "      <td>0.63</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>RAD51B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>996</th>\n",
       "      <td>14_68331716_TTAGA_T</td>\n",
       "      <td>RAD51B</td>\n",
       "      <td>10</td>\n",
       "      <td>3</td>\n",
       "      <td>19</td>\n",
       "      <td>10</td>\n",
       "      <td>0.63</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>RAD51B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1116</th>\n",
       "      <td>16_3808046_AGTCCTT_A</td>\n",
       "      <td>CREBBP</td>\n",
       "      <td>-18</td>\n",
       "      <td>3</td>\n",
       "      <td>-13</td>\n",
       "      <td>-18</td>\n",
       "      <td>0.97</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>CREBBP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>insideAcceptorSite</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1444</th>\n",
       "      <td>22_29083975_CTGAAAATAAA_C</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>-12</td>\n",
       "      <td>-1</td>\n",
       "      <td>47</td>\n",
       "      <td>1</td>\n",
       "      <td>0.57</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1456</th>\n",
       "      <td>X_44949966_A_T</td>\n",
       "      <td>KDM6A</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>-39</td>\n",
       "      <td>29</td>\n",
       "      <td>0.99</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>KDM6A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1457</th>\n",
       "      <td>X_44949967_G_T</td>\n",
       "      <td>KDM6A</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>-40</td>\n",
       "      <td>28</td>\n",
       "      <td>0.99</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>KDM6A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        shortcut HUGO_Symbol SpliceAI_pred_DP_AG  \\\n",
       "349            1_243736225_TAC_T        AKT3                 -14   \n",
       "384             2_47641558_GTA_G        MSH2                   0   \n",
       "415   2_225378349_TGCGCCTCTGTC_T        CUL3                 -35   \n",
       "432               3_37053354_G_A        MLH1                  50   \n",
       "590               5_79968062_G_T        MSH3                   8   \n",
       "600          5_80064822_GGTAAT_G        MSH3                   0   \n",
       "654               7_95820419_A_C    SLC25A13                 -28   \n",
       "657              7_105190699_T_G       RINT1                   1   \n",
       "723                 9_432158_T_A       DOCK8                   2   \n",
       "765              9_137650127_C_T      COL5A1                  15   \n",
       "859              11_95546754_G_C       CEP57                   3   \n",
       "960              13_20763744_T_G        GJB2                  36   \n",
       "973              13_48936949_A_G         RB1                   8   \n",
       "993          14_68331714_TTTTA_T      RAD51B                  13   \n",
       "996          14_68331716_TTAGA_T      RAD51B                  10   \n",
       "1116        16_3808046_AGTCCTT_A      CREBBP                 -18   \n",
       "1444   22_29083975_CTGAAAATAAA_C       CHEK2                 -12   \n",
       "1456              X_44949966_A_T       KDM6A                  10   \n",
       "1457              X_44949967_G_T       KDM6A                   9   \n",
       "\n",
       "     SpliceAI_pred_DP_AL SpliceAI_pred_DP_DG SpliceAI_pred_DP_DL  \\\n",
       "349                    3                 -14                   3   \n",
       "384                  -26                 -25                  -1   \n",
       "415                    6                  -6                   6   \n",
       "432                  -43                 -44                  -1   \n",
       "590                    1                   8                  15   \n",
       "600                  -13                   8                   0   \n",
       "654                   50                 -28                   2   \n",
       "657                    9                 -49                  49   \n",
       "723                    8                  27                 -39   \n",
       "765                  -38                  -2                  15   \n",
       "859                   -1                   3                  -1   \n",
       "960                   -2                  36                 -16   \n",
       "973                    2                   8                  49   \n",
       "993                    6                  22                  13   \n",
       "996                    3                  19                  10   \n",
       "1116                   3                 -13                 -18   \n",
       "1444                  -1                  47                   1   \n",
       "1456                   2                 -39                  29   \n",
       "1457                   1                 -40                  28   \n",
       "\n",
       "      SpliceAI_pred_DS_AG  SpliceAI_pred_DS_AL  SpliceAI_pred_DS_DG  \\\n",
       "349                  0.00                 0.00                 0.64   \n",
       "384                  0.00                 0.00                 0.00   \n",
       "415                  0.31                 0.94                 0.00   \n",
       "432                  0.00                 0.00                 0.00   \n",
       "590                  0.97                 1.00                 0.00   \n",
       "600                  0.00                 0.00                 0.00   \n",
       "654                  0.00                 0.00                 0.00   \n",
       "657                  0.45                 0.94                 0.00   \n",
       "723                  1.00                 0.94                 0.00   \n",
       "765                  0.00                 0.00                 0.93   \n",
       "859                  0.00                 0.00                 0.05   \n",
       "960                  0.23                 0.91                 0.00   \n",
       "973                  0.60                 0.99                 0.00   \n",
       "993                  0.63                 0.99                 0.00   \n",
       "996                  0.63                 0.99                 0.00   \n",
       "1116                 0.97                 1.00                 0.00   \n",
       "1444                 0.57                 0.98                 0.00   \n",
       "1456                 0.99                 1.00                 0.00   \n",
       "1457                 0.99                 1.00                 0.00   \n",
       "\n",
       "      SpliceAI_pred_DS_DL SpliceAI_pred_SYMBOL   ClinVar_Pathogenicity  \\\n",
       "349                  1.00                 AKT3                     NaN   \n",
       "384                  1.00                 MSH2                     NaN   \n",
       "415                  0.00                 CUL3                     NaN   \n",
       "432                  0.99                 MLH1                     NaN   \n",
       "590                  0.00                 MSH3                     NaN   \n",
       "600                  0.92                 MSH3                     NaN   \n",
       "654                  0.99             SLC25A13                     NaN   \n",
       "657                  0.00                RINT1                     NaN   \n",
       "723                  0.00                DOCK8                     NaN   \n",
       "765                  0.07               COL5A1  Uncertain Significance   \n",
       "859                  1.00                CEP57                     NaN   \n",
       "960                  0.00                 GJB2  Uncertain Significance   \n",
       "973                  0.00                  RB1                     NaN   \n",
       "993                  0.00               RAD51B                     NaN   \n",
       "996                  0.00               RAD51B                     NaN   \n",
       "1116                 0.00               CREBBP                     NaN   \n",
       "1444                 0.00                CHEK2                     NaN   \n",
       "1456                 0.00                KDM6A                     NaN   \n",
       "1457                 0.00                KDM6A                     NaN   \n",
       "\n",
       "                                         ClinVar_Traits  \\\n",
       "349                                                 NaN   \n",
       "384                                                 NaN   \n",
       "415                                                 NaN   \n",
       "432                                                 NaN   \n",
       "590                                                 NaN   \n",
       "600                                                 NaN   \n",
       "654                                                 NaN   \n",
       "657                                                 NaN   \n",
       "723                                                 NaN   \n",
       "765                Ehlers-Danlos syndrome, classic type   \n",
       "859                                                 NaN   \n",
       "960   Deafness, autosomal recessive 1A;not specified...   \n",
       "973                                                 NaN   \n",
       "993                                                 NaN   \n",
       "996                                                 NaN   \n",
       "1116                                                NaN   \n",
       "1444                                                NaN   \n",
       "1456                                                NaN   \n",
       "1457                                                NaN   \n",
       "\n",
       "      PosExonRefSeqAccession         PosExon_type  \n",
       "349                        3     outsideDonorSite  \n",
       "384                        1     outsideDonorSite  \n",
       "415                        6   insideAcceptorSite  \n",
       "432                        1     outsideDonorSite  \n",
       "590                        1  outsideAcceptorSite  \n",
       "600                        0      insideDonorSite  \n",
       "654                        2     outsideDonorSite  \n",
       "657                        9  outsideAcceptorSite  \n",
       "723                        8  outsideAcceptorSite  \n",
       "765                       15      insideDonorSite  \n",
       "859                        1     outsideDonorSite  \n",
       "960                       24      5utr_startcodon  \n",
       "973                        2  outsideAcceptorSite  \n",
       "993                        6  outsideAcceptorSite  \n",
       "996                        4  outsideAcceptorSite  \n",
       "1116                       3   insideAcceptorSite  \n",
       "1444                       1  outsideAcceptorSite  \n",
       "1456                       2  outsideAcceptorSite  \n",
       "1457                       1  outsideAcceptorSite  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "igv_wo_dup[['shortcut', 'HUGO_Symbol']+search_col(igv_wo_dup, 'spliceai')+\n",
    "           search_col(igv_wo_dup, 'clinvar')+search_col(igv_wo_dup, 'posexon')][\n",
    "    ((igv_wo_dup['SpliceAI_pred_DS_AG']>0.9)|\n",
    "                   (igv_wo_dup['SpliceAI_pred_DS_AL']>0.9)|\n",
    "                   (igv_wo_dup['SpliceAI_pred_DS_DG']>0.9)|\n",
    "                   (igv_wo_dup['SpliceAI_pred_DS_DL']>0.9))&\n",
    "                  (igv_wo_dup['too_low_coverage']=='no')&\n",
    "                  (igv_wo_dup['splice_change']=='no')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "401f03d5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "796baeee",
   "metadata": {},
   "source": [
    "### Mutational hotspots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "7ad82267",
   "metadata": {},
   "outputs": [],
   "source": [
    "onk_merge = pd.read_excel('01_xlsx_files/2021-10-27_onkostar_merged.xlsx')\n",
    "onk_merge = right_type(onk_merge)\n",
    "pd.merge(onk_merge, all_vars[cols_cpra+['PID_trans', 'Control_VAF', 'Tumor_VAF', 'Control_dpALT', \n",
    "                                        'Control_dp', 'Tumor_dpALT', 'Tumor_dp', 'AF_RNA', 'AF_RNA_noSkipReads']], \n",
    "         on=['#CHROM', 'POS', 'REF', 'ALT', 'PID_trans'], how='left'\n",
    "        ).to_excel('02_output_analysis/2021-10-27_onkostar_vafs.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "8c5338fb",
   "metadata": {},
   "outputs": [],
   "source": [
    "hotspots = pd.read_excel('01_xlsx_files/hotspots_v2.xls')\n",
    "hotspots = hotspots[hotspots['Hugo_Symbol'].isin(onk_merge['HUGO_Symbol'].unique())].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "426e259f",
   "metadata": {},
   "outputs": [],
   "source": [
    "tp = list(hotspots['Samples'][(hotspots['Hugo_Symbol']=='TP53')&(hotspots['Genomic_Position'].str.startswith('17:7578370'))])[0].split(':')\n",
    "tp_num = [t.split('|') for t in tp]\n",
    "tp_num = sum([int(n) for sl in tp_num for n in sl if n.isdigit()])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0ff02462",
   "metadata": {},
   "source": [
    "### variant infos merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 417,
   "id": "198ea71c",
   "metadata": {},
   "outputs": [],
   "source": [
    "vars1 = pd.read_excel('01_xlsx_files/vars_splice.xlsx')\n",
    "\n",
    "vars2 = pd.read_excel('01_xlsx_files/new_vars_analysis_genes.xlsx').rename(columns={'shortcut':'shortcut_v'})\n",
    "vars2[['sc0', 'sc1', 'sc2', 'sc3', 'sc4']] = vars2['shortcut_v'].str.split('_', expand=True)\n",
    "vars2['shortcut'] = vars2['sc0']+'_'+vars2['sc1']+'_'+vars2['sc2']+'_'+vars2['sc3']\n",
    "vars2 = vars2.drop(columns=['sc0', 'sc1', 'sc2', 'sc3', 'sc4'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 436,
   "id": "c0572a77",
   "metadata": {},
   "outputs": [],
   "source": [
    "var_merge = pd.merge(vars1.drop(columns=['PIDs', 'kind', 'pos_last', 'pos_next']), vars2, \n",
    "                     on=['#CHROM','shortcut'], how='outer')\n",
    "var_merge[['#CHROM', 'ALT', 'REF']] = var_merge[['#CHROM', 'ALT', 'REF']].astype(str)\n",
    "var_merge['POS'] = var_merge['POS'].astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 470,
   "id": "7f324977",
   "metadata": {},
   "outputs": [],
   "source": [
    "kge_hboc_wodup = kge_hboc[cols_cpra].drop_duplicates().reset_index(drop=True)\n",
    "kge_hboc_wodup['ID_Person'] = [list(df['ID Person']) for i,df in kge_hboc.groupby(cols_cpra)]\n",
    "kge_hboc_wodup['KGE_HBOC'] = [list(df['from']) for i,df in kge_hboc.groupby(cols_cpra)]\n",
    "kge_hboc_vars = pd.merge(var_merge, kge_hboc_wodup, on=cols_cpra, how='outer')\n",
    "\n",
    "kge_hboc_vars.to_excel('02_output_analysis/vars_facts_together.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f1559220",
   "metadata": {},
   "source": [
    "### 06.10."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "18c1ee29",
   "metadata": {},
   "outputs": [],
   "source": [
    "igv_all = merge_xlsx('2021-08-19_all_vars_analyzed.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "2ebcfc6d",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>shortcut</th>\n",
       "      <th>HUGO_Symbol</th>\n",
       "      <th>max_gnomAD_AF</th>\n",
       "      <th>splice_change</th>\n",
       "      <th>ClinVar_Pathogenicity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1_23219372_G_A</td>\n",
       "      <td>EPHB2</td>\n",
       "      <td>0.00021103</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>1_27688743_T_C</td>\n",
       "      <td>MAP3K6</td>\n",
       "      <td>0.0037306</td>\n",
       "      <td>yes</td>\n",
       "      <td>Likely Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>1_45797760_T_C</td>\n",
       "      <td>MUTYH</td>\n",
       "      <td>0.0011775</td>\n",
       "      <td>yes</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>1_45797835_T_G</td>\n",
       "      <td>MUTYH</td>\n",
       "      <td>6.7534e-05</td>\n",
       "      <td>yes</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>380</th>\n",
       "      <td>2_47601174_C_T</td>\n",
       "      <td>EPCAM</td>\n",
       "      <td>4.2257e-06</td>\n",
       "      <td>yes</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>381</th>\n",
       "      <td>2_47607108_G_A</td>\n",
       "      <td>EPCAM</td>\n",
       "      <td>0.0021976</td>\n",
       "      <td>yes</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>382</th>\n",
       "      <td>2_47612302_C_G</td>\n",
       "      <td>EPCAM</td>\n",
       "      <td>0.00028683</td>\n",
       "      <td>yes</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>392</th>\n",
       "      <td>2_48033789_C_T</td>\n",
       "      <td>MSH6</td>\n",
       "      <td>6.3853e-05</td>\n",
       "      <td>yes</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>423</th>\n",
       "      <td>3_14190057_C_T</td>\n",
       "      <td>XPC</td>\n",
       "      <td>8.5326e-06</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425</th>\n",
       "      <td>3_14206927_C_A</td>\n",
       "      <td>XPC</td>\n",
       "      <td>0</td>\n",
       "      <td>yes</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>465</th>\n",
       "      <td>3_52440267_A_G</td>\n",
       "      <td>BAP1</td>\n",
       "      <td>4.222e-06</td>\n",
       "      <td>yes</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>470</th>\n",
       "      <td>3_52443725_C_G</td>\n",
       "      <td>BAP1</td>\n",
       "      <td>2.1114e-05</td>\n",
       "      <td>yes</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>637</th>\n",
       "      <td>7_66459197_A_G</td>\n",
       "      <td>SBDS</td>\n",
       "      <td>0.003971</td>\n",
       "      <td>yes</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>667</th>\n",
       "      <td>8_30916058_A_G</td>\n",
       "      <td>WRN</td>\n",
       "      <td>0.0035021</td>\n",
       "      <td>yes</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>717</th>\n",
       "      <td>8_145738768_G_T</td>\n",
       "      <td>RECQL4</td>\n",
       "      <td>9.5554e-05</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>747</th>\n",
       "      <td>9_22006246_C_G</td>\n",
       "      <td>CDKN2B</td>\n",
       "      <td>9.1e-06</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>757</th>\n",
       "      <td>9_100437871_T_C</td>\n",
       "      <td>XPA</td>\n",
       "      <td>4.2304e-06</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>811</th>\n",
       "      <td>10_99220447_C_T</td>\n",
       "      <td>MMS19</td>\n",
       "      <td>4.9383e-06</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>832</th>\n",
       "      <td>11_44255793_G_A</td>\n",
       "      <td>EXT2</td>\n",
       "      <td>2.1171e-05</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>879</th>\n",
       "      <td>11_108202604_A_C</td>\n",
       "      <td>ATM</td>\n",
       "      <td>3.1867e-05</td>\n",
       "      <td>yes</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>934</th>\n",
       "      <td>12_21624358_ATACT_A</td>\n",
       "      <td>RECQL</td>\n",
       "      <td>0.00035045</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1000</th>\n",
       "      <td>14_75498886_T_C</td>\n",
       "      <td>MLH3</td>\n",
       "      <td>6.3694e-05</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1030</th>\n",
       "      <td>14_105243112_G_T</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>2.1313e-05</td>\n",
       "      <td>yes</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1097</th>\n",
       "      <td>16_2137924_TCCCTGCAGTGCAGGAAAGGTAGGGCCGGGTGGGG_T</td>\n",
       "      <td>TSC2</td>\n",
       "      <td>0.0022738</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1107</th>\n",
       "      <td>16_2138446_G_C</td>\n",
       "      <td>TSC2</td>\n",
       "      <td>3.1871e-05</td>\n",
       "      <td>yes</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1138</th>\n",
       "      <td>17_7578370_C_T</td>\n",
       "      <td>TP53</td>\n",
       "      <td>0</td>\n",
       "      <td>yes</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1180</th>\n",
       "      <td>17_59934594_T_C</td>\n",
       "      <td>BRIP1</td>\n",
       "      <td>4.2344e-06</td>\n",
       "      <td>yes</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1184</th>\n",
       "      <td>17_74469327_G_A</td>\n",
       "      <td>RHBDF2</td>\n",
       "      <td>0.0036629</td>\n",
       "      <td>yes</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1216</th>\n",
       "      <td>17_74470455_C_T</td>\n",
       "      <td>RHBDF2</td>\n",
       "      <td>1.2732e-05</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1397</th>\n",
       "      <td>19_40744001_G_C</td>\n",
       "      <td>AKT2</td>\n",
       "      <td>0.00015921</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1408</th>\n",
       "      <td>19_50918678_G_A</td>\n",
       "      <td>POLD1</td>\n",
       "      <td>0.0016256</td>\n",
       "      <td>yes</td>\n",
       "      <td>Likely Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1442</th>\n",
       "      <td>22_24175755_G_A</td>\n",
       "      <td>SMARCB1</td>\n",
       "      <td>4.2206e-05</td>\n",
       "      <td>yes</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1446</th>\n",
       "      <td>22_29121230_C_T</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>0.0002229</td>\n",
       "      <td>yes</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              shortcut HUGO_Symbol  \\\n",
       "7                                       1_23219372_G_A       EPHB2   \n",
       "20                                      1_27688743_T_C      MAP3K6   \n",
       "79                                      1_45797760_T_C       MUTYH   \n",
       "80                                      1_45797835_T_G       MUTYH   \n",
       "380                                     2_47601174_C_T       EPCAM   \n",
       "381                                     2_47607108_G_A       EPCAM   \n",
       "382                                     2_47612302_C_G       EPCAM   \n",
       "392                                     2_48033789_C_T        MSH6   \n",
       "423                                     3_14190057_C_T         XPC   \n",
       "425                                     3_14206927_C_A         XPC   \n",
       "465                                     3_52440267_A_G        BAP1   \n",
       "470                                     3_52443725_C_G        BAP1   \n",
       "637                                     7_66459197_A_G        SBDS   \n",
       "667                                     8_30916058_A_G         WRN   \n",
       "717                                    8_145738768_G_T      RECQL4   \n",
       "747                                     9_22006246_C_G      CDKN2B   \n",
       "757                                    9_100437871_T_C         XPA   \n",
       "811                                    10_99220447_C_T       MMS19   \n",
       "832                                    11_44255793_G_A        EXT2   \n",
       "879                                   11_108202604_A_C         ATM   \n",
       "934                                12_21624358_ATACT_A       RECQL   \n",
       "1000                                   14_75498886_T_C        MLH3   \n",
       "1030                                  14_105243112_G_T        AKT1   \n",
       "1097  16_2137924_TCCCTGCAGTGCAGGAAAGGTAGGGCCGGGTGGGG_T        TSC2   \n",
       "1107                                    16_2138446_G_C        TSC2   \n",
       "1138                                    17_7578370_C_T        TP53   \n",
       "1180                                   17_59934594_T_C       BRIP1   \n",
       "1184                                   17_74469327_G_A      RHBDF2   \n",
       "1216                                   17_74470455_C_T      RHBDF2   \n",
       "1397                                   19_40744001_G_C        AKT2   \n",
       "1408                                   19_50918678_G_A       POLD1   \n",
       "1442                                   22_24175755_G_A     SMARCB1   \n",
       "1446                                   22_29121230_C_T       CHEK2   \n",
       "\n",
       "     max_gnomAD_AF splice_change   ClinVar_Pathogenicity  \n",
       "7       0.00021103           yes                     NaN  \n",
       "20       0.0037306           yes           Likely Benign  \n",
       "79       0.0011775           yes  Uncertain Significance  \n",
       "80      6.7534e-05           yes  Uncertain Significance  \n",
       "380     4.2257e-06           yes              Pathogenic  \n",
       "381      0.0021976           yes                  Benign  \n",
       "382     0.00028683           yes  Uncertain Significance  \n",
       "392     6.3853e-05           yes  Uncertain Significance  \n",
       "423     8.5326e-06           yes                     NaN  \n",
       "425              0           yes       Likely Pathogenic  \n",
       "465      4.222e-06           yes       Likely Pathogenic  \n",
       "470     2.1114e-05           yes  Uncertain Significance  \n",
       "637       0.003971           yes              Pathogenic  \n",
       "667      0.0035021           yes  Uncertain Significance  \n",
       "717     9.5554e-05           yes                     NaN  \n",
       "747        9.1e-06           yes                     NaN  \n",
       "757     4.2304e-06           yes                     NaN  \n",
       "811     4.9383e-06           yes                     NaN  \n",
       "832     2.1171e-05           yes                     NaN  \n",
       "879     3.1867e-05           yes              Pathogenic  \n",
       "934     0.00035045           yes                     NaN  \n",
       "1000    6.3694e-05           yes                     NaN  \n",
       "1030    2.1313e-05           yes  Uncertain Significance  \n",
       "1097     0.0022738           yes                     NaN  \n",
       "1107    3.1871e-05           yes  Uncertain Significance  \n",
       "1138             0           yes              Pathogenic  \n",
       "1180    4.2344e-06           yes       Likely Pathogenic  \n",
       "1184     0.0036629           yes                  Benign  \n",
       "1216    1.2732e-05           yes                     NaN  \n",
       "1397    0.00015921           yes                     NaN  \n",
       "1408     0.0016256           yes           Likely Benign  \n",
       "1442    4.2206e-05           yes  Uncertain Significance  \n",
       "1446     0.0002229           yes              Pathogenic  "
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "igv_all[['shortcut', 'HUGO_Symbol', 'max_gnomAD_AF', \n",
    "         'splice_change', 'ClinVar_Pathogenicity']][(igv_all['max_gnomAD_AF'].notnull())&(igv_all['splice_change']=='yes')&\n",
    "                           (igv_all['Splice Project Gene Priority']!='Cold (387-ACMG-MASTER)')&\n",
    "                           (~igv_all['Splice Project Gene Priority'].isin(gus))].drop_duplicates('shortcut')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "a24eb054",
   "metadata": {},
   "outputs": [],
   "source": [
    "hboc_all = pd.read_csv('01_xlsx_files/2021-10-06_HBOC_Gepado_PML_all_variants.csv', sep=';')\n",
    "hboc_all = hboc_all.dropna(axis=0, subset=['Chr', 'POS', 'REF', 'ALT']).reset_index(drop=True)\n",
    "kge_all = pd.read_csv('01_xlsx_files/2021-10-06_KGE_Gepado_PML_all_variants.csv', sep=';')\n",
    "kge_all = kge_all.dropna(axis=0, subset=['Chr', 'POS', 'REF', 'ALT']).reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "0c2bab9f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pysam as ps\n",
    "# reference sequence fasta\n",
    "ref_fasta = ps.FastaFile('/mnt/g27prist/CMTD/Stephan/bcbio_installation/genomes/Hsapiens/GRCh37/seq/GRCh37.fa')\n",
    "def DataCleaning(df):\n",
    "    dfn = df.copy()\n",
    "    # for deletion\n",
    "    \n",
    "    # position = position - 1\n",
    "    df_del = dfn[(dfn['ALT']=='-')&(dfn['REF']!='-')].copy()\n",
    "    df_del['POS'] = df_del['POS'].str.replace(' - ', '..').str.replace('-', '..')\n",
    "    df_del['newPOS'] = df_del['POS'].str.split('.', expand=True)[0].astype(int)-1\n",
    "\n",
    "    new_ref_del_l = []\n",
    "    new_alt_del_l = []\n",
    "    for d in range(len(df_del)):\n",
    "        # position of nucleotide of first element of dataframe in which 'Alternate' not filled out\n",
    "        pos_flossies = df_del.iloc[d]['newPOS']\n",
    "        # reference nucleotide at specific position\n",
    "        ref_refseq = ref_fasta.fetch(reference=str(df_del.iloc[d]['Chr']), start=pos_flossies-1, end=pos_flossies)\n",
    "        # for alternative nucleotide reference nucleotide\n",
    "        new_alt_del_l.append(ref_refseq)\n",
    "        # reference nucleotide of first element of dataframe in which 'Alternate' not filled out\n",
    "        ref_flossies = df_del.iloc[d]['REF']\n",
    "        # new reference nucleotides for vcf annotation\n",
    "        new_ref_del = ref_refseq + ref_flossies\n",
    "        new_ref_del_l.append(new_ref_del)\n",
    "    # add new columns\n",
    "    df_del['REF_n'] = new_ref_del_l\n",
    "    df_del['ALT_n'] = new_alt_del_l\n",
    "    df_del.drop(columns = ['REF', 'ALT', 'POS'], inplace = True)\n",
    "    df_del.rename(columns = {'REF_n':'REF', 'ALT_n':'ALT', 'newPOS':'POS', 'Chr':'#CHROM'}, inplace = True)\n",
    "    \n",
    "    # for insertion\n",
    "    # position = position, because something was inserted at this position\n",
    "    df_ins = dfn[(dfn['REF']=='-')].copy()\n",
    "    df_ins['POS'] = df_ins['POS'].str.replace('-', '^')\n",
    "    df_ins['newPOS'] = df_ins['POS'].str.split('^', expand=True)[0].astype(int)\n",
    "\n",
    "    new_alt_ins_l = []\n",
    "    new_ref_ins_l = []\n",
    "    for i in range(len(df_ins)):\n",
    "        # position of nucleotide of first element of dataframe in which 'Reference' missing\n",
    "        pos_flossies = df_ins.iloc[i]['newPOS']\n",
    "        # reference nucleotide at specific position\n",
    "        ref_refseq = ref_fasta.fetch(reference=str(df_ins.iloc[i]['Chr']), start=pos_flossies-1, end=pos_flossies)\n",
    "        # for reference nucleotide reference nucleotide\n",
    "        new_ref_ins_l.append(ref_refseq)\n",
    "        # reference nucleotide of first element of dataframe in which 'Reference' missing\n",
    "        alt_flossies = df_ins.iloc[i]['ALT']\n",
    "        # new alternate nucleotides for vcf annotation\n",
    "        new_alt_ins = ref_refseq + alt_flossies\n",
    "        new_alt_ins_l.append(new_alt_ins)\n",
    "    # add new columns\n",
    "    df_ins['ALT_n'] = new_alt_ins_l\n",
    "    df_ins['REF_n'] = new_ref_ins_l\n",
    "    df_ins.drop(columns = ['REF', 'ALT', 'POS'], inplace = True)\n",
    "    df_ins.rename(columns = {'REF_n':'REF', 'ALT_n':'ALT', 'newPOS':'POS', 'Chr':'#CHROM'}, inplace = True)\n",
    "    \n",
    "    df_snv = dfn[(dfn['ALT']!='-')&(dfn['REF']!='-')].copy()\n",
    "    df_snv['POS'] = df_snv['POS'].str.split('.', expand=True)[0]\n",
    "    df_snv = df_snv.rename(columns={'Chr':'#CHROM'})\n",
    "\n",
    "    dfn_new = df_snv.append([df_del, df_ins])\n",
    "    dfn_new = dfn_new.sort_index()\n",
    "    return dfn_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "ad052934",
   "metadata": {},
   "outputs": [],
   "source": [
    "kge_all_n = DataCleaning(kge_all).drop(index=[3630, 5103])\n",
    "hboc_all_n = DataCleaning(hboc_all)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "34d4db1b",
   "metadata": {},
   "outputs": [],
   "source": [
    "hboc_all_n[['#CHROM', 'ALT', 'REF']] = hboc_all_n[['#CHROM', 'ALT', 'REF']].astype(str)\n",
    "hboc_all_n['POS'] = hboc_all_n['POS'].astype(int)\n",
    "kge_all_n[['#CHROM', 'ALT', 'REF']] = kge_all_n[['#CHROM', 'ALT', 'REF']].astype(str)\n",
    "kge_all_n['POS'] = kge_all_n['POS'].astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "2d55bbed",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "kge_merge = pd.merge(kge_all_n, igv_all[cols_cpra+['splice_change', 'Splice Project Gene Priority']], how='inner', \n",
    "         on=cols_cpra).drop_duplicates(cols_cpra+['ID Person'])\n",
    "kge_merge = kge_merge[(kge_merge['Splice Project Gene Priority']!='Cold (387-ACMG-MASTER)')&\n",
    "                      (~kge_merge['Gen'].isin(gus))&(kge_merge['splice_change']=='yes')]\n",
    "kge_merge['from'] = 'KGE'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "33987a17",
   "metadata": {},
   "outputs": [],
   "source": [
    "hboc_merge = pd.merge(hboc_all_n, igv_all[cols_cpra+['splice_change', 'Splice Project Gene Priority']], how='inner', \n",
    "         on=cols_cpra).drop_duplicates(cols_cpra+['ID Person'])\n",
    "hboc_merge = hboc_merge[(hboc_merge['Splice Project Gene Priority']!='Cold (387-ACMG-MASTER)')&\n",
    "                      (~hboc_merge['Gen'].isin(gus))&(hboc_merge['splice_change']=='yes')]\n",
    "hboc_merge['from'] = 'HBOC'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "209ef078",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "kge_hboc = kge_merge.append(hboc_merge, ignore_index=True)\n",
    "kge_hboc['#CHROM'] = pd.Categorical(kge_hboc['#CHROM'],\n",
    "                                    categories=['1','2','3','4','5','6','7','8','9','10','11','12', '13', '14', \n",
    "                                                '15', '16', '17', '18', '19', '20', '21', '22', 'X', 'Y'],\n",
    "                                    ordered=True)\n",
    "kge_hboc = kge_hboc.sort_values(by=['#CHROM', 'POS', 'REF', 'ALT'], ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 333,
   "id": "9e15d327",
   "metadata": {},
   "outputs": [],
   "source": [
    "kge_hboc[cols_cpra+['Gen', 'ID Person', 'Analytik ID', 'Bewertung gesamt', 'Clin Var', 'from']\n",
    "        ].to_excel('02_output_analysis/kge_hboc_spl_vars.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4757d184",
   "metadata": {},
   "source": [
    "### 01.09."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e81f1053",
   "metadata": {},
   "outputs": [],
   "source": [
    "igv_all = merge_xlsx('2021-08-19_all_vars_analyzed.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "d8136bce",
   "metadata": {},
   "outputs": [],
   "source": [
    "newn = igv_all[cols_cpra+['HUGO_Symbol','PID_trans']][\n",
    "    (igv_all['Splice Project Gene Priority']!='Cold (387-ACMG-MASTER)')&\n",
    "    (~igv_all['HUGO_Symbol'].isin(gus))&\n",
    "    (igv_all['splice_change']=='yes')].copy().reset_index(drop=True)\n",
    "newn = newn.drop(index=[17,18,19,20,21,22,23,24,25,26,27,28,29,30,32,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,\n",
    "                 52,53,54,86,90,71,72]).reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "7ce312d6",
   "metadata": {},
   "outputs": [],
   "source": [
    "#newn.to_excel('02_output_analysis/PIDs_patient_data.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "270de706",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "246d2a4b",
   "metadata": {},
   "outputs": [],
   "source": [
    "wo_dups = igv_all.drop_duplicates(cols_cpra).copy().reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "c50bc00e",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>splice_change</th>\n",
       "      <th>too_low_coverage</th>\n",
       "      <th>PosExonRefSeqAccession</th>\n",
       "      <th>PosExon_type</th>\n",
       "      <th>ClinVar_Pathogenicity</th>\n",
       "      <th>MaxEntScan_alt</th>\n",
       "      <th>...</th>\n",
       "      <th>SpliceAI_pred_DP_AG</th>\n",
       "      <th>SpliceAI_pred_DP_AL</th>\n",
       "      <th>SpliceAI_pred_DP_DG</th>\n",
       "      <th>SpliceAI_pred_DP_DL</th>\n",
       "      <th>SpliceAI_pred_DS_AG</th>\n",
       "      <th>SpliceAI_pred_DS_AL</th>\n",
       "      <th>SpliceAI_pred_DS_DG</th>\n",
       "      <th>SpliceAI_pred_DS_DL</th>\n",
       "      <th>ada_score</th>\n",
       "      <th>rf_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>2</td>\n",
       "      <td>48033789</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>1</td>\n",
       "      <td>insideDonorSite</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>4.269</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-24</td>\n",
       "      <td>-43</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.128779</td>\n",
       "      <td>0.324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>550</th>\n",
       "      <td>22</td>\n",
       "      <td>24175755</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>4</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>9.546</td>\n",
       "      <td>...</td>\n",
       "      <td>24</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>24</td>\n",
       "      <td>0.66</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0.002</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    #CHROM       POS REF ALT splice_change too_low_coverage  \\\n",
       "103      2  48033789   C   T           yes               no   \n",
       "550     22  24175755   G   A           yes               no   \n",
       "\n",
       "     PosExonRefSeqAccession         PosExon_type   ClinVar_Pathogenicity  \\\n",
       "103                       1      insideDonorSite  Uncertain Significance   \n",
       "550                       4  outsideAcceptorSite  Uncertain Significance   \n",
       "\n",
       "     MaxEntScan_alt  ...  SpliceAI_pred_DP_AG  SpliceAI_pred_DP_AL  \\\n",
       "103           4.269  ...                    1                  -24   \n",
       "550           9.546  ...                   24                    4   \n",
       "\n",
       "    SpliceAI_pred_DP_DG SpliceAI_pred_DP_DL SpliceAI_pred_DS_AG  \\\n",
       "103                 -43                   1                0.00   \n",
       "550                   4                  24                0.66   \n",
       "\n",
       "    SpliceAI_pred_DS_AL  SpliceAI_pred_DS_DG  SpliceAI_pred_DS_DL  ada_score  \\\n",
       "103                0.00                  0.0                 0.07   0.128779   \n",
       "550                0.02                  0.0                 0.00   0.000020   \n",
       "\n",
       "     rf_score  \n",
       "103     0.324  \n",
       "550     0.002  \n",
       "\n",
       "[2 rows x 22 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wo_dups[cols_cpra+['splice_change', 'too_low_coverage', 'PosExonRefSeqAccession', 'PosExon_type',\n",
    "                   'ClinVar_Pathogenicity', 'MaxEntScan_alt',\n",
    "                   'MaxEntScan_diff', 'MaxEntScan_ref', 'SpliceAI_pred_DP_AG', 'SpliceAI_pred_DP_AL',\n",
    "                   'SpliceAI_pred_DP_DG', 'SpliceAI_pred_DP_DL', 'SpliceAI_pred_DS_AG', 'SpliceAI_pred_DS_AL',\n",
    "                    'SpliceAI_pred_DS_DG', 'SpliceAI_pred_DS_DL', 'ada_score', 'rf_score']\n",
    "       ][(wo_dups['splice_change']=='yes')&\n",
    "         ((wo_dups['SpliceAI_pred_DS_AG']<0.9)&(wo_dups['SpliceAI_pred_DS_AL']<0.9)&\n",
    "          (wo_dups['SpliceAI_pred_DS_DG']<0.9)&(wo_dups['SpliceAI_pred_DS_DL']<0.9))&\n",
    "         (wo_dups['ada_score']<0.9)&(wo_dups['rf_score']<0.9)&\n",
    "         (((wo_dups['MaxEntScan_diff']>=0)&(wo_dups['MaxEntScan_alt']<8.5))|\n",
    "          ((wo_dups['MaxEntScan_diff']<0)&(wo_dups['MaxEntScan_alt']>6.2)))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "8465e065",
   "metadata": {},
   "outputs": [],
   "source": [
    "# vcf file for QCI (09.09.2021)\n",
    "import pysam as ps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "c6463bc6",
   "metadata": {},
   "outputs": [],
   "source": [
    "igv_hot_warm = igv_all[cols_cpra][(igv_all['splice_change']=='yes')&\n",
    "                                  (igv_all['Splice Project Gene Priority']!='Cold (387-ACMG-MASTER)')&\n",
    "                                  (~igv_all['HUGO_Symbol'].isin(gus))].drop_duplicates()\n",
    "\n",
    "vcf_header = ps.VariantHeader()\n",
    "for i in igv_hot_warm['#CHROM'].unique():\n",
    "    vcf_header.add_meta('contig', items = [('ID', i)])\n",
    "vcf_header.add_meta('reference', \n",
    "                    value='/AnnotierungsProzesse/Referenzdaten/Genome/Human/hg19/hg19_Ensembl/hg19 Ensembl')\n",
    "\n",
    "vcf_out = ps.VariantFile('01_xlsx_files/hot_warm_splice.txt', 'w', header=vcf_header)\n",
    "\n",
    "for c,p,r,a in zip(igv_hot_warm['#CHROM'], igv_hot_warm['POS'], igv_hot_warm['REF'], igv_hot_warm['ALT']):\n",
    "    rec = vcf_out.new_record()\n",
    "    rec.chrom = str(c)\n",
    "    rec.start = p-1\n",
    "    rec.stop = p\n",
    "    rec.ref = r\n",
    "    rec.alts = a\n",
    "    vcf_out.write(rec)\n",
    "vcf_out.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "18ab9cd8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ALT sequences were comma-separated --> \n",
    "with open('01_xlsx_files/hot_warm_splice.vcf', 'w') as w:\n",
    "    with open('01_xlsx_files/hot_warm_splice.txt') as o:\n",
    "        for line in o.readlines():\n",
    "            if line.startswith('#'):\n",
    "                w.write(line)\n",
    "            else:\n",
    "                line = line.replace(',','')\n",
    "                w.write(line)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "390bff6c",
   "metadata": {},
   "outputs": [],
   "source": [
    "igv_all[['#CHROM', 'pos_last', 'pos_next', 'kind', 'shortcut', 'strand', 'HUGO_Symbol'\n",
    "        ]][igv_all['pos_last'].notnull()].drop_duplicates('shortcut').to_excel('03_for_cDNA/pre_for_cDNA.xlsx', \n",
    "                                                                               index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "58c9854d",
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "igv_all[['#CHROM', 'POS', 'REF', 'ALT', 'HGVSg', 'HUGO_Symbol', 'shortcut', 'frameshift', 'premat_stop_cod', 'hgsv_new', 'INFO_IGV', 'too_low_coverage',\n",
    "         'splice_change', 'kind_hboc', 'classification_hboc', 'aquisition_hboc', 'taskforce_review_hboc',\n",
    "         'date_taskforce_hboc', 'note_hboc', 'add_note_hboc', 'splice_pred_alamut_hboc', 'prediction_hboc',\n",
    "         'literature_hboc', 'evidence_level_lit_hboc', 'comment_hboc', 'recomm_action_hboc']][\n",
    "    (igv_all['splice_change']=='yes')&(igv_all['kind_hboc'].notnull())&(igv_all['too_low_coverage']=='no')\n",
    "].drop_duplicates(cols_cpra).to_excel('02_output_analysis/hboc_splice_change.xlsx')\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "981ad04d",
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "igv_all[['#CHROM', 'POS', 'REF', 'ALT', 'HGVSg', 'HUGO_Symbol', 'shortcut', 'frameshift', 'premat_stop_cod', 'hgsv_new', 'INFO_IGV', 'too_low_coverage',\n",
    "         'splice_change', 'kind_hboc', 'classification_hboc', 'aquisition_hboc', 'taskforce_review_hboc', \n",
    "         'date_taskforce_hboc', 'note_hboc', 'add_note_hboc', 'splice_pred_alamut_hboc', 'prediction_hboc', \n",
    "         'literature_hboc', 'evidence_level_lit_hboc', 'comment_hboc', 'recomm_action_hboc']][\n",
    "    (igv_all['splice_change']=='no')&(igv_all['kind_hboc'].notnull())&(igv_all['too_low_coverage']=='no')\n",
    "].drop_duplicates(cols_cpra).to_excel('02_output_analysis/hboc_no_splice_change.xlsx')\n",
    "'''"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "02fbead2",
   "metadata": {},
   "source": [
    "### 30.08."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "07521dea",
   "metadata": {},
   "outputs": [],
   "source": [
    "igv_all = merge_xlsx('2021-08-19_all_vars_analyzed.xlsx')\n",
    "nct_vars = pd.read_pickle('00_dataframes/nct_gepado_master_vars')\n",
    "\n",
    "igv_nct = pd.merge(nct_vars[cols_cpra].drop_duplicates(), igv_all, how='inner')\n",
    "\n",
    "igv_outcome = perc_cat(igv_all)\n",
    "transp_outcome = igv_outcome.transpose(copy=True).reset_index().drop(index=[7,8,9,10,11])\n",
    "transp_outcome = transp_outcome.rename(columns={i:cat for i,cat in zip(transp_outcome.columns, \n",
    "                                                                       list(transp_outcome.loc[0]))}).drop(index=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "0ff04f3a",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>SpliceAI_pred_DP_AG</th>\n",
       "      <th>SpliceAI_pred_DP_AL</th>\n",
       "      <th>SpliceAI_pred_DP_DG</th>\n",
       "      <th>SpliceAI_pred_DP_DL</th>\n",
       "      <th>SpliceAI_pred_DS_AG</th>\n",
       "      <th>SpliceAI_pred_DS_AL</th>\n",
       "      <th>SpliceAI_pred_DS_DG</th>\n",
       "      <th>SpliceAI_pred_DS_DL</th>\n",
       "      <th>SpliceAI_pred_SYMBOL</th>\n",
       "      <th>ada_score</th>\n",
       "      <th>rf_score</th>\n",
       "      <th>PosExonRefSeqAccession</th>\n",
       "      <th>PosExon_type</th>\n",
       "      <th>ClinVar_Pathogenicity</th>\n",
       "      <th>INFO_IGV</th>\n",
       "      <th>shortcut</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>8</td>\n",
       "      <td>30916058</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>29</td>\n",
       "      <td>-28</td>\n",
       "      <td>29</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.00</td>\n",
       "      <td>WRN</td>\n",
       "      <td>0.991457</td>\n",
       "      <td>0.700</td>\n",
       "      <td>1</td>\n",
       "      <td>insideDonorSite</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>new 5'ss in intron (like spliceAI said) in 2LJ...</td>\n",
       "      <td>8_30916058_A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>17</td>\n",
       "      <td>7579699</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-11</td>\n",
       "      <td>22</td>\n",
       "      <td>-11</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.25</td>\n",
       "      <td>1.00</td>\n",
       "      <td>TP53</td>\n",
       "      <td>0.999987</td>\n",
       "      <td>0.940</td>\n",
       "      <td>22</td>\n",
       "      <td>insideAcceptorSite</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exon skipping</td>\n",
       "      <td>17_7579699_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>2</td>\n",
       "      <td>48033789</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>-24</td>\n",
       "      <td>-43</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.07</td>\n",
       "      <td>MSH6</td>\n",
       "      <td>0.128779</td>\n",
       "      <td>0.324</td>\n",
       "      <td>1</td>\n",
       "      <td>insideDonorSite</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>some exon skipping, intron retention, in other...</td>\n",
       "      <td>2_48033789_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>9</td>\n",
       "      <td>100437871</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-17</td>\n",
       "      <td>-2</td>\n",
       "      <td>-15</td>\n",
       "      <td>-10</td>\n",
       "      <td>0.09</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>XPA</td>\n",
       "      <td>0.999835</td>\n",
       "      <td>0.726</td>\n",
       "      <td>2</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "      <td>NaN</td>\n",
       "      <td>acceptor loss, new 3'ss in exon, frameshift</td>\n",
       "      <td>9_100437871_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>17</td>\n",
       "      <td>59934594</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-11</td>\n",
       "      <td>-2</td>\n",
       "      <td>-2</td>\n",
       "      <td>-21</td>\n",
       "      <td>0.90</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>BRIP1</td>\n",
       "      <td>0.999963</td>\n",
       "      <td>0.830</td>\n",
       "      <td>2</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>new ss in exon, 9 bases of exon skipped</td>\n",
       "      <td>17_59934594_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>1</td>\n",
       "      <td>17345454</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>-14</td>\n",
       "      <td>-1</td>\n",
       "      <td>38</td>\n",
       "      <td>-18</td>\n",
       "      <td>0.30</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>SDHB</td>\n",
       "      <td>0.999986</td>\n",
       "      <td>0.938</td>\n",
       "      <td>1</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "      <td>NaN</td>\n",
       "      <td>new 3'ss in exon, part of exon skipped, frames...</td>\n",
       "      <td>1_17345454_C_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>16</td>\n",
       "      <td>68849662</td>\n",
       "      <td>CG</td>\n",
       "      <td>C</td>\n",
       "      <td>6</td>\n",
       "      <td>-31</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.88</td>\n",
       "      <td>1.00</td>\n",
       "      <td>CDH1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>insideDonorSite</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>donor loss, intron retention</td>\n",
       "      <td>16_68849662_CG_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>2</td>\n",
       "      <td>47601174</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>-2</td>\n",
       "      <td>13</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.28</td>\n",
       "      <td>EPCAM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>13</td>\n",
       "      <td>insideDonorSite</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>about 50 % new 5' ss in exon, frameshift, spli...</td>\n",
       "      <td>2_47601174_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>3</td>\n",
       "      <td>14212051</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>-13</td>\n",
       "      <td>-1</td>\n",
       "      <td>-3</td>\n",
       "      <td>-11</td>\n",
       "      <td>0.86</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>XPC</td>\n",
       "      <td>0.999946</td>\n",
       "      <td>0.918</td>\n",
       "      <td>1</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron retention, acceptor loss, new 3' ss in ...</td>\n",
       "      <td>3_14212051_C_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>12</td>\n",
       "      <td>133254296</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>-3</td>\n",
       "      <td>-1</td>\n",
       "      <td>-3</td>\n",
       "      <td>-1</td>\n",
       "      <td>0.88</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>POLE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>insideAcceptorSite</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>new 3'Ss in exon (as spliceAI said), in frame,...</td>\n",
       "      <td>12_133254296_C_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>16</td>\n",
       "      <td>2121936</td>\n",
       "      <td>G</td>\n",
       "      <td>GT</td>\n",
       "      <td>-1</td>\n",
       "      <td>-13</td>\n",
       "      <td>-6</td>\n",
       "      <td>-2</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.02</td>\n",
       "      <td>TSC2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>outsideDonorSite</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exon skipping, quantification difficult (ca, 1...</td>\n",
       "      <td>16_2121936_G_GT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>17</td>\n",
       "      <td>41234419</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.99</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>0.999988</td>\n",
       "      <td>0.932</td>\n",
       "      <td>2</td>\n",
       "      <td>outsideDonorSite</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>bit of intron retention, exon skipping, count ...</td>\n",
       "      <td>17_41234419_A_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>19</td>\n",
       "      <td>45871887</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-31</td>\n",
       "      <td>1</td>\n",
       "      <td>-37</td>\n",
       "      <td>1</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.99</td>\n",
       "      <td>ERCC2</td>\n",
       "      <td>0.999987</td>\n",
       "      <td>0.930</td>\n",
       "      <td>1</td>\n",
       "      <td>outsideDonorSite</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exon skipping, donor loss</td>\n",
       "      <td>19_45871887_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>3</td>\n",
       "      <td>52443569</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-38</td>\n",
       "      <td>47</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.29</td>\n",
       "      <td>1.00</td>\n",
       "      <td>BAP1</td>\n",
       "      <td>0.999982</td>\n",
       "      <td>0.940</td>\n",
       "      <td>1</td>\n",
       "      <td>outsideDonorSite</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>exon skipped</td>\n",
       "      <td>3_52443569_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>3</td>\n",
       "      <td>52440267</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>38</td>\n",
       "      <td>25</td>\n",
       "      <td>-2</td>\n",
       "      <td>2</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.61</td>\n",
       "      <td>0.08</td>\n",
       "      <td>BAP1</td>\n",
       "      <td>0.994878</td>\n",
       "      <td>0.700</td>\n",
       "      <td>2</td>\n",
       "      <td>outsideDonorSite</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>new 5'ss in exon, 12 aa skipped, donor loss, 1...</td>\n",
       "      <td>3_52440267_A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144</th>\n",
       "      <td>11</td>\n",
       "      <td>108186639</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>24</td>\n",
       "      <td>-20</td>\n",
       "      <td>24</td>\n",
       "      <td>-1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.99</td>\n",
       "      <td>ATM</td>\n",
       "      <td>0.999992</td>\n",
       "      <td>0.942</td>\n",
       "      <td>1</td>\n",
       "      <td>outsideDonorSite</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>exon skipped</td>\n",
       "      <td>11_108186639_G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>16</td>\n",
       "      <td>2138446</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>-50</td>\n",
       "      <td>1</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>TSC2</td>\n",
       "      <td>0.999984</td>\n",
       "      <td>0.840</td>\n",
       "      <td>1</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>intron retention or new 3'ss in exon, part of ...</td>\n",
       "      <td>16_2138446_G_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>22</td>\n",
       "      <td>29121230</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>25</td>\n",
       "      <td>-3</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.94</td>\n",
       "      <td>1.00</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>0.999962</td>\n",
       "      <td>0.740</td>\n",
       "      <td>1</td>\n",
       "      <td>outsideDonorSite</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>in EUBCTJ intron retention of 4 bases, in RS,,...</td>\n",
       "      <td>22_29121230_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>17</td>\n",
       "      <td>56809887</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-45</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>18</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.87</td>\n",
       "      <td>0.12</td>\n",
       "      <td>RAD51C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18</td>\n",
       "      <td>insideDonorSite</td>\n",
       "      <td>Likely Benign</td>\n",
       "      <td>new 5'ss in exon (as spliceAI said, count 333:...</td>\n",
       "      <td>17_56809887_A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164</th>\n",
       "      <td>9</td>\n",
       "      <td>135779206</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-17</td>\n",
       "      <td>-2</td>\n",
       "      <td>-41</td>\n",
       "      <td>-2</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.94</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>TSC1</td>\n",
       "      <td>0.999993</td>\n",
       "      <td>0.926</td>\n",
       "      <td>2</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>new 3' ss in exon, part of exon skipped</td>\n",
       "      <td>9_135779206_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>173</th>\n",
       "      <td>3</td>\n",
       "      <td>52439929</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>-9</td>\n",
       "      <td>-1</td>\n",
       "      <td>-11</td>\n",
       "      <td>-9</td>\n",
       "      <td>0.86</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>BAP1</td>\n",
       "      <td>0.999988</td>\n",
       "      <td>0.940</td>\n",
       "      <td>1</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "      <td>NaN</td>\n",
       "      <td>acceptor loss, new 3'ss in exon, frameshift</td>\n",
       "      <td>3_52439929_C_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175</th>\n",
       "      <td>16</td>\n",
       "      <td>3650977</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>-48</td>\n",
       "      <td>19</td>\n",
       "      <td>-39</td>\n",
       "      <td>2</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.87</td>\n",
       "      <td>SLX4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>outsideDonorSite</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exon skipping, intron retention (but intron re...</td>\n",
       "      <td>16_3650977_T_TA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>7</td>\n",
       "      <td>66459197</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>10</td>\n",
       "      <td>27</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.64</td>\n",
       "      <td>0.93</td>\n",
       "      <td>SBDS</td>\n",
       "      <td>0.999936</td>\n",
       "      <td>0.866</td>\n",
       "      <td>2</td>\n",
       "      <td>outsideDonorSite</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>new ss, part of exon skipped, (for 7TBCL somet...</td>\n",
       "      <td>7_66459197_A_G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    #CHROM        POS REF ALT SpliceAI_pred_DP_AG SpliceAI_pred_DP_AL  \\\n",
       "0        8   30916058   A   G                  29                 -28   \n",
       "35      17    7579699   C   T                 -11                  22   \n",
       "42       2   48033789   C   T                   1                 -24   \n",
       "49       9  100437871   T   C                 -17                  -2   \n",
       "57      17   59934594   T   C                 -11                  -2   \n",
       "61       1   17345454   C   G                 -14                  -1   \n",
       "65      16   68849662  CG   C                   6                 -31   \n",
       "72       2   47601174   C   T                  13                   2   \n",
       "73       3   14212051   C   G                 -13                  -1   \n",
       "77      12  133254296   C   A                  -3                  -1   \n",
       "82      16    2121936   G  GT                  -1                 -13   \n",
       "98      17   41234419   A   C                   0                  38   \n",
       "129     19   45871887   C   T                 -31                   1   \n",
       "130      3   52443569   C   T                 -38                  47   \n",
       "131      3   52440267   A   G                  38                  25   \n",
       "144     11  108186639   G   A                  24                 -20   \n",
       "145     16    2138446   G   C                  14                   1   \n",
       "149     22   29121230   C   T                   1                  25   \n",
       "153     17   56809887   A   G                 -45                  18   \n",
       "164      9  135779206   T   C                 -17                  -2   \n",
       "173      3   52439929   C   A                  -9                  -1   \n",
       "175     16    3650977   T  TA                 -48                  19   \n",
       "183      7   66459197   A   G                  10                  27   \n",
       "\n",
       "    SpliceAI_pred_DP_DG SpliceAI_pred_DP_DL  SpliceAI_pred_DS_AG  \\\n",
       "0                    29                   1                 0.00   \n",
       "35                  -11                   1                 0.00   \n",
       "42                  -43                   1                 0.00   \n",
       "49                  -15                 -10                 0.09   \n",
       "57                   -2                 -21                 0.90   \n",
       "61                   38                 -18                 0.30   \n",
       "65                    6                   0                 0.00   \n",
       "72                   -2                  13                 0.00   \n",
       "73                   -3                 -11                 0.86   \n",
       "77                   -3                  -1                 0.88   \n",
       "82                   -6                  -2                 0.00   \n",
       "98                    0                   2                 0.00   \n",
       "129                 -37                   1                 0.01   \n",
       "130                   8                   1                 0.00   \n",
       "131                  -2                   2                 0.00   \n",
       "144                  24                  -1                 0.00   \n",
       "145                 -50                   1                 0.22   \n",
       "149                  -3                   1                 0.00   \n",
       "153                   0                  18                 0.11   \n",
       "164                 -41                  -2                 0.04   \n",
       "173                 -11                  -9                 0.86   \n",
       "175                 -39                   2                 0.00   \n",
       "183                  10                   2                 0.00   \n",
       "\n",
       "     SpliceAI_pred_DS_AL  SpliceAI_pred_DS_DG  SpliceAI_pred_DS_DL  \\\n",
       "0                   0.00                 0.06                 0.00   \n",
       "35                  0.00                 0.25                 1.00   \n",
       "42                  0.00                 0.00                 0.07   \n",
       "49                  1.00                 0.00                 0.00   \n",
       "57                  0.99                 0.00                 0.00   \n",
       "61                  0.98                 0.00                 0.00   \n",
       "65                  0.00                 0.88                 1.00   \n",
       "72                  0.00                 0.95                 0.28   \n",
       "73                  1.00                 0.00                 0.00   \n",
       "77                  0.00                 0.00                 0.00   \n",
       "82                  0.00                 0.07                 0.02   \n",
       "98                  0.00                 0.00                 0.99   \n",
       "129                 0.00                 0.18                 0.99   \n",
       "130                 0.00                 0.29                 1.00   \n",
       "131                 0.00                 0.61                 0.08   \n",
       "144                 0.00                 0.01                 0.99   \n",
       "145                 0.97                 0.00                 0.00   \n",
       "149                 0.00                 0.94                 1.00   \n",
       "153                 0.00                 0.87                 0.12   \n",
       "164                 0.94                 0.00                 0.00   \n",
       "173                 0.99                 0.00                 0.00   \n",
       "175                 0.00                 0.00                 0.87   \n",
       "183                 0.00                 0.64                 0.93   \n",
       "\n",
       "    SpliceAI_pred_SYMBOL  ada_score  rf_score  PosExonRefSeqAccession  \\\n",
       "0                    WRN   0.991457     0.700                       1   \n",
       "35                  TP53   0.999987     0.940                      22   \n",
       "42                  MSH6   0.128779     0.324                       1   \n",
       "49                   XPA   0.999835     0.726                       2   \n",
       "57                 BRIP1   0.999963     0.830                       2   \n",
       "61                  SDHB   0.999986     0.938                       1   \n",
       "65                  CDH1        NaN       NaN                       0   \n",
       "72                 EPCAM        NaN       NaN                      13   \n",
       "73                   XPC   0.999946     0.918                       1   \n",
       "77                  POLE        NaN       NaN                       9   \n",
       "82                  TSC2        NaN       NaN                       1   \n",
       "98                 BRCA1   0.999988     0.932                       2   \n",
       "129                ERCC2   0.999987     0.930                       1   \n",
       "130                 BAP1   0.999982     0.940                       1   \n",
       "131                 BAP1   0.994878     0.700                       2   \n",
       "144                  ATM   0.999992     0.942                       1   \n",
       "145                 TSC2   0.999984     0.840                       1   \n",
       "149                CHEK2   0.999962     0.740                       1   \n",
       "153               RAD51C        NaN       NaN                      18   \n",
       "164                 TSC1   0.999993     0.926                       2   \n",
       "173                 BAP1   0.999988     0.940                       1   \n",
       "175                 SLX4        NaN       NaN                       3   \n",
       "183                 SBDS   0.999936     0.866                       2   \n",
       "\n",
       "            PosExon_type   ClinVar_Pathogenicity  \\\n",
       "0        insideDonorSite  Uncertain Significance   \n",
       "35    insideAcceptorSite                     NaN   \n",
       "42       insideDonorSite  Uncertain Significance   \n",
       "49   outsideAcceptorSite                     NaN   \n",
       "57   outsideAcceptorSite       Likely Pathogenic   \n",
       "61   outsideAcceptorSite                     NaN   \n",
       "65       insideDonorSite              Pathogenic   \n",
       "72       insideDonorSite              Pathogenic   \n",
       "73   outsideAcceptorSite                     NaN   \n",
       "77    insideAcceptorSite  Uncertain Significance   \n",
       "82      outsideDonorSite                     NaN   \n",
       "98      outsideDonorSite              Pathogenic   \n",
       "129     outsideDonorSite                     NaN   \n",
       "130     outsideDonorSite              Pathogenic   \n",
       "131     outsideDonorSite       Likely Pathogenic   \n",
       "144     outsideDonorSite       Likely Pathogenic   \n",
       "145  outsideAcceptorSite  Uncertain Significance   \n",
       "149     outsideDonorSite              Pathogenic   \n",
       "153      insideDonorSite           Likely Benign   \n",
       "164  outsideAcceptorSite  Uncertain Significance   \n",
       "173  outsideAcceptorSite                     NaN   \n",
       "175     outsideDonorSite                     NaN   \n",
       "183     outsideDonorSite              Pathogenic   \n",
       "\n",
       "                                              INFO_IGV          shortcut  \n",
       "0    new 5'ss in intron (like spliceAI said) in 2LJ...    8_30916058_A_G  \n",
       "35                                       exon skipping    17_7579699_C_T  \n",
       "42   some exon skipping, intron retention, in other...    2_48033789_C_T  \n",
       "49         acceptor loss, new 3'ss in exon, frameshift   9_100437871_T_C  \n",
       "57            new ss in exon, 9 bases of exon skipped    17_59934594_T_C  \n",
       "61   new 3'ss in exon, part of exon skipped, frames...    1_17345454_C_G  \n",
       "65                        donor loss, intron retention  16_68849662_CG_C  \n",
       "72   about 50 % new 5' ss in exon, frameshift, spli...    2_47601174_C_T  \n",
       "73   intron retention, acceptor loss, new 3' ss in ...    3_14212051_C_G  \n",
       "77   new 3'Ss in exon (as spliceAI said), in frame,...  12_133254296_C_A  \n",
       "82   exon skipping, quantification difficult (ca, 1...   16_2121936_G_GT  \n",
       "98   bit of intron retention, exon skipping, count ...   17_41234419_A_C  \n",
       "129                          exon skipping, donor loss   19_45871887_C_T  \n",
       "130                                       exon skipped    3_52443569_C_T  \n",
       "131  new 5'ss in exon, 12 aa skipped, donor loss, 1...    3_52440267_A_G  \n",
       "144                                       exon skipped  11_108186639_G_A  \n",
       "145  intron retention or new 3'ss in exon, part of ...    16_2138446_G_C  \n",
       "149  in EUBCTJ intron retention of 4 bases, in RS,,...   22_29121230_C_T  \n",
       "153  new 5'ss in exon (as spliceAI said, count 333:...   17_56809887_A_G  \n",
       "164            new 3' ss in exon, part of exon skipped   9_135779206_T_C  \n",
       "173        acceptor loss, new 3'ss in exon, frameshift    3_52439929_C_A  \n",
       "175  exon skipping, intron retention (but intron re...   16_3650977_T_TA  \n",
       "183  new ss, part of exon skipped, (for 7TBCL somet...    7_66459197_A_G  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "igv_nct[cols_cpra+['SpliceAI_pred_DP_AG', 'SpliceAI_pred_DP_AL', 'SpliceAI_pred_DP_DG', \n",
    "                    'SpliceAI_pred_DP_DL', 'SpliceAI_pred_DS_AG', 'SpliceAI_pred_DS_AL', 'SpliceAI_pred_DS_DG', \n",
    "                    'SpliceAI_pred_DS_DL', 'SpliceAI_pred_SYMBOL', 'ada_score', 'rf_score',\n",
    "                    'PosExonRefSeqAccession', 'PosExon_type', 'ClinVar_Pathogenicity', 'INFO_IGV', 'shortcut']][\n",
    "    (igv_nct['Splice Project Gene Priority']!='Cold (387-ACMG-MASTER)')&(~igv_nct['HUGO_Symbol'].isin(gus))&\n",
    "    (igv_nct['splice_change']=='yes')&(igv_nct['too_low_coverage']=='no')].drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "b5048c88",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "81"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#[cols_cpra+['SpliceAI_pred_DP_AG', 'SpliceAI_pred_DP_AL', 'SpliceAI_pred_DP_DG', \n",
    "                    #'SpliceAI_pred_DP_DL', 'SpliceAI_pred_DS_AG', 'SpliceAI_pred_DS_AL', 'SpliceAI_pred_DS_DG', \n",
    "                    #'SpliceAI_pred_DS_DL', 'SpliceAI_pred_SYMBOL', 'ada_score', 'rf_score',\n",
    "                    #'PosExonRefSeqAccession', 'PosExon_type', 'ClinVar_Pathogenicity', 'INFO_IGV']]\n",
    "len(igv_all[(igv_all['too_low_coverage']=='no')&(igv_all['splice_change']=='no')&\n",
    "    (((igv_all['PosExonRefSeqAccession']<3)&\n",
    "     (igv_all['PosExon_type'].isin(['outsideAcceptorSite', 'outsideDonorSite'])))|\n",
    "    ((igv_all['PosExonRefSeqAccession']<2)&\n",
    "     (igv_all['PosExon_type'].isin(['insideAcceptorSite', 'insideDonorSite']))))\n",
    "    ].groupby(cols_cpra))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "4944a0bd",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>SpliceAI_pred_DP_AG</th>\n",
       "      <th>SpliceAI_pred_DP_AL</th>\n",
       "      <th>SpliceAI_pred_DP_DG</th>\n",
       "      <th>SpliceAI_pred_DP_DL</th>\n",
       "      <th>SpliceAI_pred_DS_AG</th>\n",
       "      <th>SpliceAI_pred_DS_AL</th>\n",
       "      <th>SpliceAI_pred_DS_DG</th>\n",
       "      <th>SpliceAI_pred_DS_DL</th>\n",
       "      <th>SpliceAI_pred_SYMBOL</th>\n",
       "      <th>ada_score</th>\n",
       "      <th>rf_score</th>\n",
       "      <th>PosExonRefSeqAccession</th>\n",
       "      <th>PosExon_type</th>\n",
       "      <th>ClinVar_Pathogenicity</th>\n",
       "      <th>INFO_IGV</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>392</th>\n",
       "      <td>2</td>\n",
       "      <td>48033789</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>-24</td>\n",
       "      <td>-43</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.07</td>\n",
       "      <td>MSH6</td>\n",
       "      <td>0.128779</td>\n",
       "      <td>0.324</td>\n",
       "      <td>1</td>\n",
       "      <td>insideDonorSite</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>some exon skipping, intron retention, in other...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1184</th>\n",
       "      <td>17</td>\n",
       "      <td>74469327</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>6</td>\n",
       "      <td>27</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.64</td>\n",
       "      <td>0.01</td>\n",
       "      <td>RHBDF2</td>\n",
       "      <td>0.000024</td>\n",
       "      <td>0.000</td>\n",
       "      <td>7</td>\n",
       "      <td>outsideDonorSite</td>\n",
       "      <td>Benign</td>\n",
       "      <td>new 5'ss in intron (as spliceAI said), intron ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1442</th>\n",
       "      <td>22</td>\n",
       "      <td>24175755</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>24</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>24</td>\n",
       "      <td>0.66</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>SMARCB1</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0.002</td>\n",
       "      <td>4</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>new 3'ss in exon (as spliceAI said) in 8Q2LVZ,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1443</th>\n",
       "      <td>22</td>\n",
       "      <td>24175755</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>24</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>24</td>\n",
       "      <td>0.66</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>SMARCB1</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0.002</td>\n",
       "      <td>4</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>new 3'ss in exon (as spliceAI said) in 8Q2LVZ,...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     #CHROM       POS REF ALT SpliceAI_pred_DP_AG SpliceAI_pred_DP_AL  \\\n",
       "392       2  48033789   C   T                   1                 -24   \n",
       "1184     17  74469327   G   A                   6                  27   \n",
       "1442     22  24175755   G   A                  24                   4   \n",
       "1443     22  24175755   G   A                  24                   4   \n",
       "\n",
       "     SpliceAI_pred_DP_DG SpliceAI_pred_DP_DL  SpliceAI_pred_DS_AG  \\\n",
       "392                  -43                   1                 0.00   \n",
       "1184                   2                   7                 0.00   \n",
       "1442                   4                  24                 0.66   \n",
       "1443                   4                  24                 0.66   \n",
       "\n",
       "      SpliceAI_pred_DS_AL  SpliceAI_pred_DS_DG  SpliceAI_pred_DS_DL  \\\n",
       "392                  0.00                 0.00                 0.07   \n",
       "1184                 0.00                 0.64                 0.01   \n",
       "1442                 0.02                 0.00                 0.00   \n",
       "1443                 0.02                 0.00                 0.00   \n",
       "\n",
       "     SpliceAI_pred_SYMBOL  ada_score  rf_score  PosExonRefSeqAccession  \\\n",
       "392                  MSH6   0.128779     0.324                       1   \n",
       "1184               RHBDF2   0.000024     0.000                       7   \n",
       "1442              SMARCB1   0.000020     0.002                       4   \n",
       "1443              SMARCB1   0.000020     0.002                       4   \n",
       "\n",
       "             PosExon_type   ClinVar_Pathogenicity  \\\n",
       "392       insideDonorSite  Uncertain Significance   \n",
       "1184     outsideDonorSite                  Benign   \n",
       "1442  outsideAcceptorSite  Uncertain Significance   \n",
       "1443  outsideAcceptorSite  Uncertain Significance   \n",
       "\n",
       "                                               INFO_IGV  \n",
       "392   some exon skipping, intron retention, in other...  \n",
       "1184  new 5'ss in intron (as spliceAI said), intron ...  \n",
       "1442  new 3'ss in exon (as spliceAI said) in 8Q2LVZ,...  \n",
       "1443  new 3'ss in exon (as spliceAI said) in 8Q2LVZ,...  "
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "igv_all[cols_cpra+['SpliceAI_pred_DP_AG', 'SpliceAI_pred_DP_AL', 'SpliceAI_pred_DP_DG', \n",
    "                    'SpliceAI_pred_DP_DL', 'SpliceAI_pred_DS_AG', 'SpliceAI_pred_DS_AL', 'SpliceAI_pred_DS_DG', \n",
    "                    'SpliceAI_pred_DS_DL', 'SpliceAI_pred_SYMBOL', 'ada_score', 'rf_score',\n",
    "                    'PosExonRefSeqAccession', 'PosExon_type', 'ClinVar_Pathogenicity', 'INFO_IGV']][\n",
    "    ((igv_all['SpliceAI_pred_DS_AG']<0.9)&(igv_all['SpliceAI_pred_DS_AL']<0.9)&\n",
    "             (igv_all['SpliceAI_pred_DS_DG']<0.9)&(igv_all['SpliceAI_pred_DS_DL']<0.9))&\n",
    "            (igv_all['ada_score']<0.9)&(igv_all['rf_score']<0.9)&\n",
    "            #(((igv_all['MaxEntScan_diff']<0)&(igv_all['MaxEntScan_alt']>6.2))|\n",
    "             #((igv_all['MaxEntScan_diff']>0)&(igv_all['MaxEntScan_alt']<8.5)))&\n",
    "            (igv_all['too_low_coverage']=='no')&(igv_all['splice_change']=='yes')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a7148519",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "#igv_outcome.to_excel('02_output_analysis/2021-08-19_outcome_igv.xlsx', index=False)\n",
    "#igv_outcome.to_pickle('00_dataframes/2021-08-19_igv_perc_table')\n",
    "#transp_outcome.to_pickle('00_dataframes/transp_igv_outcome')\n",
    "#igv_all.to_pickle('00_dataframes/2021-08-19_igv_outcome')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "fd20ca80",
   "metadata": {},
   "outputs": [],
   "source": [
    "#igv_outcome.to_excel('02_output_analysis/2021-08-19_outcome_igv.xlsx')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3d7e577b",
   "metadata": {},
   "source": [
    "### 26.08."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2131831f",
   "metadata": {},
   "outputs": [],
   "source": [
    "igv_all = merge_xlsx('2021-08-18_all_vars_analyzed.xlsx')\n",
    "igv_outcome = perc_cat(igv_all)\n",
    "hboc_vars = pd.read_pickle('../04_annotations/00_dataframes/hboc_vars')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "76fd69d9",
   "metadata": {},
   "outputs": [],
   "source": [
    "search_col(igv_all, 'clinv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1dd80698",
   "metadata": {},
   "outputs": [],
   "source": [
    "hboc_igv = pd.merge(igv_all[['#CHROM', 'POS', 'REF', 'ALT', 'PID', 'splice_change', 'ClinVar_Pathogenicity']], \n",
    "                    hboc_vars[['#CHROM', 'POS', 'REF', 'ALT', 'ID_hboc']], \n",
    "                    on=cols_cpra,\n",
    "                    how='inner').drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d7f3864e",
   "metadata": {},
   "outputs": [],
   "source": [
    "hboc_igv[hboc_igv['splice_change']=='yes'].drop_duplicates(subset=cols_cpra)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2a579380",
   "metadata": {},
   "outputs": [],
   "source": [
    "len(igv_all[cols_cpra][igv_all['splice_change']=='yes'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "998f54c2",
   "metadata": {},
   "source": [
    "### 24.08."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "036577b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "igv_all = merge_xlsx('2021-08-18_all_vars_analyzed.xlsx')\n",
    "igv_outcome = perc_cat(igv_all)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "560d7df3",
   "metadata": {},
   "outputs": [],
   "source": [
    "search_col(igv_all,'hgv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "38f9031d",
   "metadata": {},
   "outputs": [],
   "source": [
    "igv_all['ClinVar_Pathogenicity'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f731908e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "asdf = igv_all[(igv_all['splice_change']=='yes')\n",
    "               &(igv_all['ClinVar_Pathogenicity'].isin(['Uncertain Significance', np.nan]))\n",
    "               &(igv_all['Splice Project Gene Priority']!='Cold (387-ACMG-MASTER)')\n",
    "               &(~igv_all['HUGO_Symbol'].isin(gus))\n",
    "               &(((igv_all['PosExonRefSeqAccession']>2)\n",
    "                  &(igv_all['PosExon_type'].isin(['outsideDonorSite', 'outsideAcceptorSite'])))\n",
    "                 |((igv_all['PosExonRefSeqAccession']>0)\n",
    "                   &(igv_all['PosExon_type'].isin(['insideDonorSite', 'insideAcceptorSite']))))\n",
    "       ]#.drop_duplicates(subset=cols_cpra)\n",
    "len(asdf[['shortcut','splice_change', 'too_low_coverage', 'PosExonRefSeqAccession', 'PosExon_type', 'INFO_IGV']])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ef92be04",
   "metadata": {},
   "outputs": [],
   "source": [
    "#xlsx_df(asdf, '2021-08-24_hot_warm_uncertain_unusual')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b1f4752d",
   "metadata": {},
   "source": [
    "### 18.08."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9153c1bb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# previously 4 duplicates -> newest version, more info...\n",
    "# newest file (2021-08-18_all_vars_analyzed.xlsx) no duplicates anymore\n",
    "igv_all = merge_xlsx('2021-08-18_all_vars_analyzed.xlsx')\n",
    "igv_outcome = perc_cat(igv_all)\n",
    "#igv_outcome.to_excel('02_output_analysis/2021-08-18_outcome_igv.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2c71cdb7",
   "metadata": {},
   "outputs": [],
   "source": [
    "#xlsx_df(igv_all, '2021-08-19_all_vars_analyzed')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "510cc7d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "#for coverage, splice_change, cat in zip(['no', 'no', 'no', 'yes', 'yes', 'yes'], \n",
    "#                                        ['yes', 'maybe', 'no', 'yes', 'maybe', 'no'], \n",
    "#                                        ['sufficient, yes', 'sufficient, maybe', 'sufficient, no', \n",
    "#                                         'not sufficient, yes', 'not sufficient, maybe', 'not sufficient, no']):\n",
    "#    print(cat, len(igv_all[((igv_all['SpliceAI_pred_DS_AG']>0.5)|\n",
    "#                   (igv_all['SpliceAI_pred_DS_AL']>0.5)|\n",
    "#                   (igv_all['SpliceAI_pred_DS_DG']>0.5)|\n",
    "#                   (igv_all['SpliceAI_pred_DS_DL']>0.5))&\n",
    "#                  (igv_all['too_low_coverage']==coverage)&\n",
    "#                  (igv_all['splice_change']==splice_change)].groupby(cols_cpra)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "712b079c",
   "metadata": {},
   "outputs": [],
   "source": [
    "transp_outcome = igv_outcome.transpose(copy=True).reset_index().drop(index=[7,8,9,10,11])\n",
    "transp_outcome = transp_outcome.rename(columns={i:cat for i,cat in zip(transp_outcome.columns, \n",
    "                                                                       list(transp_outcome.loc[0]))}).drop(index=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5376baca",
   "metadata": {},
   "outputs": [],
   "source": [
    "#transp_outcome.to_pickle('00_dataframes/transp_igv_outcome')\n",
    "#igv_all.to_pickle('00_dataframes/2021-08-19_igv_outcome')\n",
    "#igv_outcome.to_pickle('00_dataframes/2021-08-19_igv_perc_table')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d63cb8b3",
   "metadata": {},
   "source": [
    "### 12.08. new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c3720f3b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# newest version up to 2021-08-12 13:30 2021-08-12_all_vars_analyzed.xlsx\n",
    "new_vars = merge_xlsx('2021-08-12_all_vars_analyzed.xlsx')\n",
    "new_vars = make_float(new_vars)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6acdd888",
   "metadata": {},
   "outputs": [],
   "source": [
    "new_vars.to_pickle('00_dataframes/igv_vars')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cf9612ea",
   "metadata": {},
   "outputs": [],
   "source": [
    "#len(new_vars[((new_vars['SpliceAI_pred_DS_AG']>0.9)|\n",
    "#              (new_vars['SpliceAI_pred_DS_AL']>0.9)|\n",
    "#              (new_vars['SpliceAI_pred_DS_DG']>0.9)|\n",
    "#              (new_vars['SpliceAI_pred_DS_DL']>0.9))&\n",
    "#             (new_vars['too_low_coverage']=='yes')&\n",
    "#             (new_vars['splice_change']=='no')].groupby(\n",
    "#    cols_cpra))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cc401ed5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# interesting variants for further analysis on clinical patient basis\n",
    "int_vars = new_vars[(new_vars['Splice Project Gene Priority']==\n",
    "             'Hot (ACMG / MASTER)')&\n",
    "             (new_vars['too_low_coverage']=='no')&\n",
    "             (new_vars['splice_change']=='yes')]\n",
    "\n",
    "int_vars[['#CHROM', 'POS', 'REF', 'ALT', 'MaxEntScan_alt', 'MaxEntScan_diff', 'MaxEntScan_ref', \n",
    "                    'CADD_PHRED', 'SpliceAI_pred_DP_AG', 'SpliceAI_pred_DP_AL', 'SpliceAI_pred_DP_DG', \n",
    "                    'SpliceAI_pred_DP_DL', 'SpliceAI_pred_DS_AG', 'SpliceAI_pred_DS_AL', 'SpliceAI_pred_DS_DG', \n",
    "                    'SpliceAI_pred_DS_DL', 'SpliceAI_pred_SYMBOL', 'ada_score', 'rf_score',\n",
    "                    'PosExonRefSeqAccession', 'PosExon_type', 'ClinVar_Pathogenicity', \n",
    "                    'consensus_sequence_IGV_region_left', 'consensus_sequence_IGV_sequence_left', \n",
    "                    'consensus_sequence_IGV_region_right', 'consensus_sequence_IGV_sequence_right', 'INFO_IGV', \n",
    "                    'too_low_coverage', 'splice_change', 'shortcut', 'ClinVar_Traits', 'PID']].to_excel('01_xlsx_files/int_vars_pats_hot.xlsx')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "15a06f7b",
   "metadata": {},
   "source": [
    "### 12.08.2021"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a13a7c96",
   "metadata": {},
   "outputs": [],
   "source": [
    "# newest version up to 2021-08-12 10:00 2021-08-11_all_vars_analyzed.xlsx\n",
    "#new_vars = merge_xlsx('2021-08-11_all_vars_analyzed.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4b4fb4fc",
   "metadata": {},
   "outputs": [],
   "source": [
    "warm_vars = make_float(all_vars[(all_vars['Splice Project Gene Priority']=='Warm (MASTER-ACMG)')&\n",
    "                                (~all_vars['HUGO_Symbol'].isin(gus))&\n",
    "                                (~all_vars['HGVSg'].isin(new_vars['HGVSg']))].copy())\n",
    "warm_splai = warm_vars[(warm_vars['SpliceAI_pred_DS_AG']>0.4)|(warm_vars['SpliceAI_pred_DS_AL']>0.4)|\n",
    "                       (warm_vars['SpliceAI_pred_DS_DG']>0.4)|(warm_vars['SpliceAI_pred_DS_DL']>0.4)].copy()\n",
    "warm_dbsc = warm_vars[(warm_vars['ada_score']>0.6)&(warm_vars['rf_score']>0.6)&\n",
    "                      (~warm_vars['HGVSg'].isin(warm_splai['HGVSg']))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "99ced857",
   "metadata": {},
   "outputs": [],
   "source": [
    "#xlsx_df(conf_xlsx(warm_splai), 'warm_splai_o04')\n",
    "#xlsx_df(conf_xlsx(warm_dbsc), 'warm_dbsc_o06')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "feae8194",
   "metadata": {},
   "source": [
    "### 11.08. new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d35d7e22",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_xlsx = pd.read_excel('01_xlsx_files/2021-08-11_variants_seen_no_dup_new.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ddfe7c91",
   "metadata": {},
   "outputs": [],
   "source": [
    "# duplicates for specific variants\n",
    "# 434 to 428 variants\n",
    "grpd_xlsx = df_xlsx.groupby(by=cols_cpra)\n",
    "grpd_lst = [(i,len(df)) for i,df in grpd_xlsx if len(df)>1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ed68fc93",
   "metadata": {},
   "outputs": [],
   "source": [
    "# to search in each group for best keep\n",
    "#grpd_xlsx.get_group(grpd_lst[5][0])[['#CHROM', 'POS', 'REF', 'ALT', 'INFO_IGV','too_low_coverage','splice_change', \n",
    "                                     #'PIDs', 'consensus_sequence_IGV_region_left', \n",
    "                                     #'consensus_sequence_IGV_sequence_left','consensus_sequence_IGV_region_right',\n",
    "                                     #'consensus_sequence_IGV_sequence_right']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9529914d",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_xlsx = df_xlsx.drop(index=[401, 403, 387, 409, 430, 399])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c1d2a705",
   "metadata": {},
   "outputs": [],
   "source": [
    "vars_df = pd.merge(df_xlsx[['#CHROM', 'POS', 'REF', 'ALT', 'consensus_sequence_IGV_region_left',\n",
    "                            'consensus_sequence_IGV_sequence_left', 'consensus_sequence_IGV_region_right',\n",
    "                            'consensus_sequence_IGV_sequence_right', 'INFO_IGV', 'too_low_coverage',\n",
    "                            'splice_change']], all_vars, on=cols_cpra, how='left').reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ef7fc6e9",
   "metadata": {},
   "outputs": [],
   "source": [
    "vars_df['shortcut'] = vars_df['#CHROM']+'_'+vars_df['POS'].astype(str)+'_'+vars_df['REF']+'_'+vars_df['ALT']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3f329e5f",
   "metadata": {},
   "outputs": [],
   "source": [
    "#xlsx_df(vars_df, '2021-08-11_all_vars_analyzed')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a9ac9b3c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# filter out cold and gus variants\n",
    "rel_vars = vars_df[(vars_df['Splice Project Gene Priority']!='Cold (387-ACMG-MASTER)')&\n",
    "                   (~vars_df['HUGO_Symbol'].isin(gus))].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d2cb7664",
   "metadata": {},
   "outputs": [],
   "source": [
    "len(all_vars[(all_vars['Splice Project Gene Priority']!='Cold (387-ACMG-MASTER)')&\n",
    "             (~all_vars['HUGO_Symbol'].isin(gus))]['HUGO_Symbol'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "73cb5beb",
   "metadata": {},
   "outputs": [],
   "source": [
    "rel_vars[['#CHROM', 'POS', 'ALT', 'REF', 'PID', 'ClinVar_Pathogenicity', 'HGVSc', 'PosExonRefSeqAccession', 'PosExon_type', \n",
    "         'HUGO_Symbol', 'shortcut']][rel_vars['splice_change']=='yes'].drop_duplicates(cols_cpra)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7c870126",
   "metadata": {},
   "outputs": [],
   "source": [
    "vars_df[['#CHROM', 'POS', 'ALT', 'REF', 'PID', 'ClinVar_Pathogenicity', 'HGVSc', 'PosExonRefSeqAccession', 'PosExon_type', \n",
    "         'HUGO_Symbol', 'shortcut']][(vars_df['Splice Project Gene Priority']=='Hot (ACMG / MASTER)')&(vars_df['splice_change']=='yes')]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "635b9edf",
   "metadata": {},
   "source": [
    "### 11.08.2021"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e96d6cf9",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_xlsx = pd.read_excel('01_xlsx_files/2021-08-11_variants_seen_no_dup.xlsx')\n",
    "all_vars = pd.read_pickle('00_dataframes/all_vars')\n",
    "\n",
    "all_vars['#CHROM'] = pd.Categorical(all_vars['#CHROM'],\n",
    "                                    categories=['1','2','3','4','5','6','7','8','9','10','11','12', '13', '14', \n",
    "                                                '15', '16', '17', '18', '19', '20', '21', '22', 'X', 'Y'],\n",
    "                                    ordered=True)\n",
    "all_vars = all_vars.sort_values(by=['#CHROM', 'POS', 'REF', 'ALT'], ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "282ebbc8",
   "metadata": {},
   "outputs": [],
   "source": [
    "vars_df = pd.merge(df_xlsx[['#CHROM', 'POS', 'REF', 'ALT', 'consensus_sequence_IGV_region_left',\n",
    "                            'consensus_sequence_IGV_sequence_left', 'consensus_sequence_IGV_region_right',\n",
    "                            'consensus_sequence_IGV_sequence_right', 'INFO_IGV', 'too_low_coverage',\n",
    "                            'splice_change', 'shortcut']], all_vars, on=cols_cpra, how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f499f80d",
   "metadata": {},
   "outputs": [],
   "source": [
    "hot_vars = make_float(all_vars[(all_vars['Splice Project Gene Priority']=='Hot (ACMG / MASTER)')&\n",
    "                               (~all_vars['HGVSg'].isin(vars_df['HGVSg']))&\n",
    "                               (all_vars['ada_score'].notnull())].copy())\n",
    "hot_vars[['ada_score', 'rf_score']] = hot_vars[['ada_score', 'rf_score']].astype(float)\n",
    "hot_ada_rf = hot_vars[(hot_vars['ada_score']>0.1)&(hot_vars['rf_score']>0.1)].copy()\n",
    "\n",
    "for i in ['consensus_sequence_IGV_region_left', 'consensus_sequence_IGV_sequence_left',  \n",
    "          'consensus_sequence_IGV_region_right', 'consensus_sequence_IGV_sequence_right', \n",
    "          'INFO_IGV', 'too_low_coverage', 'splice_change', 'shortcut']:\n",
    "    hot_ada_rf[i] = np.nan"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c8e335e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "#xlsx_df(hot_ada_rf, 'hot_genes_adarf_o0')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "49f7d885",
   "metadata": {},
   "outputs": [],
   "source": [
    "hot_pos = hot_vars[(hot_vars['PosExonRefSeqAccession']<6)&(~hot_vars['HGVSg'].isin(hot_ada_rf['HGVSg']))].copy()\n",
    "for i in ['consensus_sequence_IGV_region_left', 'consensus_sequence_IGV_sequence_left',  \n",
    "          'consensus_sequence_IGV_region_right', 'consensus_sequence_IGV_sequence_right', \n",
    "          'INFO_IGV', 'too_low_coverage', 'splice_change', 'shortcut']:\n",
    "    hot_pos[i] = np.nan"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f1e67e70",
   "metadata": {},
   "outputs": [],
   "source": [
    "#xlsx_df(hot_pos, 'hot_genes_pos_l6')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "985912f3",
   "metadata": {},
   "source": [
    "### 10.08.2021"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "76a2ffc4",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_xlsx = pd.read_excel('01_xlsx_files/2021-08-10_variants_seen_no_dup.xlsx')\n",
    "all_vars = pd.read_pickle('00_dataframes/all_vars')\n",
    "\n",
    "all_vars['#CHROM'] = pd.Categorical(all_vars['#CHROM'],\n",
    "                                    categories=['1','2','3','4','5','6','7','8','9','10','11','12', '13', '14', \n",
    "                                                '15', '16', '17', '18', '19', '20', '21', '22', 'X', 'Y'],\n",
    "                                    ordered=True)\n",
    "all_vars = all_vars.sort_values(by=['#CHROM', 'POS', 'REF', 'ALT'], ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "11627199",
   "metadata": {},
   "outputs": [],
   "source": [
    "vars_df = pd.merge(df_xlsx[['#CHROM', 'POS', 'REF', 'ALT', 'consensus_sequence_IGV_region_left',\n",
    "                            'consensus_sequence_IGV_sequence_left', 'consensus_sequence_IGV_region_right',\n",
    "                            'consensus_sequence_IGV_sequence_right', 'INFO_IGV', 'too_low_coverage',\n",
    "                            'splice_change', 'shortcut']], all_vars, on=cols_cpra, how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ddec686c",
   "metadata": {},
   "outputs": [],
   "source": [
    "hot_vars = make_float(all_vars[(all_vars['Splice Project Gene Priority']=='Hot (ACMG / MASTER)')&\n",
    "                               (~all_vars['HGVSg'].isin(vars_df['HGVSg']))].copy())\n",
    "hot_splai = hot_vars[(hot_vars['SpliceAI_pred_DS_AG']>0.1)|(hot_vars['SpliceAI_pred_DS_AL']>0.1)|\n",
    "                     (hot_vars['SpliceAI_pred_DS_DG']>0.1)|(hot_vars['SpliceAI_pred_DS_DL']>0.1)].copy()\n",
    "\n",
    "for i in ['consensus_sequence_IGV_region_left', 'consensus_sequence_IGV_sequence_left',  \n",
    "          'consensus_sequence_IGV_region_right', 'consensus_sequence_IGV_sequence_right', \n",
    "          'INFO_IGV', 'too_low_coverage', 'splice_change', 'shortcut']:\n",
    "    hot_splai[i] = np.nan"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "73e87f13",
   "metadata": {},
   "outputs": [],
   "source": [
    "#xlsx_df(hot_splai, 'hot_genes_splai_o0')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0271073c",
   "metadata": {},
   "source": [
    "### 09.08.2021"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fc6316d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_xlsx = pd.read_excel('01_xlsx_files/2021-08-07_variants_seen_no_dup.xlsx')\n",
    "all_vars = pd.read_pickle('00_dataframes/all_vars')\n",
    "red_xlsx = pd.read_excel('01_xlsx_files/2021-08-09_redund_vars.xlsx')\n",
    "\n",
    "all_vars['#CHROM'] = pd.Categorical(all_vars['#CHROM'],\n",
    "                                    categories=['1','2','3','4','5','6','7','8','9','10','11','12', '13', '14', \n",
    "                                                '15', '16', '17', '18', '19', '20', '21', '22', 'X', 'Y'],\n",
    "                                    ordered=True)\n",
    "all_vars = all_vars.sort_values(by=['#CHROM', 'POS', 'REF', 'ALT'], ignore_index=True)\n",
    "\n",
    "\n",
    "# no duplicates in df_xlsx --> merge with all_vars\n",
    "\n",
    "vars_df = pd.merge(df_xlsx[['#CHROM', 'POS', 'REF', 'ALT', 'consensus_sequence_IGV_region_left',\n",
    "                            'consensus_sequence_IGV_sequence_left', 'consensus_sequence_IGV_region_right',\n",
    "                            'consensus_sequence_IGV_sequence_right', 'INFO_IGV', 'too_low_coverage',\n",
    "                            'splice_change', 'shortcut']], all_vars, on=cols_cpra, how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cd5d645d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# for variants in more than one patient\n",
    "grpd_all_vars = vars_df.groupby(cols_cpra)\n",
    "grpd_lst = [(i,len(df)) for i,df in grpd_all_vars if len(df)>1]\n",
    "\n",
    "redund_df = pd.DataFrame()\n",
    "for i, l in grpd_lst:\n",
    "    redund_df = redund_df.append(grpd_all_vars.get_group(i))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "22447020",
   "metadata": {},
   "outputs": [],
   "source": [
    "#xlsx_df(redund_df, '2021-08-09_redund_vars')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "581cb8f9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# combine new characterized redundant variants with other variants\n",
    "new_df_xlsx = df_xlsx[~df_xlsx['shortcut'].isin(redund_df['shortcut'])]\n",
    "new_df_xlsx = new_df_xlsx.append(red_xlsx)\n",
    "new_df_xlsx = new_df_xlsx.reset_index(drop=True)\n",
    "new_vars_df = pd.merge(new_df_xlsx[['#CHROM', 'POS', 'REF', 'ALT', 'consensus_sequence_IGV_region_left',\n",
    "                            'consensus_sequence_IGV_sequence_left', 'consensus_sequence_IGV_region_right',\n",
    "                            'consensus_sequence_IGV_sequence_right', 'INFO_IGV', 'too_low_coverage',\n",
    "                            'splice_change', 'shortcut']], all_vars, on=cols_cpra, how='left')\n",
    "new_onl_vars = new_vars_df.drop_duplicates(cols_cpra).reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d3139753",
   "metadata": {},
   "source": [
    "#### IGV splice effect seen"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "20185ab8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# splice effect seen, safe (not too low coverage, only if sure (splice_change == yes)) --> 88 variants\n",
    "# if splice_change == maybe --> 38 variants, if splice_change == no --> 136 variants\n",
    "# if too low coverage and splice_change == no --> 75 variants, splice_change == maybe --> 27 variants\n",
    "\n",
    "#len(new_onl_vars[(new_onl_vars['too_low_coverage']=='yes')&\n",
    "                 #(new_onl_vars['splice_change']=='yes')].reset_index(drop=True))\n",
    "\n",
    "new_spl_yes = new_onl_vars[(new_onl_vars['too_low_coverage']=='no')&\n",
    "                           (new_onl_vars['splice_change']=='yes')].reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "edf783b4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# 106 variants with spliceAI score > 0.9\n",
    "# 47 variants with significant effect in splicing, 18 variants no effect, 9 maybe effect\n",
    "# 32 too low coverage, 19 no effect, 13 maybe effect\n",
    "for_splai = make_float(new_onl_vars[new_onl_vars['SpliceAI_pred_DS_AG'].notnull()].copy())\n",
    "len(for_splai[((for_splai['SpliceAI_pred_DS_AG']>0.9)|(for_splai['SpliceAI_pred_DS_AL']>0.9)|\n",
    "               (for_splai['SpliceAI_pred_DS_DG']>0.9)|(for_splai['SpliceAI_pred_DS_DL']>0.9))&\n",
    "              (for_splai['too_low_coverage']=='yes')&(for_splai['splice_change']=='no')\n",
    "             ].groupby(by=cols_cpra))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9028883a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# 63 variants previously of uncertain significance or not in ClinVar with splice change and enough coverage\n",
    "for_splai[['#CHROM', 'POS', 'REF', 'ALT', 'consensus_sequence_IGV_region_left', \n",
    "           'consensus_sequence_IGV_sequence_left', 'consensus_sequence_IGV_region_right',\n",
    "           'consensus_sequence_IGV_sequence_right', 'INFO_IGV', 'too_low_coverage', 'splice_change', \n",
    "           'shortcut', 'HGVSc', 'HGVSg', 'HGVSp']][((for_splai['ClinVar_Pathogenicity']=='Uncertain Significance')|\n",
    "                         (for_splai['ClinVar_Pathogenicity']=='uncertain_significance')|\n",
    "                         (for_splai['ClinVar_Pathogenicity']=='Uncertain significance')|\n",
    "                         (for_splai['ClinVar_Pathogenicity'].isnull()))&\n",
    "                        (for_splai['too_low_coverage']=='no')&\n",
    "                        (for_splai['splice_change']=='yes')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e2157439",
   "metadata": {},
   "outputs": [],
   "source": [
    "for_MES = make_float(vars_df[vars_df['MaxEntScan_diff']].notnull().copy())\n",
    "for_MES_disr = for_MES[(for_MES['MaxEntScan_diff']>=0)&(for_MES['MaxEntScan_alt']<6.2)]\n",
    "for_MES_new = for_MES[(['MaxEntScan_diff']<0)&(['MaxEntScan_alt']>8.5)]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dd68e0f3",
   "metadata": {},
   "source": [
    "#### Variants per prioritization criterion"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b2f8c002",
   "metadata": {},
   "outputs": [],
   "source": [
    "#vars_grpd_splai = vars_grpd[vars_grpd['SpliceAI_pred_DP_AG'].notnull()].copy()\n",
    "#vars_grpd_splai = make_float(vars_grpd_splai)\n",
    "\n",
    "#vars_grpd_splai = vars_grpd_splai[cols_cpra][((vars_grpd_splai['SpliceAI_pred_DS_AG']>0.9)|\n",
    "                                              #(vars_grpd_splai['SpliceAI_pred_DS_AL']>0.9)|\n",
    "                                              #(vars_grpd_splai['SpliceAI_pred_DS_DG']>0.9)|\n",
    "                                              #(vars_grpd_splai['SpliceAI_pred_DS_DL']>0.9))\n",
    "                                    #].drop_duplicates()\n",
    "#vars_grpd_splai.reset_index(drop=True, inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "662072e8",
   "metadata": {},
   "source": [
    "## Input"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "01478a27",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_xlsx = pd.read_excel('01_xlsx_files/2021-08-05_all_variants.xlsx')\n",
    "all_vars = pd.read_pickle('00_dataframes/all_vars')\n",
    "\n",
    "all_vars['#CHROM'] = pd.Categorical(all_vars['#CHROM'],\n",
    "                                    categories=['1','2','3','4','5','6','7','8','9','10','11','12', '13', '14', \n",
    "                                                '15', '16', '17', '18', '19', '20', '21', '22', 'X', 'Y'],\n",
    "                                    ordered=True)\n",
    "all_vars = all_vars.sort_values(by=['#CHROM', 'POS', 'REF', 'ALT'], ignore_index=True)\n",
    "\n",
    "cols_cpra = ['#CHROM', 'POS', 'REF', 'ALT']\n",
    "vars_grpd = all_vars.drop_duplicates(subset=cols_cpra)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7892ffd5",
   "metadata": {},
   "source": [
    "## Drop duplicates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f3170a9b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# duplicates for specific variants\n",
    "#len(df_xlsx), len(df_xlsx.drop_duplicates(subset=cols_cpra)) #(328, 272)\n",
    "grpd_xlsx = df_xlsx.groupby(by=cols_cpra)\n",
    "grpd_lst = [(i,len(df)) for i,df in grpd_xlsx if len(df)>1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b4432b15",
   "metadata": {},
   "outputs": [],
   "source": [
    "# to search in each group for best keep\n",
    "# grpd_xlsx.get_group(grpd_lst[42][0])[['#CHROM', 'POS', 'REF', 'ALT', 'INFO_IGV','too_low_coverage','splice_change', \n",
    "                                     #'PIDs', 'consensus_sequence_IGV_region_left', \n",
    "                                     #'consensus_sequence_IGV_sequence_left','consensus_sequence_IGV_region_right',\n",
    "                                     #'consensus_sequence_IGV_sequence_right']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "428256e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "for_drop = [7, 17, 9, 36, 78, 74, 90, 91, 89, 101, 104, 102, 105, 113, 122, 148, 152, 149, 150, 155, 157, 154, \n",
    "            166, 165, 183, 177, 174, 185, 202, 220, 211, 215, 205, 207, 229, 239, 235, 241, 250, 265, 267, 264, \n",
    "            263, 268, 271, 280, 284, 287, 289, 292, 304, 314, 316, 313, 325, 327]\n",
    "df_xlsx.drop(index=for_drop, inplace=True)\n",
    "df_xlsx.reset_index(drop=True, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e847adaf",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_xlsx['shortcut'] = (df_xlsx['#CHROM'].astype(str) + '_' + df_xlsx['POS'].astype(str) + '_' + \n",
    "                       df_xlsx['REF'].astype(str) + '_' + df_xlsx['ALT'].astype(str))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "672891b8",
   "metadata": {},
   "outputs": [],
   "source": [
    "vars_df = pd.merge(df_xlsx[['#CHROM', 'POS', 'REF', 'ALT', 'consensus_sequence_IGV_region_left',\n",
    "                               'consensus_sequence_IGV_sequence_left', 'consensus_sequence_IGV_region_right',\n",
    "                               'consensus_sequence_IGV_sequence_right', 'INFO_IGV', 'too_low_coverage',\n",
    "                               'splice_change', 'shortcut']], all_vars, on=cols_cpra, how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "37784c53",
   "metadata": {},
   "outputs": [],
   "source": [
    "#xlsx_df(vars_df, 'variants_seen_no_dup')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3ef32aa7",
   "metadata": {},
   "source": [
    "## New excel files without duplicates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "58f19b98",
   "metadata": {},
   "outputs": [],
   "source": [
    "splAI_high = vars_df[vars_df['SpliceAI_pred_DS_AG'].notnull()].copy()\n",
    "splAI_high = make_float(splAI_high)\n",
    "splAI_high = splAI_high[((splAI_high['SpliceAI_pred_DS_AG']>0.7)|\n",
    "                         (splAI_high['SpliceAI_pred_DS_AL']>0.7)|\n",
    "                         (splAI_high['SpliceAI_pred_DS_DG']>0.7)|\n",
    "                         (splAI_high['SpliceAI_pred_DS_DL']>0.7))].drop_duplicates()\n",
    "\n",
    "for i in ['consensus_sequence_IGV_region_left', 'consensus_sequence_IGV_sequence_left', 'too_low_coverage',\n",
    "          'consensus_sequence_IGV_region_right', 'consensus_sequence_IGV_sequence_right', 'splice_change',\n",
    "          'INFO_IGV']:\n",
    "    splAI_high[i] = np.nan\n",
    "\n",
    "xlsx_df(splAI_high, 'new_splAI_high')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bdd95d82",
   "metadata": {},
   "outputs": [],
   "source": [
    "MES_high_disr = pd.read_excel('../04_annotations/03_prioritized_xlsx/new_MES_high_disr.xlsx')\n",
    "MES_high_disr['from_where'] = 'MES_high_disr'\n",
    "MES_high_new = pd.read_excel('../04_annotations/03_prioritized_xlsx/new_MES_high_new.xlsx')\n",
    "MES_high_new['from_where'] = 'MES_high_new'\n",
    "dbsc_high = pd.read_excel('../04_annotations/03_prioritized_xlsx/new_dbsc_high.xlsx')\n",
    "dbsc_high['from_where'] = 'dbsc_high'\n",
    "splAI_high_new = pd.read_excel('01_xlsx_files/new_splAI_high.xlsx')\n",
    "splAI_high_new['from_where'] = 'splAI_high'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "189a3d20",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_for_new = MES_high_disr.append(MES_high_new, ignore_index=True)\n",
    "all_for_new = all_for_new.append(dbsc_high, ignore_index=True)\n",
    "all_for_new = all_for_new.append(splAI_high_new, ignore_index=True).drop_duplicates(subset=cols_cpra, \n",
    "                                                                                    ignore_index=True)\n",
    "all_for_new['shortcut'] = (all_for_new['#CHROM'].astype(str) + '_' + all_for_new['POS'].astype(str) + '_' + \n",
    "                           all_for_new['REF'].astype(str) + '_' + all_for_new['ALT'].astype(str))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0938a91b",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_for_new = all_for_new[~all_for_new['shortcut'].isin(vars_df['shortcut'])].reset_index(drop=True)\n",
    "all_for_new.to_excel('01_xlsx_files/2021-08-06_rest_vars_nodup.xlsx')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "76198c98",
   "metadata": {},
   "source": [
    "## IGV splice effect seen"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e21ee0d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# splice effect seen, safe (not too low coverage, only if sure (splice_change = yes)) --> 80 variants\n",
    "spl_yes = vars_df[(vars_df['too_low_coverage']=='no')&(vars_df['splice_change']=='yes')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1c165309",
   "metadata": {},
   "outputs": [],
   "source": [
    "spl_yes_splai = spl_yes_df[spl_yes_df['SpliceAI_pred_DS_AG'].notnull()].copy()\n",
    "make_float(spl_yes_splai)\n",
    "spl_yes_splai_high = spl_yes_splai[((spl_yes_splai['SpliceAI_pred_DS_AG']>0.9)|\n",
    "            (spl_yes_splai['SpliceAI_pred_DS_AL']>0.9)|\n",
    "            (spl_yes_splai['SpliceAI_pred_DS_DG']>0.9)|\n",
    "            (spl_yes_splai['SpliceAI_pred_DS_DL']>0.9))].copy()\n",
    "len(spl_yes_splai), len(spl_yes_df), len(spl_yes_splai_high), len(spl_yes_splai_high[spl_yes_splai_high['Splice Project Gene Priority']=='Hot (ACMG / MASTER)'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2b278789",
   "metadata": {},
   "outputs": [],
   "source": [
    "# 106 variants with spliceAI score > 0.9\n",
    "# 46 variants with significant effect in splicing, 17 variants no effect, 33 too low coverage, 10 maybe effect\n",
    "for_splai = make_float(vars_df[vars_df['SpliceAI_pred_DS_AG'].notnull()].copy())\n",
    "len(for_splai[((for_splai['SpliceAI_pred_DS_AG']>0.9)|(for_splai['SpliceAI_pred_DS_AL']>0.9)|\n",
    "               (for_splai['SpliceAI_pred_DS_DG']>0.9)|(for_splai['SpliceAI_pred_DS_DL']>0.9))&\n",
    "              (for_splai['too_low_coverage']=='no')&(for_splai['splice_change']=='maybe')].groupby(by=cols_cpra))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "59aba0cc",
   "metadata": {},
   "outputs": [],
   "source": [
    "for_MES = make_float(vars_df[vars_df['MaxEntScan_diff']].notnull().copy())\n",
    "for_MES_disr = for_MES[(for_MES['MaxEntScan_diff']>=0)&(for_MES['MaxEntScan_alt']<6.2)]\n",
    "for_MES_new = for_MES[(['MaxEntScan_diff']<0)&(['MaxEntScan_alt']>8.5)]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "81d0df0e",
   "metadata": {},
   "source": [
    "## Variants per prioritization criterion"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "67a27d2f",
   "metadata": {},
   "outputs": [],
   "source": [
    "#vars_grpd_splai = vars_grpd[vars_grpd['SpliceAI_pred_DP_AG'].notnull()].copy()\n",
    "#vars_grpd_splai = make_float(vars_grpd_splai)\n",
    "\n",
    "#vars_grpd_splai = vars_grpd_splai[cols_cpra][((vars_grpd_splai['SpliceAI_pred_DS_AG']>0.9)|\n",
    "                                              #(vars_grpd_splai['SpliceAI_pred_DS_AL']>0.9)|\n",
    "                                              #(vars_grpd_splai['SpliceAI_pred_DS_DG']>0.9)|\n",
    "                                              #(vars_grpd_splai['SpliceAI_pred_DS_DL']>0.9))\n",
    "                                    #].drop_duplicates()\n",
    "#vars_grpd_splai.reset_index(drop=True, inplace=True)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
