{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "ada78249",
   "metadata": {},
   "source": [
    "To prioritize splice-relevant variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "798a5fd1",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "b70fa9f5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# function to search for specific name in column\n",
    "def search_col(inp):\n",
    "    return [i for i in naga_all.columns if inp.lower() in i.lower()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "a51594dc",
   "metadata": {},
   "outputs": [],
   "source": [
    "# function to make spliceAI prediction scores into floats\n",
    "def make_float(df):\n",
    "    for i in ['AG', 'AL', 'DG', 'DL']:\n",
    "        df[('SpliceAI_pred_DS_'+i)]=df[('SpliceAI_pred_DS_'+i)].astype(float)\n",
    "    for i in ['ada_score', 'rf_score', 'MaxEntScan_diff', 'MaxEntScan_alt', 'MaxEntScan_ref']:\n",
    "        df[i] = df[i].astype(float)\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "a73b7c80",
   "metadata": {},
   "outputs": [],
   "source": [
    "# function to make excel tables with one variant per row\n",
    "columns_df_prior = ['#CHROM', 'POS', 'REF', 'ALT', 'MaxEntScan_alt', 'MaxEntScan_diff', 'MaxEntScan_ref', \n",
    "                    'CADD_PHRED', 'SpliceAI_pred_DP_AG', 'SpliceAI_pred_DP_AL', 'SpliceAI_pred_DP_DG', \n",
    "                    'SpliceAI_pred_DP_DL', 'SpliceAI_pred_DS_AG', 'SpliceAI_pred_DS_AL', 'SpliceAI_pred_DS_DG', \n",
    "                    'SpliceAI_pred_DS_DL', 'SpliceAI_pred_SYMBOL', 'ada_score', 'rf_score',\n",
    "                    'PosExonRefSeqAccession', 'PosExon_type', 'ClinVar_Pathogenicity', 'ClinVar_Traits', 'PID']\n",
    "columns_df_xlsx = columns_df_prior[:-1]\n",
    "columns_df_xlsx.extend(['PIDs', 'ct_PIDs'])\n",
    "cols_cpra = ['#CHROM', 'POS', 'REF', 'ALT']\n",
    "\n",
    "def xlsx_df(prio_df):\n",
    "    lists = []\n",
    "    for p,df in prio_df.groupby(by=['#CHROM', 'POS', 'REF', 'ALT']):\n",
    "        df = df.reset_index(drop=True)\n",
    "        list_df = [df.iloc[0][i] for i in columns_df_prior[:-1]]\n",
    "        list_df.append(list(df['PID']))\n",
    "        lists.append(list_df)\n",
    "    return pd.DataFrame(lists, columns=columns_df_xlsx)\n",
    "\n",
    "def xlsx_vars(prio_vars_df, name):\n",
    "    merged = pd.merge(naga_all[columns_df_prior], prio_vars_df, how='inner')\n",
    "    lists = []\n",
    "    for p,df in merged.groupby(by=['#CHROM', 'POS', 'REF', 'ALT']):\n",
    "        df = df.reset_index(drop=True)\n",
    "        list_df = [df.iloc[0][i] for i in columns_df_prior[:-1]]\n",
    "        list_df.extend([list(df['PID']), len(df['PID'])])\n",
    "        lists.append(list_df)\n",
    "    new_df = pd.DataFrame(lists, columns=columns_df_xlsx)\n",
    "    new_df['#CHROM'] = pd.Categorical(new_df['#CHROM'],\n",
    "                                      categories=['1','2','3','4','5','6','7','8','9','10','11','12', '13', '14', \n",
    "                                                  '15', '16', '17', '18', '19', '20', '21', '22', 'X', 'Y'],\n",
    "                                      ordered=True)\n",
    "    new_df.sort_values(by=['ct_PIDs', '#CHROM', 'POS', 'REF', 'ALT'], inplace=True, ignore_index=True, \n",
    "                       ascending=[False, True, True, True, True])\n",
    "    new_df.drop(axis='columns', columns='ct_PIDs', inplace=True)\n",
    "    new_df.to_excel('03_prioritized_xlsx/%s.xlsx'%name, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "9abaf299",
   "metadata": {},
   "outputs": [],
   "source": [
    "# overview on variants counts for specific prioritizations\n",
    "# any spliceAI score > 0.9\n",
    "def splai_o09(df):\n",
    "    return df[((df['SpliceAI_pred_DS_AG']>0.9)|(df['SpliceAI_pred_DS_AL']>0.9)|\n",
    "               (df['SpliceAI_pred_DS_DG']>0.9)|(df['SpliceAI_pred_DS_DL']>0.9))].copy()\n",
    "\n",
    "# ada/rf score > 0.6\n",
    "def ada_rf_o06(df):\n",
    "    return df[(df['ada_score']>0.6)&\n",
    "              (df['rf_score']>0.6)].copy()\n",
    "\n",
    "# MaxEntScan high disruption of native splice site\n",
    "def MES_disr(df):\n",
    "    return df[(df['MaxEntScan_diff']>=0)&\n",
    "              (df['MaxEntScan_alt']<6.2)].copy()\n",
    "\n",
    "# MaxEntScan high possibility of creating new splice site\n",
    "def MES_new(df):\n",
    "    return df[(df['MaxEntScan_diff']<0)&\n",
    "              (df['MaxEntScan_alt']>8.5)].copy()\n",
    "\n",
    "# position relative to exon/intron boundary less than 3 and in intron\n",
    "def pos_rel_ei(df):\n",
    "    return df[((df['PosExonRefSeqAccession']<3)&\n",
    "               (df['PosExon_type'].isin(['outsideAcceptorSite', 'outsideDonorSite'])))|\n",
    "              ((df['PosExonRefSeqAccession']<2)&\n",
    "               (df['PosExon_type'].isin(['insideAcceptorSite', 'insideDonorSite'])))].copy()\n",
    "\n",
    "# variant categorization per hot/warm/cold genes\n",
    "def gene_prio(df):\n",
    "    gus = ['AKT2', 'AKT3', 'ATR', 'ATRX', 'CDKN1A', 'CDKN2B', 'CDKN2C', 'CDKN2D', 'CEBPA', 'CEP57', 'DAXX', 'EGFR', \n",
    "           'EGLN1', 'EGLN2', 'EPAS1', 'EPHB2', 'ABRAXAS1', 'FANCM', 'GPRC5A', 'HNF1A', 'HORMAD1', 'HORMAD2', 'JAK2',\n",
    "           'KIF1B', 'MAP3K1', 'MAP3K6', 'MDH2', 'MLH3', 'MMS19', 'MN1', 'MRE11', 'PALLD', 'PIK3C2G', 'PIK3R2', \n",
    "           'PMS1', 'PRF1', 'PTCH2', 'RAD50', 'RAD51', 'RECQL', 'RINT1']\n",
    "    not_gus = (set(gus) ^ set(df['HUGO_Symbol'][df['Splice Project Gene Priority']=='Warm (MASTER-ACMG)'].unique()))\n",
    "    df_lst = []\n",
    "    for i,lst in zip(['Hot (ACMG / MASTER)', 'Warm (MASTER-ACMG)', 'Warm (MASTER-ACMG)', 'Cold (387-ACMG-MASTER)'], \n",
    "                     [[''], gus, not_gus, ['']]):\n",
    "        df_lst.append(df[(df['Splice Project Gene Priority']==i)&\n",
    "                         (~df['HUGO_Symbol'].isin(lst))].copy())\n",
    "    return df_lst\n",
    "\n",
    "def var_scores(df):\n",
    "    table = pd.DataFrame(columns=['category', 'variant_count', 'row_count'])\n",
    "    for data,n in zip([splai_o09(df), ada_rf_o06(df), MES_disr(df), MES_new(df), pos_rel_ei(df)]+gene_prio(df),\n",
    "                      ['splai_o0.9', 'ada_rf_o0.6', 'MES_high_disr', 'MES_high_new', 'pos_rel_e_i', \n",
    "                       'gene_prio_hot', 'gene_prio_warm_wo_gus', 'gene_prio_gus', 'gene_prio_cold']):\n",
    "        table = table.append({'category':n, 'variant_count':len(data.drop_duplicates(subset=cols_cpra)), \n",
    "                              'row_count':len(data)}, ignore_index=True)\n",
    "    return table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "6a821da0",
   "metadata": {},
   "outputs": [],
   "source": [
    "naga_all=pd.read_pickle('00_dataframes/naga_vep_floss_hboc_splai')\n",
    "naga_all['#CHROM'] = pd.Categorical(naga_all['#CHROM'],\n",
    "                                    categories=['1','2','3','4','5','6','7','8','9','10','11','12', '13', '14', \n",
    "                                                '15', '16', '17', '18', '19', '20', '21', '22', 'X', 'Y'],\n",
    "                                    ordered=True)\n",
    "naga_all = naga_all.sort_values(by=['#CHROM', 'POS', 'REF', 'ALT'], ignore_index=True)\n",
    "naga_all['PosExon_type'] = naga_all['PosExon_type'].replace({'insideDonor':'insideDonorSite', \n",
    "                                                             'outsideDonor':'outsideDonorSite'})\n",
    "naga_all = make_float(naga_all)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "865bf950",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "5211ec15",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['RNA_EXPRESSION', 'RNA_INFO', 'RNA_VAF', 'AF_RNA', 'AF_RNA_noSkipReads']"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_col('rna')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "4e044e14",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>RNA_EXPRESSION</th>\n",
       "      <th>RNA_INFO</th>\n",
       "      <th>RNA_VAF</th>\n",
       "      <th>AF_RNA</th>\n",
       "      <th>AF_RNA_noSkipReads</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>201</th>\n",
       "      <td>1</td>\n",
       "      <td>17345454</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>exact_match</td>\n",
       "      <td>AD=0,12;VAF=1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314</th>\n",
       "      <td>1</td>\n",
       "      <td>23191718</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>not_expressed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>319</th>\n",
       "      <td>1</td>\n",
       "      <td>23222908</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exact_match</td>\n",
       "      <td>AD=6,3;VAF=0.3333333333333333</td>\n",
       "      <td>0.3333333333333333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>667</th>\n",
       "      <td>1</td>\n",
       "      <td>45797228</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exact_match</td>\n",
       "      <td>AD=15,7;VAF=0.3181818181818182</td>\n",
       "      <td>0.3181818181818182</td>\n",
       "      <td>0.30</td>\n",
       "      <td>0.35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2356</th>\n",
       "      <td>2</td>\n",
       "      <td>96920727</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>exact_match</td>\n",
       "      <td>AD=81,118;VAF=0.592964824120603</td>\n",
       "      <td>0.592964824120603</td>\n",
       "      <td>0.58</td>\n",
       "      <td>0.59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2501</th>\n",
       "      <td>2</td>\n",
       "      <td>169820808</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>exact_match</td>\n",
       "      <td>AD=12,10;VAF=0.45454545454545453</td>\n",
       "      <td>0.45454545454545453</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2508</th>\n",
       "      <td>2</td>\n",
       "      <td>169828370</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>not_expressed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2885</th>\n",
       "      <td>2</td>\n",
       "      <td>212293119</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>not_expressed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2899</th>\n",
       "      <td>2</td>\n",
       "      <td>212495336</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>not_expressed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3100</th>\n",
       "      <td>2</td>\n",
       "      <td>233164812</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>exact_match</td>\n",
       "      <td>AD=12,11;VAF=0.4782608695652174</td>\n",
       "      <td>0.4782608695652174</td>\n",
       "      <td>0.30</td>\n",
       "      <td>0.44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3276</th>\n",
       "      <td>3</td>\n",
       "      <td>10131962</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>not_expressed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3443</th>\n",
       "      <td>3</td>\n",
       "      <td>23961015</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>exact_match</td>\n",
       "      <td>AD=227,105;VAF=0.31626506024096385</td>\n",
       "      <td>0.31626506024096385</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3502</th>\n",
       "      <td>3</td>\n",
       "      <td>37059106</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exact_match</td>\n",
       "      <td>AD=0,1;VAF=1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3541</th>\n",
       "      <td>3</td>\n",
       "      <td>37067120</td>\n",
       "      <td>T</td>\n",
       "      <td>TTTTTTTTA</td>\n",
       "      <td>not_expressed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3632</th>\n",
       "      <td>3</td>\n",
       "      <td>48602216</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exact_match</td>\n",
       "      <td>AD=6,3;VAF=0.3333333333333333</td>\n",
       "      <td>0.3333333333333333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3756</th>\n",
       "      <td>3</td>\n",
       "      <td>48614349</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exact_match</td>\n",
       "      <td>AD=83,280;VAF=0.7713498622589532</td>\n",
       "      <td>0.7713498622589532</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3892</th>\n",
       "      <td>3</td>\n",
       "      <td>48624782</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>not_expressed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4334</th>\n",
       "      <td>3</td>\n",
       "      <td>121973238</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>not_expressed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4355</th>\n",
       "      <td>3</td>\n",
       "      <td>138664422</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>not_expressed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4563</th>\n",
       "      <td>3</td>\n",
       "      <td>178922370</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exact_match</td>\n",
       "      <td>AD=21,30;VAF=0.5882352941176471</td>\n",
       "      <td>0.5882352941176471</td>\n",
       "      <td>0.59</td>\n",
       "      <td>0.59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5258</th>\n",
       "      <td>5</td>\n",
       "      <td>896917</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>not_expressed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5295</th>\n",
       "      <td>5</td>\n",
       "      <td>1268758</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>not_expressed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5387</th>\n",
       "      <td>5</td>\n",
       "      <td>24537505</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>not_expressed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5445</th>\n",
       "      <td>5</td>\n",
       "      <td>31526205</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exact_match</td>\n",
       "      <td>AD=10,9;VAF=0.47368421052631576</td>\n",
       "      <td>0.47368421052631576</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6014</th>\n",
       "      <td>5</td>\n",
       "      <td>149498403</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exact_match</td>\n",
       "      <td>AD=260,217;VAF=0.4549266247379455</td>\n",
       "      <td>0.4549266247379455</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6242</th>\n",
       "      <td>5</td>\n",
       "      <td>177578009</td>\n",
       "      <td>AAC</td>\n",
       "      <td>A</td>\n",
       "      <td>position_match</td>\n",
       "      <td>AD=10,5;VAF=0.3333333333333333;Variant=5_17757...</td>\n",
       "      <td>0.3333333333333333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6989</th>\n",
       "      <td>7</td>\n",
       "      <td>42262853</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>exact_match</td>\n",
       "      <td>AD=9,14;VAF=0.6086956521739131</td>\n",
       "      <td>0.6086956521739131</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7333</th>\n",
       "      <td>7</td>\n",
       "      <td>140482812</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>not_expressed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7817</th>\n",
       "      <td>8</td>\n",
       "      <td>119122302</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>not_expressed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7857</th>\n",
       "      <td>8</td>\n",
       "      <td>145736832</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exact_match</td>\n",
       "      <td>AD=10,10;VAF=0.5</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.37</td>\n",
       "      <td>0.56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8145</th>\n",
       "      <td>9</td>\n",
       "      <td>382693</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>not_expressed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8232</th>\n",
       "      <td>9</td>\n",
       "      <td>432140</td>\n",
       "      <td>CTTT</td>\n",
       "      <td>CT</td>\n",
       "      <td>not_expressed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8331</th>\n",
       "      <td>9</td>\n",
       "      <td>2110399</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exact_match</td>\n",
       "      <td>AD=98,99;VAF=0.5025380710659898</td>\n",
       "      <td>0.5025380710659898</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8381</th>\n",
       "      <td>9</td>\n",
       "      <td>5029893</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>exact_match</td>\n",
       "      <td>AD=28,27;VAF=0.4909090909090909</td>\n",
       "      <td>0.4909090909090909</td>\n",
       "      <td>0.53</td>\n",
       "      <td>0.53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8501</th>\n",
       "      <td>9</td>\n",
       "      <td>20865973</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exact_match</td>\n",
       "      <td>AD=0,2;VAF=1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8598</th>\n",
       "      <td>9</td>\n",
       "      <td>35074513</td>\n",
       "      <td>CAG</td>\n",
       "      <td>C</td>\n",
       "      <td>not_expressed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8681</th>\n",
       "      <td>9</td>\n",
       "      <td>87325713</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>not_expressed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8730</th>\n",
       "      <td>9</td>\n",
       "      <td>98212243</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exact_match</td>\n",
       "      <td>AD=1,1;VAF=0.5</td>\n",
       "      <td>0.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8979</th>\n",
       "      <td>9</td>\n",
       "      <td>137582774</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exact_match</td>\n",
       "      <td>AD=21,5;VAF=0.19230769230769232</td>\n",
       "      <td>0.19230769230769232</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9486</th>\n",
       "      <td>10</td>\n",
       "      <td>27326133</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>exact_match</td>\n",
       "      <td>AD=0,2;VAF=1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9759</th>\n",
       "      <td>10</td>\n",
       "      <td>99218542</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exact_match</td>\n",
       "      <td>AD=4,2;VAF=0.3333333333333333</td>\n",
       "      <td>0.3333333333333333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10136</th>\n",
       "      <td>11</td>\n",
       "      <td>44193138</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>not_expressed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10633</th>\n",
       "      <td>11</td>\n",
       "      <td>118390323</td>\n",
       "      <td>ATTC</td>\n",
       "      <td>A</td>\n",
       "      <td>position_match</td>\n",
       "      <td>AD=6,1;VAF=0.14285714285714285;Variant=11_1183...</td>\n",
       "      <td>0.14285714285714285</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10673</th>\n",
       "      <td>11</td>\n",
       "      <td>118962233</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exact_match</td>\n",
       "      <td>AD=70,19;VAF=0.21348314606741572</td>\n",
       "      <td>0.21348314606741572</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10937</th>\n",
       "      <td>12</td>\n",
       "      <td>18649112</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>not_expressed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10948</th>\n",
       "      <td>12</td>\n",
       "      <td>18691266</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>not_expressed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10970</th>\n",
       "      <td>12</td>\n",
       "      <td>18715827</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>not_expressed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11027</th>\n",
       "      <td>12</td>\n",
       "      <td>21624056</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>exact_match</td>\n",
       "      <td>AD=16,13;VAF=0.4482758620689655</td>\n",
       "      <td>0.4482758620689655</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11239</th>\n",
       "      <td>12</td>\n",
       "      <td>56477541</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exact_match</td>\n",
       "      <td>AD=10,10;VAF=0.5</td>\n",
       "      <td>0.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11242</th>\n",
       "      <td>12</td>\n",
       "      <td>56477541</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exact_match</td>\n",
       "      <td>AD=12,19;VAF=0.6129032258064516</td>\n",
       "      <td>0.6129032258064516</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      #CHROM        POS   REF        ALT  RNA_EXPRESSION  \\\n",
       "201        1   17345454     C          G     exact_match   \n",
       "314        1   23191718     G          A   not_expressed   \n",
       "319        1   23222908     C          T     exact_match   \n",
       "667        1   45797228     C          T     exact_match   \n",
       "2356       2   96920727     T          C     exact_match   \n",
       "2501       2  169820808     G          T     exact_match   \n",
       "2508       2  169828370     A          G   not_expressed   \n",
       "2885       2  212293119     G          A   not_expressed   \n",
       "2899       2  212495336     G         GA   not_expressed   \n",
       "3100       2  233164812     G          T     exact_match   \n",
       "3276       3   10131962     T          C   not_expressed   \n",
       "3443       3   23961015     C          A     exact_match   \n",
       "3502       3   37059106     A          G     exact_match   \n",
       "3541       3   37067120     T  TTTTTTTTA   not_expressed   \n",
       "3632       3   48602216     C          T     exact_match   \n",
       "3756       3   48614349     C          T     exact_match   \n",
       "3892       3   48624782     G          C   not_expressed   \n",
       "4334       3  121973238     A          G   not_expressed   \n",
       "4355       3  138664422     T          A   not_expressed   \n",
       "4563       3  178922370     A          G     exact_match   \n",
       "5258       5     896917     G          A   not_expressed   \n",
       "5295       5    1268758     G          A   not_expressed   \n",
       "5387       5   24537505     G          A   not_expressed   \n",
       "5445       5   31526205     G          A     exact_match   \n",
       "6014       5  149498403     C          T     exact_match   \n",
       "6242       5  177578009   AAC          A  position_match   \n",
       "6989       7   42262853     G          T     exact_match   \n",
       "7333       7  140482812     T          C   not_expressed   \n",
       "7817       8  119122302     A          G   not_expressed   \n",
       "7857       8  145736832     G          A     exact_match   \n",
       "8145       9     382693     A          G   not_expressed   \n",
       "8232       9     432140  CTTT         CT   not_expressed   \n",
       "8331       9    2110399     C          T     exact_match   \n",
       "8381       9    5029893     C          G     exact_match   \n",
       "8501       9   20865973     A          G     exact_match   \n",
       "8598       9   35074513   CAG          C   not_expressed   \n",
       "8681       9   87325713     T          C   not_expressed   \n",
       "8730       9   98212243     G          A     exact_match   \n",
       "8979       9  137582774     C          T     exact_match   \n",
       "9486      10   27326133     T          C     exact_match   \n",
       "9759      10   99218542     G          C     exact_match   \n",
       "10136     11   44193138     A          G   not_expressed   \n",
       "10633     11  118390323  ATTC          A  position_match   \n",
       "10673     11  118962233     G          A     exact_match   \n",
       "10937     12   18649112     G          A   not_expressed   \n",
       "10948     12   18691266     A          T   not_expressed   \n",
       "10970     12   18715827     G          A   not_expressed   \n",
       "11027     12   21624056     T          C     exact_match   \n",
       "11239     12   56477541     C          T     exact_match   \n",
       "11242     12   56477541     C          T     exact_match   \n",
       "\n",
       "                                                RNA_INFO              RNA_VAF  \\\n",
       "201                                      AD=0,12;VAF=1.0                  1.0   \n",
       "314                                                  NaN                  NaN   \n",
       "319                        AD=6,3;VAF=0.3333333333333333   0.3333333333333333   \n",
       "667                       AD=15,7;VAF=0.3181818181818182   0.3181818181818182   \n",
       "2356                     AD=81,118;VAF=0.592964824120603    0.592964824120603   \n",
       "2501                    AD=12,10;VAF=0.45454545454545453  0.45454545454545453   \n",
       "2508                                                 NaN                  NaN   \n",
       "2885                                                 NaN                  NaN   \n",
       "2899                                                 NaN                  NaN   \n",
       "3100                     AD=12,11;VAF=0.4782608695652174   0.4782608695652174   \n",
       "3276                                                 NaN                  NaN   \n",
       "3443                  AD=227,105;VAF=0.31626506024096385  0.31626506024096385   \n",
       "3502                                      AD=0,1;VAF=1.0                  1.0   \n",
       "3541                                                 NaN                  NaN   \n",
       "3632                       AD=6,3;VAF=0.3333333333333333   0.3333333333333333   \n",
       "3756                    AD=83,280;VAF=0.7713498622589532   0.7713498622589532   \n",
       "3892                                                 NaN                  NaN   \n",
       "4334                                                 NaN                  NaN   \n",
       "4355                                                 NaN                  NaN   \n",
       "4563                     AD=21,30;VAF=0.5882352941176471   0.5882352941176471   \n",
       "5258                                                 NaN                  NaN   \n",
       "5295                                                 NaN                  NaN   \n",
       "5387                                                 NaN                  NaN   \n",
       "5445                     AD=10,9;VAF=0.47368421052631576  0.47368421052631576   \n",
       "6014                   AD=260,217;VAF=0.4549266247379455   0.4549266247379455   \n",
       "6242   AD=10,5;VAF=0.3333333333333333;Variant=5_17757...   0.3333333333333333   \n",
       "6989                      AD=9,14;VAF=0.6086956521739131   0.6086956521739131   \n",
       "7333                                                 NaN                  NaN   \n",
       "7817                                                 NaN                  NaN   \n",
       "7857                                    AD=10,10;VAF=0.5                  0.5   \n",
       "8145                                                 NaN                  NaN   \n",
       "8232                                                 NaN                  NaN   \n",
       "8331                     AD=98,99;VAF=0.5025380710659898   0.5025380710659898   \n",
       "8381                     AD=28,27;VAF=0.4909090909090909   0.4909090909090909   \n",
       "8501                                      AD=0,2;VAF=1.0                  1.0   \n",
       "8598                                                 NaN                  NaN   \n",
       "8681                                                 NaN                  NaN   \n",
       "8730                                      AD=1,1;VAF=0.5                  0.5   \n",
       "8979                     AD=21,5;VAF=0.19230769230769232  0.19230769230769232   \n",
       "9486                                      AD=0,2;VAF=1.0                  1.0   \n",
       "9759                       AD=4,2;VAF=0.3333333333333333   0.3333333333333333   \n",
       "10136                                                NaN                  NaN   \n",
       "10633  AD=6,1;VAF=0.14285714285714285;Variant=11_1183...  0.14285714285714285   \n",
       "10673                   AD=70,19;VAF=0.21348314606741572  0.21348314606741572   \n",
       "10937                                                NaN                  NaN   \n",
       "10948                                                NaN                  NaN   \n",
       "10970                                                NaN                  NaN   \n",
       "11027                    AD=16,13;VAF=0.4482758620689655   0.4482758620689655   \n",
       "11239                                   AD=10,10;VAF=0.5                  0.5   \n",
       "11242                    AD=12,19;VAF=0.6129032258064516   0.6129032258064516   \n",
       "\n",
       "       AF_RNA  AF_RNA_noSkipReads  \n",
       "201      0.05                1.00  \n",
       "314       NaN                 NaN  \n",
       "319       NaN                 NaN  \n",
       "667      0.30                0.35  \n",
       "2356     0.58                0.59  \n",
       "2501      NaN                 NaN  \n",
       "2508      NaN                 NaN  \n",
       "2885      NaN                 NaN  \n",
       "2899      NaN                 NaN  \n",
       "3100     0.30                0.44  \n",
       "3276      NaN                 NaN  \n",
       "3443      NaN                 NaN  \n",
       "3502      NaN                 NaN  \n",
       "3541      NaN                 NaN  \n",
       "3632      NaN                 NaN  \n",
       "3756      NaN                 NaN  \n",
       "3892      NaN                 NaN  \n",
       "4334      NaN                 NaN  \n",
       "4355      NaN                 NaN  \n",
       "4563     0.59                0.59  \n",
       "5258      NaN                 NaN  \n",
       "5295      NaN                 NaN  \n",
       "5387      NaN                 NaN  \n",
       "5445      NaN                 NaN  \n",
       "6014      NaN                 NaN  \n",
       "6242      NaN                 NaN  \n",
       "6989      NaN                 NaN  \n",
       "7333      NaN                 NaN  \n",
       "7817      NaN                 NaN  \n",
       "7857     0.37                0.56  \n",
       "8145      NaN                 NaN  \n",
       "8232      NaN                 NaN  \n",
       "8331      NaN                 NaN  \n",
       "8381     0.53                0.53  \n",
       "8501      NaN                 NaN  \n",
       "8598      NaN                 NaN  \n",
       "8681      NaN                 NaN  \n",
       "8730      NaN                 NaN  \n",
       "8979      NaN                 NaN  \n",
       "9486      NaN                 NaN  \n",
       "9759      NaN                 NaN  \n",
       "10136     NaN                 NaN  \n",
       "10633     NaN                 NaN  \n",
       "10673     NaN                 NaN  \n",
       "10937     NaN                 NaN  \n",
       "10948     NaN                 NaN  \n",
       "10970     NaN                 NaN  \n",
       "11027     NaN                 NaN  \n",
       "11239     NaN                 NaN  \n",
       "11242     NaN                 NaN  "
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "naga_all[cols_cpra+['RNA_EXPRESSION', 'RNA_INFO', 'RNA_VAF', 'AF_RNA', 'AF_RNA_noSkipReads']][\n",
    "    naga_all['RNA_EXPRESSION'].notnull()][:50]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "70835a56",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CharGer_Classification</th>\n",
       "      <th>CharGer_Summary</th>\n",
       "      <th>RNA_EXPRESSION</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>14109</th>\n",
       "      <td>17</td>\n",
       "      <td>7579699</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>PVS1(splice_donor_variant in susceptible gene ...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      #CHROM      POS REF ALT CharGer_Classification  \\\n",
       "14109     17  7579699   C   T             Pathogenic   \n",
       "\n",
       "                                         CharGer_Summary RNA_EXPRESSION  \n",
       "14109  PVS1(splice_donor_variant in susceptible gene ...            NaN  "
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "naga_all[cols_cpra+['CharGer_Classification', 'CharGer_Summary', 'RNA_EXPRESSION']][naga_all['POS']==7579699]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "cb9862d0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['PSC1(splice_acceptor_variant recessive in gene MUTYH) -- PP3(Multiple (>=2) in silico predictions of deliterious effect=dbscSNV:0.99998025568 0.78|GERP:5.3)']"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(naga_all['CharGer_Summary'][naga_all['POS']==45797760])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "2a7c4417",
   "metadata": {},
   "outputs": [],
   "source": [
    "pat_sb = pd.read_excel('03_prioritized_xlsx/2021-10-15_pml stammbaum nct gepado.xlsx')\n",
    "info_vars = pd.read_excel('03_prioritized_xlsx/2021-10-11_onkostar_merged.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "14b2a231",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>master_project</th>\n",
       "      <th>Familie</th>\n",
       "      <th>ID Person</th>\n",
       "      <th>Nachname</th>\n",
       "      <th>Geburtsdatum</th>\n",
       "      <th>Beschreibung</th>\n",
       "      <th>Dokumententyp</th>\n",
       "      <th>Datum</th>\n",
       "      <th>Stammbaum Name</th>\n",
       "      <th>ICD-O3 Histologie</th>\n",
       "      <th>Diagnose</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NCT_MASTER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12001325</td>\n",
       "      <td>S5N2XH</td>\n",
       "      <td>1953-06-13 00:00:00</td>\n",
       "      <td>NCT Import</td>\n",
       "      <td>Word-Dokument alt, Tumorboard</td>\n",
       "      <td>2018-06-12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8409/3 Malignes ekkrines Porom</td>\n",
       "      <td>Bösartige Neubildungen sonstiger ungenau bezei...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NCT_MASTER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12001325</td>\n",
       "      <td>S5N2XH</td>\n",
       "      <td>1953-06-13 00:00:00</td>\n",
       "      <td>NCT Import</td>\n",
       "      <td>Word-Dokument alt, Tumorboard</td>\n",
       "      <td>2018-06-12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8409/3 Malignes ekkrines Porom</td>\n",
       "      <td>Bösartige Neubildungen sonstiger ungenau bezei...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NCT_MASTER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12001325</td>\n",
       "      <td>S5N2XH</td>\n",
       "      <td>1953-06-13 00:00:00</td>\n",
       "      <td>NCT Import</td>\n",
       "      <td>Word-Dokument alt, Tumorboard</td>\n",
       "      <td>2018-06-12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8409/3 Malignes ekkrines Porom</td>\n",
       "      <td>Bösartige Neubildungen sonstiger ungenau bezei...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NCT_MASTER</td>\n",
       "      <td>6ERRP7</td>\n",
       "      <td>12003378</td>\n",
       "      <td>6ERRP7</td>\n",
       "      <td>1978-11-20 00:00:00</td>\n",
       "      <td>NCT Import</td>\n",
       "      <td>Excel-Tabelle Varianten aus Heidelberg</td>\n",
       "      <td>2020-08-17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8000/3 Maligne Neoplasie</td>\n",
       "      <td>Bösartige Neubildungen der Lymphknoten mehrere...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NCT_MASTER</td>\n",
       "      <td>6ERRP7</td>\n",
       "      <td>12003378</td>\n",
       "      <td>6ERRP7</td>\n",
       "      <td>1978-11-20 00:00:00</td>\n",
       "      <td>Präsentation</td>\n",
       "      <td>Präsentation Tumorboard</td>\n",
       "      <td>2020-09-18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8000/3 Maligne Neoplasie</td>\n",
       "      <td>Bösartige Neubildungen der Lymphknoten mehrere...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6782</th>\n",
       "      <td>NCT_MASTER</td>\n",
       "      <td>7KTQ4V</td>\n",
       "      <td>12002468</td>\n",
       "      <td>7KTQ4V</td>\n",
       "      <td>1996-02-10 00:00:00</td>\n",
       "      <td>Präsentation</td>\n",
       "      <td>Präsentation Tumorboard</td>\n",
       "      <td>2019-10-16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8900/3 Rhabdomyosarkom o.n.A.\\n</td>\n",
       "      <td>Bösartige Neubildungen des Bindegewebes und an...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6783</th>\n",
       "      <td>NCT_MASTER</td>\n",
       "      <td>7KTQ4V</td>\n",
       "      <td>12002468</td>\n",
       "      <td>7KTQ4V</td>\n",
       "      <td>1996-02-10 00:00:00</td>\n",
       "      <td>NCT Import</td>\n",
       "      <td>Excel-Tabelle Varianten aus Heidelberg</td>\n",
       "      <td>2019-08-05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8900/3 Rhabdomyosarkom o.n.A.\\n</td>\n",
       "      <td>Bösartige Neubildungen des Bindegewebes und an...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6784</th>\n",
       "      <td>NCT_MASTER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12001152</td>\n",
       "      <td>EW9GX2</td>\n",
       "      <td>1995-06-01 00:00:00</td>\n",
       "      <td>Excel-Tabellen</td>\n",
       "      <td>Excel-Tabelle Varianten aus Heidelberg</td>\n",
       "      <td>2018-09-26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8070/3 Plattenepithelkarzinom o.n.A.</td>\n",
       "      <td>Bösartige Neubildungen sonstiger ungenau bezei...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6785</th>\n",
       "      <td>NCT_MASTER</td>\n",
       "      <td>2M5GJ3</td>\n",
       "      <td>12004087</td>\n",
       "      <td>2M5GJ3</td>\n",
       "      <td>1951-07-30 00:00:00</td>\n",
       "      <td>NCT Import</td>\n",
       "      <td>Excel-Tabelle Varianten aus Heidelberg</td>\n",
       "      <td>2021-04-06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8852/3\\nMyxoides Liposarkom\\nWeichteilsarkom</td>\n",
       "      <td>Bösartige Neubildungen des Bindegewebes und an...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6786</th>\n",
       "      <td>NCT_MASTER</td>\n",
       "      <td>2M5GJ3</td>\n",
       "      <td>12004087</td>\n",
       "      <td>2M5GJ3</td>\n",
       "      <td>1951-07-30 00:00:00</td>\n",
       "      <td>Präsentation Tumorboard</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2021-04-22</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8852/3\\nMyxoides Liposarkom\\nWeichteilsarkom</td>\n",
       "      <td>Bösartige Neubildungen des Bindegewebes und an...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6787 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     master_project Familie  ID Person Nachname         Geburtsdatum  \\\n",
       "0        NCT_MASTER     NaN   12001325   S5N2XH  1953-06-13 00:00:00   \n",
       "1        NCT_MASTER     NaN   12001325   S5N2XH  1953-06-13 00:00:00   \n",
       "2        NCT_MASTER     NaN   12001325   S5N2XH  1953-06-13 00:00:00   \n",
       "3        NCT_MASTER  6ERRP7   12003378   6ERRP7  1978-11-20 00:00:00   \n",
       "4        NCT_MASTER  6ERRP7   12003378   6ERRP7  1978-11-20 00:00:00   \n",
       "...             ...     ...        ...      ...                  ...   \n",
       "6782     NCT_MASTER  7KTQ4V   12002468   7KTQ4V  1996-02-10 00:00:00   \n",
       "6783     NCT_MASTER  7KTQ4V   12002468   7KTQ4V  1996-02-10 00:00:00   \n",
       "6784     NCT_MASTER     NaN   12001152   EW9GX2  1995-06-01 00:00:00   \n",
       "6785     NCT_MASTER  2M5GJ3   12004087   2M5GJ3  1951-07-30 00:00:00   \n",
       "6786     NCT_MASTER  2M5GJ3   12004087   2M5GJ3  1951-07-30 00:00:00   \n",
       "\n",
       "                 Beschreibung                           Dokumententyp  \\\n",
       "0                  NCT Import           Word-Dokument alt, Tumorboard   \n",
       "1                  NCT Import           Word-Dokument alt, Tumorboard   \n",
       "2                  NCT Import           Word-Dokument alt, Tumorboard   \n",
       "3                  NCT Import  Excel-Tabelle Varianten aus Heidelberg   \n",
       "4                Präsentation                 Präsentation Tumorboard   \n",
       "...                       ...                                     ...   \n",
       "6782             Präsentation                 Präsentation Tumorboard   \n",
       "6783               NCT Import  Excel-Tabelle Varianten aus Heidelberg   \n",
       "6784           Excel-Tabellen  Excel-Tabelle Varianten aus Heidelberg   \n",
       "6785               NCT Import  Excel-Tabelle Varianten aus Heidelberg   \n",
       "6786  Präsentation Tumorboard                                     NaN   \n",
       "\n",
       "          Datum Stammbaum Name                              ICD-O3 Histologie  \\\n",
       "0    2018-06-12            NaN                 8409/3 Malignes ekkrines Porom   \n",
       "1    2018-06-12            NaN                 8409/3 Malignes ekkrines Porom   \n",
       "2    2018-06-12            NaN                 8409/3 Malignes ekkrines Porom   \n",
       "3    2020-08-17            NaN                       8000/3 Maligne Neoplasie   \n",
       "4    2020-09-18            NaN                       8000/3 Maligne Neoplasie   \n",
       "...         ...            ...                                            ...   \n",
       "6782 2019-10-16            NaN                8900/3 Rhabdomyosarkom o.n.A.\\n   \n",
       "6783 2019-08-05            NaN                8900/3 Rhabdomyosarkom o.n.A.\\n   \n",
       "6784 2018-09-26            NaN           8070/3 Plattenepithelkarzinom o.n.A.   \n",
       "6785 2021-04-06            NaN  8852/3\\nMyxoides Liposarkom\\nWeichteilsarkom    \n",
       "6786 2021-04-22            NaN  8852/3\\nMyxoides Liposarkom\\nWeichteilsarkom    \n",
       "\n",
       "                                               Diagnose  \n",
       "0     Bösartige Neubildungen sonstiger ungenau bezei...  \n",
       "1     Bösartige Neubildungen sonstiger ungenau bezei...  \n",
       "2     Bösartige Neubildungen sonstiger ungenau bezei...  \n",
       "3     Bösartige Neubildungen der Lymphknoten mehrere...  \n",
       "4     Bösartige Neubildungen der Lymphknoten mehrere...  \n",
       "...                                                 ...  \n",
       "6782  Bösartige Neubildungen des Bindegewebes und an...  \n",
       "6783  Bösartige Neubildungen des Bindegewebes und an...  \n",
       "6784  Bösartige Neubildungen sonstiger ungenau bezei...  \n",
       "6785  Bösartige Neubildungen des Bindegewebes und an...  \n",
       "6786  Bösartige Neubildungen des Bindegewebes und an...  \n",
       "\n",
       "[6787 rows x 11 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pat_sb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "c6d96a73",
   "metadata": {},
   "outputs": [],
   "source": [
    "mg = pd.merge(info_vars, pat_sb, left_on='PID_trans', right_on='Nachname', how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "075e8ce9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Studienzentrale', 'TMO HD', 'FA Berlin', 'FA Freiburg',\n",
       "       'FA Dresden', nan, 'NCT Dresden TMO', 'Frankfurt', 'Essen',\n",
       "       'TU München', 'FA Tübingen'], dtype=object)"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mg['Dokumentierende Fachabteilung'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "5ffe68ae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Word-Dokument alt, Tumorboard',\n",
       "       'Excel-Tabelle Varianten aus Heidelberg', nan,\n",
       "       'Präsentation Tumorboard',\n",
       "       'Individualanamnese NCT-Master germline',\n",
       "       'Einwilligung NCT-MASTER germline',\n",
       "       'Follow-up Dokumente NCT-Master germline',\n",
       "       'Stammbaum NCT-MASTER germline'], dtype=object)"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mg['Dokumententyp'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "f59beead",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>PID_trans</th>\n",
       "      <th>PosExonRefSeqAccession</th>\n",
       "      <th>PosExon_type</th>\n",
       "      <th>ClinVar_Pathogenicity</th>\n",
       "      <th>hgsv_new</th>\n",
       "      <th>INFO_IGV</th>\n",
       "      <th>...</th>\n",
       "      <th>Familie</th>\n",
       "      <th>ID Person</th>\n",
       "      <th>Nachname</th>\n",
       "      <th>Geburtsdatum</th>\n",
       "      <th>Beschreibung</th>\n",
       "      <th>Dokumententyp</th>\n",
       "      <th>Datum</th>\n",
       "      <th>Stammbaum Name</th>\n",
       "      <th>ICD-O3 Histologie</th>\n",
       "      <th>Diagnose</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>19</td>\n",
       "      <td>45871887</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>K4WF4R</td>\n",
       "      <td>1</td>\n",
       "      <td>outsideDonor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exon 5 skipped (of 23, length 760, RefSeq NM_0...</td>\n",
       "      <td>exon skipping, donor loss</td>\n",
       "      <td>...</td>\n",
       "      <td>K4WF4R</td>\n",
       "      <td>12002108.0</td>\n",
       "      <td>K4WF4R</td>\n",
       "      <td>1982-01-20 00:00:00</td>\n",
       "      <td>load from scanner</td>\n",
       "      <td>Stammbaum NCT-MASTER germline</td>\n",
       "      <td>2021-01-18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bösartige Neubildungen der Schilddrüse und son...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>19</td>\n",
       "      <td>50918678</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>MBK1QX</td>\n",
       "      <td>17</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "      <td>Likely Benign</td>\n",
       "      <td>exon 21 (of 27, length 1107, RefSeq NM_0012568...</td>\n",
       "      <td>intron retention in 32XTFQ, MBK1QX, PH42G6 onl...</td>\n",
       "      <td>...</td>\n",
       "      <td>MBK1QX</td>\n",
       "      <td>12001718.0</td>\n",
       "      <td>MBK1QX</td>\n",
       "      <td>1969-08-24 00:00:00</td>\n",
       "      <td>load from scanner</td>\n",
       "      <td>Stammbaum NCT-MASTER germline</td>\n",
       "      <td>2021-02-19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9732/3\\nMultiples MyelomC42.1</td>\n",
       "      <td>Multiples Myelom</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 114 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #CHROM       POS REF ALT PID_trans  PosExonRefSeqAccession  \\\n",
       "145      19  45871887   C   T    K4WF4R                       1   \n",
       "150      19  50918678   G   A    MBK1QX                      17   \n",
       "\n",
       "            PosExon_type ClinVar_Pathogenicity  \\\n",
       "145         outsideDonor                   NaN   \n",
       "150  outsideAcceptorSite         Likely Benign   \n",
       "\n",
       "                                              hgsv_new  \\\n",
       "145  exon 5 skipped (of 23, length 760, RefSeq NM_0...   \n",
       "150  exon 21 (of 27, length 1107, RefSeq NM_0012568...   \n",
       "\n",
       "                                              INFO_IGV  ... Familie  \\\n",
       "145                          exon skipping, donor loss  ...  K4WF4R   \n",
       "150  intron retention in 32XTFQ, MBK1QX, PH42G6 onl...  ...  MBK1QX   \n",
       "\n",
       "      ID Person Nachname         Geburtsdatum       Beschreibung  \\\n",
       "145  12002108.0   K4WF4R  1982-01-20 00:00:00  load from scanner   \n",
       "150  12001718.0   MBK1QX  1969-08-24 00:00:00  load from scanner   \n",
       "\n",
       "                     Dokumententyp      Datum Stammbaum Name  \\\n",
       "145  Stammbaum NCT-MASTER germline 2021-01-18            NaN   \n",
       "150  Stammbaum NCT-MASTER germline 2021-02-19            NaN   \n",
       "\n",
       "                 ICD-O3 Histologie  \\\n",
       "145                            NaN   \n",
       "150  9732/3\\nMultiples MyelomC42.1   \n",
       "\n",
       "                                              Diagnose  \n",
       "145  Bösartige Neubildungen der Schilddrüse und son...  \n",
       "150                                   Multiples Myelom  \n",
       "\n",
       "[2 rows x 114 columns]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mg[mg['Dokumententyp']=='Stammbaum NCT-MASTER germline']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4f3800e4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "f3be7044",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Splice Project Gene Priority']"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_col('prio')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "22e789f6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>SpliceAI_pred_DP_AG</th>\n",
       "      <th>SpliceAI_pred_DP_AL</th>\n",
       "      <th>SpliceAI_pred_DP_DG</th>\n",
       "      <th>SpliceAI_pred_DP_DL</th>\n",
       "      <th>SpliceAI_pred_DS_AG</th>\n",
       "      <th>SpliceAI_pred_DS_AL</th>\n",
       "      <th>SpliceAI_pred_DS_DG</th>\n",
       "      <th>SpliceAI_pred_DS_DL</th>\n",
       "      <th>SpliceAI_pred_SYMBOL</th>\n",
       "      <th>PosExonRefSeqAccession</th>\n",
       "      <th>PosExon_type</th>\n",
       "      <th>Splice Project Gene Priority</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>1</td>\n",
       "      <td>17350470</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>8</td>\n",
       "      <td>29</td>\n",
       "      <td>2</td>\n",
       "      <td>-2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.19</td>\n",
       "      <td>SDHB</td>\n",
       "      <td>2</td>\n",
       "      <td>insideDonorSite</td>\n",
       "      <td>Hot (ACMG / MASTER)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1560</th>\n",
       "      <td>1</td>\n",
       "      <td>156849947</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>-12</td>\n",
       "      <td>-33</td>\n",
       "      <td>-2</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.18</td>\n",
       "      <td>NTRK1</td>\n",
       "      <td>2</td>\n",
       "      <td>insideDonorSite</td>\n",
       "      <td>Cold (387-ACMG-MASTER)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8236</th>\n",
       "      <td>9</td>\n",
       "      <td>432187</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-10</td>\n",
       "      <td>-21</td>\n",
       "      <td>-2</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.00</td>\n",
       "      <td>DOCK8</td>\n",
       "      <td>21</td>\n",
       "      <td>insideAcceptorSite</td>\n",
       "      <td>Cold (387-ACMG-MASTER)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     #CHROM        POS REF ALT SpliceAI_pred_DP_AG SpliceAI_pred_DP_AL  \\\n",
       "207       1   17350470   G   A                   8                  29   \n",
       "1560      1  156849947   G   T                 -12                 -33   \n",
       "8236      9     432187   C   T                 -10                 -21   \n",
       "\n",
       "     SpliceAI_pred_DP_DG SpliceAI_pred_DP_DL  SpliceAI_pred_DS_AG  \\\n",
       "207                    2                  -2                  0.0   \n",
       "1560                  -2                   2                  0.0   \n",
       "8236                  -2                   2                  0.0   \n",
       "\n",
       "      SpliceAI_pred_DS_AL  SpliceAI_pred_DS_DG  SpliceAI_pred_DS_DL  \\\n",
       "207                   0.0                 0.34                 0.19   \n",
       "1560                  0.0                 0.99                 0.18   \n",
       "8236                  0.0                 0.07                 0.00   \n",
       "\n",
       "     SpliceAI_pred_SYMBOL  PosExonRefSeqAccession        PosExon_type  \\\n",
       "207                  SDHB                       2     insideDonorSite   \n",
       "1560                NTRK1                       2     insideDonorSite   \n",
       "8236                DOCK8                      21  insideAcceptorSite   \n",
       "\n",
       "     Splice Project Gene Priority  \n",
       "207           Hot (ACMG / MASTER)  \n",
       "1560       Cold (387-ACMG-MASTER)  \n",
       "8236       Cold (387-ACMG-MASTER)  "
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cadd[cols_cpra+search_col('spliceAI')+['PosExonRefSeqAccession', 'PosExon_type', 'Splice Project Gene Priority']][\n",
    "    cadd['CADD_PHRED']>50]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "0246d35d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CADD_PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>8012.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>11.669732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>9.224739</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>0.001000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>3.923250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>10.075000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>18.042500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>56.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        CADD_PHRED\n",
       "count  8012.000000\n",
       "mean     11.669732\n",
       "std       9.224739\n",
       "min       0.001000\n",
       "25%       3.923250\n",
       "50%      10.075000\n",
       "75%      18.042500\n",
       "max      56.000000"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cadd = naga_all[naga_all['CADD_PHRED'].notnull()].copy().drop_duplicates(cols_cpra)\n",
    "cadd['CADD_PHRED']=cadd['CADD_PHRED'].astype(float)\n",
    "cadd[['CADD_PHRED']].describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "6ef046bf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['SpliceAI_pred_DP_AG',\n",
       " 'SpliceAI_pred_DP_AL',\n",
       " 'SpliceAI_pred_DP_DG',\n",
       " 'SpliceAI_pred_DP_DL',\n",
       " 'SpliceAI_pred_DS_AG',\n",
       " 'SpliceAI_pred_DS_AL',\n",
       " 'SpliceAI_pred_DS_DG',\n",
       " 'SpliceAI_pred_DS_DL',\n",
       " 'SpliceAI_pred_SYMBOL']"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_col('spliceai')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "c3c10f81",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['outsideAcceptorSite', 'outsideDonorSite', 'insideDonorSite',\n",
       "       'insideAcceptorSite', '3utr_stopcodon', '5utr_startcodon'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "naga_all['PosExon_type'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "31999c43",
   "metadata": {},
   "outputs": [],
   "source": [
    "naga_all[cols_cpra+search_col('spliceAI')+search_col('maxentscan')+[\n",
    "    'ada_score', 'rf_score', 'PID_trans', 'ClinVar_Pathogenicity', 'Splice Project Gene Priority']][\n",
    "    (naga_all['PosExonRefSeqAccession']<3)&\n",
    "    (naga_all['PosExon_type'].isin(['outsideAcceptorSite', 'outsideDonorSite']))&\n",
    "    ((naga_all['SpliceAI_pred_DS_AG']<0.5)&(naga_all['SpliceAI_pred_DS_AL']<0.5)&\n",
    "     (naga_all['SpliceAI_pred_DS_DG']<0.5)&(naga_all['SpliceAI_pred_DS_DL']<0.5))\n",
    "].to_excel('03_prioritized_xlsx/no_sign_splice.xlsx')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6db7ac62",
   "metadata": {},
   "source": [
    "### 21.10.2021"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "235d4b5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "def splai(n):\n",
    "    df = naga_all[(naga_all['SpliceAI_pred_DS_AG']>n)|(naga_all['SpliceAI_pred_DS_AL']>n)|\n",
    "                  (naga_all['SpliceAI_pred_DS_DG']>n)|(naga_all['SpliceAI_pred_DS_DL']>n)].copy()\n",
    "    return (float('{:.1f}'.format(n)), len(df.drop_duplicates(['#CHROM', 'POS', 'REF', 'ALT'])), len(df))\n",
    "\n",
    "# ada/rf score > 0.6\n",
    "def ada_rf(n):\n",
    "    df = naga_all[(naga_all['ada_score']>n)&(naga_all['rf_score']>n)].copy()\n",
    "    return (float('{:.1f}'.format(n)), len(df.drop_duplicates(['#CHROM', 'POS', 'REF', 'ALT'])), len(df))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "039b619e",
   "metadata": {},
   "outputs": [],
   "source": [
    "spl = naga_all[naga_all['SpliceAI_pred_DS_AG'].notnull()].copy()\n",
    "adarf = naga_all[(naga_all['ada_score'].notnull())&(naga_all['rf_score'].notnull())]\n",
    "mes = naga_all[naga_all['MaxEntScan_diff'].notnull()].copy()\n",
    "\n",
    "df_splai = pd.DataFrame([splai(n) for n in np.arange(0,1.1,0.1)[:-1]], columns=['cutoff', 'variants', 'rows'])\n",
    "df_adarf = pd.DataFrame([ada_rf(n) for n in np.arange(0,1.1,0.1)[:-1]], columns=['cutoff', 'variants', 'rows'])\n",
    "\n",
    "# disr_ss = disrupts native splice site, diff > 0\n",
    "disr_ss_vars = mes[(mes['MaxEntScan_diff']>=0)].copy()\n",
    "disr_high_vars = disr_ss_vars[cols_cpra][disr_ss_vars['MaxEntScan_alt']<6.2]\n",
    "disr_mod_vars =disr_ss_vars[cols_cpra][(disr_ss_vars['MaxEntScan_alt']>=6.2)&\n",
    "                                     (disr_ss_vars['MaxEntScan_alt']<=8.5)]\n",
    "disr_low_vars = disr_ss_vars[cols_cpra][disr_ss_vars['MaxEntScan_alt']>8.5]\n",
    "\n",
    "# new_ss = creates new splice site, diff < 0\n",
    "new_ss_vars = mes[(mes['MaxEntScan_diff']<0)].copy()\n",
    "new_high_vars = new_ss_vars[cols_cpra][new_ss_vars['MaxEntScan_alt']>8.5]\n",
    "new_mod_vars = new_ss_vars[cols_cpra][(new_ss_vars['MaxEntScan_alt']>=6.2)&\n",
    "                                      (new_ss_vars['MaxEntScan_alt']<=8.5)]\n",
    "new_low_vars = new_ss_vars[cols_cpra][new_ss_vars['MaxEntScan_alt']<6.2]\n",
    "\n",
    "mes_dfs = [disr_high_vars, disr_mod_vars, disr_low_vars, new_high_vars, new_mod_vars, new_low_vars]\n",
    "mes_name = ['disr_high_vars', 'disr_mod_vars', 'disr_low_vars', 'new_high_vars', 'new_mod_vars', 'new_low_vars']\n",
    "\n",
    "df_mes = pd.DataFrame([(name, len(df.drop_duplicates(cols_cpra)), len(df)) for df,name in zip(mes_dfs, mes_name)], \n",
    "                      columns=['MES_category', 'variants', 'rows'])\n",
    "\n",
    "(sum_vars, sum_rows), (sum_v_spl, sum_r_spl), (sum_v_ada, sum_r_ada), (sum_v_mes, sum_r_mes) = [(\n",
    "    len(df.drop_duplicates(cols_cpra)), len(df)) for df in [naga_all, spl, adarf, mes]]\n",
    "\n",
    "df_splai = df_splai.append(pd.DataFrame([['sum', sum_v_spl, sum_r_spl]], columns=['cutoff', 'variants', 'rows'], \n",
    "                                        index=[10]))\n",
    "df_adarf = df_adarf.append(pd.DataFrame([['sum', sum_v_ada, sum_r_ada]], columns=['cutoff', 'variants', 'rows'], \n",
    "                                        index=[10]))\n",
    "df_mes = df_mes.append(pd.DataFrame([['sum', sum_v_mes, sum_r_mes]], columns=['MES_category', 'variants', 'rows'], \n",
    "                                    index=[6]))\n",
    "\n",
    "for i,v,r in zip([df_splai, df_adarf, df_mes], [sum_v_spl, sum_v_ada, sum_v_mes], [sum_r_spl, sum_r_ada, sum_r_mes]):\n",
    "    i['perc_vars'] = i['variants']/sum_vars*100\n",
    "    i['perc_rows'] = i['rows']/sum_rows*100\n",
    "    i['perc_vars_spec'] = i['variants']/v*100\n",
    "    i['perc_rows_spec'] = i['rows']/r*100\n",
    "    i[['perc_vars', 'perc_rows', 'perc_vars_spec', \n",
    "       'perc_rows_spec']] = i[['perc_vars', 'perc_rows', 'perc_vars_spec', 'perc_rows_spec']].round(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "id": "af71cbdb",
   "metadata": {},
   "outputs": [],
   "source": [
    "with pd.ExcelWriter('03_prioritized_xlsx/output_cats_prio.xlsx') as writer:\n",
    "    df_splai.to_excel(writer, sheet_name='df_splai', index=False)\n",
    "    df_adarf.to_excel(writer, sheet_name='df_adarf', index=False)\n",
    "    df_mes.to_excel(writer, sheet_name='df_mes', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "id": "9471150c",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>SpliceAI_pred_DP_AG</th>\n",
       "      <th>SpliceAI_pred_DP_AL</th>\n",
       "      <th>SpliceAI_pred_DP_DG</th>\n",
       "      <th>SpliceAI_pred_DP_DL</th>\n",
       "      <th>SpliceAI_pred_DS_AG</th>\n",
       "      <th>SpliceAI_pred_DS_AL</th>\n",
       "      <th>SpliceAI_pred_DS_DG</th>\n",
       "      <th>SpliceAI_pred_DS_DL</th>\n",
       "      <th>SpliceAI_pred_SYMBOL</th>\n",
       "      <th>ada_score</th>\n",
       "      <th>rf_score</th>\n",
       "      <th>MaxEntScan_alt</th>\n",
       "      <th>MaxEntScan_diff</th>\n",
       "      <th>MaxEntScan_ref</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1828</th>\n",
       "      <td>1</td>\n",
       "      <td>243736225</td>\n",
       "      <td>TAC</td>\n",
       "      <td>T</td>\n",
       "      <td>-14</td>\n",
       "      <td>3</td>\n",
       "      <td>-14</td>\n",
       "      <td>3</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.64</td>\n",
       "      <td>1.00</td>\n",
       "      <td>AKT3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2132</th>\n",
       "      <td>2</td>\n",
       "      <td>47601174</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>-2</td>\n",
       "      <td>13</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.28</td>\n",
       "      <td>EPCAM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2167</th>\n",
       "      <td>2</td>\n",
       "      <td>47641558</td>\n",
       "      <td>GTA</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>-26</td>\n",
       "      <td>-25</td>\n",
       "      <td>-1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>MSH2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3002</th>\n",
       "      <td>2</td>\n",
       "      <td>215617292</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>-1</td>\n",
       "      <td>-13</td>\n",
       "      <td>-1</td>\n",
       "      <td>-11</td>\n",
       "      <td>0.36</td>\n",
       "      <td>0.94</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>BARD1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.540</td>\n",
       "      <td>2.877</td>\n",
       "      <td>5.416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3041</th>\n",
       "      <td>2</td>\n",
       "      <td>225378349</td>\n",
       "      <td>TGCGCCTCTGTC</td>\n",
       "      <td>T</td>\n",
       "      <td>-35</td>\n",
       "      <td>6</td>\n",
       "      <td>-6</td>\n",
       "      <td>6</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.94</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>CUL3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3638</th>\n",
       "      <td>3</td>\n",
       "      <td>48602832</td>\n",
       "      <td>GCTGTCCTCACCTTC</td>\n",
       "      <td>G</td>\n",
       "      <td>19</td>\n",
       "      <td>11</td>\n",
       "      <td>-22</td>\n",
       "      <td>11</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.99</td>\n",
       "      <td>COL7A1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4051</th>\n",
       "      <td>3</td>\n",
       "      <td>52440844</td>\n",
       "      <td>CCCTGCAGT</td>\n",
       "      <td>C</td>\n",
       "      <td>-5</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.78</td>\n",
       "      <td>1.00</td>\n",
       "      <td>BAP1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5604</th>\n",
       "      <td>5</td>\n",
       "      <td>80064822</td>\n",
       "      <td>GGTAAT</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>-13</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.92</td>\n",
       "      <td>MSH3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7302</th>\n",
       "      <td>7</td>\n",
       "      <td>124465407</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-1</td>\n",
       "      <td>-11</td>\n",
       "      <td>-1</td>\n",
       "      <td>-11</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>POT1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7957</th>\n",
       "      <td>8</td>\n",
       "      <td>145738601</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-30</td>\n",
       "      <td>5</td>\n",
       "      <td>44</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.91</td>\n",
       "      <td>RECQL4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.079</td>\n",
       "      <td>7.603</td>\n",
       "      <td>5.523</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7982</th>\n",
       "      <td>8</td>\n",
       "      <td>145739572</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-33</td>\n",
       "      <td>42</td>\n",
       "      <td>-10</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>1.00</td>\n",
       "      <td>RECQL4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.415</td>\n",
       "      <td>8.182</td>\n",
       "      <td>9.597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9040</th>\n",
       "      <td>9</td>\n",
       "      <td>137650127</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>15</td>\n",
       "      <td>-38</td>\n",
       "      <td>-2</td>\n",
       "      <td>15</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.07</td>\n",
       "      <td>COL5A1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11028</th>\n",
       "      <td>12</td>\n",
       "      <td>21624358</td>\n",
       "      <td>ATACT</td>\n",
       "      <td>A</td>\n",
       "      <td>-29</td>\n",
       "      <td>1</td>\n",
       "      <td>-28</td>\n",
       "      <td>3</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.57</td>\n",
       "      <td>1.00</td>\n",
       "      <td>RECQL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11115</th>\n",
       "      <td>12</td>\n",
       "      <td>49424675</td>\n",
       "      <td>CCTGTTTCAG</td>\n",
       "      <td>C</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.88</td>\n",
       "      <td>0.96</td>\n",
       "      <td>KMT2D</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12047</th>\n",
       "      <td>14</td>\n",
       "      <td>68331714</td>\n",
       "      <td>TTTTA</td>\n",
       "      <td>T</td>\n",
       "      <td>13</td>\n",
       "      <td>6</td>\n",
       "      <td>22</td>\n",
       "      <td>13</td>\n",
       "      <td>0.63</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>RAD51B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12048</th>\n",
       "      <td>14</td>\n",
       "      <td>68331715</td>\n",
       "      <td>TTTA</td>\n",
       "      <td>T</td>\n",
       "      <td>12</td>\n",
       "      <td>5</td>\n",
       "      <td>21</td>\n",
       "      <td>12</td>\n",
       "      <td>0.63</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>RAD51B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12050</th>\n",
       "      <td>14</td>\n",
       "      <td>68331716</td>\n",
       "      <td>TTAGA</td>\n",
       "      <td>T</td>\n",
       "      <td>10</td>\n",
       "      <td>3</td>\n",
       "      <td>19</td>\n",
       "      <td>10</td>\n",
       "      <td>0.63</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>RAD51B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12052</th>\n",
       "      <td>14</td>\n",
       "      <td>68331717</td>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "      <td>10</td>\n",
       "      <td>3</td>\n",
       "      <td>19</td>\n",
       "      <td>18</td>\n",
       "      <td>0.63</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>RAD51B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12826</th>\n",
       "      <td>15</td>\n",
       "      <td>91352354</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>35</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.42</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>BLM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.299</td>\n",
       "      <td>5.115</td>\n",
       "      <td>5.413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13244</th>\n",
       "      <td>16</td>\n",
       "      <td>2137924</td>\n",
       "      <td>TCCCTGCAGTGCAGGAAAGGTAGGGCCGGGTGGGG</td>\n",
       "      <td>T</td>\n",
       "      <td>43</td>\n",
       "      <td>9</td>\n",
       "      <td>-4</td>\n",
       "      <td>18</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>TSC2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13411</th>\n",
       "      <td>16</td>\n",
       "      <td>3808046</td>\n",
       "      <td>AGTCC</td>\n",
       "      <td>A</td>\n",
       "      <td>-18</td>\n",
       "      <td>3</td>\n",
       "      <td>-11</td>\n",
       "      <td>-18</td>\n",
       "      <td>0.96</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>CREBBP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13412</th>\n",
       "      <td>16</td>\n",
       "      <td>3808046</td>\n",
       "      <td>AGTCCTT</td>\n",
       "      <td>A</td>\n",
       "      <td>-18</td>\n",
       "      <td>3</td>\n",
       "      <td>-13</td>\n",
       "      <td>-18</td>\n",
       "      <td>0.97</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>CREBBP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13802</th>\n",
       "      <td>16</td>\n",
       "      <td>68849662</td>\n",
       "      <td>CG</td>\n",
       "      <td>C</td>\n",
       "      <td>6</td>\n",
       "      <td>-31</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.88</td>\n",
       "      <td>1.00</td>\n",
       "      <td>CDH1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14156</th>\n",
       "      <td>17</td>\n",
       "      <td>7606440</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-2</td>\n",
       "      <td>-17</td>\n",
       "      <td>-2</td>\n",
       "      <td>5</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.23</td>\n",
       "      <td>WRAP53</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14299</th>\n",
       "      <td>17</td>\n",
       "      <td>29482996</td>\n",
       "      <td>TTCAGCTTCCAATA</td>\n",
       "      <td>T</td>\n",
       "      <td>21</td>\n",
       "      <td>5</td>\n",
       "      <td>29</td>\n",
       "      <td>43</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.04</td>\n",
       "      <td>NF1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15693</th>\n",
       "      <td>19</td>\n",
       "      <td>7184649</td>\n",
       "      <td>CTG</td>\n",
       "      <td>C</td>\n",
       "      <td>-21</td>\n",
       "      <td>-1</td>\n",
       "      <td>-21</td>\n",
       "      <td>-15</td>\n",
       "      <td>0.48</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>INSR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15985</th>\n",
       "      <td>19</td>\n",
       "      <td>10277266</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>37</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.93</td>\n",
       "      <td>DNMT1</td>\n",
       "      <td>0.000095</td>\n",
       "      <td>0.006</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16492</th>\n",
       "      <td>19</td>\n",
       "      <td>18272006</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-34</td>\n",
       "      <td>48</td>\n",
       "      <td>-2</td>\n",
       "      <td>2</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.00</td>\n",
       "      <td>PIK3R2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17518</th>\n",
       "      <td>22</td>\n",
       "      <td>29083975</td>\n",
       "      <td>CTGAAAATAAA</td>\n",
       "      <td>C</td>\n",
       "      <td>-12</td>\n",
       "      <td>-1</td>\n",
       "      <td>47</td>\n",
       "      <td>1</td>\n",
       "      <td>0.57</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17866</th>\n",
       "      <td>X</td>\n",
       "      <td>44949962</td>\n",
       "      <td>TCCTAGCC</td>\n",
       "      <td>T</td>\n",
       "      <td>14</td>\n",
       "      <td>6</td>\n",
       "      <td>-35</td>\n",
       "      <td>33</td>\n",
       "      <td>0.98</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>KDM6A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      #CHROM        POS                                  REF ALT  \\\n",
       "1828       1  243736225                                  TAC   T   \n",
       "2132       2   47601174                                    C   T   \n",
       "2167       2   47641558                                  GTA   G   \n",
       "3002       2  215617292                                    G   C   \n",
       "3041       2  225378349                         TGCGCCTCTGTC   T   \n",
       "3638       3   48602832                      GCTGTCCTCACCTTC   G   \n",
       "4051       3   52440844                            CCCTGCAGT   C   \n",
       "5604       5   80064822                               GGTAAT   G   \n",
       "7302       7  124465407                                    T   C   \n",
       "7957       8  145738601                                    C   T   \n",
       "7982       8  145739572                                    C   T   \n",
       "9040       9  137650127                                    C   T   \n",
       "11028     12   21624358                                ATACT   A   \n",
       "11115     12   49424675                           CCTGTTTCAG   C   \n",
       "12047     14   68331714                                TTTTA   T   \n",
       "12048     14   68331715                                 TTTA   T   \n",
       "12050     14   68331716                                TTAGA   T   \n",
       "12052     14   68331717                                   TA   T   \n",
       "12826     15   91352354                                    T   G   \n",
       "13244     16    2137924  TCCCTGCAGTGCAGGAAAGGTAGGGCCGGGTGGGG   T   \n",
       "13411     16    3808046                                AGTCC   A   \n",
       "13412     16    3808046                              AGTCCTT   A   \n",
       "13802     16   68849662                                   CG   C   \n",
       "14156     17    7606440                                    C   T   \n",
       "14299     17   29482996                       TTCAGCTTCCAATA   T   \n",
       "15693     19    7184649                                  CTG   C   \n",
       "15985     19   10277266                                    G   A   \n",
       "16492     19   18272006                                    C   T   \n",
       "17518     22   29083975                          CTGAAAATAAA   C   \n",
       "17866      X   44949962                             TCCTAGCC   T   \n",
       "\n",
       "      SpliceAI_pred_DP_AG SpliceAI_pred_DP_AL SpliceAI_pred_DP_DG  \\\n",
       "1828                  -14                   3                 -14   \n",
       "2132                   13                   2                  -2   \n",
       "2167                    0                 -26                 -25   \n",
       "3002                   -1                 -13                  -1   \n",
       "3041                  -35                   6                  -6   \n",
       "3638                   19                  11                 -22   \n",
       "4051                   -5                   1                   9   \n",
       "5604                    0                 -13                   8   \n",
       "7302                   -1                 -11                  -1   \n",
       "7957                  -30                   5                  44   \n",
       "7982                  -33                  42                 -10   \n",
       "9040                   15                 -38                  -2   \n",
       "11028                 -29                   1                 -28   \n",
       "11115                  10                   1                  10   \n",
       "12047                  13                   6                  22   \n",
       "12048                  12                   5                  21   \n",
       "12050                  10                   3                  19   \n",
       "12052                  10                   3                  19   \n",
       "12826                   1                  13                  13   \n",
       "13244                  43                   9                  -4   \n",
       "13411                 -18                   3                 -11   \n",
       "13412                 -18                   3                 -13   \n",
       "13802                   6                 -31                   6   \n",
       "14156                  -2                 -17                  -2   \n",
       "14299                  21                   5                  29   \n",
       "15693                 -21                  -1                 -21   \n",
       "15985                  37                   2                   2   \n",
       "16492                 -34                  48                  -2   \n",
       "17518                 -12                  -1                  47   \n",
       "17866                  14                   6                 -35   \n",
       "\n",
       "      SpliceAI_pred_DP_DL  SpliceAI_pred_DS_AG  SpliceAI_pred_DS_AL  \\\n",
       "1828                    3                 0.00                 0.00   \n",
       "2132                   13                 0.00                 0.00   \n",
       "2167                   -1                 0.00                 0.00   \n",
       "3002                  -11                 0.36                 0.94   \n",
       "3041                    6                 0.31                 0.94   \n",
       "3638                   11                 0.16                 0.00   \n",
       "4051                    1                 0.00                 0.00   \n",
       "5604                    0                 0.00                 0.00   \n",
       "7302                  -11                 0.97                 0.00   \n",
       "7957                    0                 0.00                 0.00   \n",
       "7982                    1                 0.00                 0.00   \n",
       "9040                   15                 0.00                 0.00   \n",
       "11028                   3                 0.00                 0.00   \n",
       "11115                   1                 0.00                 0.00   \n",
       "12047                  13                 0.63                 0.99   \n",
       "12048                  12                 0.63                 0.99   \n",
       "12050                  10                 0.63                 0.99   \n",
       "12052                  18                 0.63                 0.99   \n",
       "12826                  35                 0.96                 0.42   \n",
       "13244                  18                 0.02                 0.00   \n",
       "13411                 -18                 0.96                 1.00   \n",
       "13412                 -18                 0.97                 1.00   \n",
       "13802                   0                 0.00                 0.00   \n",
       "14156                   5                 0.00                 0.00   \n",
       "14299                  43                 1.00                 1.00   \n",
       "15693                 -15                 0.48                 1.00   \n",
       "15985                   8                 0.00                 0.00   \n",
       "16492                   2                 0.00                 0.00   \n",
       "17518                   1                 0.57                 0.98   \n",
       "17866                  33                 0.98                 1.00   \n",
       "\n",
       "       SpliceAI_pred_DS_DG  SpliceAI_pred_DS_DL SpliceAI_pred_SYMBOL  \\\n",
       "1828                  0.64                 1.00                 AKT3   \n",
       "2132                  0.95                 0.28                EPCAM   \n",
       "2167                  0.00                 1.00                 MSH2   \n",
       "3002                  0.00                 0.00                BARD1   \n",
       "3041                  0.00                 0.00                 CUL3   \n",
       "3638                  0.01                 0.99               COL7A1   \n",
       "4051                  0.78                 1.00                 BAP1   \n",
       "5604                  0.00                 0.92                 MSH3   \n",
       "7302                  0.00                 0.00                 POT1   \n",
       "7957                  0.00                 0.91               RECQL4   \n",
       "7982                  0.02                 1.00               RECQL4   \n",
       "9040                  0.93                 0.07               COL5A1   \n",
       "11028                 0.57                 1.00                RECQL   \n",
       "11115                 0.88                 0.96                KMT2D   \n",
       "12047                 0.00                 0.00               RAD51B   \n",
       "12048                 0.00                 0.00               RAD51B   \n",
       "12050                 0.00                 0.00               RAD51B   \n",
       "12052                 0.00                 0.00               RAD51B   \n",
       "12826                 0.00                 0.00                  BLM   \n",
       "13244                 0.00                 1.00                 TSC2   \n",
       "13411                 0.00                 0.00               CREBBP   \n",
       "13412                 0.00                 0.00               CREBBP   \n",
       "13802                 0.88                 1.00                 CDH1   \n",
       "14156                 0.95                 0.23               WRAP53   \n",
       "14299                 0.00                 0.04                  NF1   \n",
       "15693                 0.00                 0.00                 INSR   \n",
       "15985                 0.10                 0.93                DNMT1   \n",
       "16492                 0.98                 0.00               PIK3R2   \n",
       "17518                 0.00                 0.00                CHEK2   \n",
       "17866                 0.00                 0.00                KDM6A   \n",
       "\n",
       "       ada_score  rf_score  MaxEntScan_alt  MaxEntScan_diff  MaxEntScan_ref  \n",
       "1828         NaN       NaN             NaN              NaN             NaN  \n",
       "2132         NaN       NaN             NaN              NaN             NaN  \n",
       "2167         NaN       NaN             NaN              NaN             NaN  \n",
       "3002         NaN       NaN           2.540            2.877           5.416  \n",
       "3041         NaN       NaN             NaN              NaN             NaN  \n",
       "3638         NaN       NaN             NaN              NaN             NaN  \n",
       "4051         NaN       NaN             NaN              NaN             NaN  \n",
       "5604         NaN       NaN             NaN              NaN             NaN  \n",
       "7302         NaN       NaN             NaN              NaN             NaN  \n",
       "7957         NaN       NaN          -2.079            7.603           5.523  \n",
       "7982         NaN       NaN           1.415            8.182           9.597  \n",
       "9040         NaN       NaN             NaN              NaN             NaN  \n",
       "11028        NaN       NaN             NaN              NaN             NaN  \n",
       "11115        NaN       NaN             NaN              NaN             NaN  \n",
       "12047        NaN       NaN             NaN              NaN             NaN  \n",
       "12048        NaN       NaN             NaN              NaN             NaN  \n",
       "12050        NaN       NaN             NaN              NaN             NaN  \n",
       "12052        NaN       NaN             NaN              NaN             NaN  \n",
       "12826        NaN       NaN           0.299            5.115           5.413  \n",
       "13244        NaN       NaN             NaN              NaN             NaN  \n",
       "13411        NaN       NaN             NaN              NaN             NaN  \n",
       "13412        NaN       NaN             NaN              NaN             NaN  \n",
       "13802        NaN       NaN             NaN              NaN             NaN  \n",
       "14156        NaN       NaN             NaN              NaN             NaN  \n",
       "14299        NaN       NaN             NaN              NaN             NaN  \n",
       "15693        NaN       NaN             NaN              NaN             NaN  \n",
       "15985   0.000095     0.006             NaN              NaN             NaN  \n",
       "16492        NaN       NaN             NaN              NaN             NaN  \n",
       "17518        NaN       NaN             NaN              NaN             NaN  \n",
       "17866        NaN       NaN             NaN              NaN             NaN  "
      ]
     },
     "execution_count": 156,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n=0.9\n",
    "naga_all[cols_cpra+search_col('spliceai')+['ada_score', 'rf_score']+search_col('maxe')][\n",
    "    ((naga_all['SpliceAI_pred_DS_AG']>n)|(naga_all['SpliceAI_pred_DS_AL']>n)|(naga_all['SpliceAI_pred_DS_DG']>n)\n",
    "     |(naga_all['SpliceAI_pred_DS_DL']>n))&\n",
    "    (((naga_all['ada_score'].isnull())|(naga_all['rf_score'].isnull()))|(naga_all['MaxEntScan_diff'].isnull()))\n",
    "].drop_duplicates(cols_cpra)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f84806af",
   "metadata": {},
   "source": [
    "### Overview count of variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "2677c50c",
   "metadata": {},
   "outputs": [],
   "source": [
    "ov_all = var_scores(naga_all)\n",
    "#ov_all.to_excel('03_prioritized_xlsx/overview_var_counts_all.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6e367b19",
   "metadata": {},
   "source": [
    "# For grouped variants"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4e542393",
   "metadata": {},
   "source": [
    "## Variants grouped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "de4c6ef9",
   "metadata": {},
   "outputs": [],
   "source": [
    "vars_grpd = naga_all.drop_duplicates(subset=cols_cpra)\n",
    "#len(naga_all), len(vars_grpd)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a0bf9a85",
   "metadata": {},
   "source": [
    "## SpliceAI grouped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "f28e6b88",
   "metadata": {},
   "outputs": [],
   "source": [
    "vars_grpd_splai = vars_grpd[vars_grpd['SpliceAI_pred_DP_AG'].notnull()].copy()\n",
    "vars_grpd_splai = make_float(vars_grpd_splai)\n",
    "\n",
    "vars_grpd_splai = vars_grpd_splai[cols_cpra][((vars_grpd_splai['SpliceAI_pred_DS_AG']>0.9)|\n",
    "                                              (vars_grpd_splai['SpliceAI_pred_DS_AL']>0.9)|\n",
    "                                              (vars_grpd_splai['SpliceAI_pred_DS_DG']>0.9)|\n",
    "                                              (vars_grpd_splai['SpliceAI_pred_DS_DL']>0.9))\n",
    "                                    ].drop_duplicates()\n",
    "vars_grpd_splai.reset_index(drop=True, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "4c9270f1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "219"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(naga_all[((naga_all['SpliceAI_pred_DS_AG']>0.9)|(naga_all['SpliceAI_pred_DS_AL']>0.9)|\n",
    "              (naga_all['SpliceAI_pred_DS_DG']>0.9)|(naga_all['SpliceAI_pred_DS_DL']>0.9))].drop_duplicates())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b5b03eed",
   "metadata": {},
   "source": [
    "## MaxEntScan grouped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a00ef6bb",
   "metadata": {},
   "outputs": [],
   "source": [
    "vars_grpd_MaxEntScan = vars_grpd[columns_df_prior][vars_grpd['MaxEntScan_alt'].notnull()].copy()\n",
    "vars_grpd_MaxEntScan[['MaxEntScan_alt', 'MaxEntScan_diff', 'MaxEntScan_ref']] = vars_grpd_MaxEntScan[[\n",
    "    'MaxEntScan_alt', 'MaxEntScan_diff', 'MaxEntScan_ref']].astype(float)\n",
    "\n",
    "# disr_ss = disrupts native splice site, diff > 0\n",
    "disr_ss_vars = vars_grpd_MaxEntScan[(vars_grpd_MaxEntScan['MaxEntScan_diff']>=0)].copy()\n",
    "disr_high_vars = disr_ss_vars[cols_cpra][disr_ss_vars['MaxEntScan_alt']<6.2]\n",
    "disr_mod_vars =disr_ss_vars[cols_cpra][(disr_ss_vars['MaxEntScan_alt']>=6.2)&\n",
    "                                     (disr_ss_vars['MaxEntScan_alt']<=8.5)]\n",
    "disr_low_vars = disr_ss_vars[cols_cpra][disr_ss_vars['MaxEntScan_alt']>8.5]\n",
    "\n",
    "#disr_high_xlsx, disr_mod_xlsx, disr_low_xlsx = [xlsx_df(i) for i in [disr_high, disr_mod, disr_low]]\n",
    "\n",
    "# new_ss = creates new splice site, diff < 0\n",
    "new_ss_vars = vars_grpd_MaxEntScan[(vars_grpd_MaxEntScan['MaxEntScan_diff']<0)].copy()\n",
    "new_high_vars = new_ss_vars[cols_cpra][new_ss_vars['MaxEntScan_alt']>8.5]\n",
    "new_mod_vars = new_ss_vars[cols_cpra][(new_ss_vars['MaxEntScan_alt']>=6.2)&\n",
    "                                      (new_ss_vars['MaxEntScan_alt']<=8.5)]\n",
    "new_low_vars = new_ss_vars[cols_cpra][new_ss_vars['MaxEntScan_alt']<6.2]\n",
    "\n",
    "# reset index for all dfs\n",
    "for df in [disr_high_vars, disr_low_vars, disr_mod_vars, new_high_vars, new_low_vars, new_mod_vars]:\n",
    "    df.reset_index(drop=True, inplace=True)\n",
    "\n",
    "#new_high_xlsx, new_mod_xlsx, new_low_xlsx = [xlsx_df(i) for i in [new_high, new_mod, new_low]]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "005ac265",
   "metadata": {},
   "source": [
    "## dbscSNV (ada score, rf score) grouped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1d67dfdc",
   "metadata": {},
   "outputs": [],
   "source": [
    "vars_grpd_dbscSNV = vars_grpd[columns_df_prior][vars_grpd['ada_score'].notnull()&\n",
    "                                                vars_grpd['rf_score'].notnull()].copy()\n",
    "vars_grpd_dbscSNV[['ada_score', 'rf_score']] = vars_grpd_dbscSNV[['ada_score', 'rf_score']].astype(float)\n",
    "dbscSNV_high_vars = vars_grpd_dbscSNV[cols_cpra][(vars_grpd_dbscSNV['ada_score']>0.6)&\n",
    "                                                 (vars_grpd_dbscSNV['rf_score']>0.6)].copy()\n",
    "dbscSNV_high_vars.reset_index(drop=True, inplace=True)\n",
    "#dbscSNV_high_xlsx = xlsx_df(dbscSNV_high)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "84c98e49",
   "metadata": {},
   "source": [
    "## Variant position rel. to splice site grouped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d042ed2c",
   "metadata": {},
   "outputs": [],
   "source": [
    "vars_grpd_pos_rel_df = vars_grpd[cols_cpra][(vars_grpd['PosExon_type'].isin(['outsideAcceptorSite', \n",
    "                                                        'outsideDonor']))&(vars_grpd['PosExonRefSeqAccession']<3)]\n",
    "vars_grpd_pos_rel_df.reset_index(drop=True, inplace=True)\n",
    "#pos_rel_df_xlsx = xlsx_df(pos_rel_df)\n",
    "#pos_rel_df_xlsx.to_excel('03_prioritized_xlsx/var_pos_ss.xlsx', index=False, sheet_name='Pos<3,oA,oD')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6d3f27a5",
   "metadata": {},
   "source": [
    "## ClinVar characterization grouped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5e14348e",
   "metadata": {},
   "outputs": [],
   "source": [
    "clinvar_unknown_vars = vars_grpd[cols_cpra][(vars_grpd['ClinVar_Pathogenicity']=='Uncertain Significance')|\n",
    "                                            (vars_grpd['ClinVar_Pathogenicity']=='uncertain_significance')|\n",
    "                                            (vars_grpd['ClinVar_Pathogenicity']=='Uncertain significance')|\n",
    "                                            (vars_grpd['ClinVar_Pathogenicity']=='not_provided')|\n",
    "                                            (vars_grpd['ClinVar_Pathogenicity'].isnull())]\n",
    "clinvar_benign_vars = vars_grpd[cols_cpra][(vars_grpd['ClinVar_Pathogenicity']=='Benign')|\n",
    "                                           (vars_grpd['ClinVar_Pathogenicity']=='Likely Benign')]\n",
    "clinvar_patho_vars = vars_grpd[cols_cpra][(vars_grpd['ClinVar_Pathogenicity']=='Pathogenic')|\n",
    "                                          (vars_grpd['ClinVar_Pathogenicity']=='Likely Pathogenic')]\n",
    "for df in [clinvar_unknown_vars, clinvar_benign_vars, clinvar_patho_vars]:\n",
    "    df.reset_index(drop=True, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fa9a16f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "controls_cols = ['#CHROM', 'POS', 'REF', 'ALT', 'MaxEntScan_alt', 'MaxEntScan_diff', 'MaxEntScan_ref', \n",
    "                 'CADD_PHRED', 'SpliceAI_pred_DP_AG', 'SpliceAI_pred_DP_AL', 'SpliceAI_pred_DP_DG', \n",
    "                 'SpliceAI_pred_DP_DL', 'SpliceAI_pred_DS_AG', 'SpliceAI_pred_DS_AL', 'SpliceAI_pred_DS_DG', \n",
    "                 'SpliceAI_pred_DS_DL', 'SpliceAI_pred_SYMBOL', 'ada_score', 'rf_score',\n",
    "                 'PosExonRefSeqAccession', 'PosExon_type', 'ClinVar_Pathogenicity', 'ClinVar_Traits', \n",
    "                 'VEP_Most_Severe_Consequence', 'PID']\n",
    "controls_xlsx_cols = controls_cols[:-1]\n",
    "controls_xlsx_cols.extend(['PIDs', 'ct_PIDs'])\n",
    "\n",
    "clinvar_benign_vars2 = vars_grpd[cols_cpra][(vars_grpd['ClinVar_Pathogenicity']=='Benign')|\n",
    "                                           (vars_grpd['ClinVar_Pathogenicity']=='Likely Benign')]\n",
    "clinvar_patho_vars2 = vars_grpd[cols_cpra][(vars_grpd['ClinVar_Pathogenicity']=='Pathogenic')|\n",
    "                                          (vars_grpd['ClinVar_Pathogenicity']=='Likely Pathogenic')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0306e5f3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# with VEP most severe consequence for pathogenic variants as controls\n",
    "merged2 = pd.merge(naga_all[controls_cols], clinvar_patho_vars2, how='inner')\n",
    "lists2 = []\n",
    "for p,df2 in merged2.groupby(by=['#CHROM', 'POS', 'REF', 'ALT']):\n",
    "    df2 = df2.reset_index(drop=True)\n",
    "    list_df2 = [df2.iloc[0][i] for i in controls_cols[:-1]]\n",
    "    list_df2.extend([list(df2['PID']), len(df2['PID'])])\n",
    "    lists2.append(list_df2)\n",
    "new_df2 = pd.DataFrame(lists2, columns=controls_xlsx_cols)\n",
    "new_df2['#CHROM'] = pd.Categorical(new_df2['#CHROM'],\n",
    "                                    categories=['1','2','3','4','5','6','7','8','9','10','11','12', '13', '14', \n",
    "                                                '15', '16', '17', '18', '19', '20', '21', '22', 'X', 'Y'],\n",
    "                                    ordered=True)\n",
    "new_df2.sort_values(by=['ct_PIDs', '#CHROM', 'POS', 'REF', 'ALT'], inplace=True, ignore_index=True, \n",
    "                    ascending=[False, True, True, True, True])\n",
    "new_df2.drop(axis='columns', columns='ct_PIDs', inplace=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2493bbcb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# RNA expression data for specific variants\n",
    "pd.merge(merged2[['#CHROM', 'POS', 'REF', 'ALT', 'PID']], naga_all, how='left')[['#CHROM', 'POS', 'REF', 'ALT', \n",
    "         'PID', 'Control_VAF', 'Tumor_VAF', 'exome_control_AF', 'local_control_AF', 'RNA_VAF', 'AF_C', 'AF_T', \n",
    "         'AF_RNA', 'AF_RNA_noSkipReads', 'Tumor_dpALT', 'Control_dpALT', 'Alt_T', 'Alt_C', 'Cov_T', 'Cov_C']][\n",
    "    (pd.merge(merged2[['#CHROM', 'POS', 'REF', 'ALT', 'PID']], naga_all, how='left')['AF_C'].notnull())&\n",
    "    (pd.merge(merged2[['#CHROM', 'POS', 'REF', 'ALT', 'PID']], naga_all, how='left')['VEP_Most_Severe_Consequence']\n",
    "     =='splice_acceptor_variant')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "70d4b542",
   "metadata": {},
   "outputs": [],
   "source": [
    "#new_df2.to_excel('03_prioritized_xlsx/clinvar_patho_vep.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a4615c7d",
   "metadata": {},
   "source": [
    "## Arnes variant table with positive controls by Gepado"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a45c4417",
   "metadata": {},
   "outputs": [],
   "source": [
    "arne_pos = pd.read_excel('03_prioritized_xlsx/arne_pos_crtls.xlsx')\n",
    "arne_pos[['#CHROM', 'REF', 'ALT']] = arne_pos[['#CHROM', 'REF', 'ALT']].astype(str)\n",
    "arne_pos['POS'] = arne_pos['POS'].astype(int)\n",
    "arne_pos = arne_pos.rename(columns={'PID':'PID_trans'})\n",
    "arne_pos_vars = arne_pos[cols_cpra].drop_duplicates().reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "88cef6ee",
   "metadata": {},
   "outputs": [],
   "source": [
    "# variants of Arnes table (28) not all in Nagas table (14)\n",
    "arne_naga = pd.merge((arne_pos[['#CHROM', 'POS', 'REF', 'ALT']]).drop_duplicates(ignore_index=True), vars_grpd, \n",
    "                     how='inner')\n",
    "xlsx_vars(arne_naga, 'arne_naga_vars')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "465117c9",
   "metadata": {},
   "outputs": [],
   "source": [
    "#arne_naga[~arne_naga['HGVSg'].isin(pd.merge(ol_splai_mes_dbscSNV, naga_all, how='inner')['HGVSg'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e819383f",
   "metadata": {},
   "outputs": [],
   "source": [
    "#ol_splai_mes_dbscSNV_naga['ClinVar_Pathogenicity'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dbe6503b",
   "metadata": {},
   "source": [
    "## Hot/Cold splice project gene priority grouped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ed31a9b5",
   "metadata": {},
   "outputs": [],
   "source": [
    "hot_gene_vars = vars_grpd[cols_cpra][vars_grpd['Splice Project Gene Priority']==\n",
    "                                     'Hot (ACMG / MASTER)'].reset_index(drop=True)\n",
    "warm_gene_vars = vars_grpd[cols_cpra][vars_grpd['Splice Project Gene Priority']==\n",
    "                                      'Warm (MASTER-ACMG)'].reset_index(drop=True)\n",
    "cold_gene_vars = vars_grpd[cols_cpra][vars_grpd['Splice Project Gene Priority']==\n",
    "                                      'Cold (387-ACMG-MASTER)'].reset_index(drop=True)\n",
    "#len(vars_grpd), len(hot_gene_vars), len(warm_gene_vars), len(cold_gene_vars)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1ecaf2ab",
   "metadata": {},
   "source": [
    "## gnomAD AF grouped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "98931874",
   "metadata": {},
   "outputs": [],
   "source": [
    "# previously filtered by Naga -> only variants with AF < 0.005\n",
    "ga_af_vars = vars_grpd[vars_grpd['max_gnomAD_AF'].notnull()].copy()\n",
    "ga_af_vars['max_gnomAD_AF'] = ga_af_vars['max_gnomAD_AF'].astype(float)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eb1ace3c",
   "metadata": {},
   "source": [
    "## Variants without splice prediction scores"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "38ac0306",
   "metadata": {},
   "outputs": [],
   "source": [
    "no_splAI = vars_grpd[vars_grpd['SpliceAI_pred_DP_AG'].isnull()]\n",
    "no_MES = vars_grpd[vars_grpd['MaxEntScan_alt'].isnull()]\n",
    "no_dbsc = vars_grpd[vars_grpd['rf_score'].isnull()|vars_grpd['ada_score'].isnull()]\n",
    "\n",
    "len(vars_grpd), len(no_dbsc), len(no_MES), len(no_splAI)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5e64783b",
   "metadata": {},
   "outputs": [],
   "source": [
    "#no_splAI[(no_splAI['ALT'].str.len()==1)&(no_splAI['REF'].str.len()==1)]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "23cc0b11",
   "metadata": {},
   "source": [
    "## Overlap of criteria grouped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "49e06ef2",
   "metadata": {},
   "outputs": [],
   "source": [
    "#len(vars_grpd_splai), len(disr_high_vars), len(dbscSNV_high_vars), len(vars_grpd_pos_rel_df), len(clinvar_patho_vars)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "60fa129c",
   "metadata": {},
   "outputs": [],
   "source": [
    "ol_splai_mes = pd.merge(vars_grpd_splai, disr_high_vars, how='inner')\n",
    "ol_splai_mes_dbscSNV = pd.merge(ol_splai_mes, dbscSNV_high_vars, how='inner')\n",
    "ol_splai_mes_dbscSNV_hot = pd.merge(ol_splai_mes_dbscSNV, hot_gene_vars, how='inner')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0a390713",
   "metadata": {},
   "source": [
    "## Evaluated variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "92115c9d",
   "metadata": {},
   "outputs": [],
   "source": [
    "new_splAI_high = pd.merge(vars_grpd_splai, vars_grpd, how='left'\n",
    "                         )[~pd.merge(vars_grpd_splai, vars_grpd, how='left')['HGVSg'].isin(\n",
    "    pd.merge(ol_splai_mes_dbscSNV, vars_grpd, how='left')['HGVSg'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a2f2b4bc",
   "metadata": {},
   "outputs": [],
   "source": [
    "new_MES_high_disr = pd.merge(disr_high_vars, vars_grpd, how='left'\n",
    "                         )[~pd.merge(disr_high_vars, vars_grpd, how='left')['HGVSg'].isin(\n",
    "    pd.merge(ol_splai_mes_dbscSNV, vars_grpd, how='left')['HGVSg'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b72f7b1d",
   "metadata": {},
   "outputs": [],
   "source": [
    "new_MES_high_new = pd.merge(new_high_vars, vars_grpd, how='left'\n",
    "                         )[~pd.merge(new_high_vars, vars_grpd, how='left')['HGVSg'].isin(\n",
    "    pd.merge(ol_splai_mes_dbscSNV, vars_grpd, how='left')['HGVSg'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7334e947",
   "metadata": {},
   "outputs": [],
   "source": [
    "new_dbsc_high = pd.merge(dbscSNV_high_vars, vars_grpd, how='left'\n",
    "                         )[~pd.merge(dbscSNV_high_vars, vars_grpd, how='left')['HGVSg'].isin(\n",
    "    pd.merge(ol_splai_mes_dbscSNV, vars_grpd, how='left')['HGVSg'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f425490a",
   "metadata": {},
   "outputs": [],
   "source": [
    "new_pos_rel = pd.merge(vars_grpd_pos_rel_df, vars_grpd, how='left'\n",
    "                         )[~pd.merge(vars_grpd_pos_rel_df, vars_grpd, how='left')['HGVSg'].isin(\n",
    "    pd.merge(ol_splai_mes_dbscSNV, vars_grpd, how='left')['HGVSg'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ae722600",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "65faa24f",
   "metadata": {},
   "source": [
    "## Make .xlsx with selected columns, PIDs for grouped variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a87bd4fd",
   "metadata": {},
   "outputs": [],
   "source": [
    "# dfs with all important information to .xlsx\n",
    "#for df, n in [(vars_grpd_splai, 'spliceAI_high'), (disr_high_vars, 'MaxEntScan_high_disr'), \n",
    "              #(new_high_vars, 'MaxEntScan_high_new_ss'), (dbscSNV_high_vars, 'dbscSNV_high'), \n",
    "              #(vars_grpd_pos_rel_df, 'pos_rel_IntExBound_under3'), (clinvar_benign_vars, 'clinvar_benign'), \n",
    "              #(clinvar_patho_vars, 'clinvar_patho'), (ol_splai_mes_dbscSNV_naga, 'merge_splAI_MES_dbsc')]:\n",
    "    #xlsx_vars(df, n)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ee055f6e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#for df, n in [(new_splAI_high, 'new_splAI_high'), (new_MES_high_disr, 'new_MES_high_disr'), \n",
    "              #(new_MES_high_new, 'new_MES_high_new'), (new_dbsc_high, 'new_dbsc_high'), \n",
    "              #(new_pos_rel, 'new_pos_rel')]:\n",
    "    #xlsx_vars(df, n)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a7651b4d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# merged df to .xlsx\n",
    "#xlsx_vars(ol_splai_mes_dbscSNV_hot, 'merge_splAI_MES_dbsc_hot')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f549e63a",
   "metadata": {},
   "source": [
    "# For original df, all rows"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2bf835b6",
   "metadata": {},
   "source": [
    "## SpliceAI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fbb98d5f",
   "metadata": {},
   "outputs": [],
   "source": [
    "spl_ai = naga_all[naga_all['SpliceAI_pred_DP_AG'].notnull()].copy()\n",
    "spl_ai = make_float(spl_ai)\n",
    "\n",
    "# how many rows in each cutoff\n",
    "splAI_ct = [(round(i,1), len(spl_ai[(spl_ai['SpliceAI_pred_DS_AG']>i)|(spl_ai['SpliceAI_pred_DS_AL']>i)|\n",
    "                            (spl_ai['SpliceAI_pred_DS_DG']>i)|(spl_ai['SpliceAI_pred_DS_DL']>i)]))\n",
    "            for i in np.arange(0,1,0.1)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "22661b34",
   "metadata": {},
   "outputs": [],
   "source": [
    "# high spliceAI scores (>0.9)\n",
    "spl_ai_df = spl_ai[columns_df_prior][((spl_ai['SpliceAI_pred_DS_AG']>0.9)|(spl_ai['SpliceAI_pred_DS_AL']>0.9)|\n",
    "                                      (spl_ai['SpliceAI_pred_DS_DG']>0.9)|(spl_ai['SpliceAI_pred_DS_DL']>0.9))\n",
    "                                    ].drop_duplicates()\n",
    "spl_ai_df_xlsx = xlsx_df(spl_ai_df)\n",
    "spl_ai_df_xlsx.to_excel('03_prioritized_xlsx/splAI_vars.xlsx', index=False, sheet_name='SpliceAI>0.9')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a18562fb",
   "metadata": {},
   "outputs": [],
   "source": [
    "spl_ai_gain = spl_ai[columns_df_prior][(spl_ai['PosExonRefSeqAccession']>2)&\n",
    "                                       ((spl_ai['SpliceAI_pred_DS_AG']>0.9)|(spl_ai['SpliceAI_pred_DS_AL']>0.9)|\n",
    "                                        (spl_ai['SpliceAI_pred_DS_DG']>0.9)|(spl_ai['SpliceAI_pred_DS_DL']>0.9))\n",
    "                                      ].copy()\n",
    "spl_ai_gain_xlsx = xlsx_df(spl_ai_gain)\n",
    "spl_ai_gain_xlsx.to_excel('03_prioritized_xlsx/splAI_gain.xlsx', index=False, sheet_name='SpliceAI>0.9,pos>2')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4e16f557",
   "metadata": {},
   "source": [
    "## MaxEntScan"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "851bda3c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# differentiation between: (1) variant disrupts native splice site and (2) variant creates new splice site\n",
    "MaxEntScan = naga_all[columns_df_prior][naga_all['MaxEntScan_alt'].notnull()].copy()\n",
    "MaxEntScan[['MaxEntScan_alt', 'MaxEntScan_diff', 'MaxEntScan_ref']] = MaxEntScan[['MaxEntScan_alt', \n",
    "                                                                                  'MaxEntScan_diff', \n",
    "                                                                                  'MaxEntScan_ref']].astype(float)\n",
    "# disr_ss = disrupts native splice site, diff > 0\n",
    "disr_ss = MaxEntScan[(MaxEntScan['MaxEntScan_diff']>=0)]\n",
    "disr_high = disr_ss[disr_ss['MaxEntScan_alt']<6.2]\n",
    "disr_mod = disr_ss[(disr_ss['MaxEntScan_alt']>=6.2)&(disr_ss['MaxEntScan_alt']<=8.5)]\n",
    "disr_low = disr_ss[disr_ss['MaxEntScan_alt']>8.5]\n",
    "\n",
    "disr_high_xlsx, disr_mod_xlsx, disr_low_xlsx = [xlsx_df(i) for i in [disr_high, disr_mod, disr_low]]\n",
    "\n",
    "# new_ss = creates new splice site, diff < 0\n",
    "new_ss = MaxEntScan[(MaxEntScan['MaxEntScan_diff']<0)]\n",
    "new_high = new_ss[new_ss['MaxEntScan_alt']>8.5]\n",
    "new_mod = new_ss[(new_ss['MaxEntScan_alt']>=6.2)&(new_ss['MaxEntScan_alt']<=8.5)]\n",
    "new_low = new_ss[new_ss['MaxEntScan_alt']<6.2]\n",
    "\n",
    "new_high_xlsx, new_mod_xlsx, new_low_xlsx = [xlsx_df(i) for i in [new_high, new_mod, new_low]]\n",
    "\n",
    "lens_MES = [(n,len(d)) for d,n in zip([disr_ss, disr_high, disr_mod, disr_low, new_ss, new_high, new_mod, new_low],\n",
    "                                      ['disr_ss', 'disr_high', 'disr_mod', 'disr_low', \n",
    "                                       'new_ss', 'new_high', 'new_mod', 'new_low'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "982bb93e",
   "metadata": {},
   "outputs": [],
   "source": [
    "writer = pd.ExcelWriter('03_prioritized_xlsx/MaxEntScan_vars.xlsx')\n",
    "disr_high_xlsx[disr_high_xlsx['PosExonRefSeqAccession']<3].to_excel(writer, index=False, \n",
    "                                                                    sheet_name='disr_high_pos<3')\n",
    "new_high_xlsx[new_high_xlsx['PosExonRefSeqAccession']>15].to_excel(writer, index=False, \n",
    "                                                                   sheet_name='new_high_pos>15')\n",
    "writer.save()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fb684147",
   "metadata": {},
   "source": [
    "## dbscSNV (ada score, rf score)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6cdad44e",
   "metadata": {},
   "outputs": [],
   "source": [
    "dbscSNV = naga_all[naga_all['ada_score'].notnull()&naga_all['rf_score'].notnull()].copy()\n",
    "dbscSNV[['ada_score', 'rf_score']] = dbscSNV[['ada_score', 'rf_score']].astype(float)\n",
    "dbscSNV_high = dbscSNV[(dbscSNV['ada_score']>0.6)&(dbscSNV['rf_score']>0.6)].copy()\n",
    "dbscSNV_high_xlsx = xlsx_df(dbscSNV_high)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "be19ea4d",
   "metadata": {},
   "source": [
    "## Variant position rel. to splice site"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f07deda9",
   "metadata": {},
   "outputs": [],
   "source": [
    "pos_rel_df = naga_all[columns_df_prior][(naga_all['PosExon_type'].isin(['outsideAcceptorSite', 'outsideDonor']))&\n",
    "                                        (naga_all['PosExonRefSeqAccession']<3)].drop_duplicates()\n",
    "pos_rel_df_xlsx = xlsx_df(pos_rel_df)\n",
    "pos_rel_df_xlsx.to_excel('03_prioritized_xlsx/var_pos_ss.xlsx', index=False, sheet_name='Pos<3,oA,oD')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "23fc6662",
   "metadata": {},
   "source": [
    "## Hot/Cold splice project gene priority"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "04151d1c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# gene priorities\n",
    "hot_gene, warm_gene, cold_gene = [naga_all[naga_all['Splice Project Gene Priority']==i].copy() for i in \n",
    "                                  ['Hot (ACMG / MASTER)', 'Warm (MASTER-ACMG)', 'Cold (387-ACMG-MASTER)']]\n",
    "hot_gene_xlsx = xlsx_df(hot_gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b59eea77",
   "metadata": {},
   "outputs": [],
   "source": [
    "# overlap hot splice gene priority, high spliceAI score > 0.9\n",
    "hot_splAI_xlsx = pd.merge(spl_ai_df_xlsx, hot_gene_xlsx[cols_cpra], how='inner')\n",
    "hot_splAI_xlsx.to_excel('03_prioritized_xlsx/hot_gene_splAI.xlsx', sheet_name='hotGenePriority,splAI>0.9', \n",
    "                        index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7316815f",
   "metadata": {},
   "source": [
    "## ClinVar characterization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4243dc04",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ClinVar groups\n",
    "# clinvar_unknown: 12,520 rows (6,533 variants)\n",
    "# clinvar_benign: 5,475 rows (1,603 variants)\n",
    "# clinvar_patho: 173 rows (101 variants)\n",
    "clinvar_unknown = naga_all[(naga_all['ClinVar_Pathogenicity']=='Uncertain Significance')|\n",
    "                           (naga_all['ClinVar_Pathogenicity']=='uncertain_significance')|\n",
    "                           (naga_all['ClinVar_Pathogenicity']=='Uncertain significance')|\n",
    "                           (naga_all['ClinVar_Pathogenicity']=='not_provided')|\n",
    "                           (naga_all['ClinVar_Pathogenicity'].isnull())]\n",
    "clinvar_benign = naga_all[(naga_all['ClinVar_Pathogenicity']=='Benign')|\n",
    "                          (naga_all['ClinVar_Pathogenicity']=='Likely Benign')]\n",
    "clinvar_patho = naga_all[(naga_all['ClinVar_Pathogenicity']=='Pathogenic')|\n",
    "                         (naga_all['ClinVar_Pathogenicity']=='Likely Pathogenic')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c8e4d35d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# for positive control: pathogenic, spliceAI>0.9 --> 63 rows, 30 variants\n",
    "path_splAI = clinvar_patho[clinvar_patho['SpliceAI_pred_DP_AG'].notnull()].copy()\n",
    "path_splAI = make_float(path_splAI)\n",
    "path_splAI = path_splAI[(path_splAI['SpliceAI_pred_DS_AG']>0.9)|(path_splAI['SpliceAI_pred_DS_AL']>0.9)|\n",
    "                        (path_splAI['SpliceAI_pred_DS_DG']>0.9)|(path_splAI['SpliceAI_pred_DS_DL']>0.9)\n",
    "                       ].reset_index(drop=True)\n",
    "path_splAI_xlsx = xlsx_df(path_splAI)\n",
    "\n",
    "# for negative control: benign, spliceAI == 0, rf/ada score < 0.001 --> 164 rows, 46 variants\n",
    "ben_splAI = clinvar_benign[(clinvar_benign['SpliceAI_pred_DP_AG'].notnull())&\n",
    "                           (clinvar_benign['ada_score'].notnull())].copy()\n",
    "ben_splAI = make_float(ben_splAI)\n",
    "ben_splAI[['ada_score', 'rf_score']] = ben_splAI[['ada_score', 'rf_score']].astype(float)\n",
    "ben_splAI = ben_splAI[(ben_splAI['SpliceAI_pred_DS_AG']==0)&(ben_splAI['SpliceAI_pred_DS_AL']==0)&\n",
    "                      (ben_splAI['SpliceAI_pred_DS_DG']==0)&(ben_splAI['SpliceAI_pred_DS_DL']==0)&\n",
    "                      (ben_splAI['MaxEntScan_diff'].notnull())&(ben_splAI['ada_score']<0.001)&\n",
    "                      (ben_splAI['rf_score']<0.001)].reset_index(drop=True)\n",
    "ben_splAI_xlsx = xlsx_df(ben_splAI)\n",
    "\n",
    "# in excel file\n",
    "writer = pd.ExcelWriter('03_prioritized_xlsx/control_vars.xlsx')\n",
    "path_splAI_xlsx.to_excel(writer, index=False, sheet_name='pos_crtl_path_high_splAI')\n",
    "ben_splAI_xlsx.to_excel(writer, index=False, sheet_name='neg_crtl_benign_low_scores')\n",
    "writer.save()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1763dd29",
   "metadata": {},
   "source": [
    "## Allele Frequency gnomAD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "749375e3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# AF by gnomAD ('max_gnomAD_AF') --> 15,174 rows (6,331 variants), all variants < 0.005 AF because of cutoff Naga\n",
    "gnomAD_af = naga_all[naga_all['max_gnomAD_AF'].notnull()].copy()\n",
    "gnomAD_af['max_gnomAD_AF'] = gnomAD_af['max_gnomAD_AF'].astype(float)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e9f2cc70",
   "metadata": {},
   "source": [
    "## Redundant variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "41e7e032",
   "metadata": {},
   "outputs": [],
   "source": [
    "red_vars = naga_all[['#CHROM', 'POS', 'REF', 'ALT']].value_counts()\n",
    "red_vars = red_vars[red_vars>5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e31e1031",
   "metadata": {},
   "outputs": [],
   "source": [
    "red_vars_df = naga_all.copy()\n",
    "red_vars_df_splAI = red_vars_df[columns_df_prior][(red_vars_df['SpliceAI_pred_DS_AG'].notnull())].copy()\n",
    "red_vars_df_splAI = make_float(red_vars_df_splAI)\n",
    "red_vars_splAI = []\n",
    "for (c,p,r,a), ct in zip(red_vars.index, red_vars):\n",
    "    df = red_vars_df_splAI[\n",
    "        ((red_vars_df_splAI['#CHROM']==c)&(red_vars_df_splAI['POS']==p)&\n",
    "         (red_vars_df_splAI['REF']==r)&(red_vars_df_splAI['ALT']==a))&\n",
    "        ((red_vars_df_splAI['SpliceAI_pred_DS_AG']>0)|(red_vars_df_splAI['SpliceAI_pred_DS_AL']>0)|\n",
    "         (red_vars_df_splAI['SpliceAI_pred_DS_DG']>0)|(red_vars_df_splAI['SpliceAI_pred_DS_DL']>0))\n",
    "    ].reset_index(drop=True)\n",
    "    if df.empty == False:\n",
    "        list_df = [df.iloc[0][i] for i in columns_df_prior[:-1]]\n",
    "        list_df.append(list(df['PID']))\n",
    "        red_vars_splAI.append(list_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c9f24d02",
   "metadata": {},
   "outputs": [],
   "source": [
    "red_vars_splAI_xlsx = pd.DataFrame(red_vars_splAI, columns=columns_df_xlsx)\n",
    "red_vars_splAI_xlsx = red_vars_splAI_xlsx.join(pd.DataFrame(red_vars_splAI_xlsx['PIDs'].to_list())\n",
    "                                              ).drop(columns='PIDs')\n",
    "red_vars_splAI_xlsx.to_excel('03_prioritized_xlsx/redundant_vars_new.xlsx', sheet_name='redundant_vars_o6', \n",
    "                             index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e09f06aa",
   "metadata": {},
   "source": [
    "## One patient, many variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f62764ea",
   "metadata": {},
   "outputs": [],
   "source": [
    "# circos plot would be possible\n",
    "by_patient = naga_all['PID'].value_counts()\n",
    "by_patient_20x = by_patient[by_patient>19]\n",
    "#len(by_patient), len(by_patient_20x)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d5d2b64e",
   "metadata": {},
   "source": [
    "## Overlap between prioritized variant categorizations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5d2d3395",
   "metadata": {},
   "outputs": [],
   "source": [
    "# overlap between spliceAI, position relative to ss --> 117 rows\n",
    "# overlap between spliceAI, position relative to ss, MaxEntScan disr_high --> 115 rows\n",
    "# overlap between spliceAI, position relative to ss, MaxEntScan disr_high, dbscSNV_high --> 114 rows, 52 variants\n",
    "splAI_rel = pd.merge(spl_ai_df[['#CHROM', 'POS', 'REF', 'ALT', 'PID']], \n",
    "                        pos_rel_df[['#CHROM', 'POS', 'REF', 'ALT', 'PID']], how='inner')\n",
    "splAI_rel_MES = pd.merge(splAI_rel, \n",
    "                         disr_high[['#CHROM', 'POS', 'REF', 'ALT', 'PID']], how='inner')\n",
    "splAI_rel_MES_dbsc = pd.merge(splAI_rel_MES, \n",
    "                              dbscSNV_high[['#CHROM', 'POS', 'REF', 'ALT', 'PID']], how = 'inner')\n",
    "naga_all_merge = pd.merge(splAI_rel_MES_dbsc, naga_all, on=['#CHROM', 'POS', 'REF', 'ALT', 'PID'], how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4cb81816",
   "metadata": {},
   "outputs": [],
   "source": [
    "# with _xlsx dataframes, variant-specific\n",
    "splAI_rel2 = pd.merge(spl_ai_df_xlsx[['#CHROM', 'POS', 'REF', 'ALT']], \n",
    "                        pos_rel_df_xlsx[['#CHROM', 'POS', 'REF', 'ALT']], how='inner')\n",
    "splAI_rel_MES2 = pd.merge(splAI_rel2, \n",
    "                         disr_high_xlsx[['#CHROM', 'POS', 'REF', 'ALT']], how='inner')\n",
    "splAI_rel_MES_dbsc2 = pd.merge(splAI_rel_MES2, \n",
    "                              dbscSNV_high_xlsx[['#CHROM', 'POS', 'REF', 'ALT']], how = 'inner')\n",
    "splAI_rel_MES_dbsc_red2 = pd.merge(splAI_rel_MES_dbsc2, \n",
    "                                   red_vars_splAI_xlsx[['#CHROM', 'POS', 'REF', 'ALT']], how = 'inner')\n",
    "#naga_all_merge2 = pd.merge(splAI_rel_MES_dbsc, naga_all, on=['#CHROM', 'POS', 'REF', 'ALT', 'PID'], how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8dfa9f57",
   "metadata": {},
   "outputs": [],
   "source": [
    "merge2 = pd.merge(splAI_rel_MES_dbsc_red2, red_vars_splAI_xlsx, on=['#CHROM', 'POS', 'REF', 'ALT'], how='left')\n",
    "merge_wona = merge2.copy()\n",
    "merge_wona.dropna(axis=1, how='all', inplace=True)\n",
    "merge_wona"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c16746f4",
   "metadata": {},
   "outputs": [],
   "source": [
    "naga_all_merge[columns_df_prior].to_excel('03_prioritized_xlsx/prioriz_naga_vars.xlsx', index=False, \n",
    "                                          sheet_name='SplAI,Pos<3,MES,dbscSNV')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5e4c2336",
   "metadata": {},
   "outputs": [],
   "source": [
    "# overlap high in spliceAI and create new splice site and position rel. to ss > 15 --> \n",
    "splAI_rel_MES_newSS = new_high[(new_high['SpliceAI_pred_DS_DG'].notnull())].copy()\n",
    "splAI_rel_MES_newSS = make_float(splAI_rel_MES_newSS)\n",
    "splAI_rel_MES_newSS =splAI_rel_MES_newSS[\n",
    "                               ((splAI_rel_MES_newSS['SpliceAI_pred_DS_AG']>0.2)|\n",
    "                                (splAI_rel_MES_newSS['SpliceAI_pred_DS_AL']>0.2)|\n",
    "                                (splAI_rel_MES_newSS['SpliceAI_pred_DS_DG']>0.2)|\n",
    "                                (splAI_rel_MES_newSS['SpliceAI_pred_DS_DL']>0.2))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "adafc357",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
