{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "843b3e60",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "56c89982",
   "metadata": {},
   "source": [
    "## Functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "69581f91",
   "metadata": {},
   "outputs": [],
   "source": [
    "# gus = genes of unknown significance\n",
    "gus = ['AKT2', 'AKT3', 'ATR', 'ATRX', 'CDKN1A', 'CDKN2B', 'CDKN2C', 'CDKN2D', 'CEBPA', 'CEP57', 'DAXX', 'EGFR', \n",
    "       'EGLN1', 'EGLN2', 'EPAS1', 'EPHB2', 'ABRAXAS1', 'FANCM', 'GPRC5A', 'HNF1A', 'HORMAD1', 'HORMAD2', 'JAK2', \n",
    "       'KIF1B', 'MAP3K1', 'MAP3K6', 'MDH2', 'MLH3', 'MMS19', 'MN1', 'MRE11', 'PALLD', 'PIK3C2G', 'PIK3R2', 'PMS1', \n",
    "       'PRF1', 'PTCH2', 'RAD50', 'RAD51', 'RECQL', 'RINT1']\n",
    "\n",
    "# make df .xlsx conform for xlsx_df function with more columns\n",
    "def conf_xlsx(df):\n",
    "    dfn = df.copy()\n",
    "    for i in ['consensus_sequence_IGV_region_left', 'consensus_sequence_IGV_sequence_left', \n",
    "              'consensus_sequence_IGV_region_right', 'consensus_sequence_IGV_sequence_right', \n",
    "              'INFO_IGV', 'too_low_coverage', 'splice_change']:\n",
    "        dfn[i] = np.nan\n",
    "    dfn['shortcut'] = dfn['#CHROM'].astype(str)+'_'+dfn['POS'].astype(str)+'_'+dfn['REF']+'_'+dfn['ALT']\n",
    "    return dfn\n",
    "\n",
    "# function to make excel tables with one variant per row\n",
    "cols_cpra = ['#CHROM', 'POS', 'REF', 'ALT']\n",
    "columns_df_prior = ['#CHROM', 'POS', 'REF', 'ALT', 'MaxEntScan_alt', 'MaxEntScan_diff', 'MaxEntScan_ref', \n",
    "                    'CADD_PHRED', 'SpliceAI_pred_DP_AG', 'SpliceAI_pred_DP_AL', 'SpliceAI_pred_DP_DG', \n",
    "                    'SpliceAI_pred_DP_DL', 'SpliceAI_pred_DS_AG', 'SpliceAI_pred_DS_AL', 'SpliceAI_pred_DS_DG', \n",
    "                    'SpliceAI_pred_DS_DL', 'SpliceAI_pred_SYMBOL', 'ada_score', 'rf_score',\n",
    "                    'PosExonRefSeqAccession', 'PosExon_type', 'ClinVar_Pathogenicity', \n",
    "                    'consensus_sequence_IGV_region_left', 'consensus_sequence_IGV_sequence_left', \n",
    "                    'consensus_sequence_IGV_region_right', 'consensus_sequence_IGV_sequence_right', 'hgsv_new',\n",
    "                    'INFO_IGV', 'too_low_coverage', 'splice_change', 'shortcut', 'ClinVar_Traits', \n",
    "                    'Splice Project Gene Priority', 'PID']\n",
    "columns_df_xlsx = columns_df_prior[:-1]\n",
    "columns_df_xlsx.extend(['PIDs', 'ct_PIDs'])\n",
    "\n",
    "def xlsx_df(prio_df, name):\n",
    "    lists = []\n",
    "    for p,df in prio_df.groupby(by=['#CHROM', 'POS', 'REF', 'ALT']):\n",
    "        df = df.reset_index(drop=True)\n",
    "        list_df = [df.iloc[0][i] for i in columns_df_prior[:-1]]\n",
    "        list_df.extend([list(df['PID']), len(df['PID'])])\n",
    "        lists.append(list_df)\n",
    "    new_df = pd.DataFrame(lists, columns=columns_df_xlsx)\n",
    "    new_df['#CHROM'] = pd.Categorical(new_df['#CHROM'],\n",
    "                                      categories=['1','2','3','4','5','6','7','8','9','10','11','12', '13', '14', \n",
    "                                                  '15', '16', '17', '18', '19', '20', '21', '22', 'X', 'Y'],\n",
    "                                      ordered=True)\n",
    "    new_df.sort_values(by=['ct_PIDs', '#CHROM', 'POS', 'REF', 'ALT'], inplace=True, ignore_index=True, \n",
    "                       ascending=[False, True, True, True, True])\n",
    "    new_df.drop(axis='columns', columns='ct_PIDs', inplace=True)\n",
    "    new_df.to_excel('01_xlsx_files/%s.xlsx'%name, index=False)\n",
    "\n",
    "def xlsx_vars(prio_vars_df, name):\n",
    "    merged = pd.merge(all_vars[columns_df_prior], prio_vars_df, how='inner')\n",
    "    lists = []\n",
    "    for p,df in merged.groupby(by=['#CHROM', 'POS', 'REF', 'ALT']):\n",
    "        df = df.reset_index(drop=True)\n",
    "        list_df = [df.iloc[0][i] for i in columns_df_prior[:-1]]\n",
    "        list_df.extend([list(df['PID']), len(df['PID'])])\n",
    "        lists.append(list_df)\n",
    "    new_df = pd.DataFrame(lists, columns=columns_df_xlsx)\n",
    "    new_df['#CHROM'] = pd.Categorical(new_df['#CHROM'],\n",
    "                                      categories=['1','2','3','4','5','6','7','8','9','10','11','12', '13', '14', \n",
    "                                                  '15', '16', '17', '18', '19', '20', '21', '22', 'X', 'Y'],\n",
    "                                      ordered=True)\n",
    "    new_df.sort_values(by=['ct_PIDs', '#CHROM', 'POS', 'REF', 'ALT'], inplace=True, ignore_index=True, \n",
    "                       ascending=[False, True, True, True, True])\n",
    "    new_df.drop(axis='columns', columns='ct_PIDs', inplace=True)\n",
    "    new_df.to_excel('01_xlsx_files/%s.xlsx'%name, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "10286278",
   "metadata": {},
   "outputs": [],
   "source": [
    "# function to search for specific name in column\n",
    "def search_col(df, inp):\n",
    "    return [i for i in df.columns if inp in i.lower()]\n",
    "\n",
    "# function to make spliceAI prediction scores into floats\n",
    "def make_float(df):\n",
    "    for i in ['AG', 'AL', 'DG', 'DL']:\n",
    "        df[('SpliceAI_pred_DS_'+i)]=df[('SpliceAI_pred_DS_'+i)].astype(float)\n",
    "    for i in ['ada_score', 'rf_score', 'MaxEntScan_diff', 'MaxEntScan_alt', 'MaxEntScan_ref']:\n",
    "        df[i] = df[i].astype(float)\n",
    "    return df\n",
    "\n",
    "# function to merge in IGV seen variant .xlsx files with all_vars with all infos\n",
    "def merge_xlsx(file_name):\n",
    "    df = pd.read_excel('01_xlsx_files/'+file_name)\n",
    "    merged = pd.merge(df[['#CHROM', 'POS', 'REF', 'ALT', 'consensus_sequence_IGV_region_left',\n",
    "                          'consensus_sequence_IGV_sequence_left', 'consensus_sequence_IGV_region_right',\n",
    "                          'consensus_sequence_IGV_sequence_right', 'exon_skipping', 'intron_retention', \n",
    "                          \"3'acceptor_loss\", \"3'acceptor_gain\", \"5'donor_loss\", \"5'donor_gain\", 'frameshift',\n",
    "                          'premat_stop_cod', 'hgsv_new', 'INFO_IGV', 'too_low_coverage',\n",
    "                          'splice_change', 'var_in_blood_perc', 'var_in_tumor_perc', 'counts_coverage', \n",
    "                          'perc_coverage', 'pos_last', 'pos_next', 'transc_pos_last', 'transc_pos_next', 'kind', \n",
    "                          'info_cDNA', 'more_consequences']], all_vars, on=cols_cpra, \n",
    "                      how='left').reset_index(drop=True)\n",
    "    merged['shortcut'] = (merged['#CHROM'].astype(str)+'_'+merged['POS'].astype(str)+\n",
    "                          '_'+merged['REF']+'_'+merged['ALT'])\n",
    "    merged['ClinVar_Pathogenicity'] = merged['ClinVar_Pathogenicity'].replace(\n",
    "        {'uncertain_significance':'Uncertain Significance', 'Uncertain significance':'Uncertain Significance'})\n",
    "    merged['PosExon_type'] = merged['PosExon_type'].replace({'insideDonor':'insideDonorSite', \n",
    "                                                             'outsideDonor':'outsideDonorSite'})\n",
    "    merged[['exon_skipping', 'intron_retention', \"3'acceptor_loss\", \"3'acceptor_gain\", \"5'donor_loss\", \n",
    "            \"5'donor_gain\"]] = merged[['exon_skipping', 'intron_retention', \"3'acceptor_loss\", \"3'acceptor_gain\", \n",
    "                                       \"5'donor_loss\", \"5'donor_gain\"]].replace({'x':True, np.nan:False})\n",
    "    merged['strand'] = merged['info_cDNA'].str.split(',', expand=True)[0]\n",
    "    merged = merged.drop(columns='info_cDNA')\n",
    "    merged['#CHROM'] = pd.Categorical(merged['#CHROM'],\n",
    "                                      categories=['1','2','3','4','5','6','7','8','9','10','11','12', '13', '14', \n",
    "                                                  '15', '16', '17', '18', '19', '20', '21', '22', 'X', 'Y'],\n",
    "                                      ordered=True)\n",
    "    merged.sort_values(by=['#CHROM', 'POS', 'REF', 'ALT'], inplace=True, ignore_index=True, \n",
    "                       ascending=True)\n",
    "    merged = make_float(merged)\n",
    "    return merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "49789acd",
   "metadata": {},
   "outputs": [],
   "source": [
    "# categorize variants into groups\n",
    "# spliceAI score > 0.9\n",
    "def splai_o09(df, coverage, splice_change):\n",
    "    return len(df[((df['SpliceAI_pred_DS_AG']>0.9)|\n",
    "                   (df['SpliceAI_pred_DS_AL']>0.9)|\n",
    "                   (df['SpliceAI_pred_DS_DG']>0.9)|\n",
    "                   (df['SpliceAI_pred_DS_DL']>0.9))&\n",
    "                  (df['too_low_coverage']==coverage)&\n",
    "                  (df['splice_change']==splice_change)].groupby(cols_cpra))\n",
    "\n",
    "# ada/rf score > 0.6\n",
    "def ada_rf_o06(df, coverage, splice_change):\n",
    "    return len(df[(df['ada_score']>0.6)&\n",
    "                  (df['rf_score']>0.6)&\n",
    "                  (df['too_low_coverage']==coverage)&\n",
    "                  (df['splice_change']==splice_change)].groupby(cols_cpra))\n",
    "\n",
    "# MaxEntScan high disruption of native splice site\n",
    "def MES_disr(df, coverage, splice_change):\n",
    "    return len(df[(df['MaxEntScan_diff']>=0)&\n",
    "                  (df['MaxEntScan_alt']<6.2)&\n",
    "                  (df['too_low_coverage']==coverage)&\n",
    "                  (df['splice_change']==splice_change)].groupby(cols_cpra))\n",
    "\n",
    "# MaxEntScan high possibility of creating new splice site\n",
    "def MES_new(df, coverage, splice_change):\n",
    "    return len(df[(df['MaxEntScan_diff']<0)&\n",
    "                  (df['MaxEntScan_alt']>8.5)&\n",
    "                  (df['too_low_coverage']==coverage)&\n",
    "                  (df['splice_change']==splice_change)].groupby(cols_cpra))\n",
    "\n",
    "# position relative to exon/intron boundary less than 3 and in intron\n",
    "def intron_pos(df, coverage, splice_change):\n",
    "    return len(df[(df['PosExonRefSeqAccession']<3)&\n",
    "                  (df['PosExon_type'].isin(['outsideAcceptorSite', 'outsideDonorSite', 'outsideDonor']))&\n",
    "                  (df['too_low_coverage']==coverage)&\n",
    "                  (df['splice_change']==splice_change)].groupby(cols_cpra))\n",
    "\n",
    "# variant categorization per hot/warm/cold genes\n",
    "def gene_prio(df):\n",
    "    hot_lst, warm_gus_lst, warm_lst, cold_lst = ({} for i in range(4))\n",
    "    for i,lst,genes in zip(['Hot (ACMG / MASTER)', 'Warm (MASTER-ACMG)', 'Warm (MASTER-ACMG)', \n",
    "                            'Cold (387-ACMG-MASTER)'], \n",
    "                           [[''], gus, list(set(df['HUGO_Symbol'][df['Splice Project Gene Priority']==\n",
    "                                                                  'Warm (MASTER-ACMG)'])^set(gus)), ['']], \n",
    "                           [hot_lst, warm_gus_lst, warm_lst, cold_lst]):\n",
    "        for coverage, splice_change, cat in zip(['no', 'no', 'no', 'yes', 'yes', 'yes'], \n",
    "                                            ['yes', 'maybe', 'no', 'yes', 'maybe', 'no'], \n",
    "                                            ['sufficient, yes', 'sufficient, maybe', 'sufficient, no', \n",
    "                                             'not sufficient, yes', 'not sufficient, maybe', 'not sufficient, no']):\n",
    "            genes[cat] = len(df[(df['Splice Project Gene Priority']==i)&\n",
    "                                (~df['HUGO_Symbol'].isin(lst))&\n",
    "                                (df['too_low_coverage']==coverage)&\n",
    "                                (df['splice_change']==splice_change)].groupby(cols_cpra))\n",
    "    newdf = pd.DataFrame([hot_lst, warm_gus_lst, warm_lst, cold_lst], \n",
    "                        index=['hot genes', 'warm genes - GUS', 'GUS', 'cold genes'])\n",
    "    newdf = newdf.reset_index(drop=False)\n",
    "    newdf = newdf.rename(columns={'index':'category'})\n",
    "    return newdf\n",
    "\n",
    "# variant categorization for specific scores\n",
    "def categ_vars(df):\n",
    "    splAI_lst, ada_rf_lst, MES_disr_lst, MES_new_lst, intron_pos_lst = ({} for i in range(5))\n",
    "    for coverage, splice_change, cat in zip(['no', 'no', 'no', 'yes', 'yes', 'yes'], \n",
    "                                            ['yes', 'maybe', 'no', 'yes', 'maybe', 'no'], \n",
    "                                            ['sufficient, yes', 'sufficient, maybe', 'sufficient, no', \n",
    "                                             'not sufficient, yes', 'not sufficient, maybe', 'not sufficient, no']):\n",
    "        splAI_lst[cat] = splai_o09(df, coverage, splice_change)\n",
    "        ada_rf_lst[cat] = ada_rf_o06(df, coverage, splice_change)\n",
    "        MES_disr_lst[cat] = MES_disr(df, coverage, splice_change)\n",
    "        MES_new_lst[cat] = MES_new(df, coverage, splice_change)\n",
    "        intron_pos_lst[cat] = intron_pos(df, coverage, splice_change)\n",
    "        \n",
    "    newdf = pd.DataFrame([splAI_lst, ada_rf_lst, MES_disr_lst, MES_new_lst, intron_pos_lst], \n",
    "                        index=['spliceAI > 0.9', 'ada, rf score > 0.6', 'MES high disruption', 'MES high new ss', \n",
    "                               'pos. rel. to ss < 3, intron'])\n",
    "    newdf = newdf.reset_index(drop=False)\n",
    "    newdf = newdf.rename(columns={'index':'category'})\n",
    "    return newdf\n",
    "\n",
    "# merge variant categorization for gene priority and specific scores, create percentages/sums\n",
    "def perc_cat(df):\n",
    "    dfs = gene_prio(df).append(categ_vars(df), ignore_index=True)\n",
    "    dfs['sum'] = (dfs['sufficient, yes']+dfs['sufficient, maybe']+dfs['sufficient, no']+\n",
    "                  dfs['not sufficient, yes']+dfs['not sufficient, maybe']+dfs['not sufficient, no'])\n",
    "    sum_row = {i:(dfs.iloc[0][i]+dfs.iloc[2][i]+dfs.iloc[3][i]) for i in dfs.columns[1:8]}\n",
    "    dfs = dfs.append({'category':'sums'}|sum_row, ignore_index=True)\n",
    "    for i in ['sufficient, yes', 'sufficient, maybe', 'sufficient, no']:\n",
    "        dfs['% '+i] = dfs[i]/(dfs['sufficient, yes']+dfs['sufficient, maybe']+dfs['sufficient, no'])\n",
    "    dfs['sufficient to not sufficient'] = ((dfs['sufficient, yes']+dfs['sufficient, maybe']+dfs['sufficient, no'])/\n",
    "                                           (dfs['sum']))\n",
    "    return dfs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "1075c58b",
   "metadata": {},
   "outputs": [],
   "source": [
    "def right_type(df):\n",
    "    dfn = df.copy()\n",
    "    dfn[['#CHROM', 'REF', 'ALT']] = dfn[['#CHROM', 'REF', 'ALT']].astype(str)\n",
    "    dfn['POS'] = dfn['POS'].astype(int)\n",
    "    return dfn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "507178ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "# dataframe with all variants\n",
    "all_vars = pd.read_pickle('00_dataframes/all_annotated_vars')\n",
    "\n",
    "all_vars['#CHROM'] = pd.Categorical(all_vars['#CHROM'],\n",
    "                                    categories=['1','2','3','4','5','6','7','8','9','10','11','12', '13', '14', \n",
    "                                                '15', '16', '17', '18', '19', '20', '21', '22', 'X', 'Y'],\n",
    "                                    ordered=True)\n",
    "all_vars = all_vars.sort_values(by=['#CHROM', 'POS', 'REF', 'ALT'], ignore_index=True)\n",
    "all_vars = right_type(all_vars)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f12b874e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e7e5ed9d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "7ad82267",
   "metadata": {},
   "outputs": [],
   "source": [
    "onk_merge = pd.read_excel('01_xlsx_files/2021-10-27_onkostar_merged.xlsx')\n",
    "onk_merge = right_type(onk_merge)\n",
    "pd.merge(onk_merge, all_vars[cols_cpra+['PID_trans', 'Control_VAF', 'Tumor_VAF', 'Control_dpALT', \n",
    "                                        'Control_dp', 'Tumor_dpALT', 'Tumor_dp', 'AF_RNA', 'AF_RNA_noSkipReads']], \n",
    "         on=['#CHROM', 'POS', 'REF', 'ALT', 'PID_trans'], how='left'\n",
    "        ).to_excel('02_output_analysis/2021-10-27_onkostar_vafs.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "0f34f241",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ClinVar_Pathogenicity']"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_col(onk_merge, 'clinvar')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "8c5338fb",
   "metadata": {},
   "outputs": [],
   "source": [
    "hotspots = pd.read_excel('01_xlsx_files/hotspots_v2.xls')\n",
    "hotspots = hotspots[hotspots['Hugo_Symbol'].isin(onk_merge['HUGO_Symbol'].unique())].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "7ab2ccfc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Hugo_Symbol', 'Amino_Acid_Position', 'log10_pvalue', 'Mutation_Count',\n",
       "       'Reference_Amino_Acid', 'Total_Mutations_in_Gene',\n",
       "       'Median_Allele_Freq_Rank', 'Allele_Freq_Rank', 'Variant_Amino_Acid',\n",
       "       'Codon_Change', 'Genomic_Position', 'Detailed_Cancer_Types',\n",
       "       'Organ_Types', 'Tri-nucleotides', 'Mutability', 'mu_protein',\n",
       "       'Total_Samples', 'Analysis_Type', 'qvalue', 'tm', 'qvalue_pancan',\n",
       "       'Is_repeat', 'seq', 'length', 'align100', 'pad12entropy',\n",
       "       'pad24entropy', 'pad36entropy', 'TP', 'reason', 'n_MSK', 'n_Retro',\n",
       "       'judgement', 'inNBT', 'inOncokb', 'ref', 'qvaluect', 'ct', 'Samples'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hotspots.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "764e1283",
   "metadata": {},
   "outputs": [],
   "source": [
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "bd207272",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "re.findall('48881547', str(list(hotspots['Genomic_Position'][hotspots['Hugo_Symbol']=='RB1'])))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "9d7e8e92",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"['13:48947629_19', '13:48936983_15|13:48936984_1', '13:48936983_15|13:48936984_1', '13:48941648_14', '13:48955382_3|13:48954379_3|13:48955381_2|13:48954378_2', '13:48954299_3|13:48954221_3|13:48954300_2|13:48954222_1', '13:48955538_13', '13:49039505_6|13:49047495_2|13:49047494_2|13:49039506_1', '13:49030486_3|13:49033823_2|13:49033822_2|13:49030487_1', '13:48953728_5|13:48953729_2', '13:48916852_3|13:48919214_2|13:48916851_2', '13:49027168_10|13:49027167_1', '13:48923160_5|13:48934152_4', '13:48881414_3|13:48878187_2|13:48881415_1|13:48878186_1', '13:49033971_3|13:49033970_3', '13:49027248_4|13:49027249_2|13:49030339_1', '13:48916734_6|13:48916733_2', '13:48953760_9', '13:49039340_3|13:49039339_3', '13:48954198_9', '13:49039374_8|13:49039375_2|13:49039373_1', '13:49039374_8|13:49039375_2|13:49039373_1', '13:48953730_10', '13:48953787_3|13:48953788_2', '13:49030343_2|13:49030342_2', '13:49030343_2|13:49030342_2', '13:48951053_8|13:48951052_1']\""
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "str(list(hotspots['Genomic_Position'][hotspots['Hugo_Symbol']=='RB1']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "0be3f466",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Amino_Acid_Position</th>\n",
       "      <th>log10_pvalue</th>\n",
       "      <th>Mutation_Count</th>\n",
       "      <th>Reference_Amino_Acid</th>\n",
       "      <th>Total_Mutations_in_Gene</th>\n",
       "      <th>Median_Allele_Freq_Rank</th>\n",
       "      <th>Allele_Freq_Rank</th>\n",
       "      <th>Variant_Amino_Acid</th>\n",
       "      <th>Codon_Change</th>\n",
       "      <th>Genomic_Position</th>\n",
       "      <th>Detailed_Cancer_Types</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2742</th>\n",
       "      <td>RECQL4</td>\n",
       "      <td>X766_splice</td>\n",
       "      <td>-3.916713</td>\n",
       "      <td>6</td>\n",
       "      <td>splice:6</td>\n",
       "      <td>370</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>splice:6</td>\n",
       "      <td>Cgg/Ggg:6</td>\n",
       "      <td>8:145738768_6</td>\n",
       "      <td>coadread:683:4|dlbcl:246:2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Hugo_Symbol Amino_Acid_Position  log10_pvalue  Mutation_Count  \\\n",
       "2742      RECQL4         X766_splice     -3.916713               6   \n",
       "\n",
       "     Reference_Amino_Acid  Total_Mutations_in_Gene  Median_Allele_Freq_Rank  \\\n",
       "2742             splice:6                      370                      NaN   \n",
       "\n",
       "     Allele_Freq_Rank Variant_Amino_Acid Codon_Change Genomic_Position  \\\n",
       "2742              NaN           splice:6    Cgg/Ggg:6    8:145738768_6   \n",
       "\n",
       "           Detailed_Cancer_Types  \n",
       "2742  coadread:683:4|dlbcl:246:2  "
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hotspots[['Hugo_Symbol', 'Amino_Acid_Position', 'log10_pvalue', 'Mutation_Count',\n",
    "       'Reference_Amino_Acid', 'Total_Mutations_in_Gene',\n",
    "       'Median_Allele_Freq_Rank', 'Allele_Freq_Rank', 'Variant_Amino_Acid',\n",
    "       'Codon_Change', 'Genomic_Position', 'Detailed_Cancer_Types']][\n",
    "    (hotspots['Hugo_Symbol']=='RECQL4')&(hotspots['Genomic_Position'].str.startswith('8:145738768_6'))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "426e259f",
   "metadata": {},
   "outputs": [],
   "source": [
    "tp = list(hotspots['Samples'][(hotspots['Hugo_Symbol']=='TP53')&(hotspots['Genomic_Position'].str.startswith('17:7578370'))])[0].split(':')\n",
    "tp_num = [t.split('|') for t in tp]\n",
    "tp_num = sum([int(n) for sl in tp_num for n in sl if n.isdigit()])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "b399d736",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>HUGO_Symbol</th>\n",
       "      <th>ClinVar_Pathogenicity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>8</td>\n",
       "      <td>145738768</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>RECQL4</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>9</td>\n",
       "      <td>135779206</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>TSC1</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>11</td>\n",
       "      <td>108186639</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ATM</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>11</td>\n",
       "      <td>108202604</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>ATM</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>11</td>\n",
       "      <td>108224490</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>ATM</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>12</td>\n",
       "      <td>133254296</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>POLE</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>13</td>\n",
       "      <td>48881547</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>RB1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>13</td>\n",
       "      <td>48953712</td>\n",
       "      <td>CTTTTT</td>\n",
       "      <td>C</td>\n",
       "      <td>RB1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>14</td>\n",
       "      <td>105243112</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>16</td>\n",
       "      <td>68849662</td>\n",
       "      <td>CG</td>\n",
       "      <td>C</td>\n",
       "      <td>CDH1</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>17</td>\n",
       "      <td>7578370</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TP53</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>17</td>\n",
       "      <td>7579699</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TP53</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>17</td>\n",
       "      <td>29482996</td>\n",
       "      <td>TTCAGCTTCCAATA</td>\n",
       "      <td>T</td>\n",
       "      <td>NF1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>17</td>\n",
       "      <td>56809887</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>RAD51C</td>\n",
       "      <td>Likely Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>19</td>\n",
       "      <td>45871887</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>ERCC2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>22</td>\n",
       "      <td>24175755</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SMARCB1</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>22</td>\n",
       "      <td>29121230</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   #CHROM        POS             REF ALT HUGO_Symbol   ClinVar_Pathogenicity\n",
       "19      8  145738768               G   T      RECQL4                     NaN\n",
       "21      9  135779206               T   C        TSC1  Uncertain Significance\n",
       "24     11  108186639               G   A         ATM       Likely Pathogenic\n",
       "25     11  108202604               A   C         ATM              Pathogenic\n",
       "27     11  108224490               T   G         ATM  Uncertain Significance\n",
       "28     12  133254296               C   A        POLE  Uncertain Significance\n",
       "29     13   48881547               G   A         RB1                     NaN\n",
       "30     13   48953712          CTTTTT   C         RB1                     NaN\n",
       "31     14  105243112               G   T        AKT1  Uncertain Significance\n",
       "39     16   68849662              CG   C        CDH1              Pathogenic\n",
       "40     17    7578370               C   T        TP53              Pathogenic\n",
       "41     17    7579699               C   T        TP53                     NaN\n",
       "42     17   29482996  TTCAGCTTCCAATA   T         NF1                     NaN\n",
       "45     17   56809887               A   G      RAD51C           Likely Benign\n",
       "50     19   45871887               C   T       ERCC2                     NaN\n",
       "54     22   24175755               G   A     SMARCB1  Uncertain Significance\n",
       "56     22   29121230               C   T       CHEK2              Pathogenic"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "onk_merge[cols_cpra+['HUGO_Symbol', 'ClinVar_Pathogenicity']][\n",
    "    onk_merge['HUGO_Symbol'].isin(hotspots['Hugo_Symbol'])\n",
    "].drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b4bebb67",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "08cb2328",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "0ff02462",
   "metadata": {},
   "source": [
    "### variant infos merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 417,
   "id": "198ea71c",
   "metadata": {},
   "outputs": [],
   "source": [
    "vars1 = pd.read_excel('01_xlsx_files/vars_splice.xlsx')\n",
    "\n",
    "vars2 = pd.read_excel('01_xlsx_files/new_vars_analysis_genes.xlsx').rename(columns={'shortcut':'shortcut_v'})\n",
    "vars2[['sc0', 'sc1', 'sc2', 'sc3', 'sc4']] = vars2['shortcut_v'].str.split('_', expand=True)\n",
    "vars2['shortcut'] = vars2['sc0']+'_'+vars2['sc1']+'_'+vars2['sc2']+'_'+vars2['sc3']\n",
    "vars2 = vars2.drop(columns=['sc0', 'sc1', 'sc2', 'sc3', 'sc4'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 436,
   "id": "c0572a77",
   "metadata": {},
   "outputs": [],
   "source": [
    "var_merge = pd.merge(vars1.drop(columns=['PIDs', 'kind', 'pos_last', 'pos_next']), vars2, \n",
    "                     on=['#CHROM','shortcut'], how='outer')\n",
    "var_merge[['#CHROM', 'ALT', 'REF']] = var_merge[['#CHROM', 'ALT', 'REF']].astype(str)\n",
    "var_merge['POS'] = var_merge['POS'].astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 470,
   "id": "7f324977",
   "metadata": {},
   "outputs": [],
   "source": [
    "kge_hboc_wodup = kge_hboc[cols_cpra].drop_duplicates().reset_index(drop=True)\n",
    "kge_hboc_wodup['ID_Person'] = [list(df['ID Person']) for i,df in kge_hboc.groupby(cols_cpra)]\n",
    "kge_hboc_wodup['KGE_HBOC'] = [list(df['from']) for i,df in kge_hboc.groupby(cols_cpra)]\n",
    "kge_hboc_vars = pd.merge(var_merge, kge_hboc_wodup, on=cols_cpra, how='outer')\n",
    "\n",
    "kge_hboc_vars.to_excel('02_output_analysis/vars_facts_together.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "1244fd6d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>Control_GT</th>\n",
       "      <th>...</th>\n",
       "      <th>taskforce_review_hboc</th>\n",
       "      <th>date_taskforce_hboc</th>\n",
       "      <th>note_hboc</th>\n",
       "      <th>add_note_hboc</th>\n",
       "      <th>splice_pred_alamut_hboc</th>\n",
       "      <th>prediction_hboc</th>\n",
       "      <th>literature_hboc</th>\n",
       "      <th>evidence_level_lit_hboc</th>\n",
       "      <th>comment_hboc</th>\n",
       "      <th>recomm_action_hboc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>15095</th>\n",
       "      <td>17</td>\n",
       "      <td>59763193</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>290</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.29;FR=0.5;HP=1;HapScore=2;MGOF=2;MMLQ=32...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>0/1:-20.69,0,-26.99:1:99:17:8</td>\n",
       "      <td>...</td>\n",
       "      <td>nein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15096</th>\n",
       "      <td>17</td>\n",
       "      <td>59793404</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1926</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.14;FR=0.5;HP=3;HapScore=1;MGOF=19;MMLQ=4...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-124.7,0,-140.1:19:99:82:39</td>\n",
       "      <td>...</td>\n",
       "      <td>nein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15097</th>\n",
       "      <td>17</td>\n",
       "      <td>59820479</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>2957</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.12;FR=0.75;HP=1;HapScore=1;MGOF=3;MMLQ=3...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-195.71,0,-137.91:2:99:90:45</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15098</th>\n",
       "      <td>17</td>\n",
       "      <td>59821950</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>2143</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.22;FR=0.5;HP=4;HapScore=1;MGOF=8;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-113.96,0,-115.76:8:99:68:34</td>\n",
       "      <td>...</td>\n",
       "      <td>nein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15099</th>\n",
       "      <td>17</td>\n",
       "      <td>59853755</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2471</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.13;FR=0.5;HP=2;HapScore=2;MGOF=8;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>0/1:-112.82,0,-145.32:4:99:77:35</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7970238#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15100</th>\n",
       "      <td>17</td>\n",
       "      <td>59853755</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2166</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.19;FR=0.5;HP=2;HapScore=1;MGOF=6;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>0/1:-143.22,0,-137.62:6:99:90:46</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7970238#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15101</th>\n",
       "      <td>17</td>\n",
       "      <td>59853755</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2862</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.13;FR=0.5;HP=2;HapScore=1;MGOF=7;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>0/1:-116.55,0,-117.05:7:99:74:37</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7970238#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15102</th>\n",
       "      <td>17</td>\n",
       "      <td>59853755</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2639</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.18;FR=0.5;HP=2;HapScore=2;MGOF=4;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>0/1:-118.51,0,-117.61:4:99:72:37</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7970238#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15103</th>\n",
       "      <td>17</td>\n",
       "      <td>59853755</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2965</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.23;FR=0.5;HP=2;HapScore=1;MGOF=4;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>0/1:-143.24,0,-124.44:2:99:87:44</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7970238#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15104</th>\n",
       "      <td>17</td>\n",
       "      <td>59853755</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2082</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.24;FR=0.5;HP=2;HapScore=2;MGOF=2;MMLQ=41...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>0/1:-109.83,0,-112.13:2:99:65:32</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7970238#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15105</th>\n",
       "      <td>17</td>\n",
       "      <td>59853755</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2851</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.13;FR=0.5;HP=2;HapScore=2;MGOF=5;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>0/1:-131.61,0,-149.81:5:99:86:41</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7970238#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15106</th>\n",
       "      <td>17</td>\n",
       "      <td>59853755</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2768</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.3;FR=0.5;HP=2;HapScore=2;MGOF=1;MMLQ=41;...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>0/1:-106.43,0,-136.43:1:99:78:34</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7970238#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15107</th>\n",
       "      <td>17</td>\n",
       "      <td>59853755</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2965</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.09;FR=0.5;HP=2;HapScore=1;MGOF=6;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>0/1:-99.68,0,-127.78:6:99:70:31</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7970238#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15108</th>\n",
       "      <td>17</td>\n",
       "      <td>59853755</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1405</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.38;FR=0.4999;HP=2;HapScore=1;MGOF=1;MMLQ...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>0/1:-141.83,0,-91.73:0:99:74:44</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7970238#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15109</th>\n",
       "      <td>17</td>\n",
       "      <td>59853755</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2251</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.1;FR=0.5;HP=2;HapScore=2;MGOF=8;MMLQ=37;...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>0/1:-108.16,0,-113.76:5:99:70:35</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7970238#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15110</th>\n",
       "      <td>17</td>\n",
       "      <td>59853755</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2965</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.14;FR=0.5;HP=2;HapScore=1;MGOF=2;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>0/1:-251.17,0,-163.77:2:99:135:79</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7970238#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15111</th>\n",
       "      <td>17</td>\n",
       "      <td>59853755</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2965</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.27;FR=0.5;HP=2;HapScore=1;MGOF=3;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>0/1:-132.32,0,-129.32:3:99:87:42</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7970238#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15112</th>\n",
       "      <td>17</td>\n",
       "      <td>59853755</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2965</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.22;FR=0.5;HP=2;HapScore=1;MGOF=5;MMLQ=41...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>0/1:-184.85,0,-158.75:5:99:104:54</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7970238#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15113</th>\n",
       "      <td>17</td>\n",
       "      <td>59853755</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2719</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.13;FR=0.5;HP=2;HapScore=2;MGOF=7;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>0/1:-139.24,0,-142.94:7:99:94:49</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7970238#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15114</th>\n",
       "      <td>17</td>\n",
       "      <td>59857749</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>1241</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.14;FR=0.5;HP=1;HapScore=2;MGOF=8;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>0/1:-45.67,0,-69.97:8:99:43:19</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15115</th>\n",
       "      <td>17</td>\n",
       "      <td>59858343</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2955</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.11;FR=0.5;HP=2;HapScore=1;MGOF=7;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-118.53,0,-117.33:7:99:74:35</td>\n",
       "      <td>...</td>\n",
       "      <td>nein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15116</th>\n",
       "      <td>17</td>\n",
       "      <td>59858377</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>2695</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.17;FR=0.5;HP=2;HapScore=1;MGOF=5;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>0/1:-121.2,0,-120.8:5:99:81:40</td>\n",
       "      <td>...</td>\n",
       "      <td>nein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15117</th>\n",
       "      <td>17</td>\n",
       "      <td>59878603</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2494</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.24;FR=0.5;HP=2;HapScore=2;MGOF=7;MMLQ=41...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>0/1:-136.72,0,-119.62:6:99:81:42</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15118</th>\n",
       "      <td>17</td>\n",
       "      <td>59885813</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>2551</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.13;FR=0.5;HP=3;HapScore=2;MGOF=6;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>0/1:-127.66,0,-107.16:6:99:82:44</td>\n",
       "      <td>...</td>\n",
       "      <td>nein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15119</th>\n",
       "      <td>17</td>\n",
       "      <td>59885813</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>1423</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.35;FR=0.5;HP=3;HapScore=2;MGOF=4;MMLQ=41...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>0/1:-94.57,0,-94.57:4:99:64:31</td>\n",
       "      <td>...</td>\n",
       "      <td>nein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15120</th>\n",
       "      <td>17</td>\n",
       "      <td>59886108</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>2965</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.37;FR=0.5;HP=3;HapScore=2;MGOF=1;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-109.13,0,-119.73:1:99:77:35</td>\n",
       "      <td>...</td>\n",
       "      <td>nein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15121</th>\n",
       "      <td>17</td>\n",
       "      <td>59886116</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>2657</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.22;FR=0.5;HP=4;HapScore=2;MGOF=9;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-114.72,0,-142.42:9:99:81:35</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15122</th>\n",
       "      <td>17</td>\n",
       "      <td>59886128</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>1737</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.44;FR=0.5;HP=2;HapScore=1;MGOF=0;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>0/1:-59.77,0,-32.37:0:99:34:23</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15123</th>\n",
       "      <td>17</td>\n",
       "      <td>59924463</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>2782</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.18;FR=0.5;HP=2;HapScore=2;MGOF=9;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-142.23,0,-80.53:6:99:70:45</td>\n",
       "      <td>...</td>\n",
       "      <td>nein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15124</th>\n",
       "      <td>17</td>\n",
       "      <td>59924572</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2560</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.15;FR=0.5;HP=4;HapScore=2;MGOF=7;MMLQ=41...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-125.77,0,-142.27:7:99:83:39</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7929646#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) FLOSSIES: 84x i...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15125</th>\n",
       "      <td>17</td>\n",
       "      <td>59924572</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1040</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.38;FR=0.5;HP=4;HapScore=2;MGOF=1;MMLQ=41...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-42.8,0,-58.1:1:99:40:15</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7929646#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) FLOSSIES: 84x i...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15126</th>\n",
       "      <td>17</td>\n",
       "      <td>59924572</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2965</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.16;FR=0.5;HP=4;HapScore=1;MGOF=4;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-182.97,0,-150.57:4:99:104:55</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7929646#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) FLOSSIES: 84x i...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15127</th>\n",
       "      <td>17</td>\n",
       "      <td>59924572</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1917</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.32;FR=0.5;HP=4;HapScore=1;MGOF=3;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-85.56,0,-62.26:1:99:47:28</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7929646#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) FLOSSIES: 84x i...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15128</th>\n",
       "      <td>17</td>\n",
       "      <td>59924572</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1828</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.2;FR=0.5;HP=4;HapScore=2;MGOF=6;MMLQ=37;...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-67.97,0,-121.57:6:99:59:24</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7929646#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) FLOSSIES: 84x i...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15129</th>\n",
       "      <td>17</td>\n",
       "      <td>59924572</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2965</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.15;FR=0.5;HP=4;HapScore=2;MGOF=5;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-171.36,0,-134.06:5:99:95:53</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7929646#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) FLOSSIES: 84x i...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15130</th>\n",
       "      <td>17</td>\n",
       "      <td>59924572</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>729</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.37;FR=0.5;HP=4;HapScore=1;MGOF=2;MMLQ=41...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-46.56,0,-83.06:1:99:47:19</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7929646#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) FLOSSIES: 84x i...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15131</th>\n",
       "      <td>17</td>\n",
       "      <td>59924572</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1679</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.26;FR=0.5;HP=4;HapScore=2;MGOF=1;MMLQ=41...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-56.96,0,-74.16:1:99:50:21</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7929646#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) FLOSSIES: 84x i...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15132</th>\n",
       "      <td>17</td>\n",
       "      <td>59924572</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2534</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.18;FR=0.5;HP=4;HapScore=2;MGOF=5;MMLQ=41...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-108.16,0,-111.76:5:99:70:32</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7929646#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) FLOSSIES: 84x i...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15133</th>\n",
       "      <td>17</td>\n",
       "      <td>59924572</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1137</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.31;FR=0.5;HP=4;HapScore=2;MGOF=1;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-58.1,0,-33.2:1:99:35:22</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7929646#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) FLOSSIES: 84x i...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15134</th>\n",
       "      <td>17</td>\n",
       "      <td>59924572</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2872</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.29;FR=0.5;HP=4;HapScore=1;MGOF=1;MMLQ=41...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-101.65,0,-82.55:0:99:71:37</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7929646#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) FLOSSIES: 84x i...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15135</th>\n",
       "      <td>17</td>\n",
       "      <td>59924572</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1506</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.3;FR=0.5;HP=4;HapScore=2;MGOF=8;MMLQ=41;...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-61.69,0,-78.19:8:99:47:23</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7929646#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) FLOSSIES: 84x i...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15136</th>\n",
       "      <td>17</td>\n",
       "      <td>59924572</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2965</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.14;FR=0.5;HP=4;HapScore=1;MGOF=8;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-114.5,0,-127.1:5:99:81:40</td>\n",
       "      <td>...</td>\n",
       "      <td>ja</td>\n",
       "      <td>19.03.2019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#Dublette zu SEQ-ID: 7929646#</td>\n",
       "      <td>unauffällig</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2 x hom in gnomAD non cancer) FLOSSIES: 84x i...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15137</th>\n",
       "      <td>17</td>\n",
       "      <td>59926500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2257</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.17;FR=0.5;HP=3;HapScore=1;MGOF=8;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-110.95,0,-144.35:7:99:82:37</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15138</th>\n",
       "      <td>17</td>\n",
       "      <td>59926633</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>1635</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.19;FR=0.5;HP=8;HapScore=1;MGOF=15;MMLQ=3...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-86.73,0,-33.23:6:99:85:43</td>\n",
       "      <td>...</td>\n",
       "      <td>nein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15139</th>\n",
       "      <td>17</td>\n",
       "      <td>59926633</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>1073</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.22;FR=0.5;HP=8;HapScore=2;MGOF=8;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-56.82,0,-68.82:4:99:89:31</td>\n",
       "      <td>...</td>\n",
       "      <td>nein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15140</th>\n",
       "      <td>17</td>\n",
       "      <td>59926633</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>1459</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.22;FR=0.5;HP=8;HapScore=2;MGOF=7;MMLQ=32...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-81.64,0,-25.64:7:99:66:40</td>\n",
       "      <td>...</td>\n",
       "      <td>nein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15141</th>\n",
       "      <td>17</td>\n",
       "      <td>59934594</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>2965</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.24;FR=0.5;HP=5;HapScore=2;MGOF=1;MMLQ=39...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-134.13,0,-102.03:1:99:84:49</td>\n",
       "      <td>...</td>\n",
       "      <td>nein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15142</th>\n",
       "      <td>17</td>\n",
       "      <td>59938819</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1236</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.28;FR=0.5;HP=2;HapScore=2;MGOF=2;MMLQ=37...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>1/0:-63.08,0,-17.98:2:99:29:21</td>\n",
       "      <td>...</td>\n",
       "      <td>nein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15143</th>\n",
       "      <td>17</td>\n",
       "      <td>59938922</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>986</td>\n",
       "      <td>PASS</td>\n",
       "      <td>BRF=0.31;FR=0.5;HP=2;HapScore=2;MGOF=2;MMLQ=32...</td>\n",
       "      <td>GT:GL:GOF:GQ:NR:NV</td>\n",
       "      <td>0/1:-45,0,-42.5:2:99:31:15</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>49 rows × 134 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #CHROM       POS  ID REF ALT  QUAL FILTER  \\\n",
       "15095     17  59763193 NaN   A   G   290   PASS   \n",
       "15096     17  59793404 NaN   G   C  1926   PASS   \n",
       "15097     17  59820479 NaN   T  TA  2957   PASS   \n",
       "15098     17  59821950 NaN   T   C  2143   PASS   \n",
       "15099     17  59853755 NaN   C   T  2471   PASS   \n",
       "15100     17  59853755 NaN   C   T  2166   PASS   \n",
       "15101     17  59853755 NaN   C   T  2862   PASS   \n",
       "15102     17  59853755 NaN   C   T  2639   PASS   \n",
       "15103     17  59853755 NaN   C   T  2965   PASS   \n",
       "15104     17  59853755 NaN   C   T  2082   PASS   \n",
       "15105     17  59853755 NaN   C   T  2851   PASS   \n",
       "15106     17  59853755 NaN   C   T  2768   PASS   \n",
       "15107     17  59853755 NaN   C   T  2965   PASS   \n",
       "15108     17  59853755 NaN   C   T  1405   PASS   \n",
       "15109     17  59853755 NaN   C   T  2251   PASS   \n",
       "15110     17  59853755 NaN   C   T  2965   PASS   \n",
       "15111     17  59853755 NaN   C   T  2965   PASS   \n",
       "15112     17  59853755 NaN   C   T  2965   PASS   \n",
       "15113     17  59853755 NaN   C   T  2719   PASS   \n",
       "15114     17  59857749 NaN   A   T  1241   PASS   \n",
       "15115     17  59858343 NaN   G   A  2955   PASS   \n",
       "15116     17  59858377 NaN   A   G  2695   PASS   \n",
       "15117     17  59878603 NaN   C   T  2494   PASS   \n",
       "15118     17  59885813 NaN   A   T  2551   PASS   \n",
       "15119     17  59885813 NaN   A   T  1423   PASS   \n",
       "15120     17  59886108 NaN   T   C  2965   PASS   \n",
       "15121     17  59886116 NaN   G   C  2657   PASS   \n",
       "15122     17  59886128 NaN   A   T  1737   PASS   \n",
       "15123     17  59924463 NaN   T   C  2782   PASS   \n",
       "15124     17  59924572 NaN   G   A  2560   PASS   \n",
       "15125     17  59924572 NaN   G   A  1040   PASS   \n",
       "15126     17  59924572 NaN   G   A  2965   PASS   \n",
       "15127     17  59924572 NaN   G   A  1917   PASS   \n",
       "15128     17  59924572 NaN   G   A  1828   PASS   \n",
       "15129     17  59924572 NaN   G   A  2965   PASS   \n",
       "15130     17  59924572 NaN   G   A   729   PASS   \n",
       "15131     17  59924572 NaN   G   A  1679   PASS   \n",
       "15132     17  59924572 NaN   G   A  2534   PASS   \n",
       "15133     17  59924572 NaN   G   A  1137   PASS   \n",
       "15134     17  59924572 NaN   G   A  2872   PASS   \n",
       "15135     17  59924572 NaN   G   A  1506   PASS   \n",
       "15136     17  59924572 NaN   G   A  2965   PASS   \n",
       "15137     17  59926500 NaN   G   A  2257   PASS   \n",
       "15138     17  59926633 NaN   T  TA  1635   PASS   \n",
       "15139     17  59926633 NaN   T  TA  1073   PASS   \n",
       "15140     17  59926633 NaN   T  TA  1459   PASS   \n",
       "15141     17  59934594 NaN   T   C  2965   PASS   \n",
       "15142     17  59938819 NaN   T   C  1236   PASS   \n",
       "15143     17  59938922 NaN   G   T   986   PASS   \n",
       "\n",
       "                                                    INFO              FORMAT  \\\n",
       "15095  BRF=0.29;FR=0.5;HP=1;HapScore=2;MGOF=2;MMLQ=32...  GT:GL:GOF:GQ:NR:NV   \n",
       "15096  BRF=0.14;FR=0.5;HP=3;HapScore=1;MGOF=19;MMLQ=4...  GT:GL:GOF:GQ:NR:NV   \n",
       "15097  BRF=0.12;FR=0.75;HP=1;HapScore=1;MGOF=3;MMLQ=3...  GT:GL:GOF:GQ:NR:NV   \n",
       "15098  BRF=0.22;FR=0.5;HP=4;HapScore=1;MGOF=8;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "15099  BRF=0.13;FR=0.5;HP=2;HapScore=2;MGOF=8;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "15100  BRF=0.19;FR=0.5;HP=2;HapScore=1;MGOF=6;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "15101  BRF=0.13;FR=0.5;HP=2;HapScore=1;MGOF=7;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "15102  BRF=0.18;FR=0.5;HP=2;HapScore=2;MGOF=4;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "15103  BRF=0.23;FR=0.5;HP=2;HapScore=1;MGOF=4;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "15104  BRF=0.24;FR=0.5;HP=2;HapScore=2;MGOF=2;MMLQ=41...  GT:GL:GOF:GQ:NR:NV   \n",
       "15105  BRF=0.13;FR=0.5;HP=2;HapScore=2;MGOF=5;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "15106  BRF=0.3;FR=0.5;HP=2;HapScore=2;MGOF=1;MMLQ=41;...  GT:GL:GOF:GQ:NR:NV   \n",
       "15107  BRF=0.09;FR=0.5;HP=2;HapScore=1;MGOF=6;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "15108  BRF=0.38;FR=0.4999;HP=2;HapScore=1;MGOF=1;MMLQ...  GT:GL:GOF:GQ:NR:NV   \n",
       "15109  BRF=0.1;FR=0.5;HP=2;HapScore=2;MGOF=8;MMLQ=37;...  GT:GL:GOF:GQ:NR:NV   \n",
       "15110  BRF=0.14;FR=0.5;HP=2;HapScore=1;MGOF=2;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "15111  BRF=0.27;FR=0.5;HP=2;HapScore=1;MGOF=3;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "15112  BRF=0.22;FR=0.5;HP=2;HapScore=1;MGOF=5;MMLQ=41...  GT:GL:GOF:GQ:NR:NV   \n",
       "15113  BRF=0.13;FR=0.5;HP=2;HapScore=2;MGOF=7;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "15114  BRF=0.14;FR=0.5;HP=1;HapScore=2;MGOF=8;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "15115  BRF=0.11;FR=0.5;HP=2;HapScore=1;MGOF=7;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "15116  BRF=0.17;FR=0.5;HP=2;HapScore=1;MGOF=5;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "15117  BRF=0.24;FR=0.5;HP=2;HapScore=2;MGOF=7;MMLQ=41...  GT:GL:GOF:GQ:NR:NV   \n",
       "15118  BRF=0.13;FR=0.5;HP=3;HapScore=2;MGOF=6;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "15119  BRF=0.35;FR=0.5;HP=3;HapScore=2;MGOF=4;MMLQ=41...  GT:GL:GOF:GQ:NR:NV   \n",
       "15120  BRF=0.37;FR=0.5;HP=3;HapScore=2;MGOF=1;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "15121  BRF=0.22;FR=0.5;HP=4;HapScore=2;MGOF=9;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "15122  BRF=0.44;FR=0.5;HP=2;HapScore=1;MGOF=0;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "15123  BRF=0.18;FR=0.5;HP=2;HapScore=2;MGOF=9;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "15124  BRF=0.15;FR=0.5;HP=4;HapScore=2;MGOF=7;MMLQ=41...  GT:GL:GOF:GQ:NR:NV   \n",
       "15125  BRF=0.38;FR=0.5;HP=4;HapScore=2;MGOF=1;MMLQ=41...  GT:GL:GOF:GQ:NR:NV   \n",
       "15126  BRF=0.16;FR=0.5;HP=4;HapScore=1;MGOF=4;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "15127  BRF=0.32;FR=0.5;HP=4;HapScore=1;MGOF=3;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "15128  BRF=0.2;FR=0.5;HP=4;HapScore=2;MGOF=6;MMLQ=37;...  GT:GL:GOF:GQ:NR:NV   \n",
       "15129  BRF=0.15;FR=0.5;HP=4;HapScore=2;MGOF=5;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "15130  BRF=0.37;FR=0.5;HP=4;HapScore=1;MGOF=2;MMLQ=41...  GT:GL:GOF:GQ:NR:NV   \n",
       "15131  BRF=0.26;FR=0.5;HP=4;HapScore=2;MGOF=1;MMLQ=41...  GT:GL:GOF:GQ:NR:NV   \n",
       "15132  BRF=0.18;FR=0.5;HP=4;HapScore=2;MGOF=5;MMLQ=41...  GT:GL:GOF:GQ:NR:NV   \n",
       "15133  BRF=0.31;FR=0.5;HP=4;HapScore=2;MGOF=1;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "15134  BRF=0.29;FR=0.5;HP=4;HapScore=1;MGOF=1;MMLQ=41...  GT:GL:GOF:GQ:NR:NV   \n",
       "15135  BRF=0.3;FR=0.5;HP=4;HapScore=2;MGOF=8;MMLQ=41;...  GT:GL:GOF:GQ:NR:NV   \n",
       "15136  BRF=0.14;FR=0.5;HP=4;HapScore=1;MGOF=8;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "15137  BRF=0.17;FR=0.5;HP=3;HapScore=1;MGOF=8;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "15138  BRF=0.19;FR=0.5;HP=8;HapScore=1;MGOF=15;MMLQ=3...  GT:GL:GOF:GQ:NR:NV   \n",
       "15139  BRF=0.22;FR=0.5;HP=8;HapScore=2;MGOF=8;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "15140  BRF=0.22;FR=0.5;HP=8;HapScore=2;MGOF=7;MMLQ=32...  GT:GL:GOF:GQ:NR:NV   \n",
       "15141  BRF=0.24;FR=0.5;HP=5;HapScore=2;MGOF=1;MMLQ=39...  GT:GL:GOF:GQ:NR:NV   \n",
       "15142  BRF=0.28;FR=0.5;HP=2;HapScore=2;MGOF=2;MMLQ=37...  GT:GL:GOF:GQ:NR:NV   \n",
       "15143  BRF=0.31;FR=0.5;HP=2;HapScore=2;MGOF=2;MMLQ=32...  GT:GL:GOF:GQ:NR:NV   \n",
       "\n",
       "                              Control_GT  ... taskforce_review_hboc  \\\n",
       "15095      0/1:-20.69,0,-26.99:1:99:17:8  ...                  nein   \n",
       "15096    1/0:-124.7,0,-140.1:19:99:82:39  ...                  nein   \n",
       "15097   1/0:-195.71,0,-137.91:2:99:90:45  ...                   NaN   \n",
       "15098   1/0:-113.96,0,-115.76:8:99:68:34  ...                  nein   \n",
       "15099   0/1:-112.82,0,-145.32:4:99:77:35  ...                    ja   \n",
       "15100   0/1:-143.22,0,-137.62:6:99:90:46  ...                    ja   \n",
       "15101   0/1:-116.55,0,-117.05:7:99:74:37  ...                    ja   \n",
       "15102   0/1:-118.51,0,-117.61:4:99:72:37  ...                    ja   \n",
       "15103   0/1:-143.24,0,-124.44:2:99:87:44  ...                    ja   \n",
       "15104   0/1:-109.83,0,-112.13:2:99:65:32  ...                    ja   \n",
       "15105   0/1:-131.61,0,-149.81:5:99:86:41  ...                    ja   \n",
       "15106   0/1:-106.43,0,-136.43:1:99:78:34  ...                    ja   \n",
       "15107    0/1:-99.68,0,-127.78:6:99:70:31  ...                    ja   \n",
       "15108    0/1:-141.83,0,-91.73:0:99:74:44  ...                    ja   \n",
       "15109   0/1:-108.16,0,-113.76:5:99:70:35  ...                    ja   \n",
       "15110  0/1:-251.17,0,-163.77:2:99:135:79  ...                    ja   \n",
       "15111   0/1:-132.32,0,-129.32:3:99:87:42  ...                    ja   \n",
       "15112  0/1:-184.85,0,-158.75:5:99:104:54  ...                    ja   \n",
       "15113   0/1:-139.24,0,-142.94:7:99:94:49  ...                    ja   \n",
       "15114     0/1:-45.67,0,-69.97:8:99:43:19  ...                   NaN   \n",
       "15115   1/0:-118.53,0,-117.33:7:99:74:35  ...                  nein   \n",
       "15116     0/1:-121.2,0,-120.8:5:99:81:40  ...                  nein   \n",
       "15117   0/1:-136.72,0,-119.62:6:99:81:42  ...                   NaN   \n",
       "15118   0/1:-127.66,0,-107.16:6:99:82:44  ...                  nein   \n",
       "15119     0/1:-94.57,0,-94.57:4:99:64:31  ...                  nein   \n",
       "15120   1/0:-109.13,0,-119.73:1:99:77:35  ...                  nein   \n",
       "15121   1/0:-114.72,0,-142.42:9:99:81:35  ...                   NaN   \n",
       "15122     0/1:-59.77,0,-32.37:0:99:34:23  ...                   NaN   \n",
       "15123    1/0:-142.23,0,-80.53:6:99:70:45  ...                  nein   \n",
       "15124   1/0:-125.77,0,-142.27:7:99:83:39  ...                    ja   \n",
       "15125       1/0:-42.8,0,-58.1:1:99:40:15  ...                    ja   \n",
       "15126  1/0:-182.97,0,-150.57:4:99:104:55  ...                    ja   \n",
       "15127     1/0:-85.56,0,-62.26:1:99:47:28  ...                    ja   \n",
       "15128    1/0:-67.97,0,-121.57:6:99:59:24  ...                    ja   \n",
       "15129   1/0:-171.36,0,-134.06:5:99:95:53  ...                    ja   \n",
       "15130     1/0:-46.56,0,-83.06:1:99:47:19  ...                    ja   \n",
       "15131     1/0:-56.96,0,-74.16:1:99:50:21  ...                    ja   \n",
       "15132   1/0:-108.16,0,-111.76:5:99:70:32  ...                    ja   \n",
       "15133       1/0:-58.1,0,-33.2:1:99:35:22  ...                    ja   \n",
       "15134    1/0:-101.65,0,-82.55:0:99:71:37  ...                    ja   \n",
       "15135     1/0:-61.69,0,-78.19:8:99:47:23  ...                    ja   \n",
       "15136     1/0:-114.5,0,-127.1:5:99:81:40  ...                    ja   \n",
       "15137   1/0:-110.95,0,-144.35:7:99:82:37  ...                   NaN   \n",
       "15138     1/0:-86.73,0,-33.23:6:99:85:43  ...                  nein   \n",
       "15139     1/0:-56.82,0,-68.82:4:99:89:31  ...                  nein   \n",
       "15140     1/0:-81.64,0,-25.64:7:99:66:40  ...                  nein   \n",
       "15141   1/0:-134.13,0,-102.03:1:99:84:49  ...                  nein   \n",
       "15142     1/0:-63.08,0,-17.98:2:99:29:21  ...                  nein   \n",
       "15143         0/1:-45,0,-42.5:2:99:31:15  ...                   NaN   \n",
       "\n",
       "      date_taskforce_hboc note_hboc                  add_note_hboc  \\\n",
       "15095                 NaN       NaN                            NaN   \n",
       "15096                 NaN       NaN                            NaN   \n",
       "15097                 NaN       NaN                            NaN   \n",
       "15098                 NaN       NaN                            NaN   \n",
       "15099          19.03.2019       NaN  #Dublette zu SEQ-ID: 7970238#   \n",
       "15100          19.03.2019       NaN  #Dublette zu SEQ-ID: 7970238#   \n",
       "15101          19.03.2019       NaN  #Dublette zu SEQ-ID: 7970238#   \n",
       "15102          19.03.2019       NaN  #Dublette zu SEQ-ID: 7970238#   \n",
       "15103          19.03.2019       NaN  #Dublette zu SEQ-ID: 7970238#   \n",
       "15104          19.03.2019       NaN  #Dublette zu SEQ-ID: 7970238#   \n",
       "15105          19.03.2019       NaN  #Dublette zu SEQ-ID: 7970238#   \n",
       "15106          19.03.2019       NaN  #Dublette zu SEQ-ID: 7970238#   \n",
       "15107          19.03.2019       NaN  #Dublette zu SEQ-ID: 7970238#   \n",
       "15108          19.03.2019       NaN  #Dublette zu SEQ-ID: 7970238#   \n",
       "15109          19.03.2019       NaN  #Dublette zu SEQ-ID: 7970238#   \n",
       "15110          19.03.2019       NaN  #Dublette zu SEQ-ID: 7970238#   \n",
       "15111          19.03.2019       NaN  #Dublette zu SEQ-ID: 7970238#   \n",
       "15112          19.03.2019       NaN  #Dublette zu SEQ-ID: 7970238#   \n",
       "15113          19.03.2019       NaN  #Dublette zu SEQ-ID: 7970238#   \n",
       "15114                 NaN       NaN                            NaN   \n",
       "15115                 NaN       NaN                            NaN   \n",
       "15116                 NaN       NaN                            NaN   \n",
       "15117                 NaN       NaN                            NaN   \n",
       "15118                 NaN       NaN                            NaN   \n",
       "15119                 NaN       NaN                            NaN   \n",
       "15120                 NaN       NaN                            NaN   \n",
       "15121                 NaN       NaN                            NaN   \n",
       "15122                 NaN       NaN                            NaN   \n",
       "15123                 NaN       NaN                            NaN   \n",
       "15124          19.03.2019       NaN  #Dublette zu SEQ-ID: 7929646#   \n",
       "15125          19.03.2019       NaN  #Dublette zu SEQ-ID: 7929646#   \n",
       "15126          19.03.2019       NaN  #Dublette zu SEQ-ID: 7929646#   \n",
       "15127          19.03.2019       NaN  #Dublette zu SEQ-ID: 7929646#   \n",
       "15128          19.03.2019       NaN  #Dublette zu SEQ-ID: 7929646#   \n",
       "15129          19.03.2019       NaN  #Dublette zu SEQ-ID: 7929646#   \n",
       "15130          19.03.2019       NaN  #Dublette zu SEQ-ID: 7929646#   \n",
       "15131          19.03.2019       NaN  #Dublette zu SEQ-ID: 7929646#   \n",
       "15132          19.03.2019       NaN  #Dublette zu SEQ-ID: 7929646#   \n",
       "15133          19.03.2019       NaN  #Dublette zu SEQ-ID: 7929646#   \n",
       "15134          19.03.2019       NaN  #Dublette zu SEQ-ID: 7929646#   \n",
       "15135          19.03.2019       NaN  #Dublette zu SEQ-ID: 7929646#   \n",
       "15136          19.03.2019       NaN  #Dublette zu SEQ-ID: 7929646#   \n",
       "15137                 NaN       NaN                            NaN   \n",
       "15138                 NaN       NaN                            NaN   \n",
       "15139                 NaN       NaN                            NaN   \n",
       "15140                 NaN       NaN                            NaN   \n",
       "15141                 NaN       NaN                            NaN   \n",
       "15142                 NaN       NaN                            NaN   \n",
       "15143                 NaN       NaN                            NaN   \n",
       "\n",
       "      splice_pred_alamut_hboc prediction_hboc  \\\n",
       "15095                     NaN             NaN   \n",
       "15096                     NaN             NaN   \n",
       "15097                     NaN             NaN   \n",
       "15098                     NaN             NaN   \n",
       "15099             unauffällig             NaN   \n",
       "15100             unauffällig             NaN   \n",
       "15101             unauffällig             NaN   \n",
       "15102             unauffällig             NaN   \n",
       "15103             unauffällig             NaN   \n",
       "15104             unauffällig             NaN   \n",
       "15105             unauffällig             NaN   \n",
       "15106             unauffällig             NaN   \n",
       "15107             unauffällig             NaN   \n",
       "15108             unauffällig             NaN   \n",
       "15109             unauffällig             NaN   \n",
       "15110             unauffällig             NaN   \n",
       "15111             unauffällig             NaN   \n",
       "15112             unauffällig             NaN   \n",
       "15113             unauffällig             NaN   \n",
       "15114                     NaN             NaN   \n",
       "15115                     NaN             NaN   \n",
       "15116                     NaN             NaN   \n",
       "15117                     NaN             NaN   \n",
       "15118                     NaN             NaN   \n",
       "15119                     NaN             NaN   \n",
       "15120                     NaN             NaN   \n",
       "15121                     NaN             NaN   \n",
       "15122                     NaN             NaN   \n",
       "15123                     NaN             NaN   \n",
       "15124             unauffällig             NaN   \n",
       "15125             unauffällig             NaN   \n",
       "15126             unauffällig             NaN   \n",
       "15127             unauffällig             NaN   \n",
       "15128             unauffällig             NaN   \n",
       "15129             unauffällig             NaN   \n",
       "15130             unauffällig             NaN   \n",
       "15131             unauffällig             NaN   \n",
       "15132             unauffällig             NaN   \n",
       "15133             unauffällig             NaN   \n",
       "15134             unauffällig             NaN   \n",
       "15135             unauffällig             NaN   \n",
       "15136             unauffällig             NaN   \n",
       "15137                     NaN             NaN   \n",
       "15138                     NaN             NaN   \n",
       "15139                     NaN             NaN   \n",
       "15140                     NaN             NaN   \n",
       "15141                     NaN             NaN   \n",
       "15142                     NaN             NaN   \n",
       "15143                     NaN             NaN   \n",
       "\n",
       "                                         literature_hboc  \\\n",
       "15095                                                NaN   \n",
       "15096                                                NaN   \n",
       "15097                                                NaN   \n",
       "15098                                                NaN   \n",
       "15099  (2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...   \n",
       "15100  (2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...   \n",
       "15101  (2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...   \n",
       "15102  (2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...   \n",
       "15103  (2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...   \n",
       "15104  (2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...   \n",
       "15105  (2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...   \n",
       "15106  (2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...   \n",
       "15107  (2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...   \n",
       "15108  (2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...   \n",
       "15109  (2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...   \n",
       "15110  (2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...   \n",
       "15111  (2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...   \n",
       "15112  (2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...   \n",
       "15113  (2 x hom in gnomAD non cancer) ExAc NFE:0.41 P...   \n",
       "15114                                                NaN   \n",
       "15115                                                NaN   \n",
       "15116                                                NaN   \n",
       "15117                                                NaN   \n",
       "15118                                                NaN   \n",
       "15119                                                NaN   \n",
       "15120                                                NaN   \n",
       "15121                                                NaN   \n",
       "15122                                                NaN   \n",
       "15123                                                NaN   \n",
       "15124  (2 x hom in gnomAD non cancer) FLOSSIES: 84x i...   \n",
       "15125  (2 x hom in gnomAD non cancer) FLOSSIES: 84x i...   \n",
       "15126  (2 x hom in gnomAD non cancer) FLOSSIES: 84x i...   \n",
       "15127  (2 x hom in gnomAD non cancer) FLOSSIES: 84x i...   \n",
       "15128  (2 x hom in gnomAD non cancer) FLOSSIES: 84x i...   \n",
       "15129  (2 x hom in gnomAD non cancer) FLOSSIES: 84x i...   \n",
       "15130  (2 x hom in gnomAD non cancer) FLOSSIES: 84x i...   \n",
       "15131  (2 x hom in gnomAD non cancer) FLOSSIES: 84x i...   \n",
       "15132  (2 x hom in gnomAD non cancer) FLOSSIES: 84x i...   \n",
       "15133  (2 x hom in gnomAD non cancer) FLOSSIES: 84x i...   \n",
       "15134  (2 x hom in gnomAD non cancer) FLOSSIES: 84x i...   \n",
       "15135  (2 x hom in gnomAD non cancer) FLOSSIES: 84x i...   \n",
       "15136  (2 x hom in gnomAD non cancer) FLOSSIES: 84x i...   \n",
       "15137                                                NaN   \n",
       "15138                                                NaN   \n",
       "15139                                                NaN   \n",
       "15140                                                NaN   \n",
       "15141                                                NaN   \n",
       "15142                                                NaN   \n",
       "15143                                                NaN   \n",
       "\n",
       "      evidence_level_lit_hboc comment_hboc recomm_action_hboc  \n",
       "15095                     NaN          NaN                NaN  \n",
       "15096                     NaN          NaN                NaN  \n",
       "15097                     NaN          NaN                NaN  \n",
       "15098                     NaN          NaN                NaN  \n",
       "15099                     NaN          NaN                NaN  \n",
       "15100                     NaN          NaN                NaN  \n",
       "15101                     NaN          NaN                NaN  \n",
       "15102                     NaN          NaN                NaN  \n",
       "15103                     NaN          NaN                NaN  \n",
       "15104                     NaN          NaN                NaN  \n",
       "15105                     NaN          NaN                NaN  \n",
       "15106                     NaN          NaN                NaN  \n",
       "15107                     NaN          NaN                NaN  \n",
       "15108                     NaN          NaN                NaN  \n",
       "15109                     NaN          NaN                NaN  \n",
       "15110                     NaN          NaN                NaN  \n",
       "15111                     NaN          NaN                NaN  \n",
       "15112                     NaN          NaN                NaN  \n",
       "15113                     NaN          NaN                NaN  \n",
       "15114                     NaN          NaN                NaN  \n",
       "15115                     NaN          NaN                NaN  \n",
       "15116                     NaN          NaN                NaN  \n",
       "15117                     NaN          NaN                NaN  \n",
       "15118                     NaN          NaN                NaN  \n",
       "15119                     NaN          NaN                NaN  \n",
       "15120                     NaN          NaN                NaN  \n",
       "15121                     NaN          NaN                NaN  \n",
       "15122                     NaN          NaN                NaN  \n",
       "15123                     NaN          NaN                NaN  \n",
       "15124                     NaN          NaN                NaN  \n",
       "15125                     NaN          NaN                NaN  \n",
       "15126                     NaN          NaN                NaN  \n",
       "15127                     NaN          NaN                NaN  \n",
       "15128                     NaN          NaN                NaN  \n",
       "15129                     NaN          NaN                NaN  \n",
       "15130                     NaN          NaN                NaN  \n",
       "15131                     NaN          NaN                NaN  \n",
       "15132                     NaN          NaN                NaN  \n",
       "15133                     NaN          NaN                NaN  \n",
       "15134                     NaN          NaN                NaN  \n",
       "15135                     NaN          NaN                NaN  \n",
       "15136                     NaN          NaN                NaN  \n",
       "15137                     NaN          NaN                NaN  \n",
       "15138                     NaN          NaN                NaN  \n",
       "15139                     NaN          NaN                NaN  \n",
       "15140                     NaN          NaN                NaN  \n",
       "15141                     NaN          NaN                NaN  \n",
       "15142                     NaN          NaN                NaN  \n",
       "15143                     NaN          NaN                NaN  \n",
       "\n",
       "[49 rows x 134 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_vars[all_vars['HUGO_Symbol']=='BRIP1']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f1559220",
   "metadata": {},
   "source": [
    "### 06.10."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "18c1ee29",
   "metadata": {},
   "outputs": [],
   "source": [
    "igv_all = merge_xlsx('2021-08-19_all_vars_analyzed.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "2ebcfc6d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>shortcut</th>\n",
       "      <th>HUGO_Symbol</th>\n",
       "      <th>max_gnomAD_AF</th>\n",
       "      <th>splice_change</th>\n",
       "      <th>ClinVar_Pathogenicity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1_23219372_G_A</td>\n",
       "      <td>EPHB2</td>\n",
       "      <td>0.00021103</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>1_27688743_T_C</td>\n",
       "      <td>MAP3K6</td>\n",
       "      <td>0.0037306</td>\n",
       "      <td>yes</td>\n",
       "      <td>Likely Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>1_45797760_T_C</td>\n",
       "      <td>MUTYH</td>\n",
       "      <td>0.0011775</td>\n",
       "      <td>yes</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>1_45797835_T_G</td>\n",
       "      <td>MUTYH</td>\n",
       "      <td>6.7534e-05</td>\n",
       "      <td>yes</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>380</th>\n",
       "      <td>2_47601174_C_T</td>\n",
       "      <td>EPCAM</td>\n",
       "      <td>4.2257e-06</td>\n",
       "      <td>yes</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>381</th>\n",
       "      <td>2_47607108_G_A</td>\n",
       "      <td>EPCAM</td>\n",
       "      <td>0.0021976</td>\n",
       "      <td>yes</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>382</th>\n",
       "      <td>2_47612302_C_G</td>\n",
       "      <td>EPCAM</td>\n",
       "      <td>0.00028683</td>\n",
       "      <td>yes</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>392</th>\n",
       "      <td>2_48033789_C_T</td>\n",
       "      <td>MSH6</td>\n",
       "      <td>6.3853e-05</td>\n",
       "      <td>yes</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>423</th>\n",
       "      <td>3_14190057_C_T</td>\n",
       "      <td>XPC</td>\n",
       "      <td>8.5326e-06</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425</th>\n",
       "      <td>3_14206927_C_A</td>\n",
       "      <td>XPC</td>\n",
       "      <td>0</td>\n",
       "      <td>yes</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>465</th>\n",
       "      <td>3_52440267_A_G</td>\n",
       "      <td>BAP1</td>\n",
       "      <td>4.222e-06</td>\n",
       "      <td>yes</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>470</th>\n",
       "      <td>3_52443725_C_G</td>\n",
       "      <td>BAP1</td>\n",
       "      <td>2.1114e-05</td>\n",
       "      <td>yes</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>637</th>\n",
       "      <td>7_66459197_A_G</td>\n",
       "      <td>SBDS</td>\n",
       "      <td>0.003971</td>\n",
       "      <td>yes</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>667</th>\n",
       "      <td>8_30916058_A_G</td>\n",
       "      <td>WRN</td>\n",
       "      <td>0.0035021</td>\n",
       "      <td>yes</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>717</th>\n",
       "      <td>8_145738768_G_T</td>\n",
       "      <td>RECQL4</td>\n",
       "      <td>9.5554e-05</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>747</th>\n",
       "      <td>9_22006246_C_G</td>\n",
       "      <td>CDKN2B</td>\n",
       "      <td>9.1e-06</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>757</th>\n",
       "      <td>9_100437871_T_C</td>\n",
       "      <td>XPA</td>\n",
       "      <td>4.2304e-06</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>811</th>\n",
       "      <td>10_99220447_C_T</td>\n",
       "      <td>MMS19</td>\n",
       "      <td>4.9383e-06</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>832</th>\n",
       "      <td>11_44255793_G_A</td>\n",
       "      <td>EXT2</td>\n",
       "      <td>2.1171e-05</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>879</th>\n",
       "      <td>11_108202604_A_C</td>\n",
       "      <td>ATM</td>\n",
       "      <td>3.1867e-05</td>\n",
       "      <td>yes</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>934</th>\n",
       "      <td>12_21624358_ATACT_A</td>\n",
       "      <td>RECQL</td>\n",
       "      <td>0.00035045</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1000</th>\n",
       "      <td>14_75498886_T_C</td>\n",
       "      <td>MLH3</td>\n",
       "      <td>6.3694e-05</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1030</th>\n",
       "      <td>14_105243112_G_T</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>2.1313e-05</td>\n",
       "      <td>yes</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1097</th>\n",
       "      <td>16_2137924_TCCCTGCAGTGCAGGAAAGGTAGGGCCGGGTGGGG_T</td>\n",
       "      <td>TSC2</td>\n",
       "      <td>0.0022738</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1107</th>\n",
       "      <td>16_2138446_G_C</td>\n",
       "      <td>TSC2</td>\n",
       "      <td>3.1871e-05</td>\n",
       "      <td>yes</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1138</th>\n",
       "      <td>17_7578370_C_T</td>\n",
       "      <td>TP53</td>\n",
       "      <td>0</td>\n",
       "      <td>yes</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1180</th>\n",
       "      <td>17_59934594_T_C</td>\n",
       "      <td>BRIP1</td>\n",
       "      <td>4.2344e-06</td>\n",
       "      <td>yes</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1184</th>\n",
       "      <td>17_74469327_G_A</td>\n",
       "      <td>RHBDF2</td>\n",
       "      <td>0.0036629</td>\n",
       "      <td>yes</td>\n",
       "      <td>Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1216</th>\n",
       "      <td>17_74470455_C_T</td>\n",
       "      <td>RHBDF2</td>\n",
       "      <td>1.2732e-05</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1397</th>\n",
       "      <td>19_40744001_G_C</td>\n",
       "      <td>AKT2</td>\n",
       "      <td>0.00015921</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1408</th>\n",
       "      <td>19_50918678_G_A</td>\n",
       "      <td>POLD1</td>\n",
       "      <td>0.0016256</td>\n",
       "      <td>yes</td>\n",
       "      <td>Likely Benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1442</th>\n",
       "      <td>22_24175755_G_A</td>\n",
       "      <td>SMARCB1</td>\n",
       "      <td>4.2206e-05</td>\n",
       "      <td>yes</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1446</th>\n",
       "      <td>22_29121230_C_T</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>0.0002229</td>\n",
       "      <td>yes</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              shortcut HUGO_Symbol  \\\n",
       "7                                       1_23219372_G_A       EPHB2   \n",
       "20                                      1_27688743_T_C      MAP3K6   \n",
       "79                                      1_45797760_T_C       MUTYH   \n",
       "80                                      1_45797835_T_G       MUTYH   \n",
       "380                                     2_47601174_C_T       EPCAM   \n",
       "381                                     2_47607108_G_A       EPCAM   \n",
       "382                                     2_47612302_C_G       EPCAM   \n",
       "392                                     2_48033789_C_T        MSH6   \n",
       "423                                     3_14190057_C_T         XPC   \n",
       "425                                     3_14206927_C_A         XPC   \n",
       "465                                     3_52440267_A_G        BAP1   \n",
       "470                                     3_52443725_C_G        BAP1   \n",
       "637                                     7_66459197_A_G        SBDS   \n",
       "667                                     8_30916058_A_G         WRN   \n",
       "717                                    8_145738768_G_T      RECQL4   \n",
       "747                                     9_22006246_C_G      CDKN2B   \n",
       "757                                    9_100437871_T_C         XPA   \n",
       "811                                    10_99220447_C_T       MMS19   \n",
       "832                                    11_44255793_G_A        EXT2   \n",
       "879                                   11_108202604_A_C         ATM   \n",
       "934                                12_21624358_ATACT_A       RECQL   \n",
       "1000                                   14_75498886_T_C        MLH3   \n",
       "1030                                  14_105243112_G_T        AKT1   \n",
       "1097  16_2137924_TCCCTGCAGTGCAGGAAAGGTAGGGCCGGGTGGGG_T        TSC2   \n",
       "1107                                    16_2138446_G_C        TSC2   \n",
       "1138                                    17_7578370_C_T        TP53   \n",
       "1180                                   17_59934594_T_C       BRIP1   \n",
       "1184                                   17_74469327_G_A      RHBDF2   \n",
       "1216                                   17_74470455_C_T      RHBDF2   \n",
       "1397                                   19_40744001_G_C        AKT2   \n",
       "1408                                   19_50918678_G_A       POLD1   \n",
       "1442                                   22_24175755_G_A     SMARCB1   \n",
       "1446                                   22_29121230_C_T       CHEK2   \n",
       "\n",
       "     max_gnomAD_AF splice_change   ClinVar_Pathogenicity  \n",
       "7       0.00021103           yes                     NaN  \n",
       "20       0.0037306           yes           Likely Benign  \n",
       "79       0.0011775           yes  Uncertain Significance  \n",
       "80      6.7534e-05           yes  Uncertain Significance  \n",
       "380     4.2257e-06           yes              Pathogenic  \n",
       "381      0.0021976           yes                  Benign  \n",
       "382     0.00028683           yes  Uncertain Significance  \n",
       "392     6.3853e-05           yes  Uncertain Significance  \n",
       "423     8.5326e-06           yes                     NaN  \n",
       "425              0           yes       Likely Pathogenic  \n",
       "465      4.222e-06           yes       Likely Pathogenic  \n",
       "470     2.1114e-05           yes  Uncertain Significance  \n",
       "637       0.003971           yes              Pathogenic  \n",
       "667      0.0035021           yes  Uncertain Significance  \n",
       "717     9.5554e-05           yes                     NaN  \n",
       "747        9.1e-06           yes                     NaN  \n",
       "757     4.2304e-06           yes                     NaN  \n",
       "811     4.9383e-06           yes                     NaN  \n",
       "832     2.1171e-05           yes                     NaN  \n",
       "879     3.1867e-05           yes              Pathogenic  \n",
       "934     0.00035045           yes                     NaN  \n",
       "1000    6.3694e-05           yes                     NaN  \n",
       "1030    2.1313e-05           yes  Uncertain Significance  \n",
       "1097     0.0022738           yes                     NaN  \n",
       "1107    3.1871e-05           yes  Uncertain Significance  \n",
       "1138             0           yes              Pathogenic  \n",
       "1180    4.2344e-06           yes       Likely Pathogenic  \n",
       "1184     0.0036629           yes                  Benign  \n",
       "1216    1.2732e-05           yes                     NaN  \n",
       "1397    0.00015921           yes                     NaN  \n",
       "1408     0.0016256           yes           Likely Benign  \n",
       "1442    4.2206e-05           yes  Uncertain Significance  \n",
       "1446     0.0002229           yes              Pathogenic  "
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "igv_all[['shortcut', 'HUGO_Symbol', 'max_gnomAD_AF', \n",
    "         'splice_change', 'ClinVar_Pathogenicity']][(igv_all['max_gnomAD_AF'].notnull())&(igv_all['splice_change']=='yes')&\n",
    "                           (igv_all['Splice Project Gene Priority']!='Cold (387-ACMG-MASTER)')&\n",
    "                           (~igv_all['Splice Project Gene Priority'].isin(gus))].drop_duplicates('shortcut')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "id": "a24eb054",
   "metadata": {},
   "outputs": [],
   "source": [
    "hboc_all = pd.read_csv('01_xlsx_files/2021-10-06_HBOC_Gepado_PML_all_variants.csv', sep=';')\n",
    "hboc_all = hboc_all.dropna(axis=0, subset=['Chr', 'POS', 'REF', 'ALT']).reset_index(drop=True)\n",
    "kge_all = pd.read_csv('01_xlsx_files/2021-10-06_KGE_Gepado_PML_all_variants.csv', sep=';')\n",
    "kge_all = kge_all.dropna(axis=0, subset=['Chr', 'POS', 'REF', 'ALT']).reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 252,
   "id": "0c2bab9f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pysam as ps\n",
    "# reference sequence fasta\n",
    "ref_fasta = ps.FastaFile('/mnt/g27prist/CMTD/Stephan/bcbio_installation/genomes/Hsapiens/GRCh37/seq/GRCh37.fa')\n",
    "def DataCleaning(df):\n",
    "    dfn = df.copy()\n",
    "    # for deletion\n",
    "    \n",
    "    # position = position - 1\n",
    "    df_del = dfn[(dfn['ALT']=='-')&(dfn['REF']!='-')].copy()\n",
    "    df_del['POS'] = df_del['POS'].str.replace(' - ', '..').str.replace('-', '..')\n",
    "    df_del['newPOS'] = df_del['POS'].str.split('.', expand=True)[0].astype(int)-1\n",
    "\n",
    "    new_ref_del_l = []\n",
    "    new_alt_del_l = []\n",
    "    for d in range(len(df_del)):\n",
    "        # position of nucleotide of first element of dataframe in which 'Alternate' not filled out\n",
    "        pos_flossies = df_del.iloc[d]['newPOS']\n",
    "        # reference nucleotide at specific position\n",
    "        ref_refseq = ref_fasta.fetch(reference=str(df_del.iloc[d]['Chr']), start=pos_flossies-1, end=pos_flossies)\n",
    "        # for alternative nucleotide reference nucleotide\n",
    "        new_alt_del_l.append(ref_refseq)\n",
    "        # reference nucleotide of first element of dataframe in which 'Alternate' not filled out\n",
    "        ref_flossies = df_del.iloc[d]['REF']\n",
    "        # new reference nucleotides for vcf annotation\n",
    "        new_ref_del = ref_refseq + ref_flossies\n",
    "        new_ref_del_l.append(new_ref_del)\n",
    "    # add new columns\n",
    "    df_del['REF_n'] = new_ref_del_l\n",
    "    df_del['ALT_n'] = new_alt_del_l\n",
    "    df_del.drop(columns = ['REF', 'ALT', 'POS'], inplace = True)\n",
    "    df_del.rename(columns = {'REF_n':'REF', 'ALT_n':'ALT', 'newPOS':'POS', 'Chr':'#CHROM'}, inplace = True)\n",
    "    \n",
    "    # for insertion\n",
    "    # position = position, because something was inserted at this position\n",
    "    df_ins = dfn[(dfn['REF']=='-')].copy()\n",
    "    df_ins['POS'] = df_ins['POS'].str.replace('-', '^')\n",
    "    df_ins['newPOS'] = df_ins['POS'].str.split('^', expand=True)[0].astype(int)\n",
    "\n",
    "    new_alt_ins_l = []\n",
    "    new_ref_ins_l = []\n",
    "    for i in range(len(df_ins)):\n",
    "        # position of nucleotide of first element of dataframe in which 'Reference' missing\n",
    "        pos_flossies = df_ins.iloc[i]['newPOS']\n",
    "        # reference nucleotide at specific position\n",
    "        ref_refseq = ref_fasta.fetch(reference=str(df_ins.iloc[i]['Chr']), start=pos_flossies-1, end=pos_flossies)\n",
    "        # for reference nucleotide reference nucleotide\n",
    "        new_ref_ins_l.append(ref_refseq)\n",
    "        # reference nucleotide of first element of dataframe in which 'Reference' missing\n",
    "        alt_flossies = df_ins.iloc[i]['ALT']\n",
    "        # new alternate nucleotides for vcf annotation\n",
    "        new_alt_ins = ref_refseq + alt_flossies\n",
    "        new_alt_ins_l.append(new_alt_ins)\n",
    "    # add new columns\n",
    "    df_ins['ALT_n'] = new_alt_ins_l\n",
    "    df_ins['REF_n'] = new_ref_ins_l\n",
    "    df_ins.drop(columns = ['REF', 'ALT', 'POS'], inplace = True)\n",
    "    df_ins.rename(columns = {'REF_n':'REF', 'ALT_n':'ALT', 'newPOS':'POS', 'Chr':'#CHROM'}, inplace = True)\n",
    "    \n",
    "    df_snv = dfn[(dfn['ALT']!='-')&(dfn['REF']!='-')].copy()\n",
    "    df_snv['POS'] = df_snv['POS'].str.split('.', expand=True)[0]\n",
    "    df_snv = df_snv.rename(columns={'Chr':'#CHROM'})\n",
    "\n",
    "    dfn_new = df_snv.append([df_del, df_ins])\n",
    "    dfn_new = dfn_new.sort_index()\n",
    "    return dfn_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 283,
   "id": "ad052934",
   "metadata": {},
   "outputs": [],
   "source": [
    "kge_all_n = DataCleaning(kge_all).drop(index=[3630, 5103])\n",
    "hboc_all_n = DataCleaning(hboc_all)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 284,
   "id": "34d4db1b",
   "metadata": {},
   "outputs": [],
   "source": [
    "hboc_all_n[['#CHROM', 'ALT', 'REF']] = hboc_all_n[['#CHROM', 'ALT', 'REF']].astype(str)\n",
    "hboc_all_n['POS'] = hboc_all_n['POS'].astype(int)\n",
    "kge_all_n[['#CHROM', 'ALT', 'REF']] = kge_all_n[['#CHROM', 'ALT', 'REF']].astype(str)\n",
    "kge_all_n['POS'] = kge_all_n['POS'].astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 305,
   "id": "2d55bbed",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "kge_merge = pd.merge(kge_all_n, igv_all[cols_cpra+['splice_change', 'Splice Project Gene Priority']], how='inner', \n",
    "         on=cols_cpra).drop_duplicates(cols_cpra+['ID Person'])\n",
    "kge_merge = kge_merge[(kge_merge['Splice Project Gene Priority']!='Cold (387-ACMG-MASTER)')&\n",
    "                      (~kge_merge['Gen'].isin(gus))&(kge_merge['splice_change']=='yes')]\n",
    "kge_merge['from'] = 'KGE'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 307,
   "id": "33987a17",
   "metadata": {},
   "outputs": [],
   "source": [
    "hboc_merge = pd.merge(hboc_all_n, igv_all[cols_cpra+['splice_change', 'Splice Project Gene Priority']], how='inner', \n",
    "         on=cols_cpra).drop_duplicates(cols_cpra+['ID Person'])\n",
    "hboc_merge = hboc_merge[(hboc_merge['Splice Project Gene Priority']!='Cold (387-ACMG-MASTER)')&\n",
    "                      (~hboc_merge['Gen'].isin(gus))&(hboc_merge['splice_change']=='yes')]\n",
    "hboc_merge['from'] = 'HBOC'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 319,
   "id": "209ef078",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "kge_hboc = kge_merge.append(hboc_merge, ignore_index=True)\n",
    "kge_hboc['#CHROM'] = pd.Categorical(kge_hboc['#CHROM'],\n",
    "                                    categories=['1','2','3','4','5','6','7','8','9','10','11','12', '13', '14', \n",
    "                                                '15', '16', '17', '18', '19', '20', '21', '22', 'X', 'Y'],\n",
    "                                    ordered=True)\n",
    "kge_hboc = kge_hboc.sort_values(by=['#CHROM', 'POS', 'REF', 'ALT'], ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 333,
   "id": "9e15d327",
   "metadata": {},
   "outputs": [],
   "source": [
    "kge_hboc[cols_cpra+['Gen', 'ID Person', 'Analytik ID', 'Bewertung gesamt', 'Clin Var', 'from']\n",
    "        ].to_excel('02_output_analysis/kge_hboc_spl_vars.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4757d184",
   "metadata": {},
   "source": [
    "### 01.09."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e81f1053",
   "metadata": {},
   "outputs": [],
   "source": [
    "igv_all = merge_xlsx('2021-08-19_all_vars_analyzed.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "d8136bce",
   "metadata": {},
   "outputs": [],
   "source": [
    "newn = igv_all[cols_cpra+['HUGO_Symbol','PID_trans']][\n",
    "    (igv_all['Splice Project Gene Priority']!='Cold (387-ACMG-MASTER)')&\n",
    "    (~igv_all['HUGO_Symbol'].isin(gus))&\n",
    "    (igv_all['splice_change']=='yes')].copy().reset_index(drop=True)\n",
    "newn = newn.drop(index=[17,18,19,20,21,22,23,24,25,26,27,28,29,30,32,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,\n",
    "                 52,53,54,86,90,71,72]).reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "7ce312d6",
   "metadata": {},
   "outputs": [],
   "source": [
    "#newn.to_excel('02_output_analysis/PIDs_patient_data.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "270de706",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "246d2a4b",
   "metadata": {},
   "outputs": [],
   "source": [
    "wo_dups = igv_all.drop_duplicates(cols_cpra).copy().reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "c50bc00e",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>splice_change</th>\n",
       "      <th>too_low_coverage</th>\n",
       "      <th>PosExonRefSeqAccession</th>\n",
       "      <th>PosExon_type</th>\n",
       "      <th>ClinVar_Pathogenicity</th>\n",
       "      <th>MaxEntScan_alt</th>\n",
       "      <th>...</th>\n",
       "      <th>SpliceAI_pred_DP_AG</th>\n",
       "      <th>SpliceAI_pred_DP_AL</th>\n",
       "      <th>SpliceAI_pred_DP_DG</th>\n",
       "      <th>SpliceAI_pred_DP_DL</th>\n",
       "      <th>SpliceAI_pred_DS_AG</th>\n",
       "      <th>SpliceAI_pred_DS_AL</th>\n",
       "      <th>SpliceAI_pred_DS_DG</th>\n",
       "      <th>SpliceAI_pred_DS_DL</th>\n",
       "      <th>ada_score</th>\n",
       "      <th>rf_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>2</td>\n",
       "      <td>48033789</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>1</td>\n",
       "      <td>insideDonorSite</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>4.269</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-24</td>\n",
       "      <td>-43</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.128779</td>\n",
       "      <td>0.324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>550</th>\n",
       "      <td>22</td>\n",
       "      <td>24175755</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>4</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>9.546</td>\n",
       "      <td>...</td>\n",
       "      <td>24</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>24</td>\n",
       "      <td>0.66</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0.002</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    #CHROM       POS REF ALT splice_change too_low_coverage  \\\n",
       "103      2  48033789   C   T           yes               no   \n",
       "550     22  24175755   G   A           yes               no   \n",
       "\n",
       "     PosExonRefSeqAccession         PosExon_type   ClinVar_Pathogenicity  \\\n",
       "103                       1      insideDonorSite  Uncertain Significance   \n",
       "550                       4  outsideAcceptorSite  Uncertain Significance   \n",
       "\n",
       "     MaxEntScan_alt  ...  SpliceAI_pred_DP_AG  SpliceAI_pred_DP_AL  \\\n",
       "103           4.269  ...                    1                  -24   \n",
       "550           9.546  ...                   24                    4   \n",
       "\n",
       "    SpliceAI_pred_DP_DG SpliceAI_pred_DP_DL SpliceAI_pred_DS_AG  \\\n",
       "103                 -43                   1                0.00   \n",
       "550                   4                  24                0.66   \n",
       "\n",
       "    SpliceAI_pred_DS_AL  SpliceAI_pred_DS_DG  SpliceAI_pred_DS_DL  ada_score  \\\n",
       "103                0.00                  0.0                 0.07   0.128779   \n",
       "550                0.02                  0.0                 0.00   0.000020   \n",
       "\n",
       "     rf_score  \n",
       "103     0.324  \n",
       "550     0.002  \n",
       "\n",
       "[2 rows x 22 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wo_dups[cols_cpra+['splice_change', 'too_low_coverage', 'PosExonRefSeqAccession', 'PosExon_type',\n",
    "                   'ClinVar_Pathogenicity', 'MaxEntScan_alt',\n",
    "                   'MaxEntScan_diff', 'MaxEntScan_ref', 'SpliceAI_pred_DP_AG', 'SpliceAI_pred_DP_AL',\n",
    "                   'SpliceAI_pred_DP_DG', 'SpliceAI_pred_DP_DL', 'SpliceAI_pred_DS_AG', 'SpliceAI_pred_DS_AL',\n",
    "                    'SpliceAI_pred_DS_DG', 'SpliceAI_pred_DS_DL', 'ada_score', 'rf_score']\n",
    "       ][(wo_dups['splice_change']=='yes')&\n",
    "         ((wo_dups['SpliceAI_pred_DS_AG']<0.9)&(wo_dups['SpliceAI_pred_DS_AL']<0.9)&\n",
    "          (wo_dups['SpliceAI_pred_DS_DG']<0.9)&(wo_dups['SpliceAI_pred_DS_DL']<0.9))&\n",
    "         (wo_dups['ada_score']<0.9)&(wo_dups['rf_score']<0.9)&\n",
    "         (((wo_dups['MaxEntScan_diff']>=0)&(wo_dups['MaxEntScan_alt']<8.5))|\n",
    "          ((wo_dups['MaxEntScan_diff']<0)&(wo_dups['MaxEntScan_alt']>6.2)))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "8465e065",
   "metadata": {},
   "outputs": [],
   "source": [
    "# vcf file for QCI (09.09.2021)\n",
    "import pysam as ps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "c6463bc6",
   "metadata": {},
   "outputs": [],
   "source": [
    "igv_hot_warm = igv_all[cols_cpra][(igv_all['splice_change']=='yes')&\n",
    "                                  (igv_all['Splice Project Gene Priority']!='Cold (387-ACMG-MASTER)')&\n",
    "                                  (~igv_all['HUGO_Symbol'].isin(gus))].drop_duplicates()\n",
    "\n",
    "vcf_header = ps.VariantHeader()\n",
    "for i in igv_hot_warm['#CHROM'].unique():\n",
    "    vcf_header.add_meta('contig', items = [('ID', i)])\n",
    "vcf_header.add_meta('reference', \n",
    "                    value='/AnnotierungsProzesse/Referenzdaten/Genome/Human/hg19/hg19_Ensembl/hg19 Ensembl')\n",
    "\n",
    "vcf_out = ps.VariantFile('01_xlsx_files/hot_warm_splice.txt', 'w', header=vcf_header)\n",
    "\n",
    "for c,p,r,a in zip(igv_hot_warm['#CHROM'], igv_hot_warm['POS'], igv_hot_warm['REF'], igv_hot_warm['ALT']):\n",
    "    rec = vcf_out.new_record()\n",
    "    rec.chrom = str(c)\n",
    "    rec.start = p-1\n",
    "    rec.stop = p\n",
    "    rec.ref = r\n",
    "    rec.alts = a\n",
    "    vcf_out.write(rec)\n",
    "vcf_out.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "18ab9cd8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ALT sequences were comma-separated --> \n",
    "with open('01_xlsx_files/hot_warm_splice.vcf', 'w') as w:\n",
    "    with open('01_xlsx_files/hot_warm_splice.txt') as o:\n",
    "        for line in o.readlines():\n",
    "            if line.startswith('#'):\n",
    "                w.write(line)\n",
    "            else:\n",
    "                line = line.replace(',','')\n",
    "                w.write(line)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "390bff6c",
   "metadata": {},
   "outputs": [],
   "source": [
    "igv_all[['#CHROM', 'pos_last', 'pos_next', 'kind', 'shortcut', 'strand', 'HUGO_Symbol'\n",
    "        ]][igv_all['pos_last'].notnull()].drop_duplicates('shortcut').to_excel('03_for_cDNA/pre_for_cDNA.xlsx', \n",
    "                                                                               index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "58c9854d",
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "igv_all[['#CHROM', 'POS', 'REF', 'ALT', 'HGVSg', 'HUGO_Symbol', 'shortcut', 'frameshift', 'premat_stop_cod', 'hgsv_new', 'INFO_IGV', 'too_low_coverage',\n",
    "         'splice_change', 'kind_hboc', 'classification_hboc', 'aquisition_hboc', 'taskforce_review_hboc',\n",
    "         'date_taskforce_hboc', 'note_hboc', 'add_note_hboc', 'splice_pred_alamut_hboc', 'prediction_hboc',\n",
    "         'literature_hboc', 'evidence_level_lit_hboc', 'comment_hboc', 'recomm_action_hboc']][\n",
    "    (igv_all['splice_change']=='yes')&(igv_all['kind_hboc'].notnull())&(igv_all['too_low_coverage']=='no')\n",
    "].drop_duplicates(cols_cpra).to_excel('02_output_analysis/hboc_splice_change.xlsx')\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "981ad04d",
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "igv_all[['#CHROM', 'POS', 'REF', 'ALT', 'HGVSg', 'HUGO_Symbol', 'shortcut', 'frameshift', 'premat_stop_cod', 'hgsv_new', 'INFO_IGV', 'too_low_coverage',\n",
    "         'splice_change', 'kind_hboc', 'classification_hboc', 'aquisition_hboc', 'taskforce_review_hboc', \n",
    "         'date_taskforce_hboc', 'note_hboc', 'add_note_hboc', 'splice_pred_alamut_hboc', 'prediction_hboc', \n",
    "         'literature_hboc', 'evidence_level_lit_hboc', 'comment_hboc', 'recomm_action_hboc']][\n",
    "    (igv_all['splice_change']=='no')&(igv_all['kind_hboc'].notnull())&(igv_all['too_low_coverage']=='no')\n",
    "].drop_duplicates(cols_cpra).to_excel('02_output_analysis/hboc_no_splice_change.xlsx')\n",
    "'''"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "02fbead2",
   "metadata": {},
   "source": [
    "### 30.08."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "07521dea",
   "metadata": {},
   "outputs": [],
   "source": [
    "igv_all = merge_xlsx('2021-08-19_all_vars_analyzed.xlsx')\n",
    "nct_vars = pd.read_pickle('00_dataframes/nct_gepado_master_vars')\n",
    "\n",
    "igv_nct = pd.merge(nct_vars[cols_cpra].drop_duplicates(), igv_all, how='inner')\n",
    "\n",
    "igv_outcome = perc_cat(igv_all)\n",
    "transp_outcome = igv_outcome.transpose(copy=True).reset_index().drop(index=[7,8,9,10,11])\n",
    "transp_outcome = transp_outcome.rename(columns={i:cat for i,cat in zip(transp_outcome.columns, \n",
    "                                                                       list(transp_outcome.loc[0]))}).drop(index=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "0ff04f3a",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>SpliceAI_pred_DP_AG</th>\n",
       "      <th>SpliceAI_pred_DP_AL</th>\n",
       "      <th>SpliceAI_pred_DP_DG</th>\n",
       "      <th>SpliceAI_pred_DP_DL</th>\n",
       "      <th>SpliceAI_pred_DS_AG</th>\n",
       "      <th>SpliceAI_pred_DS_AL</th>\n",
       "      <th>SpliceAI_pred_DS_DG</th>\n",
       "      <th>SpliceAI_pred_DS_DL</th>\n",
       "      <th>SpliceAI_pred_SYMBOL</th>\n",
       "      <th>ada_score</th>\n",
       "      <th>rf_score</th>\n",
       "      <th>PosExonRefSeqAccession</th>\n",
       "      <th>PosExon_type</th>\n",
       "      <th>ClinVar_Pathogenicity</th>\n",
       "      <th>INFO_IGV</th>\n",
       "      <th>shortcut</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>8</td>\n",
       "      <td>30916058</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>29</td>\n",
       "      <td>-28</td>\n",
       "      <td>29</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.00</td>\n",
       "      <td>WRN</td>\n",
       "      <td>0.991457</td>\n",
       "      <td>0.700</td>\n",
       "      <td>1</td>\n",
       "      <td>insideDonorSite</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>new 5'ss in intron (like spliceAI said) in 2LJ...</td>\n",
       "      <td>8_30916058_A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>17</td>\n",
       "      <td>7579699</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-11</td>\n",
       "      <td>22</td>\n",
       "      <td>-11</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.25</td>\n",
       "      <td>1.00</td>\n",
       "      <td>TP53</td>\n",
       "      <td>0.999987</td>\n",
       "      <td>0.940</td>\n",
       "      <td>22</td>\n",
       "      <td>insideAcceptorSite</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exon skipping</td>\n",
       "      <td>17_7579699_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>2</td>\n",
       "      <td>48033789</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>-24</td>\n",
       "      <td>-43</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.07</td>\n",
       "      <td>MSH6</td>\n",
       "      <td>0.128779</td>\n",
       "      <td>0.324</td>\n",
       "      <td>1</td>\n",
       "      <td>insideDonorSite</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>some exon skipping, intron retention, in other...</td>\n",
       "      <td>2_48033789_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>9</td>\n",
       "      <td>100437871</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-17</td>\n",
       "      <td>-2</td>\n",
       "      <td>-15</td>\n",
       "      <td>-10</td>\n",
       "      <td>0.09</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>XPA</td>\n",
       "      <td>0.999835</td>\n",
       "      <td>0.726</td>\n",
       "      <td>2</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "      <td>NaN</td>\n",
       "      <td>acceptor loss, new 3'ss in exon, frameshift</td>\n",
       "      <td>9_100437871_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>17</td>\n",
       "      <td>59934594</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-11</td>\n",
       "      <td>-2</td>\n",
       "      <td>-2</td>\n",
       "      <td>-21</td>\n",
       "      <td>0.90</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>BRIP1</td>\n",
       "      <td>0.999963</td>\n",
       "      <td>0.830</td>\n",
       "      <td>2</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>new ss in exon, 9 bases of exon skipped</td>\n",
       "      <td>17_59934594_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>1</td>\n",
       "      <td>17345454</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>-14</td>\n",
       "      <td>-1</td>\n",
       "      <td>38</td>\n",
       "      <td>-18</td>\n",
       "      <td>0.30</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>SDHB</td>\n",
       "      <td>0.999986</td>\n",
       "      <td>0.938</td>\n",
       "      <td>1</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "      <td>NaN</td>\n",
       "      <td>new 3'ss in exon, part of exon skipped, frames...</td>\n",
       "      <td>1_17345454_C_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>16</td>\n",
       "      <td>68849662</td>\n",
       "      <td>CG</td>\n",
       "      <td>C</td>\n",
       "      <td>6</td>\n",
       "      <td>-31</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.88</td>\n",
       "      <td>1.00</td>\n",
       "      <td>CDH1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>insideDonorSite</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>donor loss, intron retention</td>\n",
       "      <td>16_68849662_CG_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>2</td>\n",
       "      <td>47601174</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>-2</td>\n",
       "      <td>13</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.28</td>\n",
       "      <td>EPCAM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>13</td>\n",
       "      <td>insideDonorSite</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>about 50 % new 5' ss in exon, frameshift, spli...</td>\n",
       "      <td>2_47601174_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>3</td>\n",
       "      <td>14212051</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>-13</td>\n",
       "      <td>-1</td>\n",
       "      <td>-3</td>\n",
       "      <td>-11</td>\n",
       "      <td>0.86</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>XPC</td>\n",
       "      <td>0.999946</td>\n",
       "      <td>0.918</td>\n",
       "      <td>1</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "      <td>NaN</td>\n",
       "      <td>intron retention, acceptor loss, new 3' ss in ...</td>\n",
       "      <td>3_14212051_C_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>12</td>\n",
       "      <td>133254296</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>-3</td>\n",
       "      <td>-1</td>\n",
       "      <td>-3</td>\n",
       "      <td>-1</td>\n",
       "      <td>0.88</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>POLE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>insideAcceptorSite</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>new 3'Ss in exon (as spliceAI said), in frame,...</td>\n",
       "      <td>12_133254296_C_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>16</td>\n",
       "      <td>2121936</td>\n",
       "      <td>G</td>\n",
       "      <td>GT</td>\n",
       "      <td>-1</td>\n",
       "      <td>-13</td>\n",
       "      <td>-6</td>\n",
       "      <td>-2</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.02</td>\n",
       "      <td>TSC2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>outsideDonorSite</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exon skipping, quantification difficult (ca, 1...</td>\n",
       "      <td>16_2121936_G_GT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>17</td>\n",
       "      <td>41234419</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.99</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>0.999988</td>\n",
       "      <td>0.932</td>\n",
       "      <td>2</td>\n",
       "      <td>outsideDonorSite</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>bit of intron retention, exon skipping, count ...</td>\n",
       "      <td>17_41234419_A_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>19</td>\n",
       "      <td>45871887</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-31</td>\n",
       "      <td>1</td>\n",
       "      <td>-37</td>\n",
       "      <td>1</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.99</td>\n",
       "      <td>ERCC2</td>\n",
       "      <td>0.999987</td>\n",
       "      <td>0.930</td>\n",
       "      <td>1</td>\n",
       "      <td>outsideDonorSite</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exon skipping, donor loss</td>\n",
       "      <td>19_45871887_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>3</td>\n",
       "      <td>52443569</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-38</td>\n",
       "      <td>47</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.29</td>\n",
       "      <td>1.00</td>\n",
       "      <td>BAP1</td>\n",
       "      <td>0.999982</td>\n",
       "      <td>0.940</td>\n",
       "      <td>1</td>\n",
       "      <td>outsideDonorSite</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>exon skipped</td>\n",
       "      <td>3_52443569_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>3</td>\n",
       "      <td>52440267</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>38</td>\n",
       "      <td>25</td>\n",
       "      <td>-2</td>\n",
       "      <td>2</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.61</td>\n",
       "      <td>0.08</td>\n",
       "      <td>BAP1</td>\n",
       "      <td>0.994878</td>\n",
       "      <td>0.700</td>\n",
       "      <td>2</td>\n",
       "      <td>outsideDonorSite</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>new 5'ss in exon, 12 aa skipped, donor loss, 1...</td>\n",
       "      <td>3_52440267_A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144</th>\n",
       "      <td>11</td>\n",
       "      <td>108186639</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>24</td>\n",
       "      <td>-20</td>\n",
       "      <td>24</td>\n",
       "      <td>-1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.99</td>\n",
       "      <td>ATM</td>\n",
       "      <td>0.999992</td>\n",
       "      <td>0.942</td>\n",
       "      <td>1</td>\n",
       "      <td>outsideDonorSite</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>exon skipped</td>\n",
       "      <td>11_108186639_G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>16</td>\n",
       "      <td>2138446</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>-50</td>\n",
       "      <td>1</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>TSC2</td>\n",
       "      <td>0.999984</td>\n",
       "      <td>0.840</td>\n",
       "      <td>1</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>intron retention or new 3'ss in exon, part of ...</td>\n",
       "      <td>16_2138446_G_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>22</td>\n",
       "      <td>29121230</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>25</td>\n",
       "      <td>-3</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.94</td>\n",
       "      <td>1.00</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>0.999962</td>\n",
       "      <td>0.740</td>\n",
       "      <td>1</td>\n",
       "      <td>outsideDonorSite</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>in EUBCTJ intron retention of 4 bases, in RS,,...</td>\n",
       "      <td>22_29121230_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>17</td>\n",
       "      <td>56809887</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-45</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>18</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.87</td>\n",
       "      <td>0.12</td>\n",
       "      <td>RAD51C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18</td>\n",
       "      <td>insideDonorSite</td>\n",
       "      <td>Likely Benign</td>\n",
       "      <td>new 5'ss in exon (as spliceAI said, count 333:...</td>\n",
       "      <td>17_56809887_A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164</th>\n",
       "      <td>9</td>\n",
       "      <td>135779206</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-17</td>\n",
       "      <td>-2</td>\n",
       "      <td>-41</td>\n",
       "      <td>-2</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.94</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>TSC1</td>\n",
       "      <td>0.999993</td>\n",
       "      <td>0.926</td>\n",
       "      <td>2</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>new 3' ss in exon, part of exon skipped</td>\n",
       "      <td>9_135779206_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>173</th>\n",
       "      <td>3</td>\n",
       "      <td>52439929</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>-9</td>\n",
       "      <td>-1</td>\n",
       "      <td>-11</td>\n",
       "      <td>-9</td>\n",
       "      <td>0.86</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>BAP1</td>\n",
       "      <td>0.999988</td>\n",
       "      <td>0.940</td>\n",
       "      <td>1</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "      <td>NaN</td>\n",
       "      <td>acceptor loss, new 3'ss in exon, frameshift</td>\n",
       "      <td>3_52439929_C_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175</th>\n",
       "      <td>16</td>\n",
       "      <td>3650977</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>-48</td>\n",
       "      <td>19</td>\n",
       "      <td>-39</td>\n",
       "      <td>2</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.87</td>\n",
       "      <td>SLX4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>outsideDonorSite</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exon skipping, intron retention (but intron re...</td>\n",
       "      <td>16_3650977_T_TA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>7</td>\n",
       "      <td>66459197</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>10</td>\n",
       "      <td>27</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.64</td>\n",
       "      <td>0.93</td>\n",
       "      <td>SBDS</td>\n",
       "      <td>0.999936</td>\n",
       "      <td>0.866</td>\n",
       "      <td>2</td>\n",
       "      <td>outsideDonorSite</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>new ss, part of exon skipped, (for 7TBCL somet...</td>\n",
       "      <td>7_66459197_A_G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    #CHROM        POS REF ALT SpliceAI_pred_DP_AG SpliceAI_pred_DP_AL  \\\n",
       "0        8   30916058   A   G                  29                 -28   \n",
       "35      17    7579699   C   T                 -11                  22   \n",
       "42       2   48033789   C   T                   1                 -24   \n",
       "49       9  100437871   T   C                 -17                  -2   \n",
       "57      17   59934594   T   C                 -11                  -2   \n",
       "61       1   17345454   C   G                 -14                  -1   \n",
       "65      16   68849662  CG   C                   6                 -31   \n",
       "72       2   47601174   C   T                  13                   2   \n",
       "73       3   14212051   C   G                 -13                  -1   \n",
       "77      12  133254296   C   A                  -3                  -1   \n",
       "82      16    2121936   G  GT                  -1                 -13   \n",
       "98      17   41234419   A   C                   0                  38   \n",
       "129     19   45871887   C   T                 -31                   1   \n",
       "130      3   52443569   C   T                 -38                  47   \n",
       "131      3   52440267   A   G                  38                  25   \n",
       "144     11  108186639   G   A                  24                 -20   \n",
       "145     16    2138446   G   C                  14                   1   \n",
       "149     22   29121230   C   T                   1                  25   \n",
       "153     17   56809887   A   G                 -45                  18   \n",
       "164      9  135779206   T   C                 -17                  -2   \n",
       "173      3   52439929   C   A                  -9                  -1   \n",
       "175     16    3650977   T  TA                 -48                  19   \n",
       "183      7   66459197   A   G                  10                  27   \n",
       "\n",
       "    SpliceAI_pred_DP_DG SpliceAI_pred_DP_DL  SpliceAI_pred_DS_AG  \\\n",
       "0                    29                   1                 0.00   \n",
       "35                  -11                   1                 0.00   \n",
       "42                  -43                   1                 0.00   \n",
       "49                  -15                 -10                 0.09   \n",
       "57                   -2                 -21                 0.90   \n",
       "61                   38                 -18                 0.30   \n",
       "65                    6                   0                 0.00   \n",
       "72                   -2                  13                 0.00   \n",
       "73                   -3                 -11                 0.86   \n",
       "77                   -3                  -1                 0.88   \n",
       "82                   -6                  -2                 0.00   \n",
       "98                    0                   2                 0.00   \n",
       "129                 -37                   1                 0.01   \n",
       "130                   8                   1                 0.00   \n",
       "131                  -2                   2                 0.00   \n",
       "144                  24                  -1                 0.00   \n",
       "145                 -50                   1                 0.22   \n",
       "149                  -3                   1                 0.00   \n",
       "153                   0                  18                 0.11   \n",
       "164                 -41                  -2                 0.04   \n",
       "173                 -11                  -9                 0.86   \n",
       "175                 -39                   2                 0.00   \n",
       "183                  10                   2                 0.00   \n",
       "\n",
       "     SpliceAI_pred_DS_AL  SpliceAI_pred_DS_DG  SpliceAI_pred_DS_DL  \\\n",
       "0                   0.00                 0.06                 0.00   \n",
       "35                  0.00                 0.25                 1.00   \n",
       "42                  0.00                 0.00                 0.07   \n",
       "49                  1.00                 0.00                 0.00   \n",
       "57                  0.99                 0.00                 0.00   \n",
       "61                  0.98                 0.00                 0.00   \n",
       "65                  0.00                 0.88                 1.00   \n",
       "72                  0.00                 0.95                 0.28   \n",
       "73                  1.00                 0.00                 0.00   \n",
       "77                  0.00                 0.00                 0.00   \n",
       "82                  0.00                 0.07                 0.02   \n",
       "98                  0.00                 0.00                 0.99   \n",
       "129                 0.00                 0.18                 0.99   \n",
       "130                 0.00                 0.29                 1.00   \n",
       "131                 0.00                 0.61                 0.08   \n",
       "144                 0.00                 0.01                 0.99   \n",
       "145                 0.97                 0.00                 0.00   \n",
       "149                 0.00                 0.94                 1.00   \n",
       "153                 0.00                 0.87                 0.12   \n",
       "164                 0.94                 0.00                 0.00   \n",
       "173                 0.99                 0.00                 0.00   \n",
       "175                 0.00                 0.00                 0.87   \n",
       "183                 0.00                 0.64                 0.93   \n",
       "\n",
       "    SpliceAI_pred_SYMBOL  ada_score  rf_score  PosExonRefSeqAccession  \\\n",
       "0                    WRN   0.991457     0.700                       1   \n",
       "35                  TP53   0.999987     0.940                      22   \n",
       "42                  MSH6   0.128779     0.324                       1   \n",
       "49                   XPA   0.999835     0.726                       2   \n",
       "57                 BRIP1   0.999963     0.830                       2   \n",
       "61                  SDHB   0.999986     0.938                       1   \n",
       "65                  CDH1        NaN       NaN                       0   \n",
       "72                 EPCAM        NaN       NaN                      13   \n",
       "73                   XPC   0.999946     0.918                       1   \n",
       "77                  POLE        NaN       NaN                       9   \n",
       "82                  TSC2        NaN       NaN                       1   \n",
       "98                 BRCA1   0.999988     0.932                       2   \n",
       "129                ERCC2   0.999987     0.930                       1   \n",
       "130                 BAP1   0.999982     0.940                       1   \n",
       "131                 BAP1   0.994878     0.700                       2   \n",
       "144                  ATM   0.999992     0.942                       1   \n",
       "145                 TSC2   0.999984     0.840                       1   \n",
       "149                CHEK2   0.999962     0.740                       1   \n",
       "153               RAD51C        NaN       NaN                      18   \n",
       "164                 TSC1   0.999993     0.926                       2   \n",
       "173                 BAP1   0.999988     0.940                       1   \n",
       "175                 SLX4        NaN       NaN                       3   \n",
       "183                 SBDS   0.999936     0.866                       2   \n",
       "\n",
       "            PosExon_type   ClinVar_Pathogenicity  \\\n",
       "0        insideDonorSite  Uncertain Significance   \n",
       "35    insideAcceptorSite                     NaN   \n",
       "42       insideDonorSite  Uncertain Significance   \n",
       "49   outsideAcceptorSite                     NaN   \n",
       "57   outsideAcceptorSite       Likely Pathogenic   \n",
       "61   outsideAcceptorSite                     NaN   \n",
       "65       insideDonorSite              Pathogenic   \n",
       "72       insideDonorSite              Pathogenic   \n",
       "73   outsideAcceptorSite                     NaN   \n",
       "77    insideAcceptorSite  Uncertain Significance   \n",
       "82      outsideDonorSite                     NaN   \n",
       "98      outsideDonorSite              Pathogenic   \n",
       "129     outsideDonorSite                     NaN   \n",
       "130     outsideDonorSite              Pathogenic   \n",
       "131     outsideDonorSite       Likely Pathogenic   \n",
       "144     outsideDonorSite       Likely Pathogenic   \n",
       "145  outsideAcceptorSite  Uncertain Significance   \n",
       "149     outsideDonorSite              Pathogenic   \n",
       "153      insideDonorSite           Likely Benign   \n",
       "164  outsideAcceptorSite  Uncertain Significance   \n",
       "173  outsideAcceptorSite                     NaN   \n",
       "175     outsideDonorSite                     NaN   \n",
       "183     outsideDonorSite              Pathogenic   \n",
       "\n",
       "                                              INFO_IGV          shortcut  \n",
       "0    new 5'ss in intron (like spliceAI said) in 2LJ...    8_30916058_A_G  \n",
       "35                                       exon skipping    17_7579699_C_T  \n",
       "42   some exon skipping, intron retention, in other...    2_48033789_C_T  \n",
       "49         acceptor loss, new 3'ss in exon, frameshift   9_100437871_T_C  \n",
       "57            new ss in exon, 9 bases of exon skipped    17_59934594_T_C  \n",
       "61   new 3'ss in exon, part of exon skipped, frames...    1_17345454_C_G  \n",
       "65                        donor loss, intron retention  16_68849662_CG_C  \n",
       "72   about 50 % new 5' ss in exon, frameshift, spli...    2_47601174_C_T  \n",
       "73   intron retention, acceptor loss, new 3' ss in ...    3_14212051_C_G  \n",
       "77   new 3'Ss in exon (as spliceAI said), in frame,...  12_133254296_C_A  \n",
       "82   exon skipping, quantification difficult (ca, 1...   16_2121936_G_GT  \n",
       "98   bit of intron retention, exon skipping, count ...   17_41234419_A_C  \n",
       "129                          exon skipping, donor loss   19_45871887_C_T  \n",
       "130                                       exon skipped    3_52443569_C_T  \n",
       "131  new 5'ss in exon, 12 aa skipped, donor loss, 1...    3_52440267_A_G  \n",
       "144                                       exon skipped  11_108186639_G_A  \n",
       "145  intron retention or new 3'ss in exon, part of ...    16_2138446_G_C  \n",
       "149  in EUBCTJ intron retention of 4 bases, in RS,,...   22_29121230_C_T  \n",
       "153  new 5'ss in exon (as spliceAI said, count 333:...   17_56809887_A_G  \n",
       "164            new 3' ss in exon, part of exon skipped   9_135779206_T_C  \n",
       "173        acceptor loss, new 3'ss in exon, frameshift    3_52439929_C_A  \n",
       "175  exon skipping, intron retention (but intron re...   16_3650977_T_TA  \n",
       "183  new ss, part of exon skipped, (for 7TBCL somet...    7_66459197_A_G  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "igv_nct[cols_cpra+['SpliceAI_pred_DP_AG', 'SpliceAI_pred_DP_AL', 'SpliceAI_pred_DP_DG', \n",
    "                    'SpliceAI_pred_DP_DL', 'SpliceAI_pred_DS_AG', 'SpliceAI_pred_DS_AL', 'SpliceAI_pred_DS_DG', \n",
    "                    'SpliceAI_pred_DS_DL', 'SpliceAI_pred_SYMBOL', 'ada_score', 'rf_score',\n",
    "                    'PosExonRefSeqAccession', 'PosExon_type', 'ClinVar_Pathogenicity', 'INFO_IGV', 'shortcut']][\n",
    "    (igv_nct['Splice Project Gene Priority']!='Cold (387-ACMG-MASTER)')&(~igv_nct['HUGO_Symbol'].isin(gus))&\n",
    "    (igv_nct['splice_change']=='yes')&(igv_nct['too_low_coverage']=='no')].drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "b5048c88",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "81"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#[cols_cpra+['SpliceAI_pred_DP_AG', 'SpliceAI_pred_DP_AL', 'SpliceAI_pred_DP_DG', \n",
    "                    #'SpliceAI_pred_DP_DL', 'SpliceAI_pred_DS_AG', 'SpliceAI_pred_DS_AL', 'SpliceAI_pred_DS_DG', \n",
    "                    #'SpliceAI_pred_DS_DL', 'SpliceAI_pred_SYMBOL', 'ada_score', 'rf_score',\n",
    "                    #'PosExonRefSeqAccession', 'PosExon_type', 'ClinVar_Pathogenicity', 'INFO_IGV']]\n",
    "len(igv_all[(igv_all['too_low_coverage']=='no')&(igv_all['splice_change']=='no')&\n",
    "    (((igv_all['PosExonRefSeqAccession']<3)&\n",
    "     (igv_all['PosExon_type'].isin(['outsideAcceptorSite', 'outsideDonorSite'])))|\n",
    "    ((igv_all['PosExonRefSeqAccession']<2)&\n",
    "     (igv_all['PosExon_type'].isin(['insideAcceptorSite', 'insideDonorSite']))))\n",
    "    ].groupby(cols_cpra))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "4944a0bd",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>SpliceAI_pred_DP_AG</th>\n",
       "      <th>SpliceAI_pred_DP_AL</th>\n",
       "      <th>SpliceAI_pred_DP_DG</th>\n",
       "      <th>SpliceAI_pred_DP_DL</th>\n",
       "      <th>SpliceAI_pred_DS_AG</th>\n",
       "      <th>SpliceAI_pred_DS_AL</th>\n",
       "      <th>SpliceAI_pred_DS_DG</th>\n",
       "      <th>SpliceAI_pred_DS_DL</th>\n",
       "      <th>SpliceAI_pred_SYMBOL</th>\n",
       "      <th>ada_score</th>\n",
       "      <th>rf_score</th>\n",
       "      <th>PosExonRefSeqAccession</th>\n",
       "      <th>PosExon_type</th>\n",
       "      <th>ClinVar_Pathogenicity</th>\n",
       "      <th>INFO_IGV</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>392</th>\n",
       "      <td>2</td>\n",
       "      <td>48033789</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>-24</td>\n",
       "      <td>-43</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.07</td>\n",
       "      <td>MSH6</td>\n",
       "      <td>0.128779</td>\n",
       "      <td>0.324</td>\n",
       "      <td>1</td>\n",
       "      <td>insideDonorSite</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>some exon skipping, intron retention, in other...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1184</th>\n",
       "      <td>17</td>\n",
       "      <td>74469327</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>6</td>\n",
       "      <td>27</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.64</td>\n",
       "      <td>0.01</td>\n",
       "      <td>RHBDF2</td>\n",
       "      <td>0.000024</td>\n",
       "      <td>0.000</td>\n",
       "      <td>7</td>\n",
       "      <td>outsideDonorSite</td>\n",
       "      <td>Benign</td>\n",
       "      <td>new 5'ss in intron (as spliceAI said), intron ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1442</th>\n",
       "      <td>22</td>\n",
       "      <td>24175755</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>24</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>24</td>\n",
       "      <td>0.66</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>SMARCB1</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0.002</td>\n",
       "      <td>4</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>new 3'ss in exon (as spliceAI said) in 8Q2LVZ,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1443</th>\n",
       "      <td>22</td>\n",
       "      <td>24175755</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>24</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>24</td>\n",
       "      <td>0.66</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>SMARCB1</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0.002</td>\n",
       "      <td>4</td>\n",
       "      <td>outsideAcceptorSite</td>\n",
       "      <td>Uncertain Significance</td>\n",
       "      <td>new 3'ss in exon (as spliceAI said) in 8Q2LVZ,...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     #CHROM       POS REF ALT SpliceAI_pred_DP_AG SpliceAI_pred_DP_AL  \\\n",
       "392       2  48033789   C   T                   1                 -24   \n",
       "1184     17  74469327   G   A                   6                  27   \n",
       "1442     22  24175755   G   A                  24                   4   \n",
       "1443     22  24175755   G   A                  24                   4   \n",
       "\n",
       "     SpliceAI_pred_DP_DG SpliceAI_pred_DP_DL  SpliceAI_pred_DS_AG  \\\n",
       "392                  -43                   1                 0.00   \n",
       "1184                   2                   7                 0.00   \n",
       "1442                   4                  24                 0.66   \n",
       "1443                   4                  24                 0.66   \n",
       "\n",
       "      SpliceAI_pred_DS_AL  SpliceAI_pred_DS_DG  SpliceAI_pred_DS_DL  \\\n",
       "392                  0.00                 0.00                 0.07   \n",
       "1184                 0.00                 0.64                 0.01   \n",
       "1442                 0.02                 0.00                 0.00   \n",
       "1443                 0.02                 0.00                 0.00   \n",
       "\n",
       "     SpliceAI_pred_SYMBOL  ada_score  rf_score  PosExonRefSeqAccession  \\\n",
       "392                  MSH6   0.128779     0.324                       1   \n",
       "1184               RHBDF2   0.000024     0.000                       7   \n",
       "1442              SMARCB1   0.000020     0.002                       4   \n",
       "1443              SMARCB1   0.000020     0.002                       4   \n",
       "\n",
       "             PosExon_type   ClinVar_Pathogenicity  \\\n",
       "392       insideDonorSite  Uncertain Significance   \n",
       "1184     outsideDonorSite                  Benign   \n",
       "1442  outsideAcceptorSite  Uncertain Significance   \n",
       "1443  outsideAcceptorSite  Uncertain Significance   \n",
       "\n",
       "                                               INFO_IGV  \n",
       "392   some exon skipping, intron retention, in other...  \n",
       "1184  new 5'ss in intron (as spliceAI said), intron ...  \n",
       "1442  new 3'ss in exon (as spliceAI said) in 8Q2LVZ,...  \n",
       "1443  new 3'ss in exon (as spliceAI said) in 8Q2LVZ,...  "
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "igv_all[cols_cpra+['SpliceAI_pred_DP_AG', 'SpliceAI_pred_DP_AL', 'SpliceAI_pred_DP_DG', \n",
    "                    'SpliceAI_pred_DP_DL', 'SpliceAI_pred_DS_AG', 'SpliceAI_pred_DS_AL', 'SpliceAI_pred_DS_DG', \n",
    "                    'SpliceAI_pred_DS_DL', 'SpliceAI_pred_SYMBOL', 'ada_score', 'rf_score',\n",
    "                    'PosExonRefSeqAccession', 'PosExon_type', 'ClinVar_Pathogenicity', 'INFO_IGV']][\n",
    "    ((igv_all['SpliceAI_pred_DS_AG']<0.9)&(igv_all['SpliceAI_pred_DS_AL']<0.9)&\n",
    "             (igv_all['SpliceAI_pred_DS_DG']<0.9)&(igv_all['SpliceAI_pred_DS_DL']<0.9))&\n",
    "            (igv_all['ada_score']<0.9)&(igv_all['rf_score']<0.9)&\n",
    "            #(((igv_all['MaxEntScan_diff']<0)&(igv_all['MaxEntScan_alt']>6.2))|\n",
    "             #((igv_all['MaxEntScan_diff']>0)&(igv_all['MaxEntScan_alt']<8.5)))&\n",
    "            (igv_all['too_low_coverage']=='no')&(igv_all['splice_change']=='yes')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a7148519",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "#igv_outcome.to_excel('02_output_analysis/2021-08-19_outcome_igv.xlsx', index=False)\n",
    "#igv_outcome.to_pickle('00_dataframes/2021-08-19_igv_perc_table')\n",
    "#transp_outcome.to_pickle('00_dataframes/transp_igv_outcome')\n",
    "#igv_all.to_pickle('00_dataframes/2021-08-19_igv_outcome')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "fd20ca80",
   "metadata": {},
   "outputs": [],
   "source": [
    "#igv_outcome.to_excel('02_output_analysis/2021-08-19_outcome_igv.xlsx')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3d7e577b",
   "metadata": {},
   "source": [
    "### 26.08."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2131831f",
   "metadata": {},
   "outputs": [],
   "source": [
    "igv_all = merge_xlsx('2021-08-18_all_vars_analyzed.xlsx')\n",
    "igv_outcome = perc_cat(igv_all)\n",
    "hboc_vars = pd.read_pickle('../04_annotations/00_dataframes/hboc_vars')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "76fd69d9",
   "metadata": {},
   "outputs": [],
   "source": [
    "search_col(igv_all, 'clinv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1dd80698",
   "metadata": {},
   "outputs": [],
   "source": [
    "hboc_igv = pd.merge(igv_all[['#CHROM', 'POS', 'REF', 'ALT', 'PID', 'splice_change', 'ClinVar_Pathogenicity']], \n",
    "                    hboc_vars[['#CHROM', 'POS', 'REF', 'ALT', 'ID_hboc']], \n",
    "                    on=cols_cpra,\n",
    "                    how='inner').drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d7f3864e",
   "metadata": {},
   "outputs": [],
   "source": [
    "hboc_igv[hboc_igv['splice_change']=='yes'].drop_duplicates(subset=cols_cpra)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2a579380",
   "metadata": {},
   "outputs": [],
   "source": [
    "len(igv_all[cols_cpra][igv_all['splice_change']=='yes'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "998f54c2",
   "metadata": {},
   "source": [
    "### 24.08."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "036577b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "igv_all = merge_xlsx('2021-08-18_all_vars_analyzed.xlsx')\n",
    "igv_outcome = perc_cat(igv_all)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "560d7df3",
   "metadata": {},
   "outputs": [],
   "source": [
    "search_col(igv_all,'hgv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "38f9031d",
   "metadata": {},
   "outputs": [],
   "source": [
    "igv_all['ClinVar_Pathogenicity'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f731908e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "asdf = igv_all[(igv_all['splice_change']=='yes')\n",
    "               &(igv_all['ClinVar_Pathogenicity'].isin(['Uncertain Significance', np.nan]))\n",
    "               &(igv_all['Splice Project Gene Priority']!='Cold (387-ACMG-MASTER)')\n",
    "               &(~igv_all['HUGO_Symbol'].isin(gus))\n",
    "               &(((igv_all['PosExonRefSeqAccession']>2)\n",
    "                  &(igv_all['PosExon_type'].isin(['outsideDonorSite', 'outsideAcceptorSite'])))\n",
    "                 |((igv_all['PosExonRefSeqAccession']>0)\n",
    "                   &(igv_all['PosExon_type'].isin(['insideDonorSite', 'insideAcceptorSite']))))\n",
    "       ]#.drop_duplicates(subset=cols_cpra)\n",
    "len(asdf[['shortcut','splice_change', 'too_low_coverage', 'PosExonRefSeqAccession', 'PosExon_type', 'INFO_IGV']])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ef92be04",
   "metadata": {},
   "outputs": [],
   "source": [
    "#xlsx_df(asdf, '2021-08-24_hot_warm_uncertain_unusual')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b1f4752d",
   "metadata": {},
   "source": [
    "### 18.08."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9153c1bb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# previously 4 duplicates -> newest version, more info...\n",
    "# newest file (2021-08-18_all_vars_analyzed.xlsx) no duplicates anymore\n",
    "igv_all = merge_xlsx('2021-08-18_all_vars_analyzed.xlsx')\n",
    "igv_outcome = perc_cat(igv_all)\n",
    "#igv_outcome.to_excel('02_output_analysis/2021-08-18_outcome_igv.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2c71cdb7",
   "metadata": {},
   "outputs": [],
   "source": [
    "#xlsx_df(igv_all, '2021-08-19_all_vars_analyzed')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "510cc7d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "#for coverage, splice_change, cat in zip(['no', 'no', 'no', 'yes', 'yes', 'yes'], \n",
    "#                                        ['yes', 'maybe', 'no', 'yes', 'maybe', 'no'], \n",
    "#                                        ['sufficient, yes', 'sufficient, maybe', 'sufficient, no', \n",
    "#                                         'not sufficient, yes', 'not sufficient, maybe', 'not sufficient, no']):\n",
    "#    print(cat, len(igv_all[((igv_all['SpliceAI_pred_DS_AG']>0.5)|\n",
    "#                   (igv_all['SpliceAI_pred_DS_AL']>0.5)|\n",
    "#                   (igv_all['SpliceAI_pred_DS_DG']>0.5)|\n",
    "#                   (igv_all['SpliceAI_pred_DS_DL']>0.5))&\n",
    "#                  (igv_all['too_low_coverage']==coverage)&\n",
    "#                  (igv_all['splice_change']==splice_change)].groupby(cols_cpra)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "712b079c",
   "metadata": {},
   "outputs": [],
   "source": [
    "transp_outcome = igv_outcome.transpose(copy=True).reset_index().drop(index=[7,8,9,10,11])\n",
    "transp_outcome = transp_outcome.rename(columns={i:cat for i,cat in zip(transp_outcome.columns, \n",
    "                                                                       list(transp_outcome.loc[0]))}).drop(index=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5376baca",
   "metadata": {},
   "outputs": [],
   "source": [
    "#transp_outcome.to_pickle('00_dataframes/transp_igv_outcome')\n",
    "#igv_all.to_pickle('00_dataframes/2021-08-19_igv_outcome')\n",
    "#igv_outcome.to_pickle('00_dataframes/2021-08-19_igv_perc_table')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d63cb8b3",
   "metadata": {},
   "source": [
    "### 12.08. new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c3720f3b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# newest version up to 2021-08-12 13:30 2021-08-12_all_vars_analyzed.xlsx\n",
    "new_vars = merge_xlsx('2021-08-12_all_vars_analyzed.xlsx')\n",
    "new_vars = make_float(new_vars)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6acdd888",
   "metadata": {},
   "outputs": [],
   "source": [
    "new_vars.to_pickle('00_dataframes/igv_vars')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cf9612ea",
   "metadata": {},
   "outputs": [],
   "source": [
    "#len(new_vars[((new_vars['SpliceAI_pred_DS_AG']>0.9)|\n",
    "#              (new_vars['SpliceAI_pred_DS_AL']>0.9)|\n",
    "#              (new_vars['SpliceAI_pred_DS_DG']>0.9)|\n",
    "#              (new_vars['SpliceAI_pred_DS_DL']>0.9))&\n",
    "#             (new_vars['too_low_coverage']=='yes')&\n",
    "#             (new_vars['splice_change']=='no')].groupby(\n",
    "#    cols_cpra))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cc401ed5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# interesting variants for further analysis on clinical patient basis\n",
    "int_vars = new_vars[(new_vars['Splice Project Gene Priority']==\n",
    "             'Hot (ACMG / MASTER)')&\n",
    "             (new_vars['too_low_coverage']=='no')&\n",
    "             (new_vars['splice_change']=='yes')]\n",
    "\n",
    "int_vars[['#CHROM', 'POS', 'REF', 'ALT', 'MaxEntScan_alt', 'MaxEntScan_diff', 'MaxEntScan_ref', \n",
    "                    'CADD_PHRED', 'SpliceAI_pred_DP_AG', 'SpliceAI_pred_DP_AL', 'SpliceAI_pred_DP_DG', \n",
    "                    'SpliceAI_pred_DP_DL', 'SpliceAI_pred_DS_AG', 'SpliceAI_pred_DS_AL', 'SpliceAI_pred_DS_DG', \n",
    "                    'SpliceAI_pred_DS_DL', 'SpliceAI_pred_SYMBOL', 'ada_score', 'rf_score',\n",
    "                    'PosExonRefSeqAccession', 'PosExon_type', 'ClinVar_Pathogenicity', \n",
    "                    'consensus_sequence_IGV_region_left', 'consensus_sequence_IGV_sequence_left', \n",
    "                    'consensus_sequence_IGV_region_right', 'consensus_sequence_IGV_sequence_right', 'INFO_IGV', \n",
    "                    'too_low_coverage', 'splice_change', 'shortcut', 'ClinVar_Traits', 'PID']].to_excel('01_xlsx_files/int_vars_pats_hot.xlsx')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "15a06f7b",
   "metadata": {},
   "source": [
    "### 12.08.2021"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a13a7c96",
   "metadata": {},
   "outputs": [],
   "source": [
    "# newest version up to 2021-08-12 10:00 2021-08-11_all_vars_analyzed.xlsx\n",
    "#new_vars = merge_xlsx('2021-08-11_all_vars_analyzed.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4b4fb4fc",
   "metadata": {},
   "outputs": [],
   "source": [
    "warm_vars = make_float(all_vars[(all_vars['Splice Project Gene Priority']=='Warm (MASTER-ACMG)')&\n",
    "                                (~all_vars['HUGO_Symbol'].isin(gus))&\n",
    "                                (~all_vars['HGVSg'].isin(new_vars['HGVSg']))].copy())\n",
    "warm_splai = warm_vars[(warm_vars['SpliceAI_pred_DS_AG']>0.4)|(warm_vars['SpliceAI_pred_DS_AL']>0.4)|\n",
    "                       (warm_vars['SpliceAI_pred_DS_DG']>0.4)|(warm_vars['SpliceAI_pred_DS_DL']>0.4)].copy()\n",
    "warm_dbsc = warm_vars[(warm_vars['ada_score']>0.6)&(warm_vars['rf_score']>0.6)&\n",
    "                      (~warm_vars['HGVSg'].isin(warm_splai['HGVSg']))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "99ced857",
   "metadata": {},
   "outputs": [],
   "source": [
    "#xlsx_df(conf_xlsx(warm_splai), 'warm_splai_o04')\n",
    "#xlsx_df(conf_xlsx(warm_dbsc), 'warm_dbsc_o06')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "feae8194",
   "metadata": {},
   "source": [
    "### 11.08. new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d35d7e22",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_xlsx = pd.read_excel('01_xlsx_files/2021-08-11_variants_seen_no_dup_new.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ddfe7c91",
   "metadata": {},
   "outputs": [],
   "source": [
    "# duplicates for specific variants\n",
    "# 434 to 428 variants\n",
    "grpd_xlsx = df_xlsx.groupby(by=cols_cpra)\n",
    "grpd_lst = [(i,len(df)) for i,df in grpd_xlsx if len(df)>1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ed68fc93",
   "metadata": {},
   "outputs": [],
   "source": [
    "# to search in each group for best keep\n",
    "#grpd_xlsx.get_group(grpd_lst[5][0])[['#CHROM', 'POS', 'REF', 'ALT', 'INFO_IGV','too_low_coverage','splice_change', \n",
    "                                     #'PIDs', 'consensus_sequence_IGV_region_left', \n",
    "                                     #'consensus_sequence_IGV_sequence_left','consensus_sequence_IGV_region_right',\n",
    "                                     #'consensus_sequence_IGV_sequence_right']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9529914d",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_xlsx = df_xlsx.drop(index=[401, 403, 387, 409, 430, 399])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c1d2a705",
   "metadata": {},
   "outputs": [],
   "source": [
    "vars_df = pd.merge(df_xlsx[['#CHROM', 'POS', 'REF', 'ALT', 'consensus_sequence_IGV_region_left',\n",
    "                            'consensus_sequence_IGV_sequence_left', 'consensus_sequence_IGV_region_right',\n",
    "                            'consensus_sequence_IGV_sequence_right', 'INFO_IGV', 'too_low_coverage',\n",
    "                            'splice_change']], all_vars, on=cols_cpra, how='left').reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ef7fc6e9",
   "metadata": {},
   "outputs": [],
   "source": [
    "vars_df['shortcut'] = vars_df['#CHROM']+'_'+vars_df['POS'].astype(str)+'_'+vars_df['REF']+'_'+vars_df['ALT']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3f329e5f",
   "metadata": {},
   "outputs": [],
   "source": [
    "#xlsx_df(vars_df, '2021-08-11_all_vars_analyzed')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a9ac9b3c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# filter out cold and gus variants\n",
    "rel_vars = vars_df[(vars_df['Splice Project Gene Priority']!='Cold (387-ACMG-MASTER)')&\n",
    "                   (~vars_df['HUGO_Symbol'].isin(gus))].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d2cb7664",
   "metadata": {},
   "outputs": [],
   "source": [
    "len(all_vars[(all_vars['Splice Project Gene Priority']!='Cold (387-ACMG-MASTER)')&\n",
    "             (~all_vars['HUGO_Symbol'].isin(gus))]['HUGO_Symbol'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "73cb5beb",
   "metadata": {},
   "outputs": [],
   "source": [
    "rel_vars[['#CHROM', 'POS', 'ALT', 'REF', 'PID', 'ClinVar_Pathogenicity', 'HGVSc', 'PosExonRefSeqAccession', 'PosExon_type', \n",
    "         'HUGO_Symbol', 'shortcut']][rel_vars['splice_change']=='yes'].drop_duplicates(cols_cpra)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7c870126",
   "metadata": {},
   "outputs": [],
   "source": [
    "vars_df[['#CHROM', 'POS', 'ALT', 'REF', 'PID', 'ClinVar_Pathogenicity', 'HGVSc', 'PosExonRefSeqAccession', 'PosExon_type', \n",
    "         'HUGO_Symbol', 'shortcut']][(vars_df['Splice Project Gene Priority']=='Hot (ACMG / MASTER)')&(vars_df['splice_change']=='yes')]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "635b9edf",
   "metadata": {},
   "source": [
    "### 11.08.2021"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e96d6cf9",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_xlsx = pd.read_excel('01_xlsx_files/2021-08-11_variants_seen_no_dup.xlsx')\n",
    "all_vars = pd.read_pickle('00_dataframes/all_vars')\n",
    "\n",
    "all_vars['#CHROM'] = pd.Categorical(all_vars['#CHROM'],\n",
    "                                    categories=['1','2','3','4','5','6','7','8','9','10','11','12', '13', '14', \n",
    "                                                '15', '16', '17', '18', '19', '20', '21', '22', 'X', 'Y'],\n",
    "                                    ordered=True)\n",
    "all_vars = all_vars.sort_values(by=['#CHROM', 'POS', 'REF', 'ALT'], ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "282ebbc8",
   "metadata": {},
   "outputs": [],
   "source": [
    "vars_df = pd.merge(df_xlsx[['#CHROM', 'POS', 'REF', 'ALT', 'consensus_sequence_IGV_region_left',\n",
    "                            'consensus_sequence_IGV_sequence_left', 'consensus_sequence_IGV_region_right',\n",
    "                            'consensus_sequence_IGV_sequence_right', 'INFO_IGV', 'too_low_coverage',\n",
    "                            'splice_change', 'shortcut']], all_vars, on=cols_cpra, how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f499f80d",
   "metadata": {},
   "outputs": [],
   "source": [
    "hot_vars = make_float(all_vars[(all_vars['Splice Project Gene Priority']=='Hot (ACMG / MASTER)')&\n",
    "                               (~all_vars['HGVSg'].isin(vars_df['HGVSg']))&\n",
    "                               (all_vars['ada_score'].notnull())].copy())\n",
    "hot_vars[['ada_score', 'rf_score']] = hot_vars[['ada_score', 'rf_score']].astype(float)\n",
    "hot_ada_rf = hot_vars[(hot_vars['ada_score']>0.1)&(hot_vars['rf_score']>0.1)].copy()\n",
    "\n",
    "for i in ['consensus_sequence_IGV_region_left', 'consensus_sequence_IGV_sequence_left',  \n",
    "          'consensus_sequence_IGV_region_right', 'consensus_sequence_IGV_sequence_right', \n",
    "          'INFO_IGV', 'too_low_coverage', 'splice_change', 'shortcut']:\n",
    "    hot_ada_rf[i] = np.nan"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c8e335e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "#xlsx_df(hot_ada_rf, 'hot_genes_adarf_o0')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "49f7d885",
   "metadata": {},
   "outputs": [],
   "source": [
    "hot_pos = hot_vars[(hot_vars['PosExonRefSeqAccession']<6)&(~hot_vars['HGVSg'].isin(hot_ada_rf['HGVSg']))].copy()\n",
    "for i in ['consensus_sequence_IGV_region_left', 'consensus_sequence_IGV_sequence_left',  \n",
    "          'consensus_sequence_IGV_region_right', 'consensus_sequence_IGV_sequence_right', \n",
    "          'INFO_IGV', 'too_low_coverage', 'splice_change', 'shortcut']:\n",
    "    hot_pos[i] = np.nan"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f1e67e70",
   "metadata": {},
   "outputs": [],
   "source": [
    "#xlsx_df(hot_pos, 'hot_genes_pos_l6')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "985912f3",
   "metadata": {},
   "source": [
    "### 10.08.2021"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "76a2ffc4",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_xlsx = pd.read_excel('01_xlsx_files/2021-08-10_variants_seen_no_dup.xlsx')\n",
    "all_vars = pd.read_pickle('00_dataframes/all_vars')\n",
    "\n",
    "all_vars['#CHROM'] = pd.Categorical(all_vars['#CHROM'],\n",
    "                                    categories=['1','2','3','4','5','6','7','8','9','10','11','12', '13', '14', \n",
    "                                                '15', '16', '17', '18', '19', '20', '21', '22', 'X', 'Y'],\n",
    "                                    ordered=True)\n",
    "all_vars = all_vars.sort_values(by=['#CHROM', 'POS', 'REF', 'ALT'], ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "11627199",
   "metadata": {},
   "outputs": [],
   "source": [
    "vars_df = pd.merge(df_xlsx[['#CHROM', 'POS', 'REF', 'ALT', 'consensus_sequence_IGV_region_left',\n",
    "                            'consensus_sequence_IGV_sequence_left', 'consensus_sequence_IGV_region_right',\n",
    "                            'consensus_sequence_IGV_sequence_right', 'INFO_IGV', 'too_low_coverage',\n",
    "                            'splice_change', 'shortcut']], all_vars, on=cols_cpra, how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ddec686c",
   "metadata": {},
   "outputs": [],
   "source": [
    "hot_vars = make_float(all_vars[(all_vars['Splice Project Gene Priority']=='Hot (ACMG / MASTER)')&\n",
    "                               (~all_vars['HGVSg'].isin(vars_df['HGVSg']))].copy())\n",
    "hot_splai = hot_vars[(hot_vars['SpliceAI_pred_DS_AG']>0.1)|(hot_vars['SpliceAI_pred_DS_AL']>0.1)|\n",
    "                     (hot_vars['SpliceAI_pred_DS_DG']>0.1)|(hot_vars['SpliceAI_pred_DS_DL']>0.1)].copy()\n",
    "\n",
    "for i in ['consensus_sequence_IGV_region_left', 'consensus_sequence_IGV_sequence_left',  \n",
    "          'consensus_sequence_IGV_region_right', 'consensus_sequence_IGV_sequence_right', \n",
    "          'INFO_IGV', 'too_low_coverage', 'splice_change', 'shortcut']:\n",
    "    hot_splai[i] = np.nan"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "73e87f13",
   "metadata": {},
   "outputs": [],
   "source": [
    "#xlsx_df(hot_splai, 'hot_genes_splai_o0')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0271073c",
   "metadata": {},
   "source": [
    "### 09.08.2021"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fc6316d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_xlsx = pd.read_excel('01_xlsx_files/2021-08-07_variants_seen_no_dup.xlsx')\n",
    "all_vars = pd.read_pickle('00_dataframes/all_vars')\n",
    "red_xlsx = pd.read_excel('01_xlsx_files/2021-08-09_redund_vars.xlsx')\n",
    "\n",
    "all_vars['#CHROM'] = pd.Categorical(all_vars['#CHROM'],\n",
    "                                    categories=['1','2','3','4','5','6','7','8','9','10','11','12', '13', '14', \n",
    "                                                '15', '16', '17', '18', '19', '20', '21', '22', 'X', 'Y'],\n",
    "                                    ordered=True)\n",
    "all_vars = all_vars.sort_values(by=['#CHROM', 'POS', 'REF', 'ALT'], ignore_index=True)\n",
    "\n",
    "\n",
    "# no duplicates in df_xlsx --> merge with all_vars\n",
    "\n",
    "vars_df = pd.merge(df_xlsx[['#CHROM', 'POS', 'REF', 'ALT', 'consensus_sequence_IGV_region_left',\n",
    "                            'consensus_sequence_IGV_sequence_left', 'consensus_sequence_IGV_region_right',\n",
    "                            'consensus_sequence_IGV_sequence_right', 'INFO_IGV', 'too_low_coverage',\n",
    "                            'splice_change', 'shortcut']], all_vars, on=cols_cpra, how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cd5d645d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# for variants in more than one patient\n",
    "grpd_all_vars = vars_df.groupby(cols_cpra)\n",
    "grpd_lst = [(i,len(df)) for i,df in grpd_all_vars if len(df)>1]\n",
    "\n",
    "redund_df = pd.DataFrame()\n",
    "for i, l in grpd_lst:\n",
    "    redund_df = redund_df.append(grpd_all_vars.get_group(i))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "22447020",
   "metadata": {},
   "outputs": [],
   "source": [
    "#xlsx_df(redund_df, '2021-08-09_redund_vars')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "581cb8f9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# combine new characterized redundant variants with other variants\n",
    "new_df_xlsx = df_xlsx[~df_xlsx['shortcut'].isin(redund_df['shortcut'])]\n",
    "new_df_xlsx = new_df_xlsx.append(red_xlsx)\n",
    "new_df_xlsx = new_df_xlsx.reset_index(drop=True)\n",
    "new_vars_df = pd.merge(new_df_xlsx[['#CHROM', 'POS', 'REF', 'ALT', 'consensus_sequence_IGV_region_left',\n",
    "                            'consensus_sequence_IGV_sequence_left', 'consensus_sequence_IGV_region_right',\n",
    "                            'consensus_sequence_IGV_sequence_right', 'INFO_IGV', 'too_low_coverage',\n",
    "                            'splice_change', 'shortcut']], all_vars, on=cols_cpra, how='left')\n",
    "new_onl_vars = new_vars_df.drop_duplicates(cols_cpra).reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d3139753",
   "metadata": {},
   "source": [
    "#### IGV splice effect seen"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "20185ab8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# splice effect seen, safe (not too low coverage, only if sure (splice_change == yes)) --> 88 variants\n",
    "# if splice_change == maybe --> 38 variants, if splice_change == no --> 136 variants\n",
    "# if too low coverage and splice_change == no --> 75 variants, splice_change == maybe --> 27 variants\n",
    "\n",
    "#len(new_onl_vars[(new_onl_vars['too_low_coverage']=='yes')&\n",
    "                 #(new_onl_vars['splice_change']=='yes')].reset_index(drop=True))\n",
    "\n",
    "new_spl_yes = new_onl_vars[(new_onl_vars['too_low_coverage']=='no')&\n",
    "                           (new_onl_vars['splice_change']=='yes')].reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "edf783b4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# 106 variants with spliceAI score > 0.9\n",
    "# 47 variants with significant effect in splicing, 18 variants no effect, 9 maybe effect\n",
    "# 32 too low coverage, 19 no effect, 13 maybe effect\n",
    "for_splai = make_float(new_onl_vars[new_onl_vars['SpliceAI_pred_DS_AG'].notnull()].copy())\n",
    "len(for_splai[((for_splai['SpliceAI_pred_DS_AG']>0.9)|(for_splai['SpliceAI_pred_DS_AL']>0.9)|\n",
    "               (for_splai['SpliceAI_pred_DS_DG']>0.9)|(for_splai['SpliceAI_pred_DS_DL']>0.9))&\n",
    "              (for_splai['too_low_coverage']=='yes')&(for_splai['splice_change']=='no')\n",
    "             ].groupby(by=cols_cpra))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9028883a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# 63 variants previously of uncertain significance or not in ClinVar with splice change and enough coverage\n",
    "for_splai[['#CHROM', 'POS', 'REF', 'ALT', 'consensus_sequence_IGV_region_left', \n",
    "           'consensus_sequence_IGV_sequence_left', 'consensus_sequence_IGV_region_right',\n",
    "           'consensus_sequence_IGV_sequence_right', 'INFO_IGV', 'too_low_coverage', 'splice_change', \n",
    "           'shortcut', 'HGVSc', 'HGVSg', 'HGVSp']][((for_splai['ClinVar_Pathogenicity']=='Uncertain Significance')|\n",
    "                         (for_splai['ClinVar_Pathogenicity']=='uncertain_significance')|\n",
    "                         (for_splai['ClinVar_Pathogenicity']=='Uncertain significance')|\n",
    "                         (for_splai['ClinVar_Pathogenicity'].isnull()))&\n",
    "                        (for_splai['too_low_coverage']=='no')&\n",
    "                        (for_splai['splice_change']=='yes')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e2157439",
   "metadata": {},
   "outputs": [],
   "source": [
    "for_MES = make_float(vars_df[vars_df['MaxEntScan_diff']].notnull().copy())\n",
    "for_MES_disr = for_MES[(for_MES['MaxEntScan_diff']>=0)&(for_MES['MaxEntScan_alt']<6.2)]\n",
    "for_MES_new = for_MES[(['MaxEntScan_diff']<0)&(['MaxEntScan_alt']>8.5)]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dd68e0f3",
   "metadata": {},
   "source": [
    "#### Variants per prioritization criterion"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b2f8c002",
   "metadata": {},
   "outputs": [],
   "source": [
    "#vars_grpd_splai = vars_grpd[vars_grpd['SpliceAI_pred_DP_AG'].notnull()].copy()\n",
    "#vars_grpd_splai = make_float(vars_grpd_splai)\n",
    "\n",
    "#vars_grpd_splai = vars_grpd_splai[cols_cpra][((vars_grpd_splai['SpliceAI_pred_DS_AG']>0.9)|\n",
    "                                              #(vars_grpd_splai['SpliceAI_pred_DS_AL']>0.9)|\n",
    "                                              #(vars_grpd_splai['SpliceAI_pred_DS_DG']>0.9)|\n",
    "                                              #(vars_grpd_splai['SpliceAI_pred_DS_DL']>0.9))\n",
    "                                    #].drop_duplicates()\n",
    "#vars_grpd_splai.reset_index(drop=True, inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "662072e8",
   "metadata": {},
   "source": [
    "## Input"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "01478a27",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_xlsx = pd.read_excel('01_xlsx_files/2021-08-05_all_variants.xlsx')\n",
    "all_vars = pd.read_pickle('00_dataframes/all_vars')\n",
    "\n",
    "all_vars['#CHROM'] = pd.Categorical(all_vars['#CHROM'],\n",
    "                                    categories=['1','2','3','4','5','6','7','8','9','10','11','12', '13', '14', \n",
    "                                                '15', '16', '17', '18', '19', '20', '21', '22', 'X', 'Y'],\n",
    "                                    ordered=True)\n",
    "all_vars = all_vars.sort_values(by=['#CHROM', 'POS', 'REF', 'ALT'], ignore_index=True)\n",
    "\n",
    "cols_cpra = ['#CHROM', 'POS', 'REF', 'ALT']\n",
    "vars_grpd = all_vars.drop_duplicates(subset=cols_cpra)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7892ffd5",
   "metadata": {},
   "source": [
    "## Drop duplicates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f3170a9b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# duplicates for specific variants\n",
    "#len(df_xlsx), len(df_xlsx.drop_duplicates(subset=cols_cpra)) #(328, 272)\n",
    "grpd_xlsx = df_xlsx.groupby(by=cols_cpra)\n",
    "grpd_lst = [(i,len(df)) for i,df in grpd_xlsx if len(df)>1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b4432b15",
   "metadata": {},
   "outputs": [],
   "source": [
    "# to search in each group for best keep\n",
    "# grpd_xlsx.get_group(grpd_lst[42][0])[['#CHROM', 'POS', 'REF', 'ALT', 'INFO_IGV','too_low_coverage','splice_change', \n",
    "                                     #'PIDs', 'consensus_sequence_IGV_region_left', \n",
    "                                     #'consensus_sequence_IGV_sequence_left','consensus_sequence_IGV_region_right',\n",
    "                                     #'consensus_sequence_IGV_sequence_right']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "428256e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "for_drop = [7, 17, 9, 36, 78, 74, 90, 91, 89, 101, 104, 102, 105, 113, 122, 148, 152, 149, 150, 155, 157, 154, \n",
    "            166, 165, 183, 177, 174, 185, 202, 220, 211, 215, 205, 207, 229, 239, 235, 241, 250, 265, 267, 264, \n",
    "            263, 268, 271, 280, 284, 287, 289, 292, 304, 314, 316, 313, 325, 327]\n",
    "df_xlsx.drop(index=for_drop, inplace=True)\n",
    "df_xlsx.reset_index(drop=True, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e847adaf",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_xlsx['shortcut'] = (df_xlsx['#CHROM'].astype(str) + '_' + df_xlsx['POS'].astype(str) + '_' + \n",
    "                       df_xlsx['REF'].astype(str) + '_' + df_xlsx['ALT'].astype(str))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "672891b8",
   "metadata": {},
   "outputs": [],
   "source": [
    "vars_df = pd.merge(df_xlsx[['#CHROM', 'POS', 'REF', 'ALT', 'consensus_sequence_IGV_region_left',\n",
    "                               'consensus_sequence_IGV_sequence_left', 'consensus_sequence_IGV_region_right',\n",
    "                               'consensus_sequence_IGV_sequence_right', 'INFO_IGV', 'too_low_coverage',\n",
    "                               'splice_change', 'shortcut']], all_vars, on=cols_cpra, how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "37784c53",
   "metadata": {},
   "outputs": [],
   "source": [
    "#xlsx_df(vars_df, 'variants_seen_no_dup')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3ef32aa7",
   "metadata": {},
   "source": [
    "## New excel files without duplicates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "58f19b98",
   "metadata": {},
   "outputs": [],
   "source": [
    "splAI_high = vars_df[vars_df['SpliceAI_pred_DS_AG'].notnull()].copy()\n",
    "splAI_high = make_float(splAI_high)\n",
    "splAI_high = splAI_high[((splAI_high['SpliceAI_pred_DS_AG']>0.7)|\n",
    "                         (splAI_high['SpliceAI_pred_DS_AL']>0.7)|\n",
    "                         (splAI_high['SpliceAI_pred_DS_DG']>0.7)|\n",
    "                         (splAI_high['SpliceAI_pred_DS_DL']>0.7))].drop_duplicates()\n",
    "\n",
    "for i in ['consensus_sequence_IGV_region_left', 'consensus_sequence_IGV_sequence_left', 'too_low_coverage',\n",
    "          'consensus_sequence_IGV_region_right', 'consensus_sequence_IGV_sequence_right', 'splice_change',\n",
    "          'INFO_IGV']:\n",
    "    splAI_high[i] = np.nan\n",
    "\n",
    "xlsx_df(splAI_high, 'new_splAI_high')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bdd95d82",
   "metadata": {},
   "outputs": [],
   "source": [
    "MES_high_disr = pd.read_excel('../04_annotations/03_prioritized_xlsx/new_MES_high_disr.xlsx')\n",
    "MES_high_disr['from_where'] = 'MES_high_disr'\n",
    "MES_high_new = pd.read_excel('../04_annotations/03_prioritized_xlsx/new_MES_high_new.xlsx')\n",
    "MES_high_new['from_where'] = 'MES_high_new'\n",
    "dbsc_high = pd.read_excel('../04_annotations/03_prioritized_xlsx/new_dbsc_high.xlsx')\n",
    "dbsc_high['from_where'] = 'dbsc_high'\n",
    "splAI_high_new = pd.read_excel('01_xlsx_files/new_splAI_high.xlsx')\n",
    "splAI_high_new['from_where'] = 'splAI_high'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "189a3d20",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_for_new = MES_high_disr.append(MES_high_new, ignore_index=True)\n",
    "all_for_new = all_for_new.append(dbsc_high, ignore_index=True)\n",
    "all_for_new = all_for_new.append(splAI_high_new, ignore_index=True).drop_duplicates(subset=cols_cpra, \n",
    "                                                                                    ignore_index=True)\n",
    "all_for_new['shortcut'] = (all_for_new['#CHROM'].astype(str) + '_' + all_for_new['POS'].astype(str) + '_' + \n",
    "                           all_for_new['REF'].astype(str) + '_' + all_for_new['ALT'].astype(str))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0938a91b",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_for_new = all_for_new[~all_for_new['shortcut'].isin(vars_df['shortcut'])].reset_index(drop=True)\n",
    "all_for_new.to_excel('01_xlsx_files/2021-08-06_rest_vars_nodup.xlsx')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "76198c98",
   "metadata": {},
   "source": [
    "## IGV splice effect seen"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e21ee0d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# splice effect seen, safe (not too low coverage, only if sure (splice_change = yes)) --> 80 variants\n",
    "spl_yes = vars_df[(vars_df['too_low_coverage']=='no')&(vars_df['splice_change']=='yes')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1c165309",
   "metadata": {},
   "outputs": [],
   "source": [
    "spl_yes_splai = spl_yes_df[spl_yes_df['SpliceAI_pred_DS_AG'].notnull()].copy()\n",
    "make_float(spl_yes_splai)\n",
    "spl_yes_splai_high = spl_yes_splai[((spl_yes_splai['SpliceAI_pred_DS_AG']>0.9)|\n",
    "            (spl_yes_splai['SpliceAI_pred_DS_AL']>0.9)|\n",
    "            (spl_yes_splai['SpliceAI_pred_DS_DG']>0.9)|\n",
    "            (spl_yes_splai['SpliceAI_pred_DS_DL']>0.9))].copy()\n",
    "len(spl_yes_splai), len(spl_yes_df), len(spl_yes_splai_high), len(spl_yes_splai_high[spl_yes_splai_high['Splice Project Gene Priority']=='Hot (ACMG / MASTER)'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2b278789",
   "metadata": {},
   "outputs": [],
   "source": [
    "# 106 variants with spliceAI score > 0.9\n",
    "# 46 variants with significant effect in splicing, 17 variants no effect, 33 too low coverage, 10 maybe effect\n",
    "for_splai = make_float(vars_df[vars_df['SpliceAI_pred_DS_AG'].notnull()].copy())\n",
    "len(for_splai[((for_splai['SpliceAI_pred_DS_AG']>0.9)|(for_splai['SpliceAI_pred_DS_AL']>0.9)|\n",
    "               (for_splai['SpliceAI_pred_DS_DG']>0.9)|(for_splai['SpliceAI_pred_DS_DL']>0.9))&\n",
    "              (for_splai['too_low_coverage']=='no')&(for_splai['splice_change']=='maybe')].groupby(by=cols_cpra))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "59aba0cc",
   "metadata": {},
   "outputs": [],
   "source": [
    "for_MES = make_float(vars_df[vars_df['MaxEntScan_diff']].notnull().copy())\n",
    "for_MES_disr = for_MES[(for_MES['MaxEntScan_diff']>=0)&(for_MES['MaxEntScan_alt']<6.2)]\n",
    "for_MES_new = for_MES[(['MaxEntScan_diff']<0)&(['MaxEntScan_alt']>8.5)]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "81d0df0e",
   "metadata": {},
   "source": [
    "## Variants per prioritization criterion"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "67a27d2f",
   "metadata": {},
   "outputs": [],
   "source": [
    "#vars_grpd_splai = vars_grpd[vars_grpd['SpliceAI_pred_DP_AG'].notnull()].copy()\n",
    "#vars_grpd_splai = make_float(vars_grpd_splai)\n",
    "\n",
    "#vars_grpd_splai = vars_grpd_splai[cols_cpra][((vars_grpd_splai['SpliceAI_pred_DS_AG']>0.9)|\n",
    "                                              #(vars_grpd_splai['SpliceAI_pred_DS_AL']>0.9)|\n",
    "                                              #(vars_grpd_splai['SpliceAI_pred_DS_DG']>0.9)|\n",
    "                                              #(vars_grpd_splai['SpliceAI_pred_DS_DL']>0.9))\n",
    "                                    #].drop_duplicates()\n",
    "#vars_grpd_splai.reset_index(drop=True, inplace=True)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
