{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c429a05e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "from gtfparse import read_gtf\n",
    "import pysam as ps\n",
    "from Bio.Seq import Seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0fe86ed4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# which reference transcript\n",
    "hgnc = pd.read_csv('reference_transcripts.txt', sep='\\t')\n",
    "hgnc = hgnc.drop_duplicates(subset=['Approved symbol'])\n",
    "replaceDict = {'NM_003220':'NM_001042425','NM_004241':'NM_032776','NM_022976':'NM_001144914'}\n",
    "for _ in replaceDict:\n",
    "    i = hgnc[hgnc['RefSeq accession']==_].index.values[0]\n",
    "    hgnc.at[i,'RefSeq accession'] = replaceDict[_]\n",
    "\n",
    "# whole hg19 (GRCh37) reference sequence\n",
    "ref_seq = ps.FastaFile('reference_GRCh37.fa.gz')\n",
    "\n",
    "# table with variants and positions\n",
    "variants = pd.read_excel('variants_positions.xlsx')\n",
    "\n",
    "variants = pd.merge(variants, hgnc[['Approved symbol','RefSeq accession']], left_on='HUGO_Symbol', \n",
    "                    right_on='Approved symbol', how='left')\n",
    "variants = variants.drop(columns='Approved symbol')\n",
    "variants[['CHR', 'POS', 'REF', 'ALT', 'nr']] = variants['shortcut'].str.split('_', expand=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f922c368",
   "metadata": {},
   "outputs": [],
   "source": [
    "# GTF file with sequence information\n",
    "gtf_df = read_gtf('GRCh37_sequence_information.gtf')\n",
    "# only those reference transcripts within variant file\n",
    "gtf_df['transcript_id'] = gtf_df['transcript_id'].str.split('.',expand=True)[0]\n",
    "gtf_df = gtf_df[gtf_df['transcript_id'].isin(set(variants['RefSeq accession']))]\n",
    "gtf_df=gtf_df[gtf_df['feature']=='CDS']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ee30e04b",
   "metadata": {},
   "outputs": [],
   "source": [
    "def search_col(df, inp):\n",
    "    return [i for i in df.columns if inp in i.lower()]\n",
    "\n",
    "def rev_comp(seq):\n",
    "    return str(Seq(seq).reverse_complement())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c5c5aa5b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# new cDNA/protein sequence with indels and variants\n",
    "def cDNA(ind):\n",
    "    cdna = ''\n",
    "    ind_df = gtf_df[gtf_df['transcript_id']==variants.iloc[ind]['RefSeq accession']]\n",
    "    kind = variants.iloc[ind]['kind']\n",
    "    chrom = str(variants.iloc[ind]['#CHROM'])\n",
    "    var_pos = int(variants.iloc[ind]['POS'])\n",
    "    ref_len = len(variants.iloc[ind]['REF'])-1\n",
    "    pos_last = int(variants.iloc[ind]['pos_last'])\n",
    "    pos_next = int(variants.iloc[ind]['pos_next'])\n",
    "    ori = list(ind_df['strand'])[0]\n",
    "    aa_pos = 0\n",
    "    hgsv_pos = 0\n",
    "    \n",
    "    # exclude duplicated sequences\n",
    "    new_pos = 0\n",
    "    \n",
    "    # for new HGVSg in VCF\n",
    "    ref_n = ''\n",
    "    alt_n = ''\n",
    "    \n",
    "    for s,e in zip(ind_df['start'], ind_df['end']):\n",
    "        # for insertion (new pos (start/end of new exon) in intron)\n",
    "        if new_pos == 0:\n",
    "            if pos_last == e:\n",
    "                e = pos_next-1\n",
    "                new_pos +=1\n",
    "                ref_n = ref_seq.fetch(chrom, pos_last-1, pos_last)\n",
    "                alt_n = ref_seq.fetch(chrom, pos_last-1, pos_next-1)\n",
    "                hgsv_pos = pos_last\n",
    "            elif pos_next == s:\n",
    "                s = pos_last+1\n",
    "                new_pos +=1\n",
    "                ref_n = ref_seq.fetch(chrom, pos_next-2, pos_next-1)\n",
    "                alt_n = ref_n + ref_seq.fetch(chrom, pos_last, pos_next-1)\n",
    "                hgsv_pos = pos_next-1\n",
    "            # for deletion (new pos (start/end of new exon) in exon)\n",
    "            elif pos_next-1 == e:\n",
    "                e = pos_last\n",
    "                new_pos +=1\n",
    "                ref_n = ref_seq.fetch(chrom, pos_last-1, pos_next-1)\n",
    "                alt_n = ref_seq.fetch(chrom, pos_last-1, pos_last)\n",
    "                hgsv_pos = pos_last\n",
    "            elif pos_last+1 == s:\n",
    "                s = pos_next\n",
    "                new_pos +=1\n",
    "                ref_n = ref_seq.fetch(chrom, pos_last-1, pos_next-1)\n",
    "                alt_n = ref_seq.fetch(chrom, pos_last-1, pos_last)\n",
    "                hgsv_pos = pos_last\n",
    "            else:\n",
    "                pass\n",
    "        else:\n",
    "            pass\n",
    "        \n",
    "        # e.g. in exon skipping events\n",
    "        if s+1 > e:\n",
    "            cdna += ''\n",
    "        \n",
    "        # if original variant is in new cDNA\n",
    "        elif s <= var_pos and e >= var_pos:\n",
    "            # for deletions starting in exon\n",
    "            if ref_len>0:\n",
    "                if ori == '+':\n",
    "                    if var_pos+ref_len < e:\n",
    "                        cdna += ref_seq.fetch(chrom, s-1, var_pos)\n",
    "                        cdna += ref_seq.fetch(chrom, var_pos+ref_len, e)\n",
    "                    else: # var_pos+ref_len >= e\n",
    "                        cdna += ref_seq.fetch(chrom, s-1, var_pos)\n",
    "                elif ori == '-':\n",
    "                    if var_pos+ref_len < e:\n",
    "                        cdna += rev_comp(ref_seq.fetch(chrom, var_pos+ref_len, e))\n",
    "                        cdna += rev_comp(ref_seq.fetch(chrom, s-1, var_pos))\n",
    "                    else: # var_pos+ref_len >= e\n",
    "                        cdna += rev_comp(ref_seq.fetch(chrom, s-1, var_pos))\n",
    "                else:\n",
    "                    pass\n",
    "\n",
    "            # for insertions, substitutions\n",
    "            else:\n",
    "                if ori == '+':\n",
    "                    cdna += ref_seq.fetch(chrom, s-1, var_pos-1)\n",
    "                    cdna += variants.iloc[ind]['ALT']\n",
    "                    cdna += ref_seq.fetch(chrom, var_pos, e)\n",
    "                elif ori == '-':\n",
    "                    cdna += rev_comp(ref_seq.fetch(chrom, var_pos, e))\n",
    "                    cdna += rev_comp(variants.iloc[ind]['ALT'])\n",
    "                    cdna += rev_comp(ref_seq.fetch(chrom, s-1, var_pos-1))\n",
    "                else:\n",
    "                    pass\n",
    "    \n",
    "        # for deletions starting in intron\n",
    "        elif ref_len>0 and s <= var_pos+ref_len and e >= var_pos+ref_len:\n",
    "            if ori == '+':\n",
    "                cdna += ref_seq.fetch(chrom, var_pos+ref_len, e)\n",
    "            elif ori == '-':\n",
    "                cdna += rev_comp(ref_seq.fetch(chrom, var_pos+ref_len, e))\n",
    "            else:\n",
    "                pass\n",
    "        \n",
    "        elif ori == '+':\n",
    "            cdna += ref_seq.fetch(chrom, s-1, e)\n",
    "        elif ori == '-':\n",
    "            cdna += rev_comp(ref_seq.fetch(chrom, s-1, e))\n",
    "        else:\n",
    "            pass\n",
    "    return [cdna, hgsv_pos, ref_n, alt_n]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a755b834",
   "metadata": {},
   "outputs": [],
   "source": [
    "# for reference cDNA/protein sequence for comparison\n",
    "def ref_cdna(ind):\n",
    "    cdna = ''\n",
    "    chrom = str(variants.iloc[ind]['#CHROM'])\n",
    "    ind_df = gtf_df[gtf_df['transcript_id']==variants.iloc[ind]['RefSeq accession']]\n",
    "    ori = list(ind_df['strand'])[0]\n",
    "    for s,e in zip(ind_df['start'], ind_df['end']):\n",
    "        if ori == '+':\n",
    "            cdna += ref_seq.fetch(chrom, s-1, e)\n",
    "        elif ori == '-':\n",
    "            cdna += rev_comp(ref_seq.fetch(chrom, s-1, e))\n",
    "        else:\n",
    "            pass\n",
    "    return cdna\n",
    "\n",
    "# position of change in nucleic sequence\n",
    "def nuc_pos(ind):\n",
    "    alt_nuc = variants.iloc[ind]['cDNA']\n",
    "    ref_nuc = variants.iloc[ind]['ref_cDNA']\n",
    "    ct = 0\n",
    "    nuc_pos = 0\n",
    "    for a,r in zip(alt_nuc, ref_nuc[:len(alt_nuc)]):\n",
    "        ct+=1\n",
    "        if a != r:\n",
    "            nuc_pos = ct\n",
    "            break\n",
    "        else:\n",
    "            continue\n",
    "    if nuc_pos == 0 and len(alt_nuc)<len(ref_nuc):\n",
    "        nuc_pos = len(alt_nuc)+1\n",
    "    elif nuc_pos == 0 and len(alt_nuc)==len(ref_nuc):\n",
    "        nuc_pos = 'no_change'\n",
    "    return nuc_pos\n",
    "\n",
    "# position of change in aa sequence\n",
    "def prot_pos(ind):\n",
    "    alt_prot = variants.iloc[ind]['termin_prot']\n",
    "    ref_prot = variants.iloc[ind]['ref_protein_seq']\n",
    "    ct = 0\n",
    "    aa_pos = 0\n",
    "    for a,r in zip(alt_prot, ref_prot[:len(alt_prot)]):\n",
    "        ct+=1\n",
    "        if a != r:\n",
    "            aa_pos = ct\n",
    "            break\n",
    "        else:\n",
    "            continue\n",
    "    if aa_pos == 0 and len(alt_prot)<len(ref_prot):\n",
    "        aa_pos = len(alt_prot)+1\n",
    "    elif aa_pos == 0 and len(alt_prot)==len(ref_prot):\n",
    "        aa_pos = 'no_change'\n",
    "    return aa_pos\n",
    "\n",
    "# length of exons\n",
    "def exon_len(ind):\n",
    "    ind_df = gtf_df[gtf_df['transcript_id']==variants.iloc[ind]['RefSeq accession']]\n",
    "    exon_lst = []\n",
    "    bott = 0\n",
    "    for s,e,n in zip(ind_df['start'], ind_df['end'], ind_df['exon_number']):\n",
    "        exon_lst.append((n, bott+1, bott+abs(s-e)+1))\n",
    "        bott += abs(s-e)+1\n",
    "    return exon_lst, n\n",
    "\n",
    "# in which exon change occurs\n",
    "def changed_exon(ind):\n",
    "    exon = 0\n",
    "    nuc_change_pos = variants.iloc[ind]['nuc_change_pos']\n",
    "    if type(nuc_change_pos) == int:\n",
    "        for ex,s,e in variants.iloc[ind]['exons_lengths']:\n",
    "            if nuc_change_pos >= s and nuc_change_pos <= e:\n",
    "                exon = ex\n",
    "            else:\n",
    "                pass\n",
    "    return exon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d0951b98",
   "metadata": {},
   "outputs": [],
   "source": [
    "# find out next matching amino acids position\n",
    "import re\n",
    "def aas_next(ind):\n",
    "    change_pos = variants.iloc[ind]['prot_change_pos']\n",
    "    ct = 0\n",
    "    pos_next_normal = ''\n",
    "    len_indel = 0\n",
    "    \n",
    "    try:\n",
    "        if type(change_pos)==int:\n",
    "            alt1 = variants.iloc[ind]['termin_prot']\n",
    "            short_alt = alt1[change_pos-1:]\n",
    "            ref1 = variants.iloc[ind]['ref_protein_seq']\n",
    "            short_ref = ref1[change_pos-1:]\n",
    "            kind = variants.iloc[ind]['kind']\n",
    "\n",
    "            if kind=='ins':\n",
    "                if re.search(short_ref, short_alt) and len(alt1)>len(ref1):\n",
    "                    len_indel = re.search(short_ref, short_alt).regs[0][0]\n",
    "                    pos_next_normal = change_pos+len_indel\n",
    "                elif re.search(short_ref[1:], short_alt) and len(alt1)>len(ref1):\n",
    "                    len_indel = re.search(short_ref[1:], short_alt).regs[0][0]\n",
    "                    pos_next_normal = change_pos+len_indel\n",
    "                else:\n",
    "                    pos_next_normal = 'probably frameshift'\n",
    "            elif kind =='del':\n",
    "                if re.search(short_alt, short_ref) and len(ref1)>len(alt1):\n",
    "                    len_indel = re.search(short_alt, short_ref).regs[0][0]\n",
    "                    pos_next_normal = change_pos+len_indel\n",
    "                elif re.search(short_alt[1:], short_ref) and len(ref1)>len(alt1):\n",
    "                    len_indel = re.search(short_alt[1:], short_ref).regs[0][0]\n",
    "                    pos_next_normal = change_pos+len_indel\n",
    "                else:\n",
    "                    pos_next_normal = 'probably frameshift'\n",
    "            else:\n",
    "                pass\n",
    "        else:\n",
    "            pos_next_normal = 'no change'\n",
    "        if abs(len(alt1)-len(ref1)) > (len_indel+1):\n",
    "            pos_next_normal = 'probably frameshift'\n",
    "        else:\n",
    "            pass\n",
    "    except:\n",
    "        pos_next_normal = 'error'\n",
    "    return [pos_next_normal, len_indel]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9c5661b6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# new columns in variants df with cDNA, protein sequence\n",
    "variants[['cDNA', 'hgsv_pos', 'ref_n', 'alt_n']] = [cDNA(i) for i in variants.index]\n",
    "variants['protein_seq'] = [str(Seq(cDNA(i)[0]).translate()) for i in variants.index]\n",
    "# terminated protein sequence after first stop codon\n",
    "variants['termin_prot'] = variants['protein_seq'].str.split('*', expand=True)[0]\n",
    "\n",
    "# new columns in variants df with reference cDNA, protein sequence\n",
    "variants['ref_cDNA'] = [ref_cdna(i) for i in variants.index]\n",
    "variants['ref_protein_seq'] = [str(Seq(ref_cdna(i)).translate()) for i in variants.index]\n",
    "\n",
    "# length of sequences\n",
    "for i in ['cDNA', 'termin_prot', 'ref_cDNA', 'ref_protein_seq']:\n",
    "    variants[i+'_len'] = variants[i].str.len()\n",
    "\n",
    "# position of change in nucleic/aa sequence\n",
    "variants['nuc_change_pos'] = [nuc_pos(i) for i in variants.index]\n",
    "variants['prot_change_pos'] = [prot_pos(i) for i in variants.index]\n",
    "\n",
    "# position of next matching aa and length of indel\n",
    "variants[['aa_next_pos', 'len_indel_substit']] = [aas_next(i) for i in variants.index]\n",
    "\n",
    "# length of exons\n",
    "variants[['exons_lengths', 'max_exon']] = [exon_len(i) for i in variants.index]\n",
    "\n",
    "# changed exon\n",
    "variants['changed_exon'] = [changed_exon(i) for i in variants.index]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e18be458",
   "metadata": {},
   "source": [
    "#### Affected protein domains"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "77816574",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('prot_seqs_folding.txt', 'w') as o:\n",
    "    for sc, gene, alt, ref in zip(variants['shortcut'], variants['HUGO_Symbol'], \n",
    "                                  variants['termin_prot'], variants['ref_protein_seq']):\n",
    "        o.write('>%s\\n%s\\n>%s\\n%s\\n'%(sc,alt,gene,ref))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3d2085d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# import InterProScan TSVs as df-dictionary\n",
    "files = os.listdir('InterProScan_directory')\n",
    "ip_dfs = {}\n",
    "for file in files:\n",
    "    df = pd.read_csv(('InterProScan_directory/'+file), sep='\\t', \n",
    "                      names=['seq', 'ID', 'prot_len', 'where_data_from', 'domain_short', 'info_domain', 'start_pos',\n",
    "                             'end_pos', 'probab', 'sth', 'date', 'ipr_id', 'more_info', 'go_id', 'many'], \n",
    "                      usecols=['where_data_from', 'domain_short', 'info_domain', 'start_pos', 'end_pos',\n",
    "                                'ipr_id', 'more_info', 'go_id'])\n",
    "    ip_dfs[file.split('_')[1].split('.')[0].upper()] = df.sort_values(by=['start_pos', 'end_pos']\n",
    "                                                                     ).reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "566d35a6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# check, which domains are effected\n",
    "def check_domains(ind):\n",
    "    domains = []\n",
    "    gene = variants.iloc[ind]['HUGO_Symbol']\n",
    "    pos_prot_change = variants.iloc[ind]['prot_change_pos']\n",
    "    if gene in ip_dfs.keys() and type(pos_prot_change) == int:\n",
    "        ip_scan = ip_dfs[variants.iloc[ind]['HUGO_Symbol']]\n",
    "        for s,e,dom in zip(ip_scan['start_pos'], ip_scan['end_pos'], ip_scan['domain_short']):\n",
    "            if pos_prot_change >= s and pos_prot_change <= e:\n",
    "                domains.append(dom)\n",
    "    return domains\n",
    "\n",
    "# check, which domains are deleted\n",
    "def deleted_domains(ind):\n",
    "    domains = []\n",
    "    gene = variants.iloc[ind]['HUGO_Symbol']\n",
    "    pos_prot_change = variants.iloc[ind]['prot_change_pos']\n",
    "    termin_prot_len = variants.iloc[ind]['termin_prot_len']\n",
    "    if gene in ip_dfs.keys() and type(pos_prot_change) == int:\n",
    "        ip_scan = ip_dfs[variants.iloc[ind]['HUGO_Symbol']]\n",
    "        for s,e,dom in zip(ip_scan['start_pos'], ip_scan['end_pos'], ip_scan['domain_short']):\n",
    "            if termin_prot_len <= s:\n",
    "                domains.append(dom)\n",
    "    return domains\n",
    "\n",
    "# new columns with effected/deleted domains\n",
    "variants['domains_effected'] = [check_domains(i) for i in variants.index]\n",
    "variants['domains_deleted'] = [deleted_domains(i) for i in variants.index]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
